<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Andy Beck" />

<meta name="date" content="2024-06-18" />

<title>recombination_rate</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">phasing_clean</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/thandyb/phasing_clean">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">recombination_rate</h1>
<h4 class="author">Andy Beck</h4>
<h4 class="date">2024-06-18</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-06-26
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>phasing_clean/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240501code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240501)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240501code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240501)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/net/snowwhite/home/beckandy/research/phasing_clean/output/switch_errors/mb_error_rate/
</td>
<td style="text-align:left;">
output/switch_errors/mb_error_rate
</td>
</tr>
<tr>
<td style="text-align:left;">
/net/snowwhite/home/beckandy/research/phasing_clean/output/trio_phase_15/mb_error_rate/
</td>
<td style="text-align:left;">
output/trio_phase_15/mb_error_rate
</td>
</tr>
<tr>
<td style="text-align:left;">
/net/snowwhite/home/beckandy/research/phasing_clean/output/trio_phase_22/mb_error_rate/
</td>
<td style="text-align:left;">
output/trio_phase_22/mb_error_rate
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtheandybphasingcleantree653a35b3c390b4e44e856e2364f3ae6bda88e18etargetblank653a35ba">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/theandyb/phasing_clean/tree/653a35b3c390b4e44e856e2364f3ae6bda88e18e" target="_blank">653a35b</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomtheandybphasingcleantree653a35b3c390b4e44e856e2364f3ae6bda88e18etargetblank653a35ba"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/theandyb/phasing_clean/tree/653a35b3c390b4e44e856e2364f3ae6bda88e18e" target="_blank">653a35b</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .ipynb_checkpoints/
    Ignored:    chr1_10k.bed
    Ignored:    chr22_10k.bed
    Ignored:    data/1kgp/chr1/
    Ignored:    data/1kgp/chr10_pilot_mask.bed
    Ignored:    data/1kgp/chr15/
    Ignored:    data/1kgp/chr15_recomb.bed
    Ignored:    data/1kgp/chr1_recomb.bed
    Ignored:    data/1kgp/chr22/
    Ignored:    data/1kgp/chr22_recomb.bed
    Ignored:    data/1kgp/chr8/
    Ignored:    data/1kgp/chr8_recomb.bed
    Ignored:    data/1kgp/chrX.vcf.gz
    Ignored:    data/1kgp/chrX.vcf.gz.tbi
    Ignored:    data/1kgp/chrX_2504_snps_noPAR.bcf
    Ignored:    data/1kgp/chrX_2504_snps_noPAR_noSing.bcf
    Ignored:    data/1kgp/chrX_freq.tsv
    Ignored:    data/1kgp/chrX_maf.tsv
    Ignored:    data/1kgp/chrX_pilot_mask.bed
    Ignored:    data/1kgp/chrX_recomb.bed
    Ignored:    data/1kgp/exclude_samples/
    Ignored:    data/chr1.1kb.sorted.bed
    Ignored:    data/chr1.fasta
    Ignored:    data/chr1.sizes
    Ignored:    data/chr10.sizes
    Ignored:    data/chr15.1kb.sorted.bed
    Ignored:    data/chr15.fasta
    Ignored:    data/chr15.sizes
    Ignored:    data/chr15_gc1kb_pilot.bed
    Ignored:    data/chr15_mask.fasta
    Ignored:    data/chr15_mask.fasta.fai
    Ignored:    data/chr1_gc1kb_pilot.bed
    Ignored:    data/chr1_mask.fasta
    Ignored:    data/chr1_mask.fasta.fai
    Ignored:    data/chr22.1kb.sorted.bed
    Ignored:    data/chr22.fasta
    Ignored:    data/chr22.sizes
    Ignored:    data/chr22_gc1kb_pilot.bed
    Ignored:    data/chr22_mask.fasta
    Ignored:    data/chr22_mask.fasta.fai
    Ignored:    data/chr8.1kb.sorted.bed
    Ignored:    data/chr8.fasta
    Ignored:    data/chr8.sizes
    Ignored:    data/chr8_gc1kb_pilot.bed
    Ignored:    data/chr8_mask.fasta
    Ignored:    data/chr8_mask.fasta.fai
    Ignored:    data/chrX.1kb.sorted.bed
    Ignored:    data/chrX.fasta
    Ignored:    data/chrX.fasta.fai
    Ignored:    data/chrX.sizes
    Ignored:    data/chrX_gc1kb_pilot.bed
    Ignored:    data/chrX_mask.fasta
    Ignored:    data/chrX_mask.fasta.fai
    Ignored:    data/decode/
    Ignored:    data/hg38.chrom.sizes
    Ignored:    data/mask_pilot_15_fasta.bed
    Ignored:    data/mask_pilot_1_fasta.bed
    Ignored:    data/mask_pilot_22_fasta.bed
    Ignored:    data/mask_pilot_8_fasta.bed
    Ignored:    data/mask_pilot_X_fasta.bed
    Ignored:    data/ref_GRCh38.fna
    Ignored:    data/ref_GRCh38.fna.fai
    Ignored:    data/ref_GRCh38.fna.in
    Ignored:    data/ref_maps/
    Ignored:    output/switch_errors/
    Ignored:    output/trio_phase_1/
    Ignored:    output/trio_phase_15/
    Ignored:    output/trio_phase_22/
    Ignored:    output/trio_phase_8/
    Ignored:    output/vote_15/
    Ignored:    output/vote_22/
    Ignored:    output/vote_x/
    Ignored:    scratch/

Untracked files:
    Untracked:  code/batch_annotate_switch_vote.sh
    Untracked:  code/batch_compare_noTH.sh
    Untracked:  code/batch_vote_phase.sh
    Untracked:  code/batch_vote_phase_trio.sh
    Untracked:  code/vote_phase.R
    Untracked:  code/vote_phase_trio.R
    Untracked:  code/vote_summary.R

Unstaged changes:
    Modified:   README.md
    Modified:   _config.yaml

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/recombination_rate.Rmd</code>)
and HTML (<code>docs/recombination_rate.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/653a35b3c390b4e44e856e2364f3ae6bda88e18e/analysis/recombination_rate.Rmd" target="_blank">653a35b</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-06-26
</td>
<td>
wflow_publish(c("analysis/recombination_rate.Rmd",
"analysis/trio_phasing_results.Rmd"))
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.0     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ purrr::%||%()   masks base::%||%()
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
<pre class="r"><code>library(data.table)</code></pre>
<pre><code>
Attaching package: &#39;data.table&#39;

The following objects are masked from &#39;package:lubridate&#39;:

    hour, isoweek, mday, minute, month, quarter, second, wday, week,
    yday, year

The following objects are masked from &#39;package:dplyr&#39;:

    between, first, last

The following object is masked from &#39;package:purrr&#39;:

    transpose</code></pre>
<pre class="r"><code>library(gridExtra)</code></pre>
<pre><code>
Attaching package: &#39;gridExtra&#39;

The following object is masked from &#39;package:dplyr&#39;:

    combine</code></pre>
<pre class="r"><code>cbPalette &lt;- c(&quot;#999999&quot;, &quot;#E69F00&quot;, &quot;#56B4E9&quot;, &quot;#009E73&quot;, &quot;#F0E442&quot;, &quot;#0072B2&quot;, &quot;#D55E00&quot;, &quot;#CC79A7&quot;)

sd_het_dir &lt;- &quot;output/switch_errors/het_loc/annotated/&quot;
trio22_het_dir &lt;- &quot;output/trio_phase_22/het_loc/annotated/&quot;
trio15_het_dir &lt;- &quot;output/trio_phase_15/het_loc/annotated/&quot;

df_rr_x &lt;- read_tsv(&quot;data/decode/chrX_recomb.bed&quot;, col_names = c(&quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;, &quot;cM_MB&quot;))</code></pre>
<pre><code>Rows: 23976 Columns: 4
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr (1): chrom
dbl (3): start, end, cM_MB

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>df_rr_15 &lt;- read_tsv(&quot;data/decode/chr15_recomb.bed&quot;, col_names = c(&quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;, &quot;cM_MB&quot;)) </code></pre>
<pre><code>Rows: 35539 Columns: 4
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr (1): chrom
dbl (3): start, end, cM_MB

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>df_rr_22 &lt;- read_tsv(&quot;data/decode/chr22_recomb.bed&quot;, col_names = c(&quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;, &quot;cM_MB&quot;))</code></pre>
<pre><code>Rows: 18700 Columns: 4
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr (1): chrom
dbl (3): start, end, cM_MB

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>df_rr_8 &lt;- read_tsv(&quot;data/decode/chr8_recomb.bed&quot;, col_names = c(&quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;, &quot;cM_MB&quot;))</code></pre>
<pre><code>Rows: 56932 Columns: 4
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr (1): chrom
dbl (3): start, end, cM_MB

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>df_rr_1kgp_x &lt;- read_tsv(&quot;data/1kgp/chrX_recomb.bed&quot;, col_names = c(&quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;, &quot;cM_MB&quot;))</code></pre>
<pre><code>Rows: 86701 Columns: 4
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr (1): chrom
dbl (3): start, end, cM_MB

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>df_rr_1kgp_15 &lt;- read_tsv(&quot;data/1kgp/chr15_recomb.bed&quot;, col_names = c(&quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;, &quot;cM_MB&quot;)) </code></pre>
<pre><code>Rows: 85585 Columns: 4
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr (1): chrom
dbl (3): start, end, cM_MB

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>df_rr_1kgp_22 &lt;- read_tsv(&quot;data/1kgp/chr22_recomb.bed&quot;, col_names = c(&quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;, &quot;cM_MB&quot;))</code></pre>
<pre><code>Rows: 40596 Columns: 4
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr (1): chrom
dbl (3): start, end, cM_MB

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>df_rr_1kgp_8 &lt;- read_tsv(&quot;data/1kgp/chr8_recomb.bed&quot;, col_names = c(&quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;, &quot;cM_MB&quot;))</code></pre>
<pre><code>Rows: 176192 Columns: 4
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr (1): chrom
dbl (3): start, end, cM_MB

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<p>Additional information on the trio probands:</p>
<pre class="r"><code>df_subj &lt;- read_csv(&quot;data/1kgp/subject_info.csv&quot;)</code></pre>
<pre><code>Rows: 3202 Columns: 6
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (5): SAMPLE_NAME, POPULATION, SUPER, fatherID, motherID
dbl (1): sex

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>ped_df &lt;- read_table(&quot;data/1kgp/1kGP.3202_samples.pedigree_info.txt&quot;)</code></pre>
<pre><code>
── Column specification ────────────────────────────────────────────────────────
cols(
  sampleID = col_character(),
  fatherID = col_character(),
  motherID = col_character(),
  sex = col_double()
)</code></pre>
<pre class="r"><code>child_ids &lt;- ped_df %&gt;% filter(fatherID != &quot;0&quot; &amp; motherID != &quot;0&quot;) %&gt;% pull(sampleID)
unrel_ids &lt;- scan(&quot;data/1kgp/unrelated_subjects.txt&quot;, what = character())

df_subj_rel &lt;- df_subj %&gt;%
  filter(SAMPLE_NAME %in% child_ids)
df_subj_rel$id2 &lt;- 1:602</code></pre>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>Here in this document we will evaluate the relationship between
recombination rate and switch/flip error rates.</p>
</div>
<div id="single-synthetic-diploid-example" class="section level2">
<h2>Single Synthetic Diploid Example</h2>
<pre class="r"><code>load_het_pos &lt;- function(res_dir, id){
  f_name &lt;- paste0(res_dir, &quot;errors_&quot;, id, &quot;.csv&quot;)
  df &lt;- read_csv(f_name)
  return(df)
}

df_het &lt;- load_het_pos(sd_het_dir, 1)</code></pre>
<pre><code>Rows: 62449 Columns: 13
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr  (2): gt, motif
dbl (11): pos, cpg, beagle_switch, eagle_switch, shapeit_switch, beagle_flip...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code># option 1: interval join
df_het &lt;- setDT(df_het)[setDT(df_rr_x),
              on = .(pos &gt;= start, pos &lt; end), 
             cM_MB := cM_MB][] %&gt;%
  setDF()

df_het %&gt;%
  filter(is.na(cM_MB)) %&gt;%
  pull(shapeit_switch) %&gt;%
  sum()</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># option 2: closest end position
df_het &lt;- load_het_pos(sd_het_dir, 1)</code></pre>
<pre><code>Rows: 62449 Columns: 13
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr  (2): gt, motif
dbl (11): pos, cpg, beagle_switch, eagle_switch, shapeit_switch, beagle_flip...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>df_het &lt;- left_join(df_het, df_rr_x, by = join_by(closest(pos &lt;= end)))

df_het$cM_MB %&gt;% summary()</code></pre>
<pre><code>     Min.   1st Qu.    Median      Mean   3rd Qu.      Max.      NA&#39;s 
    0.000     0.000     0.020     2.100     0.212 21333.100       396 </code></pre>
<pre class="r"><code>df_het %&gt;%
  filter(is.na(cM_MB)) %&gt;%
  pull(eagle_flip) %&gt;%
  sum()</code></pre>
<pre><code>[1] 0</code></pre>
<p>If we roll with the closest end position approach, here is what the
background distribution of cM_MB values at heterozygous positions looks
like:</p>
<pre class="r"><code>df_het %&gt;%
  filter(!is.na(cM_MB)) %&gt;%
  ggplot(aes(x = cM_MB)) +
  geom_boxplot()</code></pre>
<p><img src="figure/recombination_rate.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Welp, hard to see anything; let’s look at flips:</p>
<pre class="r"><code>df_het %&gt;%
  filter(!is.na(cM_MB)) %&gt;%
  select(pos, cM_MB, beagle_flip, eagle_flip, shapeit_flip) %&gt;%
  rename(Beagle = beagle_flip, Eagle = eagle_flip, SHAPEIT = shapeit_flip) %&gt;%
  pivot_longer(Beagle:SHAPEIT, values_to = &quot;flips&quot;, names_to = &quot;method&quot;) %&gt;%
  filter(flips == 1) %&gt;%
  ggplot(aes(x = cM_MB, color = method)) +
  geom_boxplot() +
  scale_color_manual(values = cbPalette)</code></pre>
<p><img src="figure/recombination_rate.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>And switches:</p>
<pre class="r"><code>df_het %&gt;%
  filter(!is.na(cM_MB)) %&gt;%
  select(pos, cM_MB, beagle_switch, eagle_switch, shapeit_switch) %&gt;%
  rename(Beagle = beagle_switch, Eagle = eagle_switch, SHAPEIT = shapeit_switch) %&gt;%
  pivot_longer(Beagle:SHAPEIT, values_to = &quot;flips&quot;, names_to = &quot;method&quot;) %&gt;%
  filter(flips == 1) %&gt;%
  ggplot(aes(x = cM_MB, color = method)) +
  geom_boxplot() +
  scale_color_manual(values = cbPalette)</code></pre>
<p><img src="figure/recombination_rate.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="mb-bin-summary" class="section level2">
<h2>MB Bin Summary</h2>
<p>Let’s first get weighted average values for recombination in MB
bins:</p>
<pre class="r"><code>df_rr_x &lt;- df_rr_x %&gt;%
  mutate(bin = ceiling( (end - ((end - start)/2)) / 1e6),
         width = end - start)

df_rr_15 &lt;- df_rr_15 %&gt;%
  mutate(bin = ceiling( (end - ((end - start)/2)) / 1e6),
         width = end - start)

df_rr_22 &lt;- df_rr_22 %&gt;%
  mutate(bin = ceiling( (end - ((end - start)/2)) / 1e6),
         width = end - start)

df_rr_8 &lt;- df_rr_8 %&gt;%
  mutate(bin = ceiling( (end - ((end - start)/2)) / 1e6),
         width = end - start)

# 1kgp
df_rr_1kgp_x &lt;- df_rr_1kgp_x %&gt;%
  mutate(bin = ceiling( (end - ((end - start)/2)) / 1e6),
         width = end - start)

df_rr_1kgp_8 &lt;- df_rr_1kgp_8 %&gt;%
  mutate(bin = ceiling( (end - ((end - start)/2)) / 1e6),
         width = end - start)

df_rr_1kgp_15 &lt;- df_rr_1kgp_15 %&gt;%
  mutate(bin = ceiling( (end - ((end - start)/2)) / 1e6),
         width = end - start)

df_rr_1kgp_22 &lt;- df_rr_1kgp_22 %&gt;%
  mutate(bin = ceiling( (end - ((end - start)/2)) / 1e6),
         width = end - start)
# summary by bin
df_rr_x_bin &lt;- df_rr_x %&gt;%
  group_by(bin) %&gt;%
  summarise(cM_MB_w = weighted.mean(cM_MB, width, na.rm = T),
            cM_MB_max = max(cM_MB, na.rm = T),
            cM_var = sqrt(sum(width * ((cM_MB - cM_MB_w)^2), na.rm = T)/(sum(width)-1)),
            n_obs = n())

df_rr_15_bin &lt;- df_rr_15 %&gt;%
  group_by(bin) %&gt;%
  summarise(cM_MB_w = weighted.mean(cM_MB, width, na.rm = T),
            cM_MB_max = max(cM_MB, na.rm = T),
            cM_var = sqrt(sum(width * ((cM_MB - cM_MB_w)^2), na.rm = T)/(sum(width)-1)),
            n_obs = n())

df_rr_22_bin &lt;- df_rr_22 %&gt;%
  group_by(bin) %&gt;%
  summarise(cM_MB_w = weighted.mean(cM_MB, width, na.rm = T),
            cM_MB_max = max(cM_MB, na.rm = T),
            cM_var = sqrt(sum(width * ((cM_MB - cM_MB_w)^2), na.rm = T)/(sum(width)-1)),
            n_obs = n())

df_rr_8_bin &lt;- df_rr_8 %&gt;%
  group_by(bin) %&gt;%
  summarise(cM_MB_w = weighted.mean(cM_MB, width, na.rm = T),
            cM_MB_max = max(cM_MB, na.rm = T),
            cM_var = sqrt(sum(width * ((cM_MB - cM_MB_w)^2), na.rm = T)/(sum(width)-1)),
            n_obs = n())

# 1kgp
df_rr_1kgp_x_bin &lt;- df_rr_1kgp_x %&gt;%
  group_by(bin) %&gt;%
  summarise(cM_MB_w = weighted.mean(cM_MB, width, na.rm = T),
            cM_MB_max = max(cM_MB, na.rm = T),
            cM_var = sqrt(sum(width * ((cM_MB - cM_MB_w)^2), na.rm = T)/(sum(width)-1)),
            n_obs = n())

df_rr_1kgp_15_bin &lt;- df_rr_1kgp_15 %&gt;%
  group_by(bin) %&gt;%
  summarise(cM_MB_w = weighted.mean(cM_MB, width, na.rm = T),
            cM_MB_max = max(cM_MB, na.rm = T),
            cM_var = sqrt(sum(width * ((cM_MB - cM_MB_w)^2), na.rm = T)/(sum(width)-1)),
            n_obs = n())

df_rr_1kgp_22_bin &lt;- df_rr_1kgp_22 %&gt;%
  group_by(bin) %&gt;%
  summarise(cM_MB_w = weighted.mean(cM_MB, width, na.rm = T),
            cM_MB_max = max(cM_MB, na.rm = T),
            cM_var = sqrt(sum(width * ((cM_MB - cM_MB_w)^2), na.rm = T)/(sum(width)-1)),
            n_obs = n())

df_rr_1kgp_8_bin &lt;- df_rr_1kgp_8 %&gt;%
  group_by(bin) %&gt;%
  summarise(cM_MB_w = weighted.mean(cM_MB, width, na.rm = T),
            cM_MB_max = max(cM_MB, na.rm = T),
            cM_var = sqrt(sum(width * ((cM_MB - cM_MB_w)^2), na.rm = T)/(sum(width)-1)),
            n_obs = n())</code></pre>
<p>Aside: let’s compare the deCODE and 1kGP recombination rates:</p>
<pre class="r"><code>df_rr_x_bin %&gt;%
  full_join(df_rr_1kgp_x_bin, by = &quot;bin&quot;, suffix = c(&quot;_deCODE&quot;, &quot;_1kGP&quot;)) %&gt;%
  select(bin, contains(&quot;cM_MB_w&quot;)) %&gt;%
  pivot_longer(-bin, names_to = &quot;data_source&quot;, values_to = &quot;cM_MB&quot;) %&gt;%
  ggplot(aes(x = bin, y = cM_MB, color = data_source)) +
  geom_point() +
  geom_line() +
  scale_color_manual(values = cbPalette)</code></pre>
<pre><code>Warning: Removed 5 rows containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<pre><code>Warning: Removed 5 rows containing missing values or values outside the scale range
(`geom_line()`).</code></pre>
<p><img src="figure/recombination_rate.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_rr_15_bin %&gt;%
  full_join(df_rr_1kgp_15_bin, by = &quot;bin&quot;, suffix = c(&quot;_deCODE&quot;, &quot;_1kGP&quot;)) %&gt;%
  select(bin, contains(&quot;cM_MB_w&quot;)) %&gt;%
  pivot_longer(-bin, names_to = &quot;data_source&quot;, values_to = &quot;cM_MB&quot;) %&gt;%
  ggplot(aes(x = bin, y = cM_MB, color = data_source)) +
  geom_point() +
  geom_line() +
  scale_color_manual(values = cbPalette)</code></pre>
<pre><code>Warning: Removed 5 rows containing missing values or values outside the scale range
(`geom_point()`).
Removed 5 rows containing missing values or values outside the scale range
(`geom_line()`).</code></pre>
<p><img src="figure/recombination_rate.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>To do: plot some of these distributions</p>
<div id="load-in-mb-binned-results" class="section level3">
<h3>Load in MB binned results</h3>
<pre class="r"><code>x_mb_dir &lt;- &quot;/net/snowwhite/home/beckandy/research/phasing_clean/output/switch_errors/mb_error_rate/&quot;
chr15_mb_dir &lt;- &quot;/net/snowwhite/home/beckandy/research/phasing_clean/output/trio_phase_15/mb_error_rate/&quot;
chr22_mb_dir &lt;- &quot;/net/snowwhite/home/beckandy/research/phasing_clean/output/trio_phase_22/mb_error_rate/&quot;

load_mb_switch &lt;- function(res_dir, id){
  f_name &lt;- paste0(res_dir, &quot;sample_&quot;, id, &quot;.csv&quot;)
  return(read_csv(f_name, show_col_types = F))
}</code></pre>
</div>
<div id="single-sd-example" class="section level3">
<h3>Single SD Example</h3>
<pre class="r"><code>df_mb_switch &lt;- load_mb_switch(x_mb_dir, 1)

df_mb_switch &lt;- df_mb_switch %&gt;%
  left_join(df_rr_x_bin, by = &quot;bin&quot;)

df_mb_switch %&gt;%
  select(bin,cM_MB_w, n_switch_beagle, n_switch_eagle, n_switch_shapeit) %&gt;%
  rename(Beagle = n_switch_beagle, Eagle = n_switch_eagle, SHAPEIT = n_switch_shapeit) %&gt;%
  pivot_longer( Beagle:SHAPEIT, names_to = &quot;method&quot;, values_to = &quot;Switches&quot;) %&gt;%
  filter(!is.na(cM_MB_w)) %&gt;%
  ggplot(aes(x = cM_MB_w, y = Switches, colour = method)) +
  geom_point() +
  scale_color_manual(values = cbPalette)</code></pre>
<p><img src="figure/recombination_rate.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p1 &lt;- df_mb_switch %&gt;%
  select(bin,cM_MB_w, n_switch_beagle, n_switch_eagle, n_switch_shapeit) %&gt;%
  rename(Beagle = n_switch_beagle, Eagle = n_switch_eagle, SHAPEIT = n_switch_shapeit) %&gt;%
  pivot_longer( Beagle:SHAPEIT, names_to = &quot;method&quot;, values_to = &quot;Switches&quot;) %&gt;%
  ggplot(aes(x = bin, y = Switches, colour = method)) +
  geom_point() +
  geom_line() +
  scale_color_manual(values = cbPalette) +
  theme(legend.position = &quot;top&quot;)

p2 &lt;- df_mb_switch %&gt;%
  ggplot(aes(x = bin, y = cM_MB_w)) +
  geom_point() +
  geom_line()

p3 &lt;- df_mb_switch %&gt;%
  select(bin,cM_MB_w, n_flip_beagle, n_flip_eagle, n_flip_shapeit) %&gt;%
  rename(Beagle = n_flip_beagle, Eagle = n_flip_eagle, SHAPEIT = n_flip_shapeit) %&gt;%
  pivot_longer( Beagle:SHAPEIT, names_to = &quot;method&quot;, values_to = &quot;Flips&quot;) %&gt;%
  ggplot(aes(x = bin, y = Flips, colour = method)) +
  geom_point() +
  geom_line() +
  scale_color_manual(values = cbPalette) +
  guides(color=&quot;none&quot;)

grid.arrange(p1, p2, p3, ncol = 1)</code></pre>
<pre><code>Warning: Removed 2 rows containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<pre><code>Warning: Removed 2 rows containing missing values or values outside the scale range
(`geom_line()`).</code></pre>
<p><img src="figure/recombination_rate.Rmd/unnamed-chunk-10-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="summarizing-errors-by-population" class="section level3">
<h3>Summarizing Errors by Population</h3>
<pre class="r"><code>df_mb_switch_eur &lt;- load_mb_switch(x_mb_dir, 1)

for(i in 2:200){
  df_mb_switch_2 &lt;- load_mb_switch(x_mb_dir, i)
  df_mb_switch_eur &lt;- df_mb_switch_eur %&gt;%
    inner_join(df_mb_switch_2, by = &quot;bin&quot;, suffix = c(&quot;&quot;, &quot;2&quot;))
  df_mb_switch_eur[is.na(df_mb_switch_eur)] &lt;- 0
  df_mb_switch_eur &lt;- df_mb_switch_eur %&gt;%
    mutate(n_het = n_het + n_het2,
           n_switch_beagle = n_switch_beagle + n_switch_beagle2,
           n_switch_eagle = n_switch_eagle + n_switch_eagle2,
           n_switch_shapeit = n_switch_shapeit + n_switch_shapeit2,
           n_flip_beagle = n_flip_beagle + n_flip_beagle2,
           n_flip_eagle = n_flip_eagle + n_flip_eagle2,
           n_flip_shapeit = n_flip_shapeit + n_flip_shapeit2) %&gt;%
    select(!contains(&quot;2&quot;))
}</code></pre>
<pre class="r"><code>df_mb_switch_afr &lt;- load_mb_switch(x_mb_dir, 201)

for(i in 201:400){
  df_mb_switch_2 &lt;- load_mb_switch(x_mb_dir, i)
  df_mb_switch_afr &lt;- df_mb_switch_afr %&gt;%
    inner_join(df_mb_switch_2, by = &quot;bin&quot;, suffix = c(&quot;&quot;, &quot;2&quot;))
  df_mb_switch_afr[is.na(df_mb_switch_afr)] &lt;- 0
  df_mb_switch_afr &lt;- df_mb_switch_afr %&gt;%
    mutate(n_het = n_het + n_het2,
           n_switch_beagle = n_switch_beagle + n_switch_beagle2,
           n_switch_eagle = n_switch_eagle + n_switch_eagle2,
           n_switch_shapeit = n_switch_shapeit + n_switch_shapeit2,
           n_flip_beagle = n_flip_beagle + n_flip_beagle2,
           n_flip_eagle = n_flip_eagle + n_flip_eagle2,
           n_flip_shapeit = n_flip_shapeit + n_flip_shapeit2) %&gt;%
    select(!contains(&quot;2&quot;))
}</code></pre>
<p>Aside: what does the variation of <code>cM_MB</code> look like
per-bin? Let’s plot all values for a single bin in chromosome X (bin
6):</p>
<pre class="r"><code>df_rr_x %&gt;%
  filter(bin == 6) %&gt;%
  ggplot(aes(x = start, y = cM_MB)) +
  geom_point() + 
  geom_hline(yintercept = mean({df_rr_x %&gt;% filter(bin==6) %&gt;% pull(cM_MB)}, na.rm = T), colour = &quot;red&quot;)</code></pre>
<p><img src="figure/recombination_rate.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  geom_hline(yintercept = median({df_rr_x %&gt;% filter(bin==6) %&gt;% pull(cM_MB)}, na.rm = T), colour = &quot;blue&quot;)</code></pre>
<pre><code>mapping: yintercept = ~yintercept 
geom_hline: na.rm = FALSE
stat_identity: na.rm = FALSE
position_identity </code></pre>
<pre class="r"><code>df_rr_x_bin %&gt;%
  select(bin, cM_MB_w, cM_MB_max) %&gt;%
  pivot_longer(-bin, names_to = &quot;stat&quot;, values_to = &quot;val&quot;) %&gt;%
  ggplot(aes(x = bin, y = val, color = stat)) +
    geom_point()</code></pre>
<p><img src="figure/recombination_rate.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_rr_x_bin %&gt;%
  ggplot(aes(x = bin, y = cM_MB_w)) +
  geom_point()</code></pre>
<p><img src="figure/recombination_rate.Rmd/unnamed-chunk-14-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_rr_x_bin %&gt;%
  ggplot(aes(x = bin, y = cM_MB_max)) +
  geom_point()</code></pre>
<p><img src="figure/recombination_rate.Rmd/unnamed-chunk-14-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Plot EUR and AFR counts in each bin against average recombination
rate:</p>
<pre class="r"><code>df_mb_switch_eur &lt;- left_join(df_mb_switch_eur, df_rr_x_bin, by = &quot;bin&quot;)

p1 &lt;- df_mb_switch_eur %&gt;%
  select(bin, contains(&quot;switch&quot;)) %&gt;%
  pivot_longer(-bin, names_to = &quot;stat&quot;, values_to = &quot;val&quot;) %&gt;%
  ggplot(aes(x = bin, y = val, color = stat)) +
  geom_point() +
  geom_line() +
  theme(legend.position = &quot;bottom&quot;)

p2 &lt;- df_mb_switch_eur %&gt;%
  ggplot(aes(x = bin, y = cM_MB_w)) +
  geom_point() +
  geom_line()

grid.arrange(p1, p2, ncol = 1)</code></pre>
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_line()`).</code></pre>
<p><img src="figure/recombination_rate.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>coeff &lt;-  max(df_mb_switch_eur$n_switch_eagle) / max(df_mb_switch_eur$cM_MB_w, na.rm = T)

df_mb_switch_eur %&gt;%
  select(bin, contains(&quot;switch&quot;), cM_MB_w) %&gt;%
  pivot_longer(contains(&quot;switch&quot;), names_to = &quot;stat&quot;, values_to = &quot;val&quot;) %&gt;%
  ggplot(aes(x = bin)) +
    geom_line(aes(y = cM_MB_w)) +
    geom_line(aes(y = val / coeff, color = stat)) +
    scale_y_continuous(
      name = &quot;Weight Mean cM_MB&quot;, 
      sec.axis = sec_axis(~.*coeff, name = &quot;Switches&quot;)
    )</code></pre>
<pre><code>Warning: Removed 3 rows containing missing values or values outside the scale range
(`geom_line()`).</code></pre>
<p><img src="figure/recombination_rate.Rmd/unnamed-chunk-15-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>corrr::correlate({df_mb_switch_eur %&gt;% select(contains(&quot;switch&quot;), cM_MB_w)})</code></pre>
<pre><code>Correlation computed with
• Method: &#39;pearson&#39;
• Missing treated using: &#39;pairwise.complete.obs&#39;</code></pre>
<pre><code># A tibble: 4 × 5
  term             n_switch_beagle n_switch_eagle n_switch_shapeit cM_MB_w
  &lt;chr&gt;                      &lt;dbl&gt;          &lt;dbl&gt;            &lt;dbl&gt;   &lt;dbl&gt;
1 n_switch_beagle           NA              0.987            0.992   0.954
2 n_switch_eagle             0.987         NA                0.983   0.932
3 n_switch_shapeit           0.992          0.983           NA       0.954
4 cM_MB_w                    0.954          0.932            0.954  NA    </code></pre>
<pre class="r"><code>df_mb_switch_eur %&gt;%
  select(bin, contains(&quot;flip&quot;), cM_MB_w) %&gt;%
  pivot_longer(contains(&quot;flip&quot;), names_to = &quot;stat&quot;, values_to = &quot;val&quot;) %&gt;%
  ggplot(aes(x = bin)) +
    geom_line(aes(y = cM_MB_w)) +
    geom_line(aes(y = val / coeff, color = stat)) +
    scale_y_continuous(
      name = &quot;Weight Mean cM_MB&quot;, 
      sec.axis = sec_axis(~.*coeff, name = &quot;Flips&quot;)
    )</code></pre>
<pre><code>Warning: Removed 3 rows containing missing values or values outside the scale range
(`geom_line()`).</code></pre>
<p><img src="figure/recombination_rate.Rmd/unnamed-chunk-15-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>corrr::correlate({df_mb_switch_eur %&gt;% select(contains(&quot;flip&quot;), cM_MB_w)})</code></pre>
<pre><code>Correlation computed with
• Method: &#39;pearson&#39;
• Missing treated using: &#39;pairwise.complete.obs&#39;</code></pre>
<pre><code># A tibble: 4 × 5
  term           n_flip_beagle n_flip_eagle n_flip_shapeit cM_MB_w
  &lt;chr&gt;                  &lt;dbl&gt;        &lt;dbl&gt;          &lt;dbl&gt;   &lt;dbl&gt;
1 n_flip_beagle         NA            0.889          0.955   0.675
2 n_flip_eagle           0.889       NA              0.905   0.709
3 n_flip_shapeit         0.955        0.905         NA       0.669
4 cM_MB_w                0.675        0.709          0.669  NA    </code></pre>
<p>Let’s see if this is robust to choice of recombination rate
estimate:</p>
<pre class="r"><code>df_mb_switch_eur &lt;- left_join(df_mb_switch_eur, df_rr_1kgp_x_bin, by = &quot;bin&quot;, suffix = c(&quot;&quot;, &quot;_1kGP&quot;))

p1 &lt;- df_mb_switch_eur %&gt;%
  select(bin, contains(&quot;switch&quot;)) %&gt;%
  pivot_longer(-bin, names_to = &quot;stat&quot;, values_to = &quot;val&quot;) %&gt;%
  ggplot(aes(x = bin, y = val, color = stat)) +
  geom_point() +
  geom_line() +
  theme(legend.position = &quot;bottom&quot;)

p2 &lt;- df_mb_switch_eur %&gt;%
  ggplot(aes(x = bin, y = cM_MB_w_1kGP)) +
  geom_point() +
  geom_line()

grid.arrange(p1, p2, ncol = 1)</code></pre>
<p><img src="figure/recombination_rate.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>coeff &lt;-  max(df_mb_switch_eur$n_switch_eagle) / max(df_mb_switch_eur$cM_MB_w_1kGP, na.rm = T)

df_mb_switch_eur %&gt;%
  select(bin, contains(&quot;switch&quot;), cM_MB_w_1kGP) %&gt;%
  pivot_longer(contains(&quot;switch&quot;), names_to = &quot;stat&quot;, values_to = &quot;val&quot;) %&gt;%
  ggplot(aes(x = bin)) +
    geom_line(aes(y = cM_MB_w_1kGP)) +
    geom_line(aes(y = val / coeff, color = stat)) +
    scale_y_continuous(
      name = &quot;Weight Mean cM_MB&quot;, 
      sec.axis = sec_axis(~.*coeff, name = &quot;Switches&quot;)
    )</code></pre>
<p><img src="figure/recombination_rate.Rmd/unnamed-chunk-16-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>corrr::correlate({df_mb_switch_eur %&gt;% select(contains(&quot;switch&quot;), cM_MB_w_1kGP)})</code></pre>
<pre><code>Correlation computed with
• Method: &#39;pearson&#39;
• Missing treated using: &#39;pairwise.complete.obs&#39;</code></pre>
<pre><code># A tibble: 4 × 5
  term             n_switch_beagle n_switch_eagle n_switch_shapeit cM_MB_w_1kGP
  &lt;chr&gt;                      &lt;dbl&gt;          &lt;dbl&gt;            &lt;dbl&gt;        &lt;dbl&gt;
1 n_switch_beagle           NA              0.987            0.992        0.932
2 n_switch_eagle             0.987         NA                0.983        0.906
3 n_switch_shapeit           0.992          0.983           NA            0.929
4 cM_MB_w_1kGP               0.932          0.906            0.929       NA    </code></pre>
<pre class="r"><code>df_mb_switch_eur %&gt;%
  select(bin, contains(&quot;flip&quot;), cM_MB_w_1kGP) %&gt;%
  pivot_longer(contains(&quot;flip&quot;), names_to = &quot;stat&quot;, values_to = &quot;val&quot;) %&gt;%
  ggplot(aes(x = bin)) +
    geom_line(aes(y = cM_MB_w_1kGP)) +
    geom_line(aes(y = val / coeff, color = stat)) +
    scale_y_continuous(
      name = &quot;Weight Mean cM_MB&quot;, 
      sec.axis = sec_axis(~.*coeff, name = &quot;Flips&quot;)
    )</code></pre>
<p><img src="figure/recombination_rate.Rmd/unnamed-chunk-16-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>corrr::correlate({df_mb_switch_eur %&gt;% select(contains(&quot;flip&quot;), cM_MB_w_1kGP)})</code></pre>
<pre><code>Correlation computed with
• Method: &#39;pearson&#39;
• Missing treated using: &#39;pairwise.complete.obs&#39;</code></pre>
<pre><code># A tibble: 4 × 5
  term           n_flip_beagle n_flip_eagle n_flip_shapeit cM_MB_w_1kGP
  &lt;chr&gt;                  &lt;dbl&gt;        &lt;dbl&gt;          &lt;dbl&gt;        &lt;dbl&gt;
1 n_flip_beagle         NA            0.889          0.955        0.628
2 n_flip_eagle           0.889       NA              0.905        0.669
3 n_flip_shapeit         0.955        0.905         NA            0.628
4 cM_MB_w_1kGP           0.628        0.669          0.628       NA    </code></pre>
<p>And now AFR:</p>
<pre class="r"><code>df_mb_switch_afr &lt;- left_join(df_mb_switch_afr, df_rr_x_bin, by = &quot;bin&quot;)
df_mb_switch_afr &lt;- left_join(df_mb_switch_afr, df_rr_1kgp_x_bin, by = &quot;bin&quot;, suffix = c(&quot;&quot;, &quot;_1kGP&quot;))

coeff &lt;-  max(df_mb_switch_afr$n_switch_eagle) / max(df_mb_switch_afr$cM_MB_w, na.rm = T)

df_mb_switch_afr %&gt;%
  select(bin, contains(&quot;switch&quot;), cM_MB_w) %&gt;%
  pivot_longer(contains(&quot;switch&quot;), names_to = &quot;stat&quot;, values_to = &quot;val&quot;) %&gt;%
  ggplot(aes(x = bin)) +
    geom_line(aes(y = cM_MB_w)) +
    geom_line(aes(y = val / coeff, color = stat)) +
    scale_y_continuous(
      name = &quot;Weight Mean cM_MB&quot;, 
      sec.axis = sec_axis(~.*coeff, name = &quot;Switches&quot;)
    )</code></pre>
<p><img src="figure/recombination_rate.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>corrr::correlate({df_mb_switch_afr %&gt;% select(contains(&quot;switch&quot;), cM_MB_w)})</code></pre>
<pre><code>Correlation computed with
• Method: &#39;pearson&#39;
• Missing treated using: &#39;pairwise.complete.obs&#39;</code></pre>
<pre><code># A tibble: 4 × 5
  term             n_switch_beagle n_switch_eagle n_switch_shapeit cM_MB_w
  &lt;chr&gt;                      &lt;dbl&gt;          &lt;dbl&gt;            &lt;dbl&gt;   &lt;dbl&gt;
1 n_switch_beagle           NA              0.973            0.991   0.887
2 n_switch_eagle             0.973         NA                0.969   0.847
3 n_switch_shapeit           0.991          0.969           NA       0.893
4 cM_MB_w                    0.887          0.847            0.893  NA    </code></pre>
<pre class="r"><code>coeff &lt;-  max(df_mb_switch_afr$n_flip_shapeit) / max(df_mb_switch_afr$cM_MB_w, na.rm = T)

df_mb_switch_afr %&gt;%
  select(bin, contains(&quot;flip&quot;), cM_MB_w) %&gt;%
  pivot_longer(contains(&quot;flip&quot;), names_to = &quot;stat&quot;, values_to = &quot;val&quot;) %&gt;%
  ggplot(aes(x = bin)) +
    geom_line(aes(y = cM_MB_w)) +
    geom_line(aes(y = val / coeff, color = stat)) +
    scale_y_continuous(
      name = &quot;Weight Mean cM_MB&quot;, 
      sec.axis = sec_axis(~.*coeff, name = &quot;Flips&quot;)
    )</code></pre>
<p><img src="figure/recombination_rate.Rmd/unnamed-chunk-17-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>corrr::correlate({df_mb_switch_afr %&gt;% select(contains(&quot;flip&quot;), cM_MB_w)})</code></pre>
<pre><code>Correlation computed with
• Method: &#39;pearson&#39;
• Missing treated using: &#39;pairwise.complete.obs&#39;</code></pre>
<pre><code># A tibble: 4 × 5
  term           n_flip_beagle n_flip_eagle n_flip_shapeit cM_MB_w
  &lt;chr&gt;                  &lt;dbl&gt;        &lt;dbl&gt;          &lt;dbl&gt;   &lt;dbl&gt;
1 n_flip_beagle         NA            0.950          0.971   0.596
2 n_flip_eagle           0.950       NA              0.958   0.630
3 n_flip_shapeit         0.971        0.958         NA       0.597
4 cM_MB_w                0.596        0.630          0.597  NA    </code></pre>
<div id="chromosome-15" class="section level4">
<h4>Chromosome 15</h4>
<pre class="r"><code>eur_proband_ids &lt;- df_subj_rel %&gt;% filter(SUPER == &quot;EUR&quot;) %&gt;% pull(id2)

df_mb_switch_eur &lt;- load_mb_switch(chr15_mb_dir, eur_proband_ids[1])

for(i in 2:length(eur_proband_ids)){
  id &lt;- eur_proband_ids[i]
  df_mb_switch_2 &lt;- load_mb_switch(chr15_mb_dir, id)
  df_mb_switch_eur &lt;- df_mb_switch_eur %&gt;%
    inner_join(df_mb_switch_2, by = &quot;bin&quot;, suffix = c(&quot;&quot;, &quot;2&quot;))
  df_mb_switch_eur[is.na(df_mb_switch_eur)] &lt;- 0
  df_mb_switch_eur &lt;- df_mb_switch_eur %&gt;%
    mutate(n_het = n_het + n_het2,
           n_switch_beagle = n_switch_beagle + n_switch_beagle2,
           n_switch_eagle = n_switch_eagle + n_switch_eagle2,
           n_switch_shapeit = n_switch_shapeit + n_switch_shapeit2,
           n_flip_beagle = n_flip_beagle + n_flip_beagle2,
           n_flip_eagle = n_flip_eagle + n_flip_eagle2,
           n_flip_shapeit = n_flip_shapeit + n_flip_shapeit2) %&gt;%
    select(!contains(&quot;2&quot;))
}</code></pre>
<pre class="r"><code>df_mb_switch_eur &lt;- left_join(df_mb_switch_eur, df_rr_15_bin, by = &quot;bin&quot;)
df_mb_switch_eur &lt;- left_join(df_mb_switch_eur, df_rr_1kgp_15_bin, by = &quot;bin&quot;, suffix = c(&quot;&quot;, &quot;_1kGP&quot;))

coeff &lt;-  max({df_mb_switch_eur %&gt;% select(contains(&quot;switch&quot;))}) / max(df_mb_switch_eur$cM_MB_w, na.rm = T)

df_mb_switch_eur %&gt;%
  select(bin, contains(&quot;switch&quot;), cM_MB_w) %&gt;%
  pivot_longer(contains(&quot;switch&quot;), names_to = &quot;stat&quot;, values_to = &quot;val&quot;) %&gt;%
  ggplot(aes(x = bin)) +
    geom_line(aes(y = cM_MB_w)) +
    geom_line(aes(y = val / coeff, color = stat)) +
    scale_y_continuous(
      name = &quot;Weight Mean cM_MB&quot;, 
      sec.axis = sec_axis(~.*coeff, name = &quot;Switches&quot;)
    )</code></pre>
<pre><code>Warning: Removed 15 rows containing missing values or values outside the scale range
(`geom_line()`).</code></pre>
<p><img src="figure/recombination_rate.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>corrr::correlate({df_mb_switch_eur %&gt;% select(contains(&quot;switch&quot;), cM_MB_w)})</code></pre>
<pre><code>Correlation computed with
• Method: &#39;pearson&#39;
• Missing treated using: &#39;pairwise.complete.obs&#39;</code></pre>
<pre><code># A tibble: 4 × 5
  term             n_switch_beagle n_switch_eagle n_switch_shapeit cM_MB_w
  &lt;chr&gt;                      &lt;dbl&gt;          &lt;dbl&gt;            &lt;dbl&gt;   &lt;dbl&gt;
1 n_switch_beagle           NA              0.992            0.996   0.739
2 n_switch_eagle             0.992         NA                0.994   0.756
3 n_switch_shapeit           0.996          0.994           NA       0.774
4 cM_MB_w                    0.739          0.756            0.774  NA    </code></pre>
<pre class="r"><code>coeff &lt;-  max({df_mb_switch_eur %&gt;% select(contains(&quot;flip&quot;))}) / max(df_mb_switch_eur$cM_MB_w, na.rm = T)

df_mb_switch_eur %&gt;%
  select(bin, contains(&quot;flip&quot;), cM_MB_w) %&gt;%
  pivot_longer(contains(&quot;flip&quot;), names_to = &quot;stat&quot;, values_to = &quot;val&quot;) %&gt;%
  ggplot(aes(x = bin)) +
    geom_line(aes(y = cM_MB_w)) +
    geom_line(aes(y = val / coeff, color = stat)) +
    scale_y_continuous(
      name = &quot;Weight Mean cM_MB&quot;, 
      sec.axis = sec_axis(~.*coeff, name = &quot;Flips&quot;)
    )</code></pre>
<pre><code>Warning: Removed 15 rows containing missing values or values outside the scale range
(`geom_line()`).</code></pre>
<p><img src="figure/recombination_rate.Rmd/unnamed-chunk-19-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>corrr::correlate({df_mb_switch_eur %&gt;% select(contains(&quot;flip&quot;), cM_MB_w)})</code></pre>
<pre><code>Correlation computed with
• Method: &#39;pearson&#39;
• Missing treated using: &#39;pairwise.complete.obs&#39;</code></pre>
<pre><code># A tibble: 4 × 5
  term           n_flip_beagle n_flip_eagle n_flip_shapeit cM_MB_w
  &lt;chr&gt;                  &lt;dbl&gt;        &lt;dbl&gt;          &lt;dbl&gt;   &lt;dbl&gt;
1 n_flip_beagle         NA            0.975          0.987   0.574
2 n_flip_eagle           0.975       NA              0.972   0.540
3 n_flip_shapeit         0.987        0.972         NA       0.545
4 cM_MB_w                0.574        0.540          0.545  NA    </code></pre>
<p>It looks like RR and error rates are much less correlated here.</p>
</div>
</div>
</div>
<div id="triple-het-rates" class="section level2">
<h2>Triple Het Rates</h2>
<p>Let’s instead compare the chromosome 15 error rates to the rate of
triple hets in MB bins. First, let’s sum the number of triple het sites
in each trio in each MB bin</p>
<pre class="r"><code>triple_het_dir &lt;- &quot;data/1kgp/chr15/triple_het/&quot;

df_th_eur &lt;- read_tsv(paste0(triple_het_dir, 
                             &quot;sample_&quot;, 
                             eur_proband_ids[1],
                             &quot;.tsv&quot;), 
                      col_names = c(&quot;chr&quot;, &quot;pos&quot;),
                      show_col_types = FALSE) %&gt;%
  mutate(bin = ceiling(pos / 1e6)) %&gt;%
  group_by(bin) %&gt;%
  summarize(n_th = n()) %&gt;%
  select(bin, n_th)

for(i in 2:length(eur_proband_ids)){
  df_th_eur2 &lt;- read_tsv(paste0(triple_het_dir, 
                             &quot;sample_&quot;, 
                             eur_proband_ids[i],
                             &quot;.tsv&quot;), 
                      col_names = c(&quot;chr&quot;, &quot;pos&quot;),
                      show_col_types = FALSE) %&gt;%
    mutate(bin = ceiling(pos / 1e6)) %&gt;%
    group_by(bin) %&gt;%
    summarize(n_th = n()) %&gt;%
    select(bin, n_th)
  
  df_th_eur &lt;- full_join(df_th_eur, df_th_eur2, by = &quot;bin&quot;, suffix = c(&quot;&quot;,&quot;2&quot;)) %&gt;%
    replace_na(list(&quot;n_th&quot; = 0, &quot;n_th2&quot; = 0)) %&gt;%
    mutate(n_th = n_th + n_th2) %&gt;%
    select(-n_th2)

}</code></pre>
<p>Let’s see how error counts per bin looks like plotted along with
triple het counts per bin:</p>
<pre class="r"><code>df_mb_switch_eur &lt;- full_join(df_mb_switch_eur, df_th_eur, by = &quot;bin&quot;)

coeff &lt;-  max(df_mb_switch_eur$n_th, na.rm = T) / 
  max({df_mb_switch_eur %&gt;% select(contains(&quot;switch&quot;))}, na.rm=T)

df_mb_switch_eur %&gt;%
  select(bin, contains(&quot;switch&quot;), n_th) %&gt;%
  pivot_longer(contains(&quot;switch&quot;), names_to = &quot;stat&quot;, values_to = &quot;val&quot;) %&gt;%
  ggplot(aes(x = bin)) +
    geom_line(aes(y = n_th / coeff)) +
    geom_line(aes(y = val, color = stat)) +
    scale_y_continuous(
      name = &quot;Switches&quot;, 
      sec.axis = sec_axis(~.*coeff, name = &quot;Count Triple Hets&quot;)
    )</code></pre>
<p><img src="figure/recombination_rate.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>corrr::correlate({df_mb_switch_eur %&gt;% select(contains(&quot;switch&quot;), n_th)})</code></pre>
<pre><code>Correlation computed with
• Method: &#39;pearson&#39;
• Missing treated using: &#39;pairwise.complete.obs&#39;</code></pre>
<pre><code># A tibble: 4 × 5
  term             n_switch_beagle n_switch_eagle n_switch_shapeit     n_th
  &lt;chr&gt;                      &lt;dbl&gt;          &lt;dbl&gt;            &lt;dbl&gt;    &lt;dbl&gt;
1 n_switch_beagle         NA               0.992            0.996  -0.00346
2 n_switch_eagle           0.992          NA                0.994   0.0423 
3 n_switch_shapeit         0.996           0.994           NA       0.0113 
4 n_th                    -0.00346         0.0423           0.0113 NA      </code></pre>
<pre class="r"><code>coeff &lt;-  max(df_mb_switch_eur$n_th, na.rm = T) / max(df_mb_switch_eur$n_flip_shapeit, na.rm=T)

df_mb_switch_eur %&gt;%
  select(bin, contains(&quot;flip&quot;), cM_MB_w) %&gt;%
  pivot_longer(contains(&quot;flip&quot;), names_to = &quot;stat&quot;, values_to = &quot;val&quot;) %&gt;%
  ggplot(aes(x = bin)) +
    geom_line(aes(y = cM_MB_w)) +
    geom_line(aes(y = val / coeff, color = stat)) +
    scale_y_continuous(
      name = &quot;Weight Mean cM_MB&quot;, 
      sec.axis = sec_axis(~.*coeff, name = &quot;Flips&quot;)
    )</code></pre>
<pre><code>Warning: Removed 18 rows containing missing values or values outside the scale range
(`geom_line()`).</code></pre>
<p><img src="figure/recombination_rate.Rmd/unnamed-chunk-21-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>corrr::correlate({df_mb_switch_eur %&gt;% select(contains(&quot;flip&quot;), cM_MB_w)})</code></pre>
<pre><code>Correlation computed with
• Method: &#39;pearson&#39;
• Missing treated using: &#39;pairwise.complete.obs&#39;</code></pre>
<pre><code># A tibble: 4 × 5
  term           n_flip_beagle n_flip_eagle n_flip_shapeit cM_MB_w
  &lt;chr&gt;                  &lt;dbl&gt;        &lt;dbl&gt;          &lt;dbl&gt;   &lt;dbl&gt;
1 n_flip_beagle         NA            0.975          0.987   0.574
2 n_flip_eagle           0.975       NA              0.972   0.540
3 n_flip_shapeit         0.987        0.972         NA       0.545
4 cM_MB_w                0.574        0.540          0.545  NA    </code></pre>
<div id="chr-22" class="section level3">
<h3>Chr 22</h3>
<pre class="r"><code>triple_het_dir &lt;- &quot;data/1kgp/chr22/triple_het/&quot;

df_th_eur &lt;- read_tsv(paste0(triple_het_dir, 
                             &quot;sample_&quot;, 
                             eur_proband_ids[1],
                             &quot;.tsv&quot;), 
                      col_names = c(&quot;chr&quot;, &quot;pos&quot;),
                      show_col_types = FALSE) %&gt;%
  mutate(bin = ceiling(pos / 1e6)) %&gt;%
  group_by(bin) %&gt;%
  summarize(n_th = n()) %&gt;%
  select(bin, n_th)

for(i in 2:length(eur_proband_ids)){
  df_th_eur2 &lt;- read_tsv(paste0(triple_het_dir, 
                             &quot;sample_&quot;, 
                             eur_proband_ids[i],
                             &quot;.tsv&quot;), 
                      col_names = c(&quot;chr&quot;, &quot;pos&quot;),
                      show_col_types = FALSE) %&gt;%
    mutate(bin = ceiling(pos / 1e6)) %&gt;%
    group_by(bin) %&gt;%
    summarize(n_th = n()) %&gt;%
    select(bin, n_th)
  
  df_th_eur &lt;- full_join(df_th_eur, df_th_eur2, by = &quot;bin&quot;, suffix = c(&quot;&quot;,&quot;2&quot;)) %&gt;%
    replace_na(list(&quot;n_th&quot; = 0, &quot;n_th2&quot; = 0)) %&gt;%
    mutate(n_th = n_th + n_th2) %&gt;%
    select(-n_th2)

}

eur_proband_ids &lt;- df_subj_rel %&gt;% filter(SUPER == &quot;EUR&quot;) %&gt;% pull(id2)

df_mb_switch_eur &lt;- load_mb_switch(chr22_mb_dir, eur_proband_ids[1])

for(i in 2:length(eur_proband_ids)){
  id &lt;- eur_proband_ids[i]
  df_mb_switch_2 &lt;- load_mb_switch(chr22_mb_dir, id)
  df_mb_switch_eur &lt;- df_mb_switch_eur %&gt;%
    inner_join(df_mb_switch_2, by = &quot;bin&quot;, suffix = c(&quot;&quot;, &quot;2&quot;))
  df_mb_switch_eur[is.na(df_mb_switch_eur)] &lt;- 0
  df_mb_switch_eur &lt;- df_mb_switch_eur %&gt;%
    mutate(n_het = n_het + n_het2,
           n_switch_beagle = n_switch_beagle + n_switch_beagle2,
           n_switch_eagle = n_switch_eagle + n_switch_eagle2,
           n_switch_shapeit = n_switch_shapeit + n_switch_shapeit2,
           n_flip_beagle = n_flip_beagle + n_flip_beagle2,
           n_flip_eagle = n_flip_eagle + n_flip_eagle2,
           n_flip_shapeit = n_flip_shapeit + n_flip_shapeit2) %&gt;%
    select(!contains(&quot;2&quot;))
}</code></pre>
<pre class="r"><code>df_mb_switch_eur &lt;- left_join(df_mb_switch_eur, df_rr_22_bin, by = &quot;bin&quot;)
df_mb_switch_eur &lt;- left_join(df_mb_switch_eur, df_rr_1kgp_22_bin, by = &quot;bin&quot;, suffix = c(&quot;&quot;, &quot;_1kGP&quot;))

coeff &lt;-  max({df_mb_switch_eur %&gt;% select(contains(&quot;switch&quot;))}, na.rm = T) /
  max(df_mb_switch_eur$cM_MB_w_1kGP, na.rm = T)

df_mb_switch_eur %&gt;%
  select(bin, contains(&quot;switch&quot;), cM_MB_w_1kGP) %&gt;%
  pivot_longer(contains(&quot;switch&quot;), names_to = &quot;stat&quot;, values_to = &quot;val&quot;) %&gt;%
  ggplot(aes(x = bin)) +
    geom_line(aes(y = cM_MB_w_1kGP)) +
    geom_line(aes(y = val / coeff, color = stat)) +
    scale_y_continuous(
      name = &quot;Weight Mean cM_MB&quot;, 
      sec.axis = sec_axis(~.*coeff, name = &quot;Switches&quot;)
    )</code></pre>
<pre><code>Warning: Removed 9 rows containing missing values or values outside the scale range
(`geom_line()`).</code></pre>
<p><img src="figure/recombination_rate.Rmd/unnamed-chunk-23-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>corrr::correlate({df_mb_switch_eur %&gt;% select(contains(&quot;switch&quot;), cM_MB_w_1kGP)})</code></pre>
<pre><code>Correlation computed with
• Method: &#39;pearson&#39;
• Missing treated using: &#39;pairwise.complete.obs&#39;</code></pre>
<pre><code># A tibble: 4 × 5
  term             n_switch_beagle n_switch_eagle n_switch_shapeit cM_MB_w_1kGP
  &lt;chr&gt;                      &lt;dbl&gt;          &lt;dbl&gt;            &lt;dbl&gt;        &lt;dbl&gt;
1 n_switch_beagle           NA              0.991            0.994        0.701
2 n_switch_eagle             0.991         NA                0.988        0.750
3 n_switch_shapeit           0.994          0.988           NA            0.742
4 cM_MB_w_1kGP               0.701          0.750            0.742       NA    </code></pre>
<pre class="r"><code>df_mb_switch_eur &lt;- full_join(df_mb_switch_eur, df_th_eur, by = &quot;bin&quot;)

coeff &lt;-  max(df_mb_switch_eur$n_th, na.rm = T) / 
  max({df_mb_switch_eur %&gt;% select(contains(&quot;switch&quot;))}, na.rm=T)

df_mb_switch_eur %&gt;%
  select(bin, contains(&quot;switch&quot;), n_th) %&gt;%
  pivot_longer(contains(&quot;switch&quot;), names_to = &quot;stat&quot;, values_to = &quot;val&quot;) %&gt;%
  ggplot(aes(x = bin)) +
    geom_line(aes(y = n_th / coeff)) +
    geom_line(aes(y = val, color = stat)) +
    scale_y_continuous(
      name = &quot;Switches&quot;, 
      sec.axis = sec_axis(~.*coeff, name = &quot;Count Triple Hets&quot;)
    )</code></pre>
<p><img src="figure/recombination_rate.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>corrr::correlate({df_mb_switch_eur %&gt;% select(contains(&quot;switch&quot;), n_th)})</code></pre>
<pre><code>Correlation computed with
• Method: &#39;pearson&#39;
• Missing treated using: &#39;pairwise.complete.obs&#39;</code></pre>
<pre><code># A tibble: 4 × 5
  term             n_switch_beagle n_switch_eagle n_switch_shapeit   n_th
  &lt;chr&gt;                      &lt;dbl&gt;          &lt;dbl&gt;            &lt;dbl&gt;  &lt;dbl&gt;
1 n_switch_beagle           NA              0.991            0.994 -0.151
2 n_switch_eagle             0.991         NA                0.988 -0.105
3 n_switch_shapeit           0.994          0.988           NA     -0.138
4 n_th                      -0.151         -0.105           -0.138 NA    </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 20.04.6 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3;  LAPACK version 3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: America/New_York
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] gridExtra_2.3     data.table_1.15.4 lubridate_1.9.3   forcats_1.0.0    
 [5] stringr_1.5.1     dplyr_1.1.4       purrr_1.0.2       readr_2.1.5      
 [9] tidyr_1.3.1       tibble_3.2.1      ggplot2_3.5.0     tidyverse_2.0.0  
[13] workflowr_1.7.1  

loaded via a namespace (and not attached):
 [1] gtable_0.3.5      xfun_0.42         bslib_0.7.0       processx_3.8.3   
 [5] corrr_0.4.4       callr_3.7.5       tzdb_0.4.0        vctrs_0.6.5      
 [9] tools_4.4.1       ps_1.7.6          generics_0.1.3    parallel_4.4.1   
[13] fansi_1.0.6       highr_0.11        pkgconfig_2.0.3   lifecycle_1.0.4  
[17] compiler_4.4.1    farver_2.1.2      git2r_0.33.0      munsell_0.5.1    
[21] getPass_0.2-4     httpuv_1.6.14     htmltools_0.5.8   sass_0.4.9       
[25] yaml_2.3.8        later_1.3.2       pillar_1.9.0      crayon_1.5.2     
[29] jquerylib_0.1.4   whisker_0.4.1     cachem_1.1.0      tidyselect_1.2.0 
[33] digest_0.6.34     stringi_1.8.3     labeling_0.4.3    rprojroot_2.0.4  
[37] fastmap_1.2.0     grid_4.4.1        archive_1.1.8     colorspace_2.1-0 
[41] cli_3.6.2         magrittr_2.0.3    utf8_1.2.4        withr_3.0.0      
[45] scales_1.3.0      promises_1.3.0    bit64_4.0.5       timechange_0.3.0 
[49] rmarkdown_2.27    httr_1.4.7        bit_4.0.5         hms_1.1.3        
[53] evaluate_0.24.0   knitr_1.45        rlang_1.1.3       Rcpp_1.0.12      
[57] glue_1.7.0        rstudioapi_0.15.0 vroom_1.6.5       jsonlite_1.8.8   
[61] R6_2.5.1          fs_1.6.3         </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
