<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Andy Beck" />

<meta name="date" content="2024-02-20" />

<title>venn_diagram</title>

<script src="site_libs/header-attrs-2.26/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">phasing_clean</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/thandyb/phasing_clean">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">venn_diagram</h1>
<h4 class="author">Andy Beck</h4>
<h4 class="date">2024-02-20</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-06-03
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>phasing_clean/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240501code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240501)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240501code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240501)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtheandybphasingcleantreeaaaebd1311d2982b32c8af00f535ba79a81cbb53targetblankaaaebd1a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/theandyb/phasing_clean/tree/aaaebd1311d2982b32c8af00f535ba79a81cbb53" target="_blank">aaaebd1</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomtheandybphasingcleantreeaaaebd1311d2982b32c8af00f535ba79a81cbb53targetblankaaaebd1a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/theandyb/phasing_clean/tree/aaaebd1311d2982b32c8af00f535ba79a81cbb53" target="_blank">aaaebd1</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .ipynb_checkpoints/
    Ignored:    chr22_10k.bed
    Ignored:    data/1kgp/chr10_pilot_mask.bed
    Ignored:    data/1kgp/chr15/
    Ignored:    data/1kgp/chr22/
    Ignored:    data/1kgp/chr8/
    Ignored:    data/1kgp/chrX.vcf.gz
    Ignored:    data/1kgp/chrX.vcf.gz.tbi
    Ignored:    data/1kgp/chrX_2504_snps_noPAR.bcf
    Ignored:    data/1kgp/chrX_2504_snps_noPAR_noSing.bcf
    Ignored:    data/1kgp/chrX_freq.tsv
    Ignored:    data/1kgp/chrX_maf.tsv
    Ignored:    data/1kgp/chrX_pilot_mask.bed
    Ignored:    data/chr10.sizes
    Ignored:    data/chr15.1kb.sorted.bed
    Ignored:    data/chr15.fasta
    Ignored:    data/chr15.sizes
    Ignored:    data/chr15_gc1kb_pilot.bed
    Ignored:    data/chr15_mask.fasta
    Ignored:    data/chr15_mask.fasta.fai
    Ignored:    data/chr22.sizes
    Ignored:    data/chr8.sizes
    Ignored:    data/chrX.1kb.sorted.bed
    Ignored:    data/chrX.fasta
    Ignored:    data/chrX.sizes
    Ignored:    data/chrX_gc1kb_pilot.bed
    Ignored:    data/chrX_mask.fasta
    Ignored:    data/chrX_mask.fasta.fai
    Ignored:    data/decode/
    Ignored:    data/hg38.chrom.sizes
    Ignored:    data/mask_pilot_15_fasta.bed
    Ignored:    data/mask_pilot_X_fasta.bed
    Ignored:    data/ref_GRCh38.fna
    Ignored:    data/ref_GRCh38.fna.fai
    Ignored:    data/ref_GRCh38.fna.in
    Ignored:    data/ref_maps/
    Ignored:    output/switch_errors/
    Ignored:    output/trio_phase/
    Ignored:    output/trio_phase_15/
    Ignored:    output/trio_phase_22/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/sd_error_overlap.Rmd</code>) and
HTML (<code>docs/sd_error_overlap.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/aaaebd1311d2982b32c8af00f535ba79a81cbb53/analysis/sd_error_overlap.Rmd" target="_blank">aaaebd1</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-06-03
</td>
<td>
wflow_publish("analysis/sd_error_overlap.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/phasing_clean/808cfb808faf1a67f36a54d323404932439cf9ba/docs/sd_error_overlap.html" target="_blank">808cfb8</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/phasing_clean/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/sd_error_overlap.html" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/487e8bfddd230b30f0cc1d0041b8e3639eee78d3/analysis/sd_error_overlap.Rmd" target="_blank">487e8bf</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
<td>
wflow_publish(c("analysis/index.Rmd", "analysis/sd_error_overlap.Rmd"))
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>rr_df %&gt;%
  ggplot(aes(x = pos_start, y = rr)) +
  geom_point()</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>In this document we will explore the construction of a “venn diagram”
of the overlap of phasing errors across three methods over 1000
synthetic diploids. Intuitively, this makes sense for a single synthetic
diploid: we partition all errors into categories determined by which
method(s) produced the error. The potential complication is if we wish
to look only at particular classes of errors, e.g. switches or flips.
For an example of what can happen, suppose a heterozygous location is
the site of a switch in Beagle. That same position could be the location
of a switch in Eagle or SHAPEIT, or it may have been correctly phased.
Those are not exhaustive, however; another potential outcome is that
same location is the latter half of a flip error in either Eagle or
SHAPEIT. Is this still considered an overlapping switch?</p>
<p>For a first pass, I’ll be a bit exhaustive in terms of the categories
of overlap for the errors in a single synthetic diploid.</p>
<pre class="r"><code>load_het_pos &lt;- function(id, het_loc_dir){
  fname &lt;- paste0(het_loc_dir, &quot;pair_&quot;, id, &quot;.csv&quot;)
  return(read_csv(fname, show_col_types = FALSE))
}

load_switch_df &lt;- function(id, switch_dir){
  fname &lt;- paste0(switch_dir, &quot;switch_&quot;, id, &quot;.csv&quot;)
  results &lt;- read_csv(fname, show_col_types = FALSE)
  flip_loc &lt;- get_flip_pos(results)
  results$is_flip &lt;- (results$pos_start %in% flip_loc) | (results$pos_end %in% flip_loc)
  results$flip &lt;- results$pos_end %in% flip_loc
  # results &lt;- results %&gt;%
  #   filter(flip | (!is_flip &amp; !flip))
  # return(results %&gt;% select(-is_flip))
  return(results)
}

annotate_het_list &lt;- function(id, het_loc_dir, beagle_dir, eagle_dir, shapeit_dir, df_freq){
  result &lt;- load_het_pos(id, het_loc_dir)
  df_b &lt;- load_switch_df(id, beagle_dir)
  df_e &lt;- load_switch_df(id, eagle_dir)
  df_s &lt;- load_switch_df(id, shapeit_dir)
  
  result$switch_b &lt;- result$pos %in% {df_b %&gt;% filter(!is_flip) %&gt;% pull(pos_end)}
  result$flip_b &lt;- result$pos %in% {df_b %&gt;% filter(flip) %&gt;% pull(pos_end)}
  result$flip2_b &lt;- result$pos %in% {df_b %&gt;% filter(is_flip &amp; !flip) %&gt;% pull(pos_end)}
  
  result$switch_e &lt;- result$pos %in% {df_e %&gt;% filter(!is_flip) %&gt;% pull(pos_end)}
  result$flip_e &lt;- result$pos %in% {df_e %&gt;% filter(flip) %&gt;% pull(pos_end)}
  result$flip2_e &lt;- result$pos %in% {df_e %&gt;% filter(is_flip &amp; !flip) %&gt;% pull(pos_end)}
  
  result$switch_s &lt;- result$pos %in% {df_s %&gt;% filter(!is_flip) %&gt;% pull(pos_end)}
  result$flip_s &lt;- result$pos %in% {df_s %&gt;% filter(flip) %&gt;% pull(pos_end)}
  result$flip2_s &lt;- result$pos %in% {df_s %&gt;% filter(is_flip &amp; !flip) %&gt;% pull(pos_end)}
  
  result &lt;- left_join(result,
                       {df_freq %&gt;%
                           select(pos, maf)},
                       by = c(&quot;pos&quot; = &quot;pos&quot;))
  
  return(result)
}

df &lt;- annotate_het_list(1, het_loc_dir, beagle_switch_dir, eagle_switch_dir, shapeit_switch_dir, df_freq)</code></pre>
<p>For the <code>ggVennDiagram</code> library, we will want lists of
positions for each error type (note that here we “double count” flips by
including the positions of both involved heterozygous sites):</p>
<pre class="r"><code>x &lt;- list(beagle = {df %&gt;% filter(switch_b | flip_b | flip2_b) %&gt;% pull(pos)},
          eagle = {df %&gt;% filter(switch_e | flip_e | flip2_e) %&gt;% pull(pos)},
          shapeit = {df %&gt;% filter(switch_s | flip_s | flip2_s) %&gt;% pull(pos)})
ggVennDiagram(x, label = &quot;count&quot;) + scale_fill_gradient(low=&quot;grey90&quot;,high = &quot;red&quot;)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Another way to do this would be to simply consider switches and flips
separately, not worrying about cases where a switch in one method
overlaps a flip in another.</p>
<pre class="r"><code># switches
x &lt;- list(beagle = {df %&gt;% filter(switch_b) %&gt;% pull(pos)},
          eagle =  {df %&gt;% filter(switch_e) %&gt;% pull(pos)},
          shapeit =  {df %&gt;% filter(switch_s) %&gt;% pull(pos)})
p1 &lt;- ggVennDiagram(x, label = &quot;count&quot;) + scale_fill_gradient(low=&quot;grey90&quot;,high = &quot;blue&quot;)

# flips
x &lt;- list(beagle = {df %&gt;% filter(flip_b) %&gt;% pull(pos)},
          eagle =  {df %&gt;% filter(flip_e) %&gt;% pull(pos)},
          shapeit =  {df %&gt;% filter(flip_s) %&gt;% pull(pos)})
p2 &lt;- ggVennDiagram(x, label = &quot;count&quot;) + scale_fill_gradient(low=&quot;grey90&quot;,high = &quot;green&quot;)

gridExtra::grid.arrange(p1, p2, ncol=2)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-5-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Or, hear me out: we say nuts to the above, and just do the very
simple criteria of:</p>
<ol style="list-style-type: decimal">
<li>Switches are shared only with switches that occur at the same
position</li>
<li>Flips are shared only with flips that start at the same
position</li>
<li>Everything else is distinct as far as we care</li>
</ol>
<p>So for each SD we get two venn diagrams with 3 circles.</p>
<pre class="r"><code>x &lt;- list(beagle = {df %&gt;% filter(switch_b) %&gt;% pull(pos)},
          eagle = {df %&gt;% filter(switch_e) %&gt;% pull(pos)},
          shapeit = {df %&gt;% filter(switch_s) %&gt;% pull(pos)})
p1 &lt;- ggVennDiagram(x, label = &quot;count&quot;) + 
  scale_fill_gradient(low=&quot;grey90&quot;,high = &quot;green&quot;) + 
  ggtitle(&quot;Switches&quot;) + 
  theme(plot.title = element_text(hjust = 0.5))

x &lt;- list(beagle = {df %&gt;% filter(flip_b) %&gt;% pull(pos)},
          eagle = {df %&gt;% filter(flip_e) %&gt;% pull(pos)},
          shapeit = {df %&gt;% filter(flip_s) %&gt;% pull(pos)})
p2 &lt;- ggVennDiagram(x, label=&quot;count&quot;) + 
  scale_fill_gradient(low=&quot;grey90&quot;,high = &quot;blue&quot;) + 
  ggtitle(&quot;Flips&quot;) + 
  theme(plot.title = element_text(hjust = 0.5))

gridExtra::grid.arrange(p1,p2,ncol=2)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-6-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Naturally, we may ask how many intersections of errors do we miss by
considering only matching error types and, in the case of flips, only
matching directly overlapping errors. First, we look at how many
switches from each method overlap flips from the other two methods:</p>
<pre class="r"><code>x &lt;- list(beagle = {df %&gt;% filter(switch_b) %&gt;% pull(pos)},
          eagle_flip = {df %&gt;% filter(flip_e | flip2_e) %&gt;% pull(pos)},
          eagle_switch = {df %&gt;% filter(switch_e) %&gt;% pull(pos)},
          shapeit_switch = {df %&gt;% filter(switch_s) %&gt;% pull(pos)},
          shapeit_flip = {df %&gt;% filter(flip_s | flip2_s) %&gt;% pull(pos)})
p1 &lt;- ggVennDiagram(x, label = &quot;count&quot;) + scale_fill_gradient(low=&quot;grey90&quot;,high = &quot;purple&quot;) + 
  theme(plot.title = element_text(hjust = 0.5))

x &lt;- list(beagle = {df %&gt;% filter(switch_b) %&gt;% pull(pos)},
          eagle_flip = {df %&gt;% filter(flip_e | flip2_e) %&gt;% pull(pos)},
          shapeit_flip = {df %&gt;% filter(flip_s | flip2_s) %&gt;% pull(pos)})
p2 &lt;- ggVennDiagram(x, label = &quot;count&quot;) + scale_fill_gradient(low=&quot;grey90&quot;,high = &quot;darkgreen&quot;) + 
  theme(plot.title = element_text(hjust = 0.5))

gridExtra::grid.arrange(p1,p2,ncol=2)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-7-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>x &lt;- list(beagle_flip = {df %&gt;% filter(flip_b | flip2_b) %&gt;% pull(pos)},
          beagle_switch = {df %&gt;% filter(switch_b) %&gt;% pull(pos)},
          eagle = {df %&gt;% filter(switch_e) %&gt;% pull(pos)},
          shapeit_switch = {df %&gt;% filter(switch_s) %&gt;% pull(pos)},
          shapeit_flip = {df %&gt;% filter(flip_s | flip2_s) %&gt;% pull(pos)})
p1 &lt;- ggVennDiagram(x, label = &quot;count&quot;) + scale_fill_gradient(low=&quot;grey90&quot;,high = &quot;purple&quot;) + 
  theme(plot.title = element_text(hjust = 0.5))

x &lt;- list(beagle_flip = {df %&gt;% filter(flip_b | flip2_b) %&gt;% pull(pos)},
          eagle = {df %&gt;% filter(switch_e) %&gt;% pull(pos)},
          shapeit_flip = {df %&gt;% filter(flip_s | flip2_s) %&gt;% pull(pos)})
p2 &lt;- ggVennDiagram(x, label = &quot;count&quot;) + scale_fill_gradient(low=&quot;grey90&quot;,high = &quot;darkgreen&quot;) + 
  theme(plot.title = element_text(hjust = 0.5))

gridExtra::grid.arrange(p1,p2,ncol=2)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-8-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>x &lt;- list(beagle_flip = {df %&gt;% filter(flip_b | flip2_b) %&gt;% pull(pos)},
          beagle_switch = {df %&gt;% filter(switch_b) %&gt;% pull(pos)},
          eagle_switch = {df %&gt;% filter(switch_e) %&gt;% pull(pos)},
          eagle_flip = {df %&gt;% filter(flip_e | flip2_e) %&gt;% pull(pos)},
          shapeit = {df %&gt;% filter(switch_s) %&gt;% pull(pos)})
p1 &lt;- ggVennDiagram(x, label = &quot;count&quot;) + scale_fill_gradient(low=&quot;grey90&quot;,high = &quot;purple&quot;) + 
  theme(plot.title = element_text(hjust = 0.5))

x &lt;- list(beagle_flip = {df %&gt;% filter(flip_b | flip2_b) %&gt;% pull(pos)},
          eagle_flip = {df %&gt;% filter(flip_e | flip2_e) %&gt;% pull(pos)},
          shapeit = {df %&gt;% filter(switch_s) %&gt;% pull(pos)})
p2 &lt;- ggVennDiagram(x, label = &quot;count&quot;) + scale_fill_gradient(low=&quot;grey90&quot;,high = &quot;darkgreen&quot;) + 
  theme(plot.title = element_text(hjust = 0.5))

gridExtra::grid.arrange(p1,p2,ncol=2)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>So, if we wanted to aggregate these counts across multiple SDs, we’d
have to generate 14 numbers per SD. Is there a quick way of doing
this?</p>
<pre class="r"><code>df_switch &lt;- df %&gt;%
  filter(switch_b | switch_e | switch_s) %&gt;%
  mutate(beagle = ifelse(switch_b, &quot;Y&quot;, &quot;N&quot;),
         eagle = ifelse(switch_e, &quot;Y&quot;, &quot;N&quot;),
         shapeit = ifelse(switch_s, &quot;Y&quot;, &quot;N&quot;)) %&gt;%
  mutate(count_cat = paste(beagle, eagle, shapeit, sep=&quot;_&quot;)) %&gt;%
  pull(count_cat) %&gt;%
  table() %&gt;%
  as.data.frame() %&gt;%
  rename(count_cat = &quot;.&quot;)

df_flip &lt;- df %&gt;%
  filter(flip_b | flip_e | flip_s) %&gt;%
  mutate(beagle = ifelse(flip_b, &quot;Y&quot;, &quot;N&quot;),
         eagle = ifelse(flip_e, &quot;Y&quot;, &quot;N&quot;),
         shapeit = ifelse(flip_s, &quot;Y&quot;, &quot;N&quot;)) %&gt;%
  mutate(count_cat = paste(beagle, eagle, shapeit, sep=&quot;_&quot;)) %&gt;%
  pull(count_cat) %&gt;%
  table() %&gt;%
  as.data.frame() %&gt;%
  rename(count_cat = &quot;.&quot;)

df_switch_noflip &lt;- df %&gt;%
  filter((switch_b | switch_e | switch_s) &amp; !(flip_b | flip2_b | flip_e | flip2_e | flip_s | flip2_s)) %&gt;%
  mutate(beagle = ifelse(switch_b, &quot;Y&quot;, &quot;N&quot;),
         eagle = ifelse(switch_e, &quot;Y&quot;, &quot;N&quot;),
         shapeit = ifelse(switch_s, &quot;Y&quot;, &quot;N&quot;)) %&gt;%
  mutate(count_cat = paste(beagle, eagle, shapeit, sep=&quot;_&quot;)) %&gt;%
  pull(count_cat) %&gt;%
  table() %&gt;%
  as.data.frame() %&gt;%
  rename(count_cat = &quot;.&quot;)</code></pre>
</div>
<div id="simple-overlap" class="section level1">
<h1>Simple Overlap</h1>
<pre class="r"><code># Version which only does exact matches of flips and switches
get_error_counts &lt;- function(ids, het_loc_dir, beagle_switch_dir, eagle_switch_dir, shapeit_switch_dir, df_freq){
  n &lt;- length(ids)
  results &lt;- vector(mode = &quot;list&quot;, length = 2)
  
  df &lt;- annotate_het_list(ids[1], het_loc_dir, beagle_switch_dir, eagle_switch_dir, shapeit_switch_dir, df_freq)
  df_switch &lt;- df %&gt;%
    filter(switch_b | switch_e | switch_s) %&gt;%
    mutate(beagle = ifelse(switch_b, &quot;Y&quot;, &quot;N&quot;),
           eagle = ifelse(switch_e, &quot;Y&quot;, &quot;N&quot;),
           shapeit = ifelse(switch_s, &quot;Y&quot;, &quot;N&quot;)) %&gt;%
    mutate(count_cat = paste(beagle, eagle, shapeit, sep=&quot;_&quot;)) %&gt;%
    pull(count_cat) %&gt;%
    table() %&gt;%
    as.data.frame() %&gt;%
    rename(count_cat = &quot;.&quot;)
  
  df_flip &lt;- df %&gt;%
    filter(flip_b | flip_e | flip_s) %&gt;%
    mutate(beagle = ifelse(flip_b, &quot;Y&quot;, &quot;N&quot;),
           eagle = ifelse(flip_e, &quot;Y&quot;, &quot;N&quot;),
           shapeit = ifelse(flip_s, &quot;Y&quot;, &quot;N&quot;)) %&gt;%
    mutate(count_cat = paste(beagle, eagle, shapeit, sep=&quot;_&quot;)) %&gt;%
    pull(count_cat) %&gt;%
    table() %&gt;%
    as.data.frame() %&gt;%
    rename(count_cat = &quot;.&quot;)
  
  df_switch_noflip &lt;- df %&gt;%
    filter((switch_b | switch_e | switch_s) &amp; !(flip_b | flip2_b | flip_e | flip2_e | flip_s | flip2_s)) %&gt;%
    mutate(beagle = ifelse(switch_b, &quot;Y&quot;, &quot;N&quot;),
           eagle = ifelse(switch_e, &quot;Y&quot;, &quot;N&quot;),
           shapeit = ifelse(switch_s, &quot;Y&quot;, &quot;N&quot;)) %&gt;%
    mutate(count_cat = paste(beagle, eagle, shapeit, sep=&quot;_&quot;)) %&gt;%
    pull(count_cat) %&gt;%
    table() %&gt;%
    as.data.frame() %&gt;%
    rename(count_cat = &quot;.&quot;)
  
  for(i in 2:n){
    df &lt;- annotate_het_list(ids[i], het_loc_dir, beagle_switch_dir, eagle_switch_dir, shapeit_switch_dir, df_freq)
    df_switch2 &lt;- df %&gt;%
      filter(switch_b | switch_e | switch_s) %&gt;%
      mutate(beagle = ifelse(switch_b, &quot;Y&quot;, &quot;N&quot;),
             eagle = ifelse(switch_e, &quot;Y&quot;, &quot;N&quot;),
             shapeit = ifelse(switch_s, &quot;Y&quot;, &quot;N&quot;)) %&gt;%
      mutate(count_cat = paste(beagle, eagle, shapeit, sep=&quot;_&quot;)) %&gt;%
      pull(count_cat) %&gt;%
      table() %&gt;%
      as.data.frame() %&gt;%
      rename(count_cat = &quot;.&quot;, Freq2 = Freq)
    
    df_switch &lt;- df_switch %&gt;%
      full_join(df_switch2, by=&quot;count_cat&quot;) %&gt;%
      replace_na(list(&quot;Freq&quot; = 0, &quot;Freq2&quot; = 0)) %&gt;%
      mutate(Freq = Freq + Freq2) %&gt;%
      select(-Freq2)
      
    
    df_flip2 &lt;- df %&gt;%
      filter(flip_b | flip_e | flip_s) %&gt;%
      mutate(beagle = ifelse(flip_b, &quot;Y&quot;, &quot;N&quot;),
             eagle = ifelse(flip_e, &quot;Y&quot;, &quot;N&quot;),
             shapeit = ifelse(flip_s, &quot;Y&quot;, &quot;N&quot;)) %&gt;%
      mutate(count_cat = paste(beagle, eagle, shapeit, sep=&quot;_&quot;)) %&gt;%
      pull(count_cat) %&gt;%
      table() %&gt;%
      as.data.frame() %&gt;%
      rename(count_cat = &quot;.&quot;, Freq2 = Freq)
    
    df_flip &lt;- df_flip %&gt;%
      full_join(df_flip2, by=&quot;count_cat&quot;) %&gt;%
      replace_na(list(&quot;Freq&quot; = 0, &quot;Freq2&quot; = 0)) %&gt;%
      mutate(Freq = Freq + Freq2) %&gt;%
      select(-Freq2)
    
    df_switch_noflip2 &lt;- df %&gt;%
      filter((switch_b | switch_e | switch_s) &amp; !(flip_b | flip2_b | flip_e | flip2_e | flip_s | flip2_s)) %&gt;%
      mutate(beagle = ifelse(switch_b, &quot;Y&quot;, &quot;N&quot;),
             eagle = ifelse(switch_e, &quot;Y&quot;, &quot;N&quot;),
             shapeit = ifelse(switch_s, &quot;Y&quot;, &quot;N&quot;)) %&gt;%
      mutate(count_cat = paste(beagle, eagle, shapeit, sep=&quot;_&quot;)) %&gt;%
      pull(count_cat) %&gt;%
      table() %&gt;%
      as.data.frame() %&gt;%
      rename(count_cat = &quot;.&quot;, Freq2 = Freq)
    
    df_switch_noflip &lt;- df_switch_noflip %&gt;%
      full_join(df_switch_noflip2, by=&quot;count_cat&quot;) %&gt;%
      replace_na(list(&quot;Freq&quot; = 0, &quot;Freq2&quot; = 0)) %&gt;%
      mutate(Freq = Freq + Freq2) %&gt;%
      select(-Freq2)
  }
  results[[1]] &lt;- df_switch
  results[[2]] &lt;- df_flip
  results[[3]] &lt;- df_switch_noflip
  return(results)
}</code></pre>
<div id="eur" class="section level2">
<h2>EUR</h2>
<pre class="r"><code>results_eur &lt;- get_error_counts(1:200, het_loc_dir, beagle_switch_dir, eagle_switch_dir, shapeit_switch_dir, df_freq)

results_eur[[1]] %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">count_cat</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">N_N_Y</td>
<td align="right">5930</td>
</tr>
<tr class="even">
<td align="left">N_Y_N</td>
<td align="right">14437</td>
</tr>
<tr class="odd">
<td align="left">N_Y_Y</td>
<td align="right">2314</td>
</tr>
<tr class="even">
<td align="left">Y_N_N</td>
<td align="right">7016</td>
</tr>
<tr class="odd">
<td align="left">Y_N_Y</td>
<td align="right">3050</td>
</tr>
<tr class="even">
<td align="left">Y_Y_N</td>
<td align="right">2641</td>
</tr>
<tr class="odd">
<td align="left">Y_Y_Y</td>
<td align="right">5620</td>
</tr>
</tbody>
</table>
<pre class="r"><code>results_eur[[2]] %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">count_cat</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">N_N_Y</td>
<td align="right">6631</td>
</tr>
<tr class="even">
<td align="left">N_Y_N</td>
<td align="right">6070</td>
</tr>
<tr class="odd">
<td align="left">N_Y_Y</td>
<td align="right">2422</td>
</tr>
<tr class="even">
<td align="left">Y_N_N</td>
<td align="right">4902</td>
</tr>
<tr class="odd">
<td align="left">Y_N_Y</td>
<td align="right">3717</td>
</tr>
<tr class="even">
<td align="left">Y_Y_N</td>
<td align="right">1942</td>
</tr>
<tr class="odd">
<td align="left">Y_Y_Y</td>
<td align="right">4490</td>
</tr>
</tbody>
</table>
<pre class="r"><code>results_eur[[3]] %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">count_cat</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">N_N_Y</td>
<td align="right">5181</td>
</tr>
<tr class="even">
<td align="left">N_Y_N</td>
<td align="right">9584</td>
</tr>
<tr class="odd">
<td align="left">N_Y_Y</td>
<td align="right">2045</td>
</tr>
<tr class="even">
<td align="left">Y_N_N</td>
<td align="right">5361</td>
</tr>
<tr class="odd">
<td align="left">Y_N_Y</td>
<td align="right">2726</td>
</tr>
<tr class="even">
<td align="left">Y_Y_N</td>
<td align="right">2226</td>
</tr>
<tr class="odd">
<td align="left">Y_Y_Y</td>
<td align="right">5620</td>
</tr>
</tbody>
</table>
<pre class="r"><code>count_to_ven &lt;- function(x){
  df &lt;- x %&gt;%
    mutate(beagle = ifelse(str_sub(count_cat, 1, 1)==&quot;Y&quot;, 1, 0),
           eagle = ifelse(str_sub(count_cat, 3, 3)==&quot;Y&quot;, 1, 0),
           shapeit = ifelse(str_sub(count_cat, 5, 5)== &quot;Y&quot;, 1, 0)) %&gt;%
    select(beagle, eagle, shapeit, Freq)
  
  n_b &lt;- df %&gt;% filter(beagle == 1, eagle == 0, shapeit == 0) %&gt;% pull(Freq)
  n_e &lt;- df %&gt;% filter(beagle == 0, eagle == 1, shapeit == 0) %&gt;% pull(Freq)
  n_s &lt;- df %&gt;% filter(beagle == 0, eagle == 0, shapeit == 1) %&gt;% pull(Freq)
  n_b_e &lt;- df %&gt;% filter(beagle == 1, eagle == 1, shapeit == 0) %&gt;% pull(Freq)
  n_b_s &lt;- df %&gt;% filter(beagle == 1, eagle == 0, shapeit == 1) %&gt;% pull(Freq)
  n_e_s &lt;- df %&gt;% filter(beagle == 0, eagle == 1, shapeit == 1) %&gt;% pull(Freq)
  n_b_e_s &lt;- df %&gt;% filter(beagle == 1, eagle == 1, shapeit == 1) %&gt;% pull(Freq)
  
  data.frame(beagle = c(rep(TRUE, n_b), rep(FALSE, n_e),rep(FALSE, n_s), rep(TRUE, n_b_e), rep(TRUE, n_b_s), rep(FALSE, n_e_s) , rep(TRUE, n_b_e_s)),
             eagle = c(rep(FALSE, n_b), rep(TRUE, n_e),rep(FALSE, n_s), rep(TRUE, n_b_e), rep(FALSE, n_b_s), rep(TRUE, n_e_s) , rep(TRUE, n_b_e_s)),
             shapeit = c(rep(FALSE, n_b), rep(FALSE, n_e), rep(TRUE, n_s), rep(FALSE, n_b_e), rep(TRUE, n_b_s), rep(TRUE, n_e_s) , rep(TRUE, n_b_e_s)) 
             ) %&gt;%
    vennCounts() %&gt;%
    vennDiagram() %&gt;% print()
}

count_to_ven(results_eur[[1]])</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-13-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>NULL</code></pre>
<pre class="r"><code>count_to_ven(results_eur[[2]])</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-13-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-2">
Past versions of unnamed-chunk-13-2.png
</button>
</p>
<div id="fig-unnamed-chunk-13-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-13-2.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>NULL</code></pre>
<pre class="r"><code>count_to_ven(results_eur[[3]])</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-13-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-3">
Past versions of unnamed-chunk-13-3.png
</button>
</p>
<div id="fig-unnamed-chunk-13-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-13-3.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>NULL</code></pre>
</div>
<div id="afr" class="section level2">
<h2>AFR</h2>
<pre class="r"><code>results_afr &lt;- get_error_counts(201:400, het_loc_dir, beagle_switch_dir, eagle_switch_dir, shapeit_switch_dir, df_freq)

results_afr[[1]] %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">count_cat</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">N_N_Y</td>
<td align="right">5202</td>
</tr>
<tr class="even">
<td align="left">N_Y_N</td>
<td align="right">17861</td>
</tr>
<tr class="odd">
<td align="left">N_Y_Y</td>
<td align="right">1855</td>
</tr>
<tr class="even">
<td align="left">Y_N_N</td>
<td align="right">7291</td>
</tr>
<tr class="odd">
<td align="left">Y_N_Y</td>
<td align="right">3084</td>
</tr>
<tr class="even">
<td align="left">Y_Y_N</td>
<td align="right">2102</td>
</tr>
<tr class="odd">
<td align="left">Y_Y_Y</td>
<td align="right">5239</td>
</tr>
</tbody>
</table>
<pre class="r"><code>results_afr[[2]] %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">count_cat</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">N_N_Y</td>
<td align="right">8866</td>
</tr>
<tr class="even">
<td align="left">N_Y_N</td>
<td align="right">8395</td>
</tr>
<tr class="odd">
<td align="left">N_Y_Y</td>
<td align="right">3519</td>
</tr>
<tr class="even">
<td align="left">Y_N_N</td>
<td align="right">6401</td>
</tr>
<tr class="odd">
<td align="left">Y_N_Y</td>
<td align="right">5689</td>
</tr>
<tr class="even">
<td align="left">Y_Y_N</td>
<td align="right">2261</td>
</tr>
<tr class="odd">
<td align="left">Y_Y_Y</td>
<td align="right">5783</td>
</tr>
</tbody>
</table>
<pre class="r"><code>results_afr[[3]] %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">count_cat</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">N_N_Y</td>
<td align="right">4529</td>
</tr>
<tr class="even">
<td align="left">N_Y_N</td>
<td align="right">11204</td>
</tr>
<tr class="odd">
<td align="left">N_Y_Y</td>
<td align="right">1614</td>
</tr>
<tr class="even">
<td align="left">Y_N_N</td>
<td align="right">5171</td>
</tr>
<tr class="odd">
<td align="left">Y_N_Y</td>
<td align="right">2776</td>
</tr>
<tr class="even">
<td align="left">Y_Y_N</td>
<td align="right">1723</td>
</tr>
<tr class="odd">
<td align="left">Y_Y_Y</td>
<td align="right">5239</td>
</tr>
</tbody>
</table>
<pre class="r"><code>count_to_ven(results_afr[[1]])</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-14-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>NULL</code></pre>
<pre class="r"><code>count_to_ven(results_afr[[2]])</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-14-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-2">
Past versions of unnamed-chunk-14-2.png
</button>
</p>
<div id="fig-unnamed-chunk-14-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-14-2.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>NULL</code></pre>
<pre class="r"><code>count_to_ven(results_afr[[3]])</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-14-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-3">
Past versions of unnamed-chunk-14-3.png
</button>
</p>
<div id="fig-unnamed-chunk-14-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-14-3.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>NULL</code></pre>
</div>
<div id="amr" class="section level2">
<h2>AMR</h2>
<pre class="r"><code>results_amr &lt;- get_error_counts(401:600, het_loc_dir, beagle_switch_dir, eagle_switch_dir, shapeit_switch_dir, df_freq)

results_amr[[1]] %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">count_cat</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">N_N_Y</td>
<td align="right">4991</td>
</tr>
<tr class="even">
<td align="left">N_Y_N</td>
<td align="right">12136</td>
</tr>
<tr class="odd">
<td align="left">N_Y_Y</td>
<td align="right">1785</td>
</tr>
<tr class="even">
<td align="left">Y_N_N</td>
<td align="right">5661</td>
</tr>
<tr class="odd">
<td align="left">Y_N_Y</td>
<td align="right">2475</td>
</tr>
<tr class="even">
<td align="left">Y_Y_N</td>
<td align="right">2160</td>
</tr>
<tr class="odd">
<td align="left">Y_Y_Y</td>
<td align="right">4466</td>
</tr>
</tbody>
</table>
<pre class="r"><code>results_amr[[2]] %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">count_cat</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">N_N_Y</td>
<td align="right">6199</td>
</tr>
<tr class="even">
<td align="left">N_Y_N</td>
<td align="right">5570</td>
</tr>
<tr class="odd">
<td align="left">N_Y_Y</td>
<td align="right">2294</td>
</tr>
<tr class="even">
<td align="left">Y_N_N</td>
<td align="right">4417</td>
</tr>
<tr class="odd">
<td align="left">Y_N_Y</td>
<td align="right">3603</td>
</tr>
<tr class="even">
<td align="left">Y_Y_N</td>
<td align="right">1811</td>
</tr>
<tr class="odd">
<td align="left">Y_Y_Y</td>
<td align="right">4013</td>
</tr>
</tbody>
</table>
<pre class="r"><code>results_amr[[3]] %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">count_cat</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">N_N_Y</td>
<td align="right">4284</td>
</tr>
<tr class="even">
<td align="left">N_Y_N</td>
<td align="right">7498</td>
</tr>
<tr class="odd">
<td align="left">N_Y_Y</td>
<td align="right">1542</td>
</tr>
<tr class="even">
<td align="left">Y_N_N</td>
<td align="right">4079</td>
</tr>
<tr class="odd">
<td align="left">Y_N_Y</td>
<td align="right">2189</td>
</tr>
<tr class="even">
<td align="left">Y_Y_N</td>
<td align="right">1799</td>
</tr>
<tr class="odd">
<td align="left">Y_Y_Y</td>
<td align="right">4466</td>
</tr>
</tbody>
</table>
<pre class="r"><code>count_to_ven(results_amr[[1]])</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-15-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>NULL</code></pre>
<pre class="r"><code>count_to_ven(results_amr[[2]])</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-15-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-2">
Past versions of unnamed-chunk-15-2.png
</button>
</p>
<div id="fig-unnamed-chunk-15-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-15-2.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>NULL</code></pre>
<pre class="r"><code>count_to_ven(results_amr[[3]])</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-15-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-3">
Past versions of unnamed-chunk-15-3.png
</button>
</p>
<div id="fig-unnamed-chunk-15-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-15-3.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>NULL</code></pre>
</div>
<div id="eas" class="section level2">
<h2>EAS</h2>
<pre class="r"><code>results_eas &lt;- get_error_counts(601:800, het_loc_dir, beagle_switch_dir, eagle_switch_dir, shapeit_switch_dir, df_freq)

results_eas[[1]] %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">count_cat</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">N_N_Y</td>
<td align="right">10730</td>
</tr>
<tr class="even">
<td align="left">N_Y_N</td>
<td align="right">22191</td>
</tr>
<tr class="odd">
<td align="left">N_Y_Y</td>
<td align="right">4244</td>
</tr>
<tr class="even">
<td align="left">Y_N_N</td>
<td align="right">12299</td>
</tr>
<tr class="odd">
<td align="left">Y_N_Y</td>
<td align="right">5171</td>
</tr>
<tr class="even">
<td align="left">Y_Y_N</td>
<td align="right">4631</td>
</tr>
<tr class="odd">
<td align="left">Y_Y_Y</td>
<td align="right">8527</td>
</tr>
</tbody>
</table>
<pre class="r"><code>results_eas[[2]] %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">count_cat</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">N_N_Y</td>
<td align="right">9027</td>
</tr>
<tr class="even">
<td align="left">N_Y_N</td>
<td align="right">9409</td>
</tr>
<tr class="odd">
<td align="left">N_Y_Y</td>
<td align="right">3357</td>
</tr>
<tr class="even">
<td align="left">Y_N_N</td>
<td align="right">7903</td>
</tr>
<tr class="odd">
<td align="left">Y_N_Y</td>
<td align="right">4990</td>
</tr>
<tr class="even">
<td align="left">Y_Y_N</td>
<td align="right">2803</td>
</tr>
<tr class="odd">
<td align="left">Y_Y_Y</td>
<td align="right">5464</td>
</tr>
</tbody>
</table>
<pre class="r"><code>results_eas[[3]] %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">count_cat</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">N_N_Y</td>
<td align="right">9100</td>
</tr>
<tr class="even">
<td align="left">N_Y_N</td>
<td align="right">15107</td>
</tr>
<tr class="odd">
<td align="left">N_Y_Y</td>
<td align="right">3635</td>
</tr>
<tr class="even">
<td align="left">Y_N_N</td>
<td align="right">9545</td>
</tr>
<tr class="odd">
<td align="left">Y_N_Y</td>
<td align="right">4540</td>
</tr>
<tr class="even">
<td align="left">Y_Y_N</td>
<td align="right">3865</td>
</tr>
<tr class="odd">
<td align="left">Y_Y_Y</td>
<td align="right">8527</td>
</tr>
</tbody>
</table>
<pre class="r"><code>count_to_ven(results_eas[[1]])</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-16-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>NULL</code></pre>
<pre class="r"><code>count_to_ven(results_eas[[2]])</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-16-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-2">
Past versions of unnamed-chunk-16-2.png
</button>
</p>
<div id="fig-unnamed-chunk-16-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-16-2.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>NULL</code></pre>
<pre class="r"><code>count_to_ven(results_eas[[3]])</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-16-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-3">
Past versions of unnamed-chunk-16-3.png
</button>
</p>
<div id="fig-unnamed-chunk-16-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-16-3.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>NULL</code></pre>
</div>
<div id="sas" class="section level2">
<h2>SAS</h2>
<pre class="r"><code>results_sas &lt;- get_error_counts(801:1000, het_loc_dir, beagle_switch_dir, eagle_switch_dir, shapeit_switch_dir, df_freq)

results_sas[[1]] %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">count_cat</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">N_N_Y</td>
<td align="right">8759</td>
</tr>
<tr class="even">
<td align="left">N_Y_N</td>
<td align="right">19682</td>
</tr>
<tr class="odd">
<td align="left">N_Y_Y</td>
<td align="right">3492</td>
</tr>
<tr class="even">
<td align="left">Y_N_N</td>
<td align="right">10692</td>
</tr>
<tr class="odd">
<td align="left">Y_N_Y</td>
<td align="right">4281</td>
</tr>
<tr class="even">
<td align="left">Y_Y_N</td>
<td align="right">3863</td>
</tr>
<tr class="odd">
<td align="left">Y_Y_Y</td>
<td align="right">7098</td>
</tr>
</tbody>
</table>
<pre class="r"><code>results_sas[[2]] %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">count_cat</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">N_N_Y</td>
<td align="right">9313</td>
</tr>
<tr class="even">
<td align="left">N_Y_N</td>
<td align="right">9225</td>
</tr>
<tr class="odd">
<td align="left">N_Y_Y</td>
<td align="right">3366</td>
</tr>
<tr class="even">
<td align="left">Y_N_N</td>
<td align="right">7465</td>
</tr>
<tr class="odd">
<td align="left">Y_N_Y</td>
<td align="right">4720</td>
</tr>
<tr class="even">
<td align="left">Y_Y_N</td>
<td align="right">2565</td>
</tr>
<tr class="odd">
<td align="left">Y_Y_Y</td>
<td align="right">5428</td>
</tr>
</tbody>
</table>
<pre class="r"><code>results_sas[[3]] %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">count_cat</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">N_N_Y</td>
<td align="right">7418</td>
</tr>
<tr class="even">
<td align="left">N_Y_N</td>
<td align="right">13228</td>
</tr>
<tr class="odd">
<td align="left">N_Y_Y</td>
<td align="right">2987</td>
</tr>
<tr class="even">
<td align="left">Y_N_N</td>
<td align="right">8326</td>
</tr>
<tr class="odd">
<td align="left">Y_N_Y</td>
<td align="right">3784</td>
</tr>
<tr class="even">
<td align="left">Y_Y_N</td>
<td align="right">3201</td>
</tr>
<tr class="odd">
<td align="left">Y_Y_Y</td>
<td align="right">7098</td>
</tr>
</tbody>
</table>
<pre class="r"><code>count_to_ven(results_sas[[1]])</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>NULL</code></pre>
<pre class="r"><code>count_to_ven(results_sas[[2]])</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-17-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>NULL</code></pre>
<pre class="r"><code>count_to_ven(results_sas[[3]])</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-17-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>NULL</code></pre>
</div>
</div>
<div id="errors-in-bins" class="section level1">
<h1>Errors in Bins</h1>
<p>To get the sense of how much the methods overlap in a larger scale,
we’ll look at the aggregate number of errors in bins across the genome
for both switches and flips. First, let’s take a look at what this
overlap looks like within a single synthetic diploid:</p>
<pre class="r"><code>bin_size &lt;- 1e6 # try 1MB
df &lt;- annotate_het_list(1, het_loc_dir, beagle_switch_dir, eagle_switch_dir, shapeit_switch_dir, df_freq) %&gt;%
  mutate(bin_id = ceiling(pos / bin_size))</code></pre>
<p>Let’s first look at switches across bins</p>
<pre class="r"><code>df %&gt;%
  group_by(bin_id) %&gt;%
  summarize(beagle = sum(switch_b),
            eagle = sum(switch_e),
            shapeit = sum(switch_s)) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;method&quot;, values_to = &quot;switches&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = switches, color = method)) +
  geom_point() +
  geom_line() +
  theme_classic()</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-1">
Past versions of unnamed-chunk-19-1.png
</button>
</p>
<div id="fig-unnamed-chunk-19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-19-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Now let’s lower our resolution and look at 10MB bins:</p>
<pre class="r"><code>bin_size &lt;- 1e7
df %&gt;%
  mutate(bin_id = ceiling(pos / bin_size)) %&gt;%
  group_by(bin_id) %&gt;%
  summarize(beagle = sum(switch_b),
            eagle = sum(switch_e),
            shapeit = sum(switch_s)) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;method&quot;, values_to = &quot;switches&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = switches, color = method)) +
  geom_point() +
  geom_line() +
  theme_classic()</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-20-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="eur-1" class="section level2">
<h2>EUR</h2>
<p>Now let’s see what this distribution looks like across all EUR SDs.
First, on the MB scale:</p>
<div id="switches" class="section level3">
<h3>Switches</h3>
<pre class="r"><code>get_errors_mb &lt;- function(ids, bin_size, het_loc_dir, beagle_switch_dir, eagle_switch_dir, shapeit_switch_dir, df_freq){
  
  final_df &lt;- annotate_het_list(ids[1], het_loc_dir, beagle_switch_dir, eagle_switch_dir, shapeit_switch_dir, df_freq) %&gt;%
    mutate(bin_id = ceiling(pos / bin_size)) %&gt;%
    group_by(bin_id) %&gt;%
    summarize(beagle_s = sum(switch_b),
              eagle_s = sum(switch_e),
              shapeit_s = sum(switch_s),
              beagle_f = sum(flip_b),
              eagle_f = sum(flip_e),
              shapeit_f = sum(flip_s),
              hets = n())
  
  for(i in 2:length(ids)){
    df2 &lt;- annotate_het_list(ids[i], het_loc_dir, beagle_switch_dir, eagle_switch_dir, shapeit_switch_dir, df_freq) %&gt;%
      mutate(bin_id = ceiling(pos / bin_size)) %&gt;%
      group_by(bin_id) %&gt;%
      summarize(beagle_s2 = sum(switch_b),
                eagle_s2 = sum(switch_e),
                shapeit_s2 = sum(switch_s),
                beagle_f2 = sum(flip_b),
              eagle_f2 = sum(flip_e),
              shapeit_f2 = sum(flip_s),
              hets2 = n())
    final_df &lt;- full_join(final_df, df2, by=&quot;bin_id&quot;) %&gt;%
      replace_na(list(&quot;beagle_s&quot; = 0, &quot;eagle_s&quot; = 0, &quot;shapeit_s&quot; = 0,
                      &quot;beagle_s2&quot; = 0, &quot;eagle_s2&quot; = 0, &quot;shapeit_s2&quot; = 0,
                      &quot;beagle_f&quot; = 0, &quot;eagle_f&quot; = 0, &quot;shapeit_f&quot; = 0,
                      &quot;beagle_f2&quot; = 0, &quot;eagle_f2&quot; = 0, &quot;shapeit_f2&quot; = 0,
                      &quot;hets&quot; = 0, &quot;hets2&quot; = 0)) %&gt;%
      mutate(beagle_s = beagle_s + beagle_s2,
             eagle_s = eagle_s + eagle_s2,
             shapeit_s = shapeit_s + shapeit_s2,
             beagle_f = beagle_f + beagle_f2,
             eagle_f = eagle_f + eagle_f2,
             shapeit_f = shapeit_f + shapeit_f2,
             hets = hets + hets2) %&gt;%
      select(bin_id, beagle_s, eagle_s, shapeit_s, beagle_f, eagle_f, shapeit_f, hets)
  }
  return(final_df %&gt;% arrange(bin_id))
}

df_eur &lt;- get_errors_mb(1:200, 1e6, het_loc_dir, beagle_switch_dir, eagle_switch_dir, shapeit_switch_dir, df_freq)

df_eur %&gt;%
  ggplot(aes(x = bin_id, y = hets)) +
  geom_point() +
  geom_line() +
  ggtitle(&quot;Aggregated Heterozygous Sites per Bin&quot;, &quot;EUR Synthetic Diploids&quot;) +
  xlab(&quot;Bin&quot;) +
  ylab(&quot;Heterozygous Sites&quot;) +
  theme_classic()</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-21-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_eur %&gt;%
  select(bin_id, beagle_s, eagle_s, shapeit_s) %&gt;%
  rename(Beagle = beagle_s, Eagle = eagle_s, SHAPEIT = shapeit_s) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Switches&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Switches, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Switches per MB Bin&quot;, &quot;EUR Synthetic Diploids&quot;)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-21-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-2">
Past versions of unnamed-chunk-21-2.png
</button>
</p>
<div id="fig-unnamed-chunk-21-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-21-2.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_eur %&gt;%
  select(bin_id, beagle_s, eagle_s, shapeit_s) %&gt;%
  rename(Beagle = beagle_s, Eagle = eagle_s, SHAPEIT = shapeit_s) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Switches&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Switches, color = Method )) +
    geom_point() +
    geom_line() +
  geom_smooth() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Switches per MB Bin&quot;, &quot;EUR Synthetic Diploids&quot;)</code></pre>
<pre><code>`geom_smooth()` using method = &#39;loess&#39; and formula = &#39;y ~ x&#39;</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-21-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_eur %&gt;%
  select(beagle_s, eagle_s, shapeit_s) %&gt;%
  cor(method = &quot;pearson&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle_s</th>
<th align="right">eagle_s</th>
<th align="right">shapeit_s</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle_s</td>
<td align="right">1.0000000</td>
<td align="right">0.9780431</td>
<td align="right">0.9916736</td>
</tr>
<tr class="even">
<td align="left">eagle_s</td>
<td align="right">0.9780431</td>
<td align="right">1.0000000</td>
<td align="right">0.9735588</td>
</tr>
<tr class="odd">
<td align="left">shapeit_s</td>
<td align="right">0.9916736</td>
<td align="right">0.9735588</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df_eur %&gt;%
  select(beagle_s, eagle_s, shapeit_s) %&gt;%
  cor(method = &quot;spearman&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle_s</th>
<th align="right">eagle_s</th>
<th align="right">shapeit_s</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle_s</td>
<td align="right">1.0000000</td>
<td align="right">0.908188</td>
<td align="right">0.9837866</td>
</tr>
<tr class="even">
<td align="left">eagle_s</td>
<td align="right">0.9081880</td>
<td align="right">1.000000</td>
<td align="right">0.8806680</td>
</tr>
<tr class="odd">
<td align="left">shapeit_s</td>
<td align="right">0.9837866</td>
<td align="right">0.880668</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<div id="adjust-for-het-sites" class="section level4">
<h4>Adjust for het sites</h4>
<pre class="r"><code>df_eur %&gt;%
  mutate(beagle = beagle_s / hets,
         eagle = eagle_s / hets,
         shapeit = shapeit_s / hets) %&gt;%
  select(bin_id, beagle, eagle, shapeit) %&gt;%
  rename(Beagle = beagle, Eagle = eagle, SHAPEIT = shapeit) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Switches&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Switches, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Switches per Heterozygous Site per MB Bin&quot;, &quot;EUR Synthetic Diploids&quot;)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-22-1">
Past versions of unnamed-chunk-22-1.png
</button>
</p>
<div id="fig-unnamed-chunk-22-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-22-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_eur %&gt;%
  mutate(beagle = beagle_s / hets,
         eagle = eagle_s / hets,
         shapeit = shapeit_s / hets) %&gt;%
  select(bin_id, beagle, eagle, shapeit) %&gt;%
  rename(Beagle = beagle, Eagle = eagle, SHAPEIT = shapeit) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Switches&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Switches, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Switches per Heterozygous Site per MB Bin&quot;, &quot;EUR Synthetic Diploids&quot;) +
  geom_smooth()</code></pre>
<pre><code>`geom_smooth()` using method = &#39;loess&#39; and formula = &#39;y ~ x&#39;</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-22-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-22-2">
Past versions of unnamed-chunk-22-2.png
</button>
</p>
<div id="fig-unnamed-chunk-22-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-22-2.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_eur %&gt;%
  mutate(beagle = beagle_s / hets,
         eagle = eagle_s / hets,
         shapeit = shapeit_s / hets) %&gt;%
  select(beagle, eagle, shapeit) %&gt;%
  cor() %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle</th>
<th align="right">eagle</th>
<th align="right">shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle</td>
<td align="right">1.0000000</td>
<td align="right">0.8253937</td>
<td align="right">0.9837559</td>
</tr>
<tr class="even">
<td align="left">eagle</td>
<td align="right">0.8253937</td>
<td align="right">1.0000000</td>
<td align="right">0.7598874</td>
</tr>
<tr class="odd">
<td align="left">shapeit</td>
<td align="right">0.9837559</td>
<td align="right">0.7598874</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df_eur %&gt;%
  mutate(beagle = beagle_s / hets,
         eagle = eagle_s / hets,
         shapeit = shapeit_s / hets) %&gt;%
  select(beagle, eagle, shapeit) %&gt;%
  cor(method = &quot;spearman&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle</th>
<th align="right">eagle</th>
<th align="right">shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle</td>
<td align="right">1.0000000</td>
<td align="right">0.9051900</td>
<td align="right">0.9782468</td>
</tr>
<tr class="even">
<td align="left">eagle</td>
<td align="right">0.9051900</td>
<td align="right">1.0000000</td>
<td align="right">0.8509725</td>
</tr>
<tr class="odd">
<td align="left">shapeit</td>
<td align="right">0.9782468</td>
<td align="right">0.8509725</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="flips" class="section level3">
<h3>Flips</h3>
<pre class="r"><code>df_eur %&gt;%
  select(bin_id, beagle_f, eagle_f, shapeit_f) %&gt;%
  rename(Beagle = beagle_f, Eagle = eagle_f, SHAPEIT = shapeit_f) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Flips&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Flips, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Flips per MB Bin&quot;, &quot;EUR Synthetic Diploids&quot;)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-23-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-1">
Past versions of unnamed-chunk-23-1.png
</button>
</p>
<div id="fig-unnamed-chunk-23-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-23-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_eur %&gt;%
  select(bin_id, beagle_f, eagle_f, shapeit_f) %&gt;%
  rename(Beagle = beagle_f, Eagle = eagle_f, SHAPEIT = shapeit_f) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Flips&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Flips, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
  geom_smooth() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Flips per MB Bin&quot;, &quot;EUR Synthetic Diploids&quot;)</code></pre>
<pre><code>`geom_smooth()` using method = &#39;loess&#39; and formula = &#39;y ~ x&#39;</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-23-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-2">
Past versions of unnamed-chunk-23-2.png
</button>
</p>
<div id="fig-unnamed-chunk-23-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-23-2.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_eur %&gt;%
  select(bin_id, beagle_f, eagle_f, shapeit_f) %&gt;%
  rename(Beagle = beagle_f, Eagle = eagle_f, SHAPEIT = shapeit_f) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Flips&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Flips, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
  geom_smooth() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Flips per MB Bin&quot;, &quot;EUR Synthetic Diploids&quot;) +
  geom_smooth()</code></pre>
<pre><code>`geom_smooth()` using method = &#39;loess&#39; and formula = &#39;y ~ x&#39;
`geom_smooth()` using method = &#39;loess&#39; and formula = &#39;y ~ x&#39;</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-23-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_eur %&gt;%
  select(beagle_f, eagle_f, shapeit_f) %&gt;%
  cor(method = &quot;pearson&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle_f</th>
<th align="right">eagle_f</th>
<th align="right">shapeit_f</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle_f</td>
<td align="right">1.0000000</td>
<td align="right">0.8778816</td>
<td align="right">0.9545333</td>
</tr>
<tr class="even">
<td align="left">eagle_f</td>
<td align="right">0.8778816</td>
<td align="right">1.0000000</td>
<td align="right">0.8953995</td>
</tr>
<tr class="odd">
<td align="left">shapeit_f</td>
<td align="right">0.9545333</td>
<td align="right">0.8953995</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df_eur %&gt;%
  select(beagle_f, eagle_f, shapeit_f) %&gt;%
  cor(method = &quot;spearman&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle_f</th>
<th align="right">eagle_f</th>
<th align="right">shapeit_f</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle_f</td>
<td align="right">1.0000000</td>
<td align="right">0.8502419</td>
<td align="right">0.9331728</td>
</tr>
<tr class="even">
<td align="left">eagle_f</td>
<td align="right">0.8502419</td>
<td align="right">1.0000000</td>
<td align="right">0.8667677</td>
</tr>
<tr class="odd">
<td align="left">shapeit_f</td>
<td align="right">0.9331728</td>
<td align="right">0.8667677</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<div id="adjust-for-het-sites-1" class="section level4">
<h4>Adjust for het sites</h4>
<pre class="r"><code>df_eur %&gt;%
  mutate(beagle = beagle_f / hets,
         eagle = eagle_f / hets,
         shapeit = shapeit_f / hets) %&gt;%
  select(bin_id, beagle, eagle, shapeit) %&gt;%
  rename(Beagle = beagle, Eagle = eagle, SHAPEIT = shapeit) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Flips&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Flips, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Flips per Heterozygous Site per MB Bin&quot;, &quot;EUR Synthetic Diploids&quot;)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-1">
Past versions of unnamed-chunk-24-1.png
</button>
</p>
<div id="fig-unnamed-chunk-24-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-24-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_eur %&gt;%
  mutate(beagle = beagle_f / hets,
         eagle = eagle_f / hets,
         shapeit = shapeit_f / hets) %&gt;%
  select(bin_id, beagle, eagle, shapeit) %&gt;%
  rename(Beagle = beagle, Eagle = eagle, SHAPEIT = shapeit) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Flips&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Flips, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Flips per Heterozygous Site per MB Bin&quot;, &quot;EUR Synthetic Diploids&quot;) +
  geom_smooth()</code></pre>
<pre><code>`geom_smooth()` using method = &#39;loess&#39; and formula = &#39;y ~ x&#39;</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-24-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-2">
Past versions of unnamed-chunk-24-2.png
</button>
</p>
<div id="fig-unnamed-chunk-24-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-24-2.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_eur %&gt;%
  mutate(beagle = beagle_f / hets,
         eagle = eagle_f / hets,
         shapeit = shapeit_f / hets) %&gt;%
  select(beagle, eagle, shapeit) %&gt;%
  cor() %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle</th>
<th align="right">eagle</th>
<th align="right">shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle</td>
<td align="right">1.0000000</td>
<td align="right">0.9613445</td>
<td align="right">0.983199</td>
</tr>
<tr class="even">
<td align="left">eagle</td>
<td align="right">0.9613445</td>
<td align="right">1.0000000</td>
<td align="right">0.953970</td>
</tr>
<tr class="odd">
<td align="left">shapeit</td>
<td align="right">0.9831990</td>
<td align="right">0.9539700</td>
<td align="right">1.000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df_eur %&gt;%
  mutate(beagle = beagle_f / hets,
         eagle = eagle_f / hets,
         shapeit = shapeit_f / hets) %&gt;%
  select(beagle, eagle, shapeit) %&gt;%
  cor(method = &quot;spearman&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle</th>
<th align="right">eagle</th>
<th align="right">shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle</td>
<td align="right">1.0000000</td>
<td align="right">0.9400662</td>
<td align="right">0.9710352</td>
</tr>
<tr class="even">
<td align="left">eagle</td>
<td align="right">0.9400662</td>
<td align="right">1.0000000</td>
<td align="right">0.9423458</td>
</tr>
<tr class="odd">
<td align="left">shapeit</td>
<td align="right">0.9710352</td>
<td align="right">0.9423458</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="higher-resolution" class="section level3">
<h3>Higher Resolution</h3>
<pre class="r"><code>df_eur &lt;- get_errors_mb(1:200, 1e5, het_loc_dir, beagle_switch_dir, eagle_switch_dir, shapeit_switch_dir, df_freq)

df_eur %&gt;%
  ggplot(aes(x = bin_id, y = hets)) +
  geom_point() +
  geom_line() +
  ggtitle(&quot;Aggregated Heterozygous Sites per Bin&quot;, &quot;EUR Synthetic Diploids&quot;) +
  xlab(&quot;Bin&quot;) +
  ylab(&quot;Heterozygous Sites&quot;) +
  theme_classic()</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-25-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-1">
Past versions of unnamed-chunk-25-1.png
</button>
</p>
<div id="fig-unnamed-chunk-25-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-25-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_eur %&gt;%
  select(bin_id, beagle_s, eagle_s, shapeit_s) %&gt;%
  rename(Beagle = beagle_s, Eagle = eagle_s, SHAPEIT = shapeit_s) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Switches&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Switches, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Switches per 100KB Bin&quot;, &quot;EUR Synthetic Diploids&quot;)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-25-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-2">
Past versions of unnamed-chunk-25-2.png
</button>
</p>
<div id="fig-unnamed-chunk-25-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-25-2.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_eur %&gt;%
  mutate(eagle_beagle = eagle_s - beagle_s,
         eagle_shapeit = eagle_s - shapeit_s) %&gt;%
  select(bin_id, eagle_beagle, eagle_shapeit) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Comparison&quot;, values_to = &quot;Difference&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Difference, color = Comparison, group = Comparison)) +
    geom_point() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  geom_hline(yintercept = 0) +
  ggtitle(&quot;Difference in Switches per 100KB Bin&quot;, &quot;EUR Synthetic Diploids&quot;) +
  geom_smooth(linewidth = 2.5, color = &quot;black&quot;, se = FALSE)+
  geom_smooth()</code></pre>
<pre><code>`geom_smooth()` using method = &#39;gam&#39; and formula = &#39;y ~ s(x, bs = &quot;cs&quot;)&#39;
`geom_smooth()` using method = &#39;gam&#39; and formula = &#39;y ~ s(x, bs = &quot;cs&quot;)&#39;</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-25-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-3">
Past versions of unnamed-chunk-25-3.png
</button>
</p>
<div id="fig-unnamed-chunk-25-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-25-3.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_eur %&gt;%
  select(bin_id, beagle_f, eagle_f, shapeit_f) %&gt;%
  rename(Beagle = beagle_f, Eagle = eagle_f, SHAPEIT = shapeit_f) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Flips&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Flips, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Flips per 100KB Bin&quot;, &quot;EUR Synthetic Diploids&quot;)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-25-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-4">
Past versions of unnamed-chunk-25-4.png
</button>
</p>
<div id="fig-unnamed-chunk-25-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-25-4.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_eur %&gt;%
  mutate(beagle = beagle_s / hets,
         eagle = eagle_s / hets,
         shapeit = shapeit_s / hets) %&gt;%
  select(bin_id, beagle, eagle, shapeit) %&gt;%
  rename(Beagle = beagle, Eagle = eagle, SHAPEIT = shapeit) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Switches&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Switches, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Switches per Heterozygous Site per 100KB Bin&quot;, &quot;EUR Synthetic Diploids&quot;)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-25-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-5">
Past versions of unnamed-chunk-25-5.png
</button>
</p>
<div id="fig-unnamed-chunk-25-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-25-5.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_eur %&gt;%
  mutate(beagle = beagle_s / hets,
         eagle = eagle_s / hets,
         shapeit = shapeit_s / hets) %&gt;%
  select(bin_id, beagle, eagle, shapeit) %&gt;%
  rename(Beagle = beagle, Eagle = eagle, SHAPEIT = shapeit) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Switches&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Switches, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Switches per Heterozygous Site per 100KB Bin&quot;, &quot;EUR Synthetic Diploids&quot;) +
  geom_smooth()</code></pre>
<pre><code>`geom_smooth()` using method = &#39;gam&#39; and formula = &#39;y ~ s(x, bs = &quot;cs&quot;)&#39;</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-25-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-6">
Past versions of unnamed-chunk-25-6.png
</button>
</p>
<div id="fig-unnamed-chunk-25-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-25-6.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_eur %&gt;%
  mutate(beagle = beagle_s / hets,
         eagle = eagle_s / hets,
         shapeit = shapeit_s / hets) %&gt;%
  select(beagle, eagle, shapeit) %&gt;%
  cor() %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle</th>
<th align="right">eagle</th>
<th align="right">shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle</td>
<td align="right">1.0000000</td>
<td align="right">0.8308736</td>
<td align="right">0.9457198</td>
</tr>
<tr class="even">
<td align="left">eagle</td>
<td align="right">0.8308736</td>
<td align="right">1.0000000</td>
<td align="right">0.8271611</td>
</tr>
<tr class="odd">
<td align="left">shapeit</td>
<td align="right">0.9457198</td>
<td align="right">0.8271611</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df_eur %&gt;%
  mutate(beagle = beagle_s / hets,
         eagle = eagle_s / hets,
         shapeit = shapeit_s / hets) %&gt;%
  select(beagle, eagle, shapeit) %&gt;%
  cor(method = &quot;spearman&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle</th>
<th align="right">eagle</th>
<th align="right">shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle</td>
<td align="right">1.0000000</td>
<td align="right">0.7764913</td>
<td align="right">0.8721816</td>
</tr>
<tr class="even">
<td align="left">eagle</td>
<td align="right">0.7764913</td>
<td align="right">1.0000000</td>
<td align="right">0.7154716</td>
</tr>
<tr class="odd">
<td align="left">shapeit</td>
<td align="right">0.8721816</td>
<td align="right">0.7154716</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df_eur %&gt;%
  mutate(beagle = beagle_f / hets,
         eagle = eagle_f / hets,
         shapeit = shapeit_f / hets) %&gt;%
  select(bin_id, beagle, eagle, shapeit) %&gt;%
  rename(Beagle = beagle, Eagle = eagle, SHAPEIT = shapeit) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Flips&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Flips, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Flips per Heterozygous Site per 100KB Bin&quot;, &quot;EUR Synthetic Diploids&quot;)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-25-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-7">
Past versions of unnamed-chunk-25-7.png
</button>
</p>
<div id="fig-unnamed-chunk-25-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-25-7.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_eur %&gt;%
  mutate(beagle = beagle_f / hets,
         eagle = eagle_f / hets,
         shapeit = shapeit_f / hets) %&gt;%
  select(bin_id, beagle, eagle, shapeit) %&gt;%
  rename(Beagle = beagle, Eagle = eagle, SHAPEIT = shapeit) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Flips&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Flips, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Flips per Heterozygous Site per 100KB Bin&quot;, &quot;EUR Synthetic Diploids&quot;) +
  geom_smooth()</code></pre>
<pre><code>`geom_smooth()` using method = &#39;gam&#39; and formula = &#39;y ~ s(x, bs = &quot;cs&quot;)&#39;</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-25-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-8">
Past versions of unnamed-chunk-25-8.png
</button>
</p>
<div id="fig-unnamed-chunk-25-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-25-8.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_eur %&gt;%
  mutate(beagle = beagle_f / hets,
         eagle = eagle_f / hets,
         shapeit = shapeit_f / hets) %&gt;%
  select(beagle, eagle, shapeit) %&gt;%
  cor() %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle</th>
<th align="right">eagle</th>
<th align="right">shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle</td>
<td align="right">1.0000000</td>
<td align="right">0.9285519</td>
<td align="right">0.9576559</td>
</tr>
<tr class="even">
<td align="left">eagle</td>
<td align="right">0.9285519</td>
<td align="right">1.0000000</td>
<td align="right">0.9458765</td>
</tr>
<tr class="odd">
<td align="left">shapeit</td>
<td align="right">0.9576559</td>
<td align="right">0.9458765</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df_eur %&gt;%
  mutate(beagle = beagle_f / hets,
         eagle = eagle_f / hets,
         shapeit = shapeit_f / hets) %&gt;%
  select(beagle, eagle, shapeit) %&gt;%
  cor(method = &quot;spearman&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle</th>
<th align="right">eagle</th>
<th align="right">shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle</td>
<td align="right">1.0000000</td>
<td align="right">0.8207662</td>
<td align="right">0.8909827</td>
</tr>
<tr class="even">
<td align="left">eagle</td>
<td align="right">0.8207662</td>
<td align="right">1.0000000</td>
<td align="right">0.8146124</td>
</tr>
<tr class="odd">
<td align="left">shapeit</td>
<td align="right">0.8909827</td>
<td align="right">0.8146124</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
</div>
<div id="lower-resolution" class="section level3">
<h3>Lower Resolution</h3>
<pre class="r"><code>df_eur &lt;- get_errors_mb(1:200, 1e7, het_loc_dir, beagle_switch_dir, eagle_switch_dir, shapeit_switch_dir, df_freq)

df_eur %&gt;%
  ggplot(aes(x = bin_id, y = hets)) +
  geom_point() +
  geom_line() +
  ggtitle(&quot;Aggregated Heterozygous Sites per Bin&quot;, &quot;EUR Synthetic Diploids&quot;) +
  xlab(&quot;Bin&quot;) +
  ylab(&quot;Heterozygous Sites&quot;) +
  theme_classic()</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-26-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-26-1">
Past versions of unnamed-chunk-26-1.png
</button>
</p>
<div id="fig-unnamed-chunk-26-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-26-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_eur %&gt;%
  select(bin_id, beagle_s, eagle_s, shapeit_s) %&gt;%
  rename(Beagle = beagle_s, Eagle = eagle_s, SHAPEIT = shapeit_s) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Switches&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Switches, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Switches per 10MB Bin&quot;, &quot;EUR Synthetic Diploids&quot;)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-26-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-26-2">
Past versions of unnamed-chunk-26-2.png
</button>
</p>
<div id="fig-unnamed-chunk-26-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-26-2.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_eur %&gt;%
  mutate(eagle_beagle = eagle_s - beagle_s,
         eagle_shapeit = eagle_s - shapeit_s) %&gt;%
  select(bin_id, eagle_beagle, eagle_shapeit) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Comparison&quot;, values_to = &quot;Difference&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Difference, color = Comparison, group = Comparison)) +
    geom_point() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  geom_hline(yintercept = 0) +
  ggtitle(&quot;Difference in Switches per 10MB Bin&quot;, &quot;EUR Synthetic Diploids&quot;) +
  geom_smooth(linewidth = 2.5, color = &quot;black&quot;, se = FALSE)+
  geom_smooth()</code></pre>
<pre><code>`geom_smooth()` using method = &#39;loess&#39; and formula = &#39;y ~ x&#39;
`geom_smooth()` using method = &#39;loess&#39; and formula = &#39;y ~ x&#39;</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-26-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-26-3">
Past versions of unnamed-chunk-26-3.png
</button>
</p>
<div id="fig-unnamed-chunk-26-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-26-3.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_eur %&gt;%
  mutate(beagle = beagle_s / hets,
         eagle = eagle_s / hets,
         shapeit = shapeit_s / hets) %&gt;%
  select(bin_id, beagle, eagle, shapeit) %&gt;%
  rename(Beagle = beagle, Eagle = eagle, SHAPEIT = shapeit) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Switches&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Switches, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Switches per Heterozygous Site per 10MB Bin&quot;, &quot;EUR Synthetic Diploids&quot;)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-26-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_eur %&gt;%
  mutate(beagle = beagle_s / hets,
         eagle = eagle_s / hets,
         shapeit = shapeit_s / hets) %&gt;%
  select(bin_id, beagle, eagle, shapeit) %&gt;%
  rename(Beagle = beagle, Eagle = eagle, SHAPEIT = shapeit) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Switches&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Switches, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Switches per Heterozygous Site per 10MB Bin&quot;, &quot;EUR Synthetic Diploids&quot;) +
  geom_smooth()</code></pre>
<pre><code>`geom_smooth()` using method = &#39;loess&#39; and formula = &#39;y ~ x&#39;</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-26-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_eur %&gt;%
  select(bin_id, beagle_f, eagle_f, shapeit_f) %&gt;%
  rename(Beagle = beagle_f, Eagle = eagle_f, SHAPEIT = shapeit_f) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Flips&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Flips, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Flips per 10MB Bin&quot;, &quot;EUR Synthetic Diploids&quot;)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-26-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_eur %&gt;%
  mutate(beagle = beagle_f / hets,
         eagle = eagle_f / hets,
         shapeit = shapeit_f / hets) %&gt;%
  select(bin_id, beagle, eagle, shapeit) %&gt;%
  rename(Beagle = beagle, Eagle = eagle, SHAPEIT = shapeit) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Flips&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Flips, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Flips per Heterozygous Site per 10MB Bin&quot;, &quot;EUR Synthetic Diploids&quot;)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-26-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_eur %&gt;%
  mutate(beagle = beagle_f / hets,
         eagle = eagle_f / hets,
         shapeit = shapeit_f / hets) %&gt;%
  select(bin_id, beagle, eagle, shapeit) %&gt;%
  rename(Beagle = beagle, Eagle = eagle, SHAPEIT = shapeit) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Flips&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Flips, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Flips per Heterozygous Site per 10MB Bin&quot;, &quot;EUR Synthetic Diploids&quot;) +
  geom_smooth()</code></pre>
<pre><code>`geom_smooth()` using method = &#39;loess&#39; and formula = &#39;y ~ x&#39;</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-26-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_eur %&gt;%
  mutate(beagle = beagle_f / hets,
         eagle = eagle_f / hets,
         shapeit = shapeit_f / hets) %&gt;%
  select(beagle, eagle, shapeit) %&gt;%
  cor() %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle</th>
<th align="right">eagle</th>
<th align="right">shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle</td>
<td align="right">1.0000000</td>
<td align="right">0.9366537</td>
<td align="right">0.9946038</td>
</tr>
<tr class="even">
<td align="left">eagle</td>
<td align="right">0.9366537</td>
<td align="right">1.0000000</td>
<td align="right">0.9571834</td>
</tr>
<tr class="odd">
<td align="left">shapeit</td>
<td align="right">0.9946038</td>
<td align="right">0.9571834</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df_eur %&gt;%
  mutate(beagle = beagle_f / hets,
         eagle = eagle_f / hets,
         shapeit = shapeit_f / hets) %&gt;%
  select(beagle, eagle, shapeit) %&gt;%
  cor(method = &quot;spearman&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle</th>
<th align="right">eagle</th>
<th align="right">shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle</td>
<td align="right">1.0000000</td>
<td align="right">0.9794118</td>
<td align="right">0.9823529</td>
</tr>
<tr class="even">
<td align="left">eagle</td>
<td align="right">0.9794118</td>
<td align="right">1.0000000</td>
<td align="right">0.9882353</td>
</tr>
<tr class="odd">
<td align="left">shapeit</td>
<td align="right">0.9823529</td>
<td align="right">0.9882353</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="afr-1" class="section level2">
<h2>AFR</h2>
<div id="switches-1" class="section level3">
<h3>Switches</h3>
<pre class="r"><code>df_afr &lt;- get_errors_mb(201:400, 1e6, het_loc_dir, beagle_switch_dir, eagle_switch_dir, shapeit_switch_dir, df_freq) %&gt;%
  arrange(bin_id)

df_afr %&gt;%
  ggplot(aes(x = bin_id, y = hets)) +
  geom_point() +
  geom_line() +
  ggtitle(&quot;Aggregated Heterozygous Sites per Bin&quot;, &quot;AFR Synthetic Diploids&quot;) +
  xlab(&quot;Bin&quot;) +
  ylab(&quot;Heterozygous Sites&quot;) +
  theme_classic()</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-27-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-27-1">
Past versions of unnamed-chunk-27-1.png
</button>
</p>
<div id="fig-unnamed-chunk-27-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-27-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_afr %&gt;%
  select(bin_id, beagle_s, eagle_s, shapeit_s) %&gt;%
  rename(Beagle = beagle_s, Eagle = eagle_s, SHAPEIT = shapeit_s) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Switches&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Switches, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Switches per MB Bin&quot;, &quot;AFR Synthetic Diploids&quot;)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-27-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_afr %&gt;%
  select(bin_id, beagle_s, eagle_s, shapeit_s) %&gt;%
  rename(Beagle = beagle_s, Eagle = eagle_s, SHAPEIT = shapeit_s) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Switches&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Switches, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  geom_smooth() +
  ggtitle(&quot;Switches per MB Bin&quot;, &quot;AFR Synthetic Diploids&quot;)</code></pre>
<pre><code>`geom_smooth()` using method = &#39;loess&#39; and formula = &#39;y ~ x&#39;</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-27-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_afr %&gt;%
  select(beagle_s, eagle_s, shapeit_s) %&gt;%
  cor(method = &quot;pearson&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle_s</th>
<th align="right">eagle_s</th>
<th align="right">shapeit_s</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle_s</td>
<td align="right">1.0000000</td>
<td align="right">0.9603503</td>
<td align="right">0.9914144</td>
</tr>
<tr class="even">
<td align="left">eagle_s</td>
<td align="right">0.9603503</td>
<td align="right">1.0000000</td>
<td align="right">0.9543517</td>
</tr>
<tr class="odd">
<td align="left">shapeit_s</td>
<td align="right">0.9914144</td>
<td align="right">0.9543517</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df_afr %&gt;%
  select(beagle_s, eagle_s, shapeit_s) %&gt;%
  cor(method = &quot;spearman&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle_s</th>
<th align="right">eagle_s</th>
<th align="right">shapeit_s</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle_s</td>
<td align="right">1.0000000</td>
<td align="right">0.8675281</td>
<td align="right">0.9761997</td>
</tr>
<tr class="even">
<td align="left">eagle_s</td>
<td align="right">0.8675281</td>
<td align="right">1.0000000</td>
<td align="right">0.8376987</td>
</tr>
<tr class="odd">
<td align="left">shapeit_s</td>
<td align="right">0.9761997</td>
<td align="right">0.8376987</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<div id="adjust-for-het-sites-2" class="section level4">
<h4>Adjust for het sites</h4>
<pre class="r"><code>df_afr %&gt;%
  mutate(beagle = beagle_s / hets,
         eagle = eagle_s / hets,
         shapeit = shapeit_s / hets) %&gt;%
  select(bin_id, beagle, eagle, shapeit) %&gt;%
  rename(Beagle = beagle, Eagle = eagle, SHAPEIT = shapeit) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Switches&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Switches, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Switches per Heterozygous Site per MB Bin&quot;, &quot;AFR Synthetic Diploids&quot;)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-28-1">
Past versions of unnamed-chunk-28-1.png
</button>
</p>
<div id="fig-unnamed-chunk-28-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-28-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_afr %&gt;%
  mutate(beagle = beagle_s / hets,
         eagle = eagle_s / hets,
         shapeit = shapeit_s / hets) %&gt;%
  select(beagle, eagle, shapeit) %&gt;%
  cor() %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle</th>
<th align="right">eagle</th>
<th align="right">shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle</td>
<td align="right">1.0000000</td>
<td align="right">0.8859861</td>
<td align="right">0.9836189</td>
</tr>
<tr class="even">
<td align="left">eagle</td>
<td align="right">0.8859861</td>
<td align="right">1.0000000</td>
<td align="right">0.8448951</td>
</tr>
<tr class="odd">
<td align="left">shapeit</td>
<td align="right">0.9836189</td>
<td align="right">0.8448951</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df_afr %&gt;%
  mutate(beagle = beagle_s / hets,
         eagle = eagle_s / hets,
         shapeit = shapeit_s / hets) %&gt;%
  select(beagle, eagle, shapeit) %&gt;%
  cor(method = &quot;spearman&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle</th>
<th align="right">eagle</th>
<th align="right">shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle</td>
<td align="right">1.0000000</td>
<td align="right">0.8586023</td>
<td align="right">0.9703590</td>
</tr>
<tr class="even">
<td align="left">eagle</td>
<td align="right">0.8586023</td>
<td align="right">1.0000000</td>
<td align="right">0.7993133</td>
</tr>
<tr class="odd">
<td align="left">shapeit</td>
<td align="right">0.9703590</td>
<td align="right">0.7993133</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="flips-1" class="section level3">
<h3>Flips</h3>
<pre class="r"><code>df_afr %&gt;%
  select(bin_id, beagle_f, eagle_f, shapeit_f) %&gt;%
  rename(Beagle = beagle_f, Eagle = eagle_f, SHAPEIT = shapeit_f) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Flips&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Flips, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Flips per MB Bin&quot;, &quot;AFR Synthetic Diploids&quot;)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-29-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-29-1">
Past versions of unnamed-chunk-29-1.png
</button>
</p>
<div id="fig-unnamed-chunk-29-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-29-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_afr %&gt;%
  select(bin_id, beagle_f, eagle_f, shapeit_f) %&gt;%
  rename(Beagle = beagle_f, Eagle = eagle_f, SHAPEIT = shapeit_f) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Flips&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Flips, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Flips per MB Bin&quot;, &quot;AFR Synthetic Diploids&quot;) +
  geom_smooth()</code></pre>
<pre><code>`geom_smooth()` using method = &#39;loess&#39; and formula = &#39;y ~ x&#39;</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-29-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_afr %&gt;%
  select(beagle_f, eagle_f, shapeit_f) %&gt;%
  cor(method = &quot;pearson&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle_f</th>
<th align="right">eagle_f</th>
<th align="right">shapeit_f</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle_f</td>
<td align="right">1.0000000</td>
<td align="right">0.9391784</td>
<td align="right">0.9679420</td>
</tr>
<tr class="even">
<td align="left">eagle_f</td>
<td align="right">0.9391784</td>
<td align="right">1.0000000</td>
<td align="right">0.9575982</td>
</tr>
<tr class="odd">
<td align="left">shapeit_f</td>
<td align="right">0.9679420</td>
<td align="right">0.9575982</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df_afr %&gt;%
  select(beagle_f, eagle_f, shapeit_f) %&gt;%
  cor(method = &quot;spearman&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle_f</th>
<th align="right">eagle_f</th>
<th align="right">shapeit_f</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle_f</td>
<td align="right">1.0000000</td>
<td align="right">0.8956666</td>
<td align="right">0.9418626</td>
</tr>
<tr class="even">
<td align="left">eagle_f</td>
<td align="right">0.8956666</td>
<td align="right">1.0000000</td>
<td align="right">0.9203105</td>
</tr>
<tr class="odd">
<td align="left">shapeit_f</td>
<td align="right">0.9418626</td>
<td align="right">0.9203105</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<div id="adjust-for-het-sites-3" class="section level4">
<h4>Adjust for het sites</h4>
<pre class="r"><code>df_afr %&gt;%
  mutate(beagle = beagle_f / hets,
         eagle = eagle_f / hets,
         shapeit = shapeit_f / hets) %&gt;%
  select(bin_id, beagle, eagle, shapeit) %&gt;%
  rename(Beagle = beagle, Eagle = eagle, SHAPEIT = shapeit) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Flips&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Flips, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Flips per Heterozygous Site per MB Bin&quot;, &quot;AFR Synthetic Diploids&quot;)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-30-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-30-1">
Past versions of unnamed-chunk-30-1.png
</button>
</p>
<div id="fig-unnamed-chunk-30-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-30-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_afr %&gt;%
  mutate(beagle = beagle_f / hets,
         eagle = eagle_f / hets,
         shapeit = shapeit_f / hets) %&gt;%
  select(beagle, eagle, shapeit) %&gt;%
  cor() %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle</th>
<th align="right">eagle</th>
<th align="right">shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle</td>
<td align="right">1.0000000</td>
<td align="right">0.9231411</td>
<td align="right">0.9715309</td>
</tr>
<tr class="even">
<td align="left">eagle</td>
<td align="right">0.9231411</td>
<td align="right">1.0000000</td>
<td align="right">0.9521697</td>
</tr>
<tr class="odd">
<td align="left">shapeit</td>
<td align="right">0.9715309</td>
<td align="right">0.9521697</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df_afr %&gt;%
  mutate(beagle = beagle_f / hets,
         eagle = eagle_f / hets,
         shapeit = shapeit_f / hets) %&gt;%
  select(beagle, eagle, shapeit) %&gt;%
  cor(method = &quot;spearman&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle</th>
<th align="right">eagle</th>
<th align="right">shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle</td>
<td align="right">1.0000000</td>
<td align="right">0.9292924</td>
<td align="right">0.9599826</td>
</tr>
<tr class="even">
<td align="left">eagle</td>
<td align="right">0.9292924</td>
<td align="right">1.0000000</td>
<td align="right">0.9419833</td>
</tr>
<tr class="odd">
<td align="left">shapeit</td>
<td align="right">0.9599826</td>
<td align="right">0.9419833</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="amr-1" class="section level2">
<h2>AMR</h2>
<div id="switches-2" class="section level3">
<h3>Switches</h3>
<pre class="r"><code>df_amr &lt;- get_errors_mb(401:600, 1e6, het_loc_dir, beagle_switch_dir, eagle_switch_dir, shapeit_switch_dir, df_freq) %&gt;%
  arrange(bin_id)

df_amr %&gt;%
  ggplot(aes(x = bin_id, y = hets)) +
  geom_point() +
  geom_line() +
  ggtitle(&quot;Aggregated Heterozygous Sites per Bin&quot;, &quot;AMR Synthetic Diploids&quot;) +
  xlab(&quot;Bin&quot;) +
  ylab(&quot;Heterozygous Sites&quot;) +
  theme_classic()</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-31-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-31-1">
Past versions of unnamed-chunk-31-1.png
</button>
</p>
<div id="fig-unnamed-chunk-31-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-31-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_amr %&gt;%
  select(bin_id, beagle_s, eagle_s, shapeit_s) %&gt;%
  rename(Beagle = beagle_s, Eagle = eagle_s, SHAPEIT = shapeit_s) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Switches&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Switches, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Switches per MB Bin&quot;, &quot;AMR Synthetic Diploids&quot;)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-31-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_amr %&gt;%
  select(bin_id, beagle_s, eagle_s, shapeit_s) %&gt;%
  rename(Beagle = beagle_s, Eagle = eagle_s, SHAPEIT = shapeit_s) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Switches&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Switches, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Switches per MB Bin&quot;, &quot;AMR Synthetic Diploids&quot;) +
  geom_smooth()</code></pre>
<pre><code>`geom_smooth()` using method = &#39;loess&#39; and formula = &#39;y ~ x&#39;</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-31-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_amr %&gt;%
  select(beagle_s, eagle_s, shapeit_s) %&gt;%
  cor(method = &quot;pearson&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle_s</th>
<th align="right">eagle_s</th>
<th align="right">shapeit_s</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle_s</td>
<td align="right">1.0000000</td>
<td align="right">0.9686700</td>
<td align="right">0.9901478</td>
</tr>
<tr class="even">
<td align="left">eagle_s</td>
<td align="right">0.9686700</td>
<td align="right">1.0000000</td>
<td align="right">0.9621078</td>
</tr>
<tr class="odd">
<td align="left">shapeit_s</td>
<td align="right">0.9901478</td>
<td align="right">0.9621078</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df_amr %&gt;%
  select(beagle_s, eagle_s, shapeit_s) %&gt;%
  cor(method = &quot;spearman&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle_s</th>
<th align="right">eagle_s</th>
<th align="right">shapeit_s</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle_s</td>
<td align="right">1.0000000</td>
<td align="right">0.8645015</td>
<td align="right">0.9731899</td>
</tr>
<tr class="even">
<td align="left">eagle_s</td>
<td align="right">0.8645015</td>
<td align="right">1.0000000</td>
<td align="right">0.8487058</td>
</tr>
<tr class="odd">
<td align="left">shapeit_s</td>
<td align="right">0.9731899</td>
<td align="right">0.8487058</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<div id="adjust-for-het-sites-4" class="section level4">
<h4>Adjust for het sites</h4>
<pre class="r"><code>df_amr %&gt;%
  mutate(beagle = beagle_s / hets,
         eagle = eagle_s / hets,
         shapeit = shapeit_s / hets) %&gt;%
  select(bin_id, beagle, eagle, shapeit) %&gt;%
  rename(Beagle = beagle, Eagle = eagle, SHAPEIT = shapeit) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Switches&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Switches, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Switches per Heterozygous Site per MB Bin&quot;, &quot;AMR Synthetic Diploids&quot;)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-32-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-32-1">
Past versions of unnamed-chunk-32-1.png
</button>
</p>
<div id="fig-unnamed-chunk-32-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-32-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_amr %&gt;%
  mutate(beagle = beagle_s / hets,
         eagle = eagle_s / hets,
         shapeit = shapeit_s / hets) %&gt;%
  select(beagle, eagle, shapeit) %&gt;%
  cor() %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle</th>
<th align="right">eagle</th>
<th align="right">shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle</td>
<td align="right">1.0000000</td>
<td align="right">0.7978346</td>
<td align="right">0.9796487</td>
</tr>
<tr class="even">
<td align="left">eagle</td>
<td align="right">0.7978346</td>
<td align="right">1.0000000</td>
<td align="right">0.7519013</td>
</tr>
<tr class="odd">
<td align="left">shapeit</td>
<td align="right">0.9796487</td>
<td align="right">0.7519013</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df_amr %&gt;%
  mutate(beagle = beagle_s / hets,
         eagle = eagle_s / hets,
         shapeit = shapeit_s / hets) %&gt;%
  select(beagle, eagle, shapeit) %&gt;%
  cor(method = &quot;spearman&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle</th>
<th align="right">eagle</th>
<th align="right">shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle</td>
<td align="right">1.0000000</td>
<td align="right">0.8677239</td>
<td align="right">0.9658313</td>
</tr>
<tr class="even">
<td align="left">eagle</td>
<td align="right">0.8677239</td>
<td align="right">1.0000000</td>
<td align="right">0.8234646</td>
</tr>
<tr class="odd">
<td align="left">shapeit</td>
<td align="right">0.9658313</td>
<td align="right">0.8234646</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="flips-2" class="section level3">
<h3>Flips</h3>
<pre class="r"><code>df_amr %&gt;%
  select(bin_id, beagle_f, eagle_f, shapeit_f) %&gt;%
  rename(Beagle = beagle_f, Eagle = eagle_f, SHAPEIT = shapeit_f) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Flips&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Flips, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Flips per MB Bin&quot;, &quot;AMR Synthetic Diploids&quot;)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-33-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-33-1">
Past versions of unnamed-chunk-33-1.png
</button>
</p>
<div id="fig-unnamed-chunk-33-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-33-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_amr %&gt;%
  select(bin_id, beagle_f, eagle_f, shapeit_f) %&gt;%
  rename(Beagle = beagle_f, Eagle = eagle_f, SHAPEIT = shapeit_f) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Flips&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Flips, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Flips per MB Bin&quot;, &quot;AMR Synthetic Diploids&quot;) +
  geom_smooth()</code></pre>
<pre><code>`geom_smooth()` using method = &#39;loess&#39; and formula = &#39;y ~ x&#39;</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-33-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_amr %&gt;%
  select(beagle_f, eagle_f, shapeit_f) %&gt;%
  cor(method = &quot;pearson&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle_f</th>
<th align="right">eagle_f</th>
<th align="right">shapeit_f</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle_f</td>
<td align="right">1.0000000</td>
<td align="right">0.8735959</td>
<td align="right">0.9556706</td>
</tr>
<tr class="even">
<td align="left">eagle_f</td>
<td align="right">0.8735959</td>
<td align="right">1.0000000</td>
<td align="right">0.8853104</td>
</tr>
<tr class="odd">
<td align="left">shapeit_f</td>
<td align="right">0.9556706</td>
<td align="right">0.8853104</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df_amr %&gt;%
  select(beagle_f, eagle_f, shapeit_f) %&gt;%
  cor(method = &quot;spearman&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle_f</th>
<th align="right">eagle_f</th>
<th align="right">shapeit_f</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle_f</td>
<td align="right">1.0000000</td>
<td align="right">0.8047865</td>
<td align="right">0.9266127</td>
</tr>
<tr class="even">
<td align="left">eagle_f</td>
<td align="right">0.8047865</td>
<td align="right">1.0000000</td>
<td align="right">0.8267259</td>
</tr>
<tr class="odd">
<td align="left">shapeit_f</td>
<td align="right">0.9266127</td>
<td align="right">0.8267259</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<div id="adjust-for-het-sites-5" class="section level4">
<h4>Adjust for het sites</h4>
<pre class="r"><code>df_amr %&gt;%
  mutate(beagle = beagle_f / hets,
         eagle = eagle_f / hets,
         shapeit = shapeit_f / hets) %&gt;%
  select(bin_id, beagle, eagle, shapeit) %&gt;%
  rename(Beagle = beagle, Eagle = eagle, SHAPEIT = shapeit) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Flips&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Flips, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Flips per Heterozygous Site per MB Bin&quot;, &quot;AMR Synthetic Diploids&quot;)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-34-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-34-1">
Past versions of unnamed-chunk-34-1.png
</button>
</p>
<div id="fig-unnamed-chunk-34-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-34-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_amr %&gt;%
  mutate(beagle = beagle_f / hets,
         eagle = eagle_f / hets,
         shapeit = shapeit_f / hets) %&gt;%
  select(beagle, eagle, shapeit) %&gt;%
  cor() %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle</th>
<th align="right">eagle</th>
<th align="right">shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle</td>
<td align="right">1.0000000</td>
<td align="right">0.9273082</td>
<td align="right">0.9857943</td>
</tr>
<tr class="even">
<td align="left">eagle</td>
<td align="right">0.9273082</td>
<td align="right">1.0000000</td>
<td align="right">0.9424208</td>
</tr>
<tr class="odd">
<td align="left">shapeit</td>
<td align="right">0.9857943</td>
<td align="right">0.9424208</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df_amr %&gt;%
  mutate(beagle = beagle_f / hets,
         eagle = eagle_f / hets,
         shapeit = shapeit_f / hets) %&gt;%
  select(beagle, eagle, shapeit) %&gt;%
  cor(method = &quot;spearman&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle</th>
<th align="right">eagle</th>
<th align="right">shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle</td>
<td align="right">1.0000000</td>
<td align="right">0.9399407</td>
<td align="right">0.9739177</td>
</tr>
<tr class="even">
<td align="left">eagle</td>
<td align="right">0.9399407</td>
<td align="right">1.0000000</td>
<td align="right">0.9435831</td>
</tr>
<tr class="odd">
<td align="left">shapeit</td>
<td align="right">0.9739177</td>
<td align="right">0.9435831</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="eas-1" class="section level2">
<h2>EAS</h2>
<div id="switches-3" class="section level3">
<h3>Switches</h3>
<pre class="r"><code>df_eas &lt;- get_errors_mb(601:800, 1e6, het_loc_dir, beagle_switch_dir, eagle_switch_dir, shapeit_switch_dir, df_freq)

df_eas %&gt;%
  ggplot(aes(x = bin_id, y = hets)) +
  geom_point() +
  geom_line() +
  ggtitle(&quot;Aggregated Heterozygous Sites per Bin&quot;, &quot;EAS Synthetic Diploids&quot;) +
  xlab(&quot;Bin&quot;) +
  ylab(&quot;Heterozygous Sites&quot;) +
  theme_classic()</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-35-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-35-1">
Past versions of unnamed-chunk-35-1.png
</button>
</p>
<div id="fig-unnamed-chunk-35-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-35-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_eas %&gt;%
  select(bin_id, beagle_s, eagle_s, shapeit_s) %&gt;%
  rename(Beagle = beagle_s, Eagle = eagle_s, SHAPEIT = shapeit_s) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Switches&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Switches, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Switches per MB Bin&quot;, &quot;EAS Synthetic Diploids&quot;)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-35-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_eas %&gt;%
  select(bin_id, beagle_s, eagle_s, shapeit_s) %&gt;%
  rename(Beagle = beagle_s, Eagle = eagle_s, SHAPEIT = shapeit_s) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Switches&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Switches, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Switches per MB Bin&quot;, &quot;EAS Synthetic Diploids&quot;) +
  geom_smooth()</code></pre>
<pre><code>`geom_smooth()` using method = &#39;loess&#39; and formula = &#39;y ~ x&#39;</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-35-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_eas %&gt;%
  select(beagle_s, eagle_s, shapeit_s) %&gt;%
  cor(method = &quot;pearson&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle_s</th>
<th align="right">eagle_s</th>
<th align="right">shapeit_s</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle_s</td>
<td align="right">1.0000000</td>
<td align="right">0.9848287</td>
<td align="right">0.9938213</td>
</tr>
<tr class="even">
<td align="left">eagle_s</td>
<td align="right">0.9848287</td>
<td align="right">1.0000000</td>
<td align="right">0.9843878</td>
</tr>
<tr class="odd">
<td align="left">shapeit_s</td>
<td align="right">0.9938213</td>
<td align="right">0.9843878</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df_eas %&gt;%
  select(beagle_s, eagle_s, shapeit_s) %&gt;%
  cor(method = &quot;spearman&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle_s</th>
<th align="right">eagle_s</th>
<th align="right">shapeit_s</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle_s</td>
<td align="right">1.0000000</td>
<td align="right">0.9404477</td>
<td align="right">0.9862468</td>
</tr>
<tr class="even">
<td align="left">eagle_s</td>
<td align="right">0.9404477</td>
<td align="right">1.0000000</td>
<td align="right">0.9360428</td>
</tr>
<tr class="odd">
<td align="left">shapeit_s</td>
<td align="right">0.9862468</td>
<td align="right">0.9360428</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<div id="adjust-for-het-sites-6" class="section level4">
<h4>Adjust for het sites</h4>
<pre class="r"><code>df_eas %&gt;%
  mutate(beagle = beagle_s / hets,
         eagle = eagle_s / hets,
         shapeit = shapeit_s / hets) %&gt;%
  select(bin_id, beagle, eagle, shapeit) %&gt;%
  rename(Beagle = beagle, Eagle = eagle, SHAPEIT = shapeit) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Switches&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Switches, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Switches per Heterozygous Site per MB Bin&quot;, &quot;EAS Synthetic Diploids&quot;)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-36-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-36-1">
Past versions of unnamed-chunk-36-1.png
</button>
</p>
<div id="fig-unnamed-chunk-36-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-36-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_eas %&gt;%
  mutate(beagle = beagle_s / hets,
         eagle = eagle_s / hets,
         shapeit = shapeit_s / hets) %&gt;%
  select(beagle, eagle, shapeit) %&gt;%
  cor() %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle</th>
<th align="right">eagle</th>
<th align="right">shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle</td>
<td align="right">1.0000000</td>
<td align="right">0.8414058</td>
<td align="right">0.9884196</td>
</tr>
<tr class="even">
<td align="left">eagle</td>
<td align="right">0.8414058</td>
<td align="right">1.0000000</td>
<td align="right">0.8061714</td>
</tr>
<tr class="odd">
<td align="left">shapeit</td>
<td align="right">0.9884196</td>
<td align="right">0.8061714</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df_eas %&gt;%
  mutate(beagle = beagle_s / hets,
         eagle = eagle_s / hets,
         shapeit = shapeit_s / hets) %&gt;%
  select(beagle, eagle, shapeit) %&gt;%
  cor(method = &quot;spearman&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle</th>
<th align="right">eagle</th>
<th align="right">shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle</td>
<td align="right">1.0000000</td>
<td align="right">0.8981039</td>
<td align="right">0.9794667</td>
</tr>
<tr class="even">
<td align="left">eagle</td>
<td align="right">0.8981039</td>
<td align="right">1.0000000</td>
<td align="right">0.8602300</td>
</tr>
<tr class="odd">
<td align="left">shapeit</td>
<td align="right">0.9794667</td>
<td align="right">0.8602300</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="flips-3" class="section level3">
<h3>Flips</h3>
<pre class="r"><code>df_eas %&gt;%
  select(bin_id, beagle_f, eagle_f, shapeit_f) %&gt;%
  rename(Beagle = beagle_f, Eagle = eagle_f, SHAPEIT = shapeit_f) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Flips&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Flips, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Flips per MB Bin&quot;, &quot;EAS Synthetic Diploids&quot;)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-37-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-37-1">
Past versions of unnamed-chunk-37-1.png
</button>
</p>
<div id="fig-unnamed-chunk-37-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-37-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_eas %&gt;%
  select(bin_id, beagle_f, eagle_f, shapeit_f) %&gt;%
  rename(Beagle = beagle_f, Eagle = eagle_f, SHAPEIT = shapeit_f) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Flips&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Flips, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Flips per MB Bin&quot;, &quot;EAS Synthetic Diploids&quot;) +
  geom_smooth()</code></pre>
<pre><code>`geom_smooth()` using method = &#39;loess&#39; and formula = &#39;y ~ x&#39;</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-37-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_eas %&gt;%
  select(beagle_f, eagle_f, shapeit_f) %&gt;%
  cor(method = &quot;pearson&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle_f</th>
<th align="right">eagle_f</th>
<th align="right">shapeit_f</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle_f</td>
<td align="right">1.0000000</td>
<td align="right">0.9333378</td>
<td align="right">0.9633606</td>
</tr>
<tr class="even">
<td align="left">eagle_f</td>
<td align="right">0.9333378</td>
<td align="right">1.0000000</td>
<td align="right">0.9357252</td>
</tr>
<tr class="odd">
<td align="left">shapeit_f</td>
<td align="right">0.9633606</td>
<td align="right">0.9357252</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df_eas %&gt;%
  select(beagle_f, eagle_f, shapeit_f) %&gt;%
  cor(method = &quot;spearman&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle_f</th>
<th align="right">eagle_f</th>
<th align="right">shapeit_f</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle_f</td>
<td align="right">1.0000000</td>
<td align="right">0.8700481</td>
<td align="right">0.9433266</td>
</tr>
<tr class="even">
<td align="left">eagle_f</td>
<td align="right">0.8700481</td>
<td align="right">1.0000000</td>
<td align="right">0.8759664</td>
</tr>
<tr class="odd">
<td align="left">shapeit_f</td>
<td align="right">0.9433266</td>
<td align="right">0.8759664</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<div id="adjust-for-het-sites-7" class="section level4">
<h4>Adjust for het sites</h4>
<pre class="r"><code>df_eas %&gt;%
  mutate(beagle = beagle_f / hets,
         eagle = eagle_f / hets,
         shapeit = shapeit_f / hets) %&gt;%
  select(bin_id, beagle, eagle, shapeit) %&gt;%
  rename(Beagle = beagle, Eagle = eagle, SHAPEIT = shapeit) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Flips&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Flips, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Flips per Heterozygous Site per MB Bin&quot;, &quot;EAS Synthetic Diploids&quot;)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-38-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-38-1">
Past versions of unnamed-chunk-38-1.png
</button>
</p>
<div id="fig-unnamed-chunk-38-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-38-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_eas %&gt;%
  mutate(beagle = beagle_f / hets,
         eagle = eagle_f / hets,
         shapeit = shapeit_f / hets) %&gt;%
  select(beagle, eagle, shapeit) %&gt;%
  cor() %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle</th>
<th align="right">eagle</th>
<th align="right">shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle</td>
<td align="right">1.000000</td>
<td align="right">0.9785240</td>
<td align="right">0.9915450</td>
</tr>
<tr class="even">
<td align="left">eagle</td>
<td align="right">0.978524</td>
<td align="right">1.0000000</td>
<td align="right">0.9835731</td>
</tr>
<tr class="odd">
<td align="left">shapeit</td>
<td align="right">0.991545</td>
<td align="right">0.9835731</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df_eas %&gt;%
  mutate(beagle = beagle_f / hets,
         eagle = eagle_f / hets,
         shapeit = shapeit_f / hets) %&gt;%
  select(beagle, eagle, shapeit) %&gt;%
  cor(method = &quot;spearman&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle</th>
<th align="right">eagle</th>
<th align="right">shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle</td>
<td align="right">1.0000000</td>
<td align="right">0.9715127</td>
<td align="right">0.9759707</td>
</tr>
<tr class="even">
<td align="left">eagle</td>
<td align="right">0.9715127</td>
<td align="right">1.0000000</td>
<td align="right">0.9672639</td>
</tr>
<tr class="odd">
<td align="left">shapeit</td>
<td align="right">0.9759707</td>
<td align="right">0.9672639</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sas-1" class="section level2">
<h2>SAS</h2>
<div id="switches-4" class="section level3">
<h3>Switches</h3>
<pre class="r"><code>df_sas &lt;- get_errors_mb(801:1000, 1e6, het_loc_dir, beagle_switch_dir, eagle_switch_dir, shapeit_switch_dir, df_freq)

df_sas %&gt;%
  ggplot(aes(x = bin_id, y = hets)) +
  geom_point() +
  geom_line() +
  ggtitle(&quot;Aggregated Heterozygous Sites per Bin&quot;, &quot;SAS Synthetic Diploids&quot;) +
  xlab(&quot;Bin&quot;) +
  ylab(&quot;Heterozygous Sites&quot;) +
  theme_classic()</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-39-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-39-1">
Past versions of unnamed-chunk-39-1.png
</button>
</p>
<div id="fig-unnamed-chunk-39-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-39-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_sas %&gt;%
  select(bin_id, beagle_s, eagle_s, shapeit_s) %&gt;%
  rename(Beagle = beagle_s, Eagle = eagle_s, SHAPEIT = shapeit_s) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Switches&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Switches, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Switches per MB Bin&quot;, &quot;SAS Synthetic Diploids&quot;)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-39-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_sas %&gt;%
  select(bin_id, beagle_s, eagle_s, shapeit_s) %&gt;%
  rename(Beagle = beagle_s, Eagle = eagle_s, SHAPEIT = shapeit_s) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Switches&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Switches, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Switches per MB Bin&quot;, &quot;SAS Synthetic Diploids&quot;) +
  geom_smooth()</code></pre>
<pre><code>`geom_smooth()` using method = &#39;loess&#39; and formula = &#39;y ~ x&#39;</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-39-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_sas %&gt;%
  select(beagle_s, eagle_s, shapeit_s) %&gt;%
  cor(method = &quot;pearson&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle_s</th>
<th align="right">eagle_s</th>
<th align="right">shapeit_s</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle_s</td>
<td align="right">1.0000000</td>
<td align="right">0.9785701</td>
<td align="right">0.9924295</td>
</tr>
<tr class="even">
<td align="left">eagle_s</td>
<td align="right">0.9785701</td>
<td align="right">1.0000000</td>
<td align="right">0.9769748</td>
</tr>
<tr class="odd">
<td align="left">shapeit_s</td>
<td align="right">0.9924295</td>
<td align="right">0.9769748</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df_sas %&gt;%
  select(beagle_s, eagle_s, shapeit_s) %&gt;%
  cor(method = &quot;spearman&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle_s</th>
<th align="right">eagle_s</th>
<th align="right">shapeit_s</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle_s</td>
<td align="right">1.0000000</td>
<td align="right">0.9155233</td>
<td align="right">0.9881456</td>
</tr>
<tr class="even">
<td align="left">eagle_s</td>
<td align="right">0.9155233</td>
<td align="right">1.0000000</td>
<td align="right">0.9096924</td>
</tr>
<tr class="odd">
<td align="left">shapeit_s</td>
<td align="right">0.9881456</td>
<td align="right">0.9096924</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<div id="adjust-for-het-sites-8" class="section level4">
<h4>Adjust for het sites</h4>
<pre class="r"><code>df_sas %&gt;%
  mutate(beagle = beagle_s / hets,
         eagle = eagle_s / hets,
         shapeit = shapeit_s / hets) %&gt;%
  select(bin_id, beagle, eagle, shapeit) %&gt;%
  rename(Beagle = beagle, Eagle = eagle, SHAPEIT = shapeit) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Switches&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Switches, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Switches per Heterozygous Site per MB Bin&quot;, &quot;SAS Synthetic Diploids&quot;)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-40-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-40-1">
Past versions of unnamed-chunk-40-1.png
</button>
</p>
<div id="fig-unnamed-chunk-40-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-40-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_sas %&gt;%
  mutate(beagle = beagle_s / hets,
         eagle = eagle_s / hets,
         shapeit = shapeit_s / hets) %&gt;%
  select(beagle, eagle, shapeit) %&gt;%
  cor() %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle</th>
<th align="right">eagle</th>
<th align="right">shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle</td>
<td align="right">1.0000000</td>
<td align="right">0.8209293</td>
<td align="right">0.9918835</td>
</tr>
<tr class="even">
<td align="left">eagle</td>
<td align="right">0.8209293</td>
<td align="right">1.0000000</td>
<td align="right">0.7939155</td>
</tr>
<tr class="odd">
<td align="left">shapeit</td>
<td align="right">0.9918835</td>
<td align="right">0.7939155</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df_sas %&gt;%
  mutate(beagle = beagle_s / hets,
         eagle = eagle_s / hets,
         shapeit = shapeit_s / hets) %&gt;%
  select(beagle, eagle, shapeit) %&gt;%
  cor(method = &quot;spearman&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle</th>
<th align="right">eagle</th>
<th align="right">shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle</td>
<td align="right">1.0000000</td>
<td align="right">0.9124364</td>
<td align="right">0.9843604</td>
</tr>
<tr class="even">
<td align="left">eagle</td>
<td align="right">0.9124364</td>
<td align="right">1.0000000</td>
<td align="right">0.8854618</td>
</tr>
<tr class="odd">
<td align="left">shapeit</td>
<td align="right">0.9843604</td>
<td align="right">0.8854618</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="flips-4" class="section level3">
<h3>Flips</h3>
<pre class="r"><code>df_sas %&gt;%
  select(bin_id, beagle_f, eagle_f, shapeit_f) %&gt;%
  rename(Beagle = beagle_f, Eagle = eagle_f, SHAPEIT = shapeit_f) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Flips&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Flips, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Flips per MB Bin&quot;, &quot;SAS Synthetic Diploids&quot;)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-41-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-41-1">
Past versions of unnamed-chunk-41-1.png
</button>
</p>
<div id="fig-unnamed-chunk-41-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-41-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_sas %&gt;%
  select(bin_id, beagle_f, eagle_f, shapeit_f) %&gt;%
  rename(Beagle = beagle_f, Eagle = eagle_f, SHAPEIT = shapeit_f) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Flips&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Flips, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Flips per MB Bin&quot;, &quot;SAS Synthetic Diploids&quot;) +
  geom_smooth()</code></pre>
<pre><code>`geom_smooth()` using method = &#39;loess&#39; and formula = &#39;y ~ x&#39;</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-41-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_sas %&gt;%
  select(beagle_f, eagle_f, shapeit_f) %&gt;%
  cor(method = &quot;pearson&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle_f</th>
<th align="right">eagle_f</th>
<th align="right">shapeit_f</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle_f</td>
<td align="right">1.0000000</td>
<td align="right">0.9025442</td>
<td align="right">0.9501650</td>
</tr>
<tr class="even">
<td align="left">eagle_f</td>
<td align="right">0.9025442</td>
<td align="right">1.0000000</td>
<td align="right">0.9230241</td>
</tr>
<tr class="odd">
<td align="left">shapeit_f</td>
<td align="right">0.9501650</td>
<td align="right">0.9230241</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df_sas %&gt;%
  select(beagle_f, eagle_f, shapeit_f) %&gt;%
  cor(method = &quot;spearman&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle_f</th>
<th align="right">eagle_f</th>
<th align="right">shapeit_f</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle_f</td>
<td align="right">1.0000000</td>
<td align="right">0.8615219</td>
<td align="right">0.9269164</td>
</tr>
<tr class="even">
<td align="left">eagle_f</td>
<td align="right">0.8615219</td>
<td align="right">1.0000000</td>
<td align="right">0.8947259</td>
</tr>
<tr class="odd">
<td align="left">shapeit_f</td>
<td align="right">0.9269164</td>
<td align="right">0.8947259</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<div id="adjust-for-het-sites-9" class="section level4">
<h4>Adjust for het sites</h4>
<pre class="r"><code>df_sas %&gt;%
  mutate(beagle = beagle_f / hets,
         eagle = eagle_f / hets,
         shapeit = shapeit_f / hets) %&gt;%
  select(bin_id, beagle, eagle, shapeit) %&gt;%
  rename(Beagle = beagle, Eagle = eagle, SHAPEIT = shapeit) %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Method&quot;, values_to = &quot;Flips&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Flips, color = Method )) +
    geom_point() +
    geom_line() +
    theme_classic() +
    xlab(&quot;Bin&quot;) +
  ggtitle(&quot;Flips per Heterozygous Site per MB Bin&quot;, &quot;SAS Synthetic Diploids&quot;)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-42-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_sas %&gt;%
  mutate(beagle = beagle_f / hets,
         eagle = eagle_f / hets,
         shapeit = shapeit_f / hets) %&gt;%
  select(beagle, eagle, shapeit) %&gt;%
  cor() %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle</th>
<th align="right">eagle</th>
<th align="right">shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle</td>
<td align="right">1.0000000</td>
<td align="right">0.9639409</td>
<td align="right">0.9870915</td>
</tr>
<tr class="even">
<td align="left">eagle</td>
<td align="right">0.9639409</td>
<td align="right">1.0000000</td>
<td align="right">0.9819211</td>
</tr>
<tr class="odd">
<td align="left">shapeit</td>
<td align="right">0.9870915</td>
<td align="right">0.9819211</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df_sas %&gt;%
  mutate(beagle = beagle_f / hets,
         eagle = eagle_f / hets,
         shapeit = shapeit_f / hets) %&gt;%
  select(beagle, eagle, shapeit) %&gt;%
  cor(method = &quot;spearman&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">beagle</th>
<th align="right">eagle</th>
<th align="right">shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beagle</td>
<td align="right">1.0000000</td>
<td align="right">0.9501603</td>
<td align="right">0.9786790</td>
</tr>
<tr class="even">
<td align="left">eagle</td>
<td align="right">0.9501603</td>
<td align="right">1.0000000</td>
<td align="right">0.9565458</td>
</tr>
<tr class="odd">
<td align="left">shapeit</td>
<td align="right">0.9786790</td>
<td align="right">0.9565458</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="population-comparisons" class="section level1">
<h1>Population Comparisons</h1>
<pre class="r"><code>df_eur &lt;- get_errors_mb(1:200, 1e6, het_loc_dir, beagle_switch_dir, eagle_switch_dir, shapeit_switch_dir, df_freq)</code></pre>
<div id="switches-5" class="section level2">
<h2>Switches</h2>
<div id="beagle" class="section level3">
<h3>Beagle</h3>
<pre class="r"><code>df &lt;- data.frame(bin_id = df_eur$bin_id,
                 AFR = df_afr$beagle_s / df_afr$hets,
                 AMR = df_amr$beagle_s / df_amr$hets,
                 EAS = df_eas$beagle_s / df_eas$hets,
                 EUR = df_eur$beagle_s / df_eur$hets,
                 SAS = df_sas$beagle_s / df_sas$hets)

df %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Population&quot;, values_to = &quot;Switches&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Switches, color = Population)) +
  geom_point() +
  geom_line() +
  ggtitle(&quot;Beagle Switch Rate per MB Bin&quot;) + 
  xlab(&quot;Bin&quot;) +
  theme_classic() +
  scale_color_manual(values = cbPalette)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-44-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-44-1">
Past versions of unnamed-chunk-44-1.png
</button>
</p>
<div id="fig-unnamed-chunk-44-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-44-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df %&gt;%
  select(-bin_id) %&gt;% 
  cor() %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">AFR</th>
<th align="right">AMR</th>
<th align="right">EAS</th>
<th align="right">EUR</th>
<th align="right">SAS</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AFR</td>
<td align="right">1.0000000</td>
<td align="right">0.8435499</td>
<td align="right">0.5662294</td>
<td align="right">0.8187448</td>
<td align="right">0.6422762</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="right">0.8435499</td>
<td align="right">1.0000000</td>
<td align="right">0.7966688</td>
<td align="right">0.9203250</td>
<td align="right">0.8305378</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="right">0.5662294</td>
<td align="right">0.7966688</td>
<td align="right">1.0000000</td>
<td align="right">0.7694371</td>
<td align="right">0.9209262</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="right">0.8187448</td>
<td align="right">0.9203250</td>
<td align="right">0.7694371</td>
<td align="right">1.0000000</td>
<td align="right">0.8518346</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="right">0.6422762</td>
<td align="right">0.8305378</td>
<td align="right">0.9209262</td>
<td align="right">0.8518346</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df %&gt;%
  select(-bin_id) %&gt;% 
  cor(method = &quot;spearman&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">AFR</th>
<th align="right">AMR</th>
<th align="right">EAS</th>
<th align="right">EUR</th>
<th align="right">SAS</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AFR</td>
<td align="right">1.0000000</td>
<td align="right">0.8420146</td>
<td align="right">0.7813001</td>
<td align="right">0.8396026</td>
<td align="right">0.8326978</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="right">0.8420146</td>
<td align="right">1.0000000</td>
<td align="right">0.8937226</td>
<td align="right">0.8910805</td>
<td align="right">0.9349529</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="right">0.7813001</td>
<td align="right">0.8937226</td>
<td align="right">1.0000000</td>
<td align="right">0.8548972</td>
<td align="right">0.9107738</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="right">0.8396026</td>
<td align="right">0.8910805</td>
<td align="right">0.8548972</td>
<td align="right">1.0000000</td>
<td align="right">0.9271628</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="right">0.8326978</td>
<td align="right">0.9349529</td>
<td align="right">0.9107738</td>
<td align="right">0.9271628</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
</div>
<div id="eagle" class="section level3">
<h3>Eagle</h3>
<pre class="r"><code>df &lt;- data.frame(bin_id = df_eur$bin_id,
                 AFR = df_afr$eagle_s / df_afr$hets,
                 AMR = df_amr$eagle_s / df_amr$hets,
                 EAS = df_eas$eagle_s / df_eas$hets,
                 EUR = df_eur$eagle_s / df_eur$hets,
                 SAS = df_sas$eagle_s / df_sas$hets)

df %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Population&quot;, values_to = &quot;Switches&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Switches, color = Population)) +
  geom_point() +
  geom_line() +
  ggtitle(&quot;Eagle Switch Rate per MB Bin&quot;) + 
  xlab(&quot;Bin&quot;) +
  theme_classic() +
  scale_color_manual(values = cbPalette)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-45-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-45-1">
Past versions of unnamed-chunk-45-1.png
</button>
</p>
<div id="fig-unnamed-chunk-45-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-45-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df %&gt;%
  select(-bin_id) %&gt;% 
  cor() %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">AFR</th>
<th align="right">AMR</th>
<th align="right">EAS</th>
<th align="right">EUR</th>
<th align="right">SAS</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AFR</td>
<td align="right">1.0000000</td>
<td align="right">0.7905961</td>
<td align="right">0.4554649</td>
<td align="right">0.7183750</td>
<td align="right">0.5529106</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="right">0.7905961</td>
<td align="right">1.0000000</td>
<td align="right">0.7073072</td>
<td align="right">0.8909319</td>
<td align="right">0.7745032</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="right">0.4554649</td>
<td align="right">0.7073072</td>
<td align="right">1.0000000</td>
<td align="right">0.7435757</td>
<td align="right">0.8972525</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="right">0.7183750</td>
<td align="right">0.8909319</td>
<td align="right">0.7435757</td>
<td align="right">1.0000000</td>
<td align="right">0.8644305</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="right">0.5529106</td>
<td align="right">0.7745032</td>
<td align="right">0.8972525</td>
<td align="right">0.8644305</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df %&gt;%
  select(-bin_id) %&gt;% 
  cor(method = &quot;spearman&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">AFR</th>
<th align="right">AMR</th>
<th align="right">EAS</th>
<th align="right">EUR</th>
<th align="right">SAS</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AFR</td>
<td align="right">1.0000000</td>
<td align="right">0.7833287</td>
<td align="right">0.6833670</td>
<td align="right">0.7528512</td>
<td align="right">0.7704043</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="right">0.7833287</td>
<td align="right">1.0000000</td>
<td align="right">0.8765423</td>
<td align="right">0.8712966</td>
<td align="right">0.9094319</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="right">0.6833670</td>
<td align="right">0.8765423</td>
<td align="right">1.0000000</td>
<td align="right">0.8332659</td>
<td align="right">0.8821506</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="right">0.7528512</td>
<td align="right">0.8712966</td>
<td align="right">0.8332659</td>
<td align="right">1.0000000</td>
<td align="right">0.8787487</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="right">0.7704043</td>
<td align="right">0.9094319</td>
<td align="right">0.8821506</td>
<td align="right">0.8787487</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
</div>
<div id="shapeit" class="section level3">
<h3>SHAPEIT</h3>
<pre class="r"><code>df &lt;- data.frame(bin_id = df_eur$bin_id,
                 AFR = df_afr$shapeit_s / df_afr$hets,
                 AMR = df_amr$shapeit_s / df_amr$hets,
                 EAS = df_eas$shapeit_s / df_eas$hets,
                 EUR = df_eur$shapeit_s / df_eur$hets,
                 SAS = df_sas$shapeit_s / df_sas$hets)

df %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Population&quot;, values_to = &quot;Switches&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Switches, color = Population)) +
  geom_point() +
  geom_line() +
  ggtitle(&quot;SHAPEIT Switch Rate per MB Bin&quot;) + 
  xlab(&quot;Bin&quot;) +
  theme_classic() +
  scale_color_manual(values = cbPalette)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-46-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-46-1">
Past versions of unnamed-chunk-46-1.png
</button>
</p>
<div id="fig-unnamed-chunk-46-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-46-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df %&gt;%
  select(-bin_id) %&gt;% 
  cor() %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">AFR</th>
<th align="right">AMR</th>
<th align="right">EAS</th>
<th align="right">EUR</th>
<th align="right">SAS</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AFR</td>
<td align="right">1.0000000</td>
<td align="right">0.8418161</td>
<td align="right">0.5611537</td>
<td align="right">0.8363133</td>
<td align="right">0.6371268</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="right">0.8418161</td>
<td align="right">1.0000000</td>
<td align="right">0.7957959</td>
<td align="right">0.9375338</td>
<td align="right">0.8346425</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="right">0.5611537</td>
<td align="right">0.7957959</td>
<td align="right">1.0000000</td>
<td align="right">0.7637327</td>
<td align="right">0.9197992</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="right">0.8363133</td>
<td align="right">0.9375338</td>
<td align="right">0.7637327</td>
<td align="right">1.0000000</td>
<td align="right">0.8458320</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="right">0.6371268</td>
<td align="right">0.8346425</td>
<td align="right">0.9197992</td>
<td align="right">0.8458320</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df %&gt;%
  select(-bin_id) %&gt;% 
  cor(method = &quot;spearman&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">AFR</th>
<th align="right">AMR</th>
<th align="right">EAS</th>
<th align="right">EUR</th>
<th align="right">SAS</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AFR</td>
<td align="right">1.0000000</td>
<td align="right">0.8799024</td>
<td align="right">0.8079923</td>
<td align="right">0.8734228</td>
<td align="right">0.8556988</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="right">0.8799024</td>
<td align="right">1.0000000</td>
<td align="right">0.9023841</td>
<td align="right">0.9245068</td>
<td align="right">0.9350471</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="right">0.8079923</td>
<td align="right">0.9023841</td>
<td align="right">1.0000000</td>
<td align="right">0.8719136</td>
<td align="right">0.9208470</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="right">0.8734228</td>
<td align="right">0.9245068</td>
<td align="right">0.8719136</td>
<td align="right">1.0000000</td>
<td align="right">0.9360265</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="right">0.8556988</td>
<td align="right">0.9350471</td>
<td align="right">0.9208470</td>
<td align="right">0.9360265</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="flips-5" class="section level2">
<h2>Flips</h2>
<div id="beagle-1" class="section level3">
<h3>Beagle</h3>
<pre class="r"><code>df &lt;- data.frame(bin_id = df_eur$bin_id,
                 AFR = df_afr$beagle_f / df_afr$hets,
                 AMR = df_amr$beagle_f / df_amr$hets,
                 EAS = df_eas$beagle_f / df_eas$hets,
                 EUR = df_eur$beagle_f / df_eur$hets,
                 SAS = df_sas$beagle_f / df_sas$hets)

df %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Population&quot;, values_to = &quot;Flips&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Flips, color = Population)) +
  geom_point() +
  geom_line() +
  ggtitle(&quot;Beagle Flip Rate per MB Bin&quot;) + 
  xlab(&quot;Bin&quot;) +
  theme_classic() +
  scale_color_manual(values = cbPalette)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-47-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-47-1">
Past versions of unnamed-chunk-47-1.png
</button>
</p>
<div id="fig-unnamed-chunk-47-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-47-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df %&gt;%
  select(-bin_id) %&gt;% 
  cor() %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">AFR</th>
<th align="right">AMR</th>
<th align="right">EAS</th>
<th align="right">EUR</th>
<th align="right">SAS</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AFR</td>
<td align="right">1.0000000</td>
<td align="right">0.7434227</td>
<td align="right">0.4958135</td>
<td align="right">0.6718893</td>
<td align="right">0.6219825</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="right">0.7434227</td>
<td align="right">1.0000000</td>
<td align="right">0.7707449</td>
<td align="right">0.8953001</td>
<td align="right">0.8711017</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="right">0.4958135</td>
<td align="right">0.7707449</td>
<td align="right">1.0000000</td>
<td align="right">0.8190342</td>
<td align="right">0.8998085</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="right">0.6718893</td>
<td align="right">0.8953001</td>
<td align="right">0.8190342</td>
<td align="right">1.0000000</td>
<td align="right">0.9454520</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="right">0.6219825</td>
<td align="right">0.8711017</td>
<td align="right">0.8998085</td>
<td align="right">0.9454520</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df %&gt;%
  select(-bin_id) %&gt;% 
  cor(method = &quot;spearman&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">AFR</th>
<th align="right">AMR</th>
<th align="right">EAS</th>
<th align="right">EUR</th>
<th align="right">SAS</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AFR</td>
<td align="right">1.0000000</td>
<td align="right">0.7068491</td>
<td align="right">0.6004636</td>
<td align="right">0.6542942</td>
<td align="right">0.7199965</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="right">0.7068491</td>
<td align="right">1.0000000</td>
<td align="right">0.8336215</td>
<td align="right">0.8154165</td>
<td align="right">0.8764169</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="right">0.6004636</td>
<td align="right">0.8336215</td>
<td align="right">1.0000000</td>
<td align="right">0.7757651</td>
<td align="right">0.8408505</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="right">0.6542942</td>
<td align="right">0.8154165</td>
<td align="right">0.7757651</td>
<td align="right">1.0000000</td>
<td align="right">0.8725828</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="right">0.7199965</td>
<td align="right">0.8764169</td>
<td align="right">0.8408505</td>
<td align="right">0.8725828</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
</div>
<div id="eagle-1" class="section level3">
<h3>Eagle</h3>
<pre class="r"><code>df &lt;- data.frame(bin_id = df_eur$bin_id,
                 AFR = df_afr$eagle_f / df_afr$hets,
                 AMR = df_amr$eagle_f / df_amr$hets,
                 EAS = df_eas$eagle_f / df_eas$hets,
                 EUR = df_eur$eagle_f / df_eur$hets,
                 SAS = df_sas$eagle_f / df_sas$hets)

df %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Population&quot;, values_to = &quot;Flips&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Flips, color = Population)) +
  geom_point() +
  geom_line() +
  ggtitle(&quot;Eagle Flip Rate per MB Bin&quot;) + 
  xlab(&quot;Bin&quot;) +
  theme_classic() +
  scale_color_manual(values = cbPalette)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-48-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-48-1">
Past versions of unnamed-chunk-48-1.png
</button>
</p>
<div id="fig-unnamed-chunk-48-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/c4c32be1e0ed55e783752772d6c8ff5a96bf2a07/docs/figure/sd_error_overlap.Rmd/unnamed-chunk-48-1.png" target="_blank">c4c32be</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df %&gt;%
  select(-bin_id) %&gt;% 
  cor() %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">AFR</th>
<th align="right">AMR</th>
<th align="right">EAS</th>
<th align="right">EUR</th>
<th align="right">SAS</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AFR</td>
<td align="right">1.0000000</td>
<td align="right">0.6505809</td>
<td align="right">0.4983784</td>
<td align="right">0.6488042</td>
<td align="right">0.5955942</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="right">0.6505809</td>
<td align="right">1.0000000</td>
<td align="right">0.7180837</td>
<td align="right">0.8009863</td>
<td align="right">0.7813080</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="right">0.4983784</td>
<td align="right">0.7180837</td>
<td align="right">1.0000000</td>
<td align="right">0.7738408</td>
<td align="right">0.8842799</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="right">0.6488042</td>
<td align="right">0.8009863</td>
<td align="right">0.7738408</td>
<td align="right">1.0000000</td>
<td align="right">0.8871337</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="right">0.5955942</td>
<td align="right">0.7813080</td>
<td align="right">0.8842799</td>
<td align="right">0.8871337</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df %&gt;%
  select(-bin_id) %&gt;% 
  cor(method = &quot;spearman&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">AFR</th>
<th align="right">AMR</th>
<th align="right">EAS</th>
<th align="right">EUR</th>
<th align="right">SAS</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AFR</td>
<td align="right">1.0000000</td>
<td align="right">0.7032625</td>
<td align="right">0.6536668</td>
<td align="right">0.7069746</td>
<td align="right">0.7857686</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="right">0.7032625</td>
<td align="right">1.0000000</td>
<td align="right">0.7828407</td>
<td align="right">0.7396340</td>
<td align="right">0.8196410</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="right">0.6536668</td>
<td align="right">0.7828407</td>
<td align="right">1.0000000</td>
<td align="right">0.7201987</td>
<td align="right">0.8241443</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="right">0.7069746</td>
<td align="right">0.7396340</td>
<td align="right">0.7201987</td>
<td align="right">1.0000000</td>
<td align="right">0.8239282</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="right">0.7857686</td>
<td align="right">0.8196410</td>
<td align="right">0.8241443</td>
<td align="right">0.8239282</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
</div>
<div id="shapeit-1" class="section level3">
<h3>SHAPEIT</h3>
<pre class="r"><code>df &lt;- data.frame(bin_id = df_eur$bin_id,
                 AFR = df_afr$shapeit_f / df_afr$hets,
                 AMR = df_amr$shapeit_f / df_amr$hets,
                 EAS = df_eas$shapeit_f / df_eas$hets,
                 EUR = df_eur$shapeit_f / df_eur$hets,
                 SAS = df_sas$shapeit_f / df_sas$hets)

df %&gt;%
  pivot_longer(-bin_id, names_to = &quot;Population&quot;, values_to = &quot;Flips&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = Flips, color = Population)) +
  geom_point() +
  geom_line() +
  ggtitle(&quot;SHAPEIT Flip Rate per MB Bin&quot;) + 
  xlab(&quot;Bin&quot;) +
  theme_classic() +
  scale_color_manual(values = cbPalette)</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-49-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df %&gt;%
  select(-bin_id) %&gt;% 
  cor() %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">AFR</th>
<th align="right">AMR</th>
<th align="right">EAS</th>
<th align="right">EUR</th>
<th align="right">SAS</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AFR</td>
<td align="right">1.0000000</td>
<td align="right">0.7355393</td>
<td align="right">0.5188931</td>
<td align="right">0.7026406</td>
<td align="right">0.6417877</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="right">0.7355393</td>
<td align="right">1.0000000</td>
<td align="right">0.7758019</td>
<td align="right">0.8879173</td>
<td align="right">0.8595218</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="right">0.5188931</td>
<td align="right">0.7758019</td>
<td align="right">1.0000000</td>
<td align="right">0.8136348</td>
<td align="right">0.9086635</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="right">0.7026406</td>
<td align="right">0.8879173</td>
<td align="right">0.8136348</td>
<td align="right">1.0000000</td>
<td align="right">0.9251909</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="right">0.6417877</td>
<td align="right">0.8595218</td>
<td align="right">0.9086635</td>
<td align="right">0.9251909</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df %&gt;%
  select(-bin_id) %&gt;% 
  cor(method = &quot;spearman&quot;) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">AFR</th>
<th align="right">AMR</th>
<th align="right">EAS</th>
<th align="right">EUR</th>
<th align="right">SAS</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AFR</td>
<td align="right">1.0000000</td>
<td align="right">0.7411014</td>
<td align="right">0.6223039</td>
<td align="right">0.7254862</td>
<td align="right">0.7606553</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="right">0.7411014</td>
<td align="right">1.0000000</td>
<td align="right">0.8372883</td>
<td align="right">0.8264831</td>
<td align="right">0.8726839</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="right">0.6223039</td>
<td align="right">0.8372883</td>
<td align="right">1.0000000</td>
<td align="right">0.7722412</td>
<td align="right">0.8318160</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="right">0.7254862</td>
<td align="right">0.8264831</td>
<td align="right">0.7722412</td>
<td align="right">1.0000000</td>
<td align="right">0.8600139</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="right">0.7606553</td>
<td align="right">0.8726839</td>
<td align="right">0.8318160</td>
<td align="right">0.8600139</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="recombination-rate" class="section level1">
<h1>Recombination Rate</h1>
<pre class="r"><code>rr_df &lt;- rr_df %&gt;%
  mutate(bin_id =  ceiling(pos_start / 1e6))

df_eur %&gt;%
  left_join({rr_df %&gt;%
      select(bin_id, rr) %&gt;%
      group_by(bin_id) %&gt;%
      summarise(rr = mean(rr, na.rm = T))},
      by = &quot;bin_id&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = rr)) +
  geom_point() +
  geom_line() +
  theme_classic() +
  xlab(&quot;Bin&quot;) +
  ylab(&quot;deCODE Recombination Rate&quot;)</code></pre>
<pre><code>Warning: Removed 2 rows containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<pre><code>Warning: Removed 2 rows containing missing values or values outside the scale range
(`geom_line()`).</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-50-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_afr %&gt;%
  left_join({rr_df %&gt;%
      select(bin_id, rr) %&gt;%
      group_by(bin_id) %&gt;%
      summarise(rr = mean(rr, na.rm = T))},
      by = &quot;bin_id&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = rr)) +
  geom_point() +
  geom_line() +
  theme_classic() +
  xlab(&quot;Bin&quot;) +
  ylab(&quot;deCODE Recombination Rate&quot;)</code></pre>
<pre><code>Warning: Removed 2 rows containing missing values or values outside the scale range
(`geom_point()`).
Removed 2 rows containing missing values or values outside the scale range
(`geom_line()`).</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-50-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_amr %&gt;%
  left_join({rr_df %&gt;%
      select(bin_id, rr) %&gt;%
      group_by(bin_id) %&gt;%
      summarise(rr = mean(rr, na.rm = T))},
      by = &quot;bin_id&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = rr)) +
  geom_point() +
  geom_line() +
  theme_classic() +
  xlab(&quot;Bin&quot;) +
  ylab(&quot;deCODE Recombination Rate&quot;)</code></pre>
<pre><code>Warning: Removed 2 rows containing missing values or values outside the scale range
(`geom_point()`).
Removed 2 rows containing missing values or values outside the scale range
(`geom_line()`).</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-50-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_eas %&gt;%
  left_join({rr_df %&gt;%
      select(bin_id, rr) %&gt;%
      group_by(bin_id) %&gt;%
      summarise(rr = mean(rr, na.rm = T))},
      by = &quot;bin_id&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = rr)) +
  geom_point() +
  geom_line() +
  theme_classic() +
  xlab(&quot;Bin&quot;) +
  ylab(&quot;deCODE Recombination Rate&quot;)</code></pre>
<pre><code>Warning: Removed 2 rows containing missing values or values outside the scale range
(`geom_point()`).
Removed 2 rows containing missing values or values outside the scale range
(`geom_line()`).</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-50-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_sas %&gt;%
  left_join({rr_df %&gt;%
      select(bin_id, rr) %&gt;%
      group_by(bin_id) %&gt;%
      summarise(rr = mean(rr, na.rm = T))},
      by = &quot;bin_id&quot;) %&gt;%
  ggplot(aes(x = bin_id, y = rr)) +
  geom_point() +
  geom_line() +
  theme_classic() +
  xlab(&quot;Bin&quot;) +
  ylab(&quot;deCODE Recombination Rate&quot;)</code></pre>
<pre><code>Warning: Removed 2 rows containing missing values or values outside the scale range
(`geom_point()`).
Removed 2 rows containing missing values or values outside the scale range
(`geom_line()`).</code></pre>
<p><img src="figure/sd_error_overlap.Rmd/unnamed-chunk-50-5.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 20.04.6 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3;  LAPACK version 3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: America/New_York
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] yaml_2.3.8          limma_3.58.1        ggVennDiagram_1.5.2
 [4] ggExtra_0.10.1      lubridate_1.9.3     forcats_1.0.0      
 [7] stringr_1.5.1       dplyr_1.1.4         purrr_1.0.2        
[10] readr_2.1.5         tidyr_1.3.1         tibble_3.2.1       
[13] ggplot2_3.5.0       tidyverse_2.0.0     workflowr_1.7.1    

loaded via a namespace (and not attached):
 [1] gtable_0.3.4      xfun_0.42         bslib_0.7.0       processx_3.8.3   
 [5] lattice_0.22-6    callr_3.7.5       tzdb_0.4.0        vctrs_0.6.5      
 [9] tools_4.4.0       ps_1.7.6          generics_0.1.3    parallel_4.4.0   
[13] fansi_1.0.6       highr_0.10        pkgconfig_2.0.3   Matrix_1.6-5     
[17] lifecycle_1.0.4   farver_2.1.1      compiler_4.4.0    git2r_0.33.0     
[21] statmod_1.5.0     munsell_0.5.1     getPass_0.2-4     httpuv_1.6.14    
[25] htmltools_0.5.8   sass_0.4.9        crayon_1.5.2      later_1.3.2      
[29] pillar_1.9.0      jquerylib_0.1.4   whisker_0.4.1     cachem_1.0.8     
[33] nlme_3.1-164      mime_0.12         tidyselect_1.2.0  digest_0.6.34    
[37] stringi_1.8.3     splines_4.4.0     labeling_0.4.3    rprojroot_2.0.4  
[41] fastmap_1.1.1     grid_4.4.0        archive_1.1.7     colorspace_2.1-0 
[45] cli_3.6.2         magrittr_2.0.3    utf8_1.2.4        withr_3.0.0      
[49] scales_1.3.0      promises_1.2.1    bit64_4.0.5       timechange_0.3.0 
[53] rmarkdown_2.26    httr_1.4.7        bit_4.0.5         gridExtra_2.3    
[57] hms_1.1.3         shiny_1.8.1       evaluate_0.23     knitr_1.45       
[61] miniUI_0.1.1.1    mgcv_1.9-1        rlang_1.1.3       Rcpp_1.0.12      
[65] xtable_1.8-4      glue_1.7.0        rstudioapi_0.15.0 vroom_1.6.5      
[69] jsonlite_1.8.8    R6_2.5.1          fs_1.6.3         </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
