<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Andy Beck" />

<meta name="date" content="2024-05-15" />

<title>Synthetic Diploid Results</title>

<script src="site_libs/header-attrs-2.26/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">phasing_clean</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/thandyb/phasing_clean">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Synthetic Diploid Results</h1>
<h4 class="author">Andy Beck</h4>
<h4 class="date">2024-05-15</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-05-29
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>phasing_clean/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240501code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240501)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240501code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240501)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtheandybphasingcleantreebb12bc6cdb894260dee736395e9700d6ebeb60b3targetblankbb12bc6a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/theandyb/phasing_clean/tree/bb12bc6cdb894260dee736395e9700d6ebeb60b3" target="_blank">bb12bc6</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomtheandybphasingcleantreebb12bc6cdb894260dee736395e9700d6ebeb60b3targetblankbb12bc6a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/theandyb/phasing_clean/tree/bb12bc6cdb894260dee736395e9700d6ebeb60b3" target="_blank">bb12bc6</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .ipynb_checkpoints/
    Ignored:    chr22_10k.bed
    Ignored:    data/1kgp/chr10_pilot_mask.bed
    Ignored:    data/1kgp/chr15/
    Ignored:    data/1kgp/chr22/
    Ignored:    data/1kgp/chr8/
    Ignored:    data/1kgp/chrX.vcf.gz
    Ignored:    data/1kgp/chrX.vcf.gz.tbi
    Ignored:    data/1kgp/chrX_2504_snps_noPAR.bcf
    Ignored:    data/1kgp/chrX_2504_snps_noPAR_noSing.bcf
    Ignored:    data/1kgp/chrX_freq.tsv
    Ignored:    data/1kgp/chrX_maf.tsv
    Ignored:    data/1kgp/chrX_pilot_mask.bed
    Ignored:    data/chr10.sizes
    Ignored:    data/chr15.1kb.sorted.bed
    Ignored:    data/chr15.fasta
    Ignored:    data/chr15.sizes
    Ignored:    data/chr15_gc1kb_pilot.bed
    Ignored:    data/chr15_mask.fasta
    Ignored:    data/chr15_mask.fasta.fai
    Ignored:    data/chr22.sizes
    Ignored:    data/chr8.sizes
    Ignored:    data/chrX.1kb.sorted.bed
    Ignored:    data/chrX.fasta
    Ignored:    data/chrX.sizes
    Ignored:    data/chrX_gc1kb_pilot.bed
    Ignored:    data/chrX_mask.fasta
    Ignored:    data/chrX_mask.fasta.fai
    Ignored:    data/hg38.chrom.sizes
    Ignored:    data/mask_pilot_15_fasta.bed
    Ignored:    data/mask_pilot_X_fasta.bed
    Ignored:    data/ref_GRCh38.fna
    Ignored:    data/ref_GRCh38.fna.fai
    Ignored:    data/ref_GRCh38.fna.in
    Ignored:    output/switch_errors/
    Ignored:    output/trio_phase/
    Ignored:    output/trio_phase_15/
    Ignored:    output/trio_phase_22/

Unstaged changes:
    Modified:   code/common_functions.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/synthetic_diploid_results.Rmd</code>) and HTML
(<code>docs/synthetic_diploid_results.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/bb12bc6cdb894260dee736395e9700d6ebeb60b3/analysis/synthetic_diploid_results.Rmd" target="_blank">bb12bc6</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
<td>
wflow_publish("analysis/synthetic_diploid_results.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/phasing_clean/101156547ea5cf5ab7d26907d94ac857ff87c51a/docs/synthetic_diploid_results.html" target="_blank">1011565</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/d1a65202d493163d338780c51650c10a173c5f84/analysis/synthetic_diploid_results.Rmd" target="_blank">d1a6520</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
<td>
wflow_publish(c("analysis/synthetic_diploid_results.Rmd"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/phasing_clean/6172512f57cc03f295aefc7978f0e9f6439169c9/docs/synthetic_diploid_results.html" target="_blank">6172512</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/0241a0998423e92ee7822bc5c16d8640cee3fa5c/analysis/synthetic_diploid_results.Rmd" target="_blank">0241a09</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
<td>
wflow_publish("analysis/synthetic_diploid_results.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/phasing_clean/808cfb808faf1a67f36a54d323404932439cf9ba/docs/synthetic_diploid_results.html" target="_blank">808cfb8</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/phasing_clean/d6dca59ce60fa926bbefcbd9ae78d81f7d52ad3c/docs/synthetic_diploid_results.html" target="_blank">d6dca59</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/f1bf0baca982366a81ca64acaee28b87c876bd22/analysis/synthetic_diploid_results.Rmd" target="_blank">f1bf0ba</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
<td>
wflow_publish(c("analysis/synthetic_diploid_results.Rmd",
"analysis/trio_phasing_results.Rmd"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/phasing_clean/8041a90f4735210b3ffd2e9a1231fd4b9735a442/docs/synthetic_diploid_results.html" target="_blank">8041a90</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/0669558eec5e85ce21377a217903f0c25210aac2/analysis/synthetic_diploid_results.Rmd" target="_blank">0669558</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
<td>
wflow_publish(c("analysis/synthetic_diploid_results.Rmd",
"analysis/trio_phasing_results.Rmd"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/phasing_clean/aa8704bbf7c47c6b3b4c265552120bc446fa974a/docs/synthetic_diploid_results.html" target="_blank">aa8704b</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/3ea0c3d698b03635ee8a14ffe973fa6a560a3166/analysis/synthetic_diploid_results.Rmd" target="_blank">3ea0c3d</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-28
</td>
<td>
wflow_publish("analysis/synthetic_diploid_results.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/phasing_clean/af46d78ebdfe3df4a908c936e9fd9cb3c6a24173/docs/synthetic_diploid_results.html" target="_blank">af46d78</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/50265e35e8983542060f656f189733f2e9558cb4/analysis/synthetic_diploid_results.Rmd" target="_blank">50265e3</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-23
</td>
<td>
wflow_publish("analysis/synthetic_diploid_results.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/phasing_clean/7f7e2da6f25f457b0fe9eb8b6b6b22b03f3be64e/docs/synthetic_diploid_results.html" target="_blank">7f7e2da</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/a13fa5ebc8a05a1d764f1b6b38b31f905ad8cd8b/analysis/synthetic_diploid_results.Rmd" target="_blank">a13fa5e</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-22
</td>
<td>
wflow_publish("analysis/synthetic_diploid_results.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/phasing_clean/f08aad384c913bead6cd6e38f457e223f33ad7ec/docs/synthetic_diploid_results.html" target="_blank">f08aad3</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/be983e0faec864a1001597ab3a3f67edf0f7b67a/analysis/synthetic_diploid_results.Rmd" target="_blank">be983e0</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-22
</td>
<td>
wflow_publish("analysis/synthetic_diploid_results.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/phasing_clean/c8e461cb8432c76ec2ef04e39efbe62fb32c8251/docs/synthetic_diploid_results.html" target="_blank">c8e461c</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/2e92c4cfad59b685df182f0c281312cd2a54e57e/analysis/synthetic_diploid_results.Rmd" target="_blank">2e92c4c</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-22
</td>
<td>
wflow_publish(c("analysis/index.Rmd",
"analysis/synthetic_diploid_results.Rmd"))
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/e3d7c36cb467feb3a86e899ebd79278f56e05641/analysis/synthetic_diploid_results.Rmd" target="_blank">e3d7c36</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-22
</td>
<td>
update 22May24
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/phasing_clean/0e0d17276b9bf7d37f7876f92a57bafb98d9ab15/docs/synthetic_diploid_results.html" target="_blank">0e0d172</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/e8cb325e12eb933ed7686661dcae741bd040aafa/analysis/synthetic_diploid_results.Rmd" target="_blank">e8cb325</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-16
</td>
<td>
wflow_publish("analysis/synthetic_diploid_results.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="libraries-and-directories" class="section level2">
<h2>Libraries and Directories</h2>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.0     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ purrr::%||%()   masks base::%||%()
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
<pre class="r"><code>library(janitor)</code></pre>
<pre><code>
Attaching package: &#39;janitor&#39;

The following objects are masked from &#39;package:stats&#39;:

    chisq.test, fisher.test</code></pre>
<pre class="r"><code>library(reactable)
source(&quot;code/common_functions.R&quot;)
library(yaml)

config_obj &lt;- yaml::read_yaml(&quot;_config.yaml&quot;)

eagle_switch_dir &lt;-   paste0(config_obj$base_dir,&quot;/output/switch_errors/switch_errors/eagle/annotated/&quot;)
shapeit_switch_dir &lt;- paste0(config_obj$base_dir,&quot;/output/switch_errors/switch_errors/shapeit/annotated/&quot;)
beagle_switch_dir &lt;-  paste0(config_obj$base_dir,&quot;/output/switch_errors/switch_errors/beagle/annotated/&quot;)
num_sites_dir &lt;-      paste0(config_obj$base_dir,&quot;/output/switch_errors/vcf_n_sites/&quot;)
het_loc_dir &lt;-        paste0(config_obj$base_dir,&quot;/output/switch_errors/het_loc/&quot;)

sample_info_df &lt;- read_csv(&quot;data/1kgp/subject_info.csv&quot;) %&gt;%
  select(SAMPLE_NAME, POPULATION, SUPER)</code></pre>
<pre><code>Rows: 3202 Columns: 6
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (5): SAMPLE_NAME, POPULATION, SUPER, fatherID, motherID
dbl (1): sex

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>pair_info_df &lt;- read_delim(&quot;data/sample_pairs.csv&quot;, col_names = c(&quot;POP&quot;, &quot;ID1&quot;, &quot;ID2&quot;))</code></pre>
<pre><code>Rows: 1000 Columns: 3
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (3): POP, ID1, ID2

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>pair_info_df &lt;- left_join(pair_info_df, sample_info_df, by = c(&quot;ID1&quot;=&quot;SAMPLE_NAME&quot;)) %&gt;%
  rename(SP = SUPER) %&gt;%
  select(-POPULATION)

# data/chrX_gc1kb_pilot.bed
gc_content_1kb &lt;- read_tsv(&quot;data/chrX_gc1kb_pilot.bed&quot;)</code></pre>
<pre><code>Rows: 156041 Columns: 12
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr  (1): #1_usercol
dbl (11): 2_usercol, 3_usercol, 4_pct_at, 5_pct_gc, 6_num_A, 7_num_C, 8_num_...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>colnames(gc_content_1kb) &lt;- c(&quot;CHR&quot;, &quot;START&quot;, &quot;END&quot;, &quot;AT&quot;, &quot;GC&quot;, &quot;A&quot;, &quot;C&quot;, &quot;G&quot;, &quot;T&quot;, &quot;TOTAL&quot;, &quot;OTHER&quot;, &quot;LENGTH&quot;)
gc_content_1kb  &lt;- gc_content_1kb %&gt;%
  mutate(bin_id = (START / 1000) + 1)

cbPalette &lt;- c(&quot;#999999&quot;, &quot;#E69F00&quot;, &quot;#56B4E9&quot;, &quot;#009E73&quot;, &quot;#F0E442&quot;, &quot;#0072B2&quot;, &quot;#D55E00&quot;, &quot;#CC79A7&quot;, &quot;#000&quot;)

maf_prop_df &lt;- read_csv(paste0(config_obj$base_dir,&quot;/output/switch_errors/maf_props.csv&quot;))</code></pre>
<pre><code>Rows: 39000 Columns: 3
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (1): category
dbl (2): proportion, id

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>pair_info_df$id &lt;- 1:1000
maf_prop_df &lt;- left_join(maf_prop_df, {pair_info_df %&gt;% select(id, SP, POP)})</code></pre>
<pre><code>Joining with `by = join_by(id)`</code></pre>
<pre class="r"><code>pair_info_df &lt;- pair_info_df %&gt;% select(id)
#regen</code></pre>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>In this document, we present summaries and figures for the
distributions of errors in phasing of our synthetic X chromosome
diploids.</p>
</div>
<div id="load-switch-and-flips-errors" class="section level2">
<h2>Load Switch and Flips Errors</h2>
<pre class="r"><code>df_vcftools &lt;- read_csv(paste0(config_obj$base_dir,&quot;/output/switch_errors/switch_errors/summary.csv&quot;))</code></pre>
<pre><code>Rows: 1000 Columns: 45
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr  (2): pop, subpop
dbl (43): pair_id, n_switch_eagle, n_switch_beagle, n_switch_shapeit, n_flip...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div id="tables" class="section level2">
<h2>Tables</h2>
<div id="mean-number-of-switches-and-flips" class="section level3">
<h3>Mean number of switches and flips</h3>
<pre class="r"><code>df_vcftools %&gt;%
  group_by(pop) %&gt;%
  summarize(mean_switch_eagle = mean(n_other_eagle),
            mean_switch_beagle = mean(n_other_beagle),
            mean_switch_shapeit = mean(n_other_shapeit),
            mean_flip_eagle = mean(n_flip_eagle),
            mean_flip_beagle = mean(n_flip_beagle),
            mean_flip_shapeit = mean(n_flip_shapeit)) %&gt;%
  knitr::kable()</code></pre>
<table>
<colgroup>
<col width="3%" />
<col width="16%" />
<col width="16%" />
<col width="17%" />
<col width="14%" />
<col width="15%" />
<col width="16%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">pop</th>
<th align="right">mean_switch_eagle</th>
<th align="right">mean_switch_beagle</th>
<th align="right">mean_switch_shapeit</th>
<th align="right">mean_flip_eagle</th>
<th align="right">mean_flip_beagle</th>
<th align="right">mean_flip_shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AFR</td>
<td align="right">135.285</td>
<td align="right">88.580</td>
<td align="right">76.900</td>
<td align="right">99.790</td>
<td align="right">100.670</td>
<td align="right">119.285</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="right">102.735</td>
<td align="right">73.810</td>
<td align="right">68.585</td>
<td align="right">68.440</td>
<td align="right">69.220</td>
<td align="right">80.545</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="right">197.965</td>
<td align="right">153.140</td>
<td align="right">143.360</td>
<td align="right">105.165</td>
<td align="right">105.800</td>
<td align="right">114.190</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="right">125.060</td>
<td align="right">91.635</td>
<td align="right">84.570</td>
<td align="right">74.620</td>
<td align="right">75.255</td>
<td align="right">86.300</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="right">170.675</td>
<td align="right">129.670</td>
<td align="right">118.150</td>
<td align="right">102.920</td>
<td align="right">100.890</td>
<td align="right">114.135</td>
</tr>
</tbody>
</table>
</div>
<div id="mean-number-of-switches-and-flips-per-mb"
class="section level3">
<h3>Mean number of switches and flips per MB</h3>
<pre class="r"><code>df_vcftools %&gt;%
  group_by(pop) %&gt;%
  summarize(mean_switch_eagle = mean(n_other_eagle / size_mb),
            mean_switch_beagle = mean(n_other_beagle / size_mb),
            mean_switch_shapeit = mean(n_other_shapeit / size_mb),
            mean_flip_eagle = mean(n_flip_eagle / size_mb),
            mean_flip_beagle = mean(n_flip_beagle / size_mb),
            mean_flip_shapeit = mean(n_flip_shapeit / size_mb)) %&gt;%
  knitr::kable()</code></pre>
<table>
<colgroup>
<col width="3%" />
<col width="16%" />
<col width="16%" />
<col width="17%" />
<col width="14%" />
<col width="15%" />
<col width="16%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">pop</th>
<th align="right">mean_switch_eagle</th>
<th align="right">mean_switch_beagle</th>
<th align="right">mean_switch_shapeit</th>
<th align="right">mean_flip_eagle</th>
<th align="right">mean_flip_beagle</th>
<th align="right">mean_flip_shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AFR</td>
<td align="right">0.9031620</td>
<td align="right">0.5913448</td>
<td align="right">0.5133726</td>
<td align="right">0.6662131</td>
<td align="right">0.6720729</td>
<td align="right">0.7963498</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="right">0.6858833</td>
<td align="right">0.4927728</td>
<td align="right">0.4578917</td>
<td align="right">0.4569195</td>
<td align="right">0.4621275</td>
<td align="right">0.5377349</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="right">1.3216428</td>
<td align="right">1.0223829</td>
<td align="right">0.9570942</td>
<td align="right">0.7020961</td>
<td align="right">0.7063356</td>
<td align="right">0.7623499</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="right">0.8348790</td>
<td align="right">0.6117393</td>
<td align="right">0.5645758</td>
<td align="right">0.4981507</td>
<td align="right">0.5023896</td>
<td align="right">0.5761263</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="right">1.1394064</td>
<td align="right">0.8656656</td>
<td align="right">0.7887532</td>
<td align="right">0.6870792</td>
<td align="right">0.6735311</td>
<td align="right">0.7619485</td>
</tr>
</tbody>
</table>
</div>
<div id="mean-number-of-switches-and-flips-per-heterozygous-site"
class="section level3">
<h3>Mean number of switches and flips per heterozygous site</h3>
<pre class="r"><code>df_vcftools %&gt;%
  group_by(pop) %&gt;%
  summarize(mean_switch_eagle = mean(n_other_eagle / n_hets),
            mean_switch_beagle = mean(n_other_beagle / n_hets),
            mean_switch_shapeit = mean(n_other_shapeit / n_hets),
            mean_flip_eagle = mean(n_flip_eagle / n_hets),
            mean_flip_beagle = mean(n_flip_beagle / n_hets),
            mean_flip_shapeit = mean(n_flip_shapeit / n_hets)) %&gt;%
  knitr::kable()</code></pre>
<table>
<colgroup>
<col width="3%" />
<col width="16%" />
<col width="16%" />
<col width="17%" />
<col width="14%" />
<col width="15%" />
<col width="16%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">pop</th>
<th align="right">mean_switch_eagle</th>
<th align="right">mean_switch_beagle</th>
<th align="right">mean_switch_shapeit</th>
<th align="right">mean_flip_eagle</th>
<th align="right">mean_flip_beagle</th>
<th align="right">mean_flip_shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AFR</td>
<td align="right">0.0013558</td>
<td align="right">0.0008880</td>
<td align="right">0.0007707</td>
<td align="right">0.0009991</td>
<td align="right">0.0010081</td>
<td align="right">0.0011945</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="right">0.0016760</td>
<td align="right">0.0012206</td>
<td align="right">0.0011439</td>
<td align="right">0.0010802</td>
<td align="right">0.0011009</td>
<td align="right">0.0012704</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="right">0.0036614</td>
<td align="right">0.0028352</td>
<td align="right">0.0026536</td>
<td align="right">0.0019427</td>
<td align="right">0.0019555</td>
<td align="right">0.0021114</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="right">0.0020636</td>
<td align="right">0.0015108</td>
<td align="right">0.0013951</td>
<td align="right">0.0012304</td>
<td align="right">0.0012423</td>
<td align="right">0.0014230</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="right">0.0026914</td>
<td align="right">0.0020451</td>
<td align="right">0.0018639</td>
<td align="right">0.0016220</td>
<td align="right">0.0015903</td>
<td align="right">0.0017988</td>
</tr>
</tbody>
</table>
</div>
<div id="mean-median-number-of-heterozygous-sites-between-errors"
class="section level3">
<h3>Mean median number of heterozygous sites between errors</h3>
<pre class="r"><code>df_vcftools %&gt;%
  group_by(pop) %&gt;%
  summarize(mean_het_switch_beagle = mean(med_hets_switch_beagle),
            mean_het_switch_eagle = mean(med_hets_switch_eagle),
            mean_het_switch_shapeit = mean(med_hets_switch_shapeit)) %&gt;%
  knitr::kable()</code></pre>
<table>
<colgroup>
<col width="5%" />
<col width="31%" />
<col width="30%" />
<col width="32%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">pop</th>
<th align="right">mean_het_switch_beagle</th>
<th align="right">mean_het_switch_eagle</th>
<th align="right">mean_het_switch_shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AFR</td>
<td align="right">472.1675</td>
<td align="right">363.4250</td>
<td align="right">472.1025</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="right">389.9675</td>
<td align="right">314.2400</td>
<td align="right">370.4475</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="right">122.4150</td>
<td align="right">109.6275</td>
<td align="right">128.7175</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="right">295.7150</td>
<td align="right">236.3000</td>
<td align="right">305.5150</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="right">175.8725</td>
<td align="right">156.7575</td>
<td align="right">185.4350</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df_vcftools %&gt;%
  group_by(pop) %&gt;%
  summarize(mean_het_flip_beagle = mean(med_hets_flip_beagle),
            mean_het_flip_eagle = mean(med_hets_flip_eagle),
            mean_het_flip_shapeit = mean(med_hets_flip_shapeit)) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">pop</th>
<th align="right">mean_het_flip_beagle</th>
<th align="right">mean_het_flip_eagle</th>
<th align="right">mean_het_flip_shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AFR</td>
<td align="right">620.1825</td>
<td align="right">601.1875</td>
<td align="right">505.3600</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="right">572.5475</td>
<td align="right">588.4750</td>
<td align="right">487.8575</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="right">271.3000</td>
<td align="right">290.6925</td>
<td align="right">256.0175</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="right">502.0550</td>
<td align="right">504.4675</td>
<td align="right">431.3800</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="right">359.3325</td>
<td align="right">356.2950</td>
<td align="right">320.4900</td>
</tr>
</tbody>
</table>
<p>Note here that the populations will also differ in regards to the
mean number of heterozygous sites in each pseudo-diploid:</p>
<pre class="r"><code>df_vcftools %&gt;%
  group_by(pop) %&gt;%
  summarize(mean_n_het = mean(n_hets),
            sd_n_het = sd(n_hets)) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">pop</th>
<th align="right">mean_n_het</th>
<th align="right">sd_n_het</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AFR</td>
<td align="right">99817.32</td>
<td align="right">3346.941</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="right">66309.95</td>
<td align="right">12057.033</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="right">54261.28</td>
<td align="right">2590.510</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="right">60595.20</td>
<td align="right">3056.270</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="right">63663.90</td>
<td align="right">2690.040</td>
</tr>
</tbody>
</table>
</div>
<div id="mean-mean-number-of-heterozygous-sites-between-errors"
class="section level3">
<h3>Mean mean number of heterozygous sites between errors</h3>
<pre class="r"><code>df_vcftools %&gt;%
  group_by(pop) %&gt;%
  summarize(mean_het_switch_beagle = mean(mean_hets_switch_beagle),
            mean_het_switch_eagle = mean(mean_hets_switch_eagle),
            mean_het_switch_shapeit = mean(mean_hets_switch_shapeit)) %&gt;%
  knitr::kable()</code></pre>
<table>
<colgroup>
<col width="5%" />
<col width="31%" />
<col width="30%" />
<col width="32%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">pop</th>
<th align="right">mean_het_switch_beagle</th>
<th align="right">mean_het_switch_eagle</th>
<th align="right">mean_het_switch_shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AFR</td>
<td align="right">1232.3628</td>
<td align="right">786.2282</td>
<td align="right">1433.5878</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="right">1230.0232</td>
<td align="right">779.9772</td>
<td align="right">1396.1901</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="right">361.4670</td>
<td align="right">277.4508</td>
<td align="right">386.9342</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="right">741.0225</td>
<td align="right">526.7262</td>
<td align="right">821.1534</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="right">545.7064</td>
<td align="right">401.1355</td>
<td align="right">619.7363</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df_vcftools %&gt;%
  group_by(pop) %&gt;%
  summarize(mean_het_flip_beagle = mean(mean_hets_flip_beagle),
            mean_het_flip_eagle = mean(mean_hets_flip_eagle),
            mean_het_flip_shapeit = mean(mean_hets_flip_shapeit)) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">pop</th>
<th align="right">mean_het_flip_beagle</th>
<th align="right">mean_het_flip_eagle</th>
<th align="right">mean_het_flip_shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AFR</td>
<td align="right">1052.3105</td>
<td align="right">1059.8692</td>
<td align="right">879.6391</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="right">1083.4175</td>
<td align="right">1048.3398</td>
<td align="right">908.3891</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="right">519.1688</td>
<td align="right">520.3449</td>
<td align="right">482.4776</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="right">861.5598</td>
<td align="right">844.5954</td>
<td align="right">753.6581</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="right">675.3881</td>
<td align="right">644.3616</td>
<td align="right">597.1906</td>
</tr>
</tbody>
</table>
</div>
<div id="errors-at-cpgs" class="section level3">
<h3>Errors at CpGs</h3>
<div id="total" class="section level4">
<h4>Total</h4>
<pre class="r"><code>df_vcftools %&gt;%
  summarize(mean_switch_eagle = mean(n_other_cpg_eagle),
            mean_switch_beagle = mean(n_other_cpg_beagle),
            mean_switch_shapeit = mean(n_other_cpg_shapeit),
            mean_flip_eagle = mean(n_flip_cpg_eagle),
            mean_flip_beagle = mean(n_flip_cpg_beagle),
            mean_flip_shapeit = mean(n_flip_cpg_shapeit)) %&gt;%
  knitr::kable()</code></pre>
<table style="width:100%;">
<colgroup>
<col width="16%" />
<col width="17%" />
<col width="18%" />
<col width="14%" />
<col width="15%" />
<col width="16%" />
</colgroup>
<thead>
<tr class="header">
<th align="right">mean_switch_eagle</th>
<th align="right">mean_switch_beagle</th>
<th align="right">mean_switch_shapeit</th>
<th align="right">mean_flip_eagle</th>
<th align="right">mean_flip_beagle</th>
<th align="right">mean_flip_shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">28.049</td>
<td align="right">17.31</td>
<td align="right">15.32</td>
<td align="right">27.184</td>
<td align="right">30.642</td>
<td align="right">34.714</td>
</tr>
</tbody>
</table>
<div id="by-population" class="section level5">
<h5>By Population</h5>
<pre class="r"><code>df_vcftools %&gt;%
  group_by(pop) %&gt;%
  summarize(mean_switch_eagle = mean(n_other_cpg_eagle),
            mean_switch_beagle = mean(n_other_cpg_beagle),
            mean_switch_shapeit = mean(n_other_cpg_shapeit),
            mean_flip_eagle = mean(n_flip_cpg_eagle),
            mean_flip_beagle = mean(n_flip_cpg_beagle),
            mean_flip_shapeit = mean(n_flip_cpg_shapeit)) %&gt;%
  knitr::kable()</code></pre>
<table>
<colgroup>
<col width="3%" />
<col width="16%" />
<col width="16%" />
<col width="17%" />
<col width="14%" />
<col width="15%" />
<col width="16%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">pop</th>
<th align="right">mean_switch_eagle</th>
<th align="right">mean_switch_beagle</th>
<th align="right">mean_switch_shapeit</th>
<th align="right">mean_flip_eagle</th>
<th align="right">mean_flip_beagle</th>
<th align="right">mean_flip_shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AFR</td>
<td align="right">26.220</td>
<td align="right">13.170</td>
<td align="right">10.895</td>
<td align="right">28.15</td>
<td align="right">32.760</td>
<td align="right">37.820</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="right">21.065</td>
<td align="right">12.195</td>
<td align="right">11.120</td>
<td align="right">22.79</td>
<td align="right">25.820</td>
<td align="right">30.035</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="right">37.305</td>
<td align="right">25.210</td>
<td align="right">23.090</td>
<td align="right">30.33</td>
<td align="right">34.840</td>
<td align="right">37.910</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="right">23.030</td>
<td align="right">14.740</td>
<td align="right">12.620</td>
<td align="right">22.81</td>
<td align="right">26.395</td>
<td align="right">29.305</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="right">32.625</td>
<td align="right">21.235</td>
<td align="right">18.875</td>
<td align="right">31.84</td>
<td align="right">33.395</td>
<td align="right">38.500</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="proportions" class="section level4">
<h4>Proportions</h4>
<pre class="r"><code>df_vcftools %&gt;%
  summarize(mean_switch_eagle = mean(n_other_cpg_eagle / n_other_eagle),
            mean_switch_beagle = mean(n_other_cpg_beagle / n_other_beagle),
            mean_switch_shapeit = mean(n_other_cpg_shapeit / n_other_shapeit),
            mean_flip_eagle = mean(n_flip_cpg_eagle / n_flip_eagle),
            mean_flip_beagle = mean(n_flip_cpg_beagle / n_flip_beagle),
            mean_flip_shapeit = mean(n_flip_cpg_shapeit / n_flip_shapeit)) %&gt;%
  knitr::kable()</code></pre>
<table style="width:100%;">
<colgroup>
<col width="16%" />
<col width="17%" />
<col width="18%" />
<col width="14%" />
<col width="15%" />
<col width="16%" />
</colgroup>
<thead>
<tr class="header">
<th align="right">mean_switch_eagle</th>
<th align="right">mean_switch_beagle</th>
<th align="right">mean_switch_shapeit</th>
<th align="right">mean_flip_eagle</th>
<th align="right">mean_flip_beagle</th>
<th align="right">mean_flip_shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">0.1930164</td>
<td align="right">0.1605653</td>
<td align="right">0.1549997</td>
<td align="right">0.3019655</td>
<td align="right">0.3402046</td>
<td align="right">0.3389541</td>
</tr>
</tbody>
</table>
<div id="by-population-1" class="section level5">
<h5>By Population</h5>
<pre class="r"><code>df_vcftools %&gt;%
  group_by(pop) %&gt;%
  summarize(mean_switch_eagle = mean(n_other_cpg_eagle / n_other_eagle),
            mean_switch_beagle = mean(n_other_cpg_beagle / n_other_beagle),
            mean_switch_shapeit = mean(n_other_cpg_shapeit / n_other_shapeit),
            mean_flip_eagle = mean(n_flip_cpg_eagle / n_flip_eagle),
            mean_flip_beagle = mean(n_flip_cpg_beagle / n_flip_beagle),
            mean_flip_shapeit = mean(n_flip_cpg_shapeit / n_flip_shapeit)) %&gt;%
  knitr::kable()</code></pre>
<table>
<colgroup>
<col width="3%" />
<col width="16%" />
<col width="16%" />
<col width="17%" />
<col width="14%" />
<col width="15%" />
<col width="16%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">pop</th>
<th align="right">mean_switch_eagle</th>
<th align="right">mean_switch_beagle</th>
<th align="right">mean_switch_shapeit</th>
<th align="right">mean_flip_eagle</th>
<th align="right">mean_flip_beagle</th>
<th align="right">mean_flip_shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AFR</td>
<td align="right">0.1941425</td>
<td align="right">0.1479915</td>
<td align="right">0.1425189</td>
<td align="right">0.2802358</td>
<td align="right">0.3240284</td>
<td align="right">0.3162838</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="right">0.2058297</td>
<td align="right">0.1634704</td>
<td align="right">0.1595254</td>
<td align="right">0.3289779</td>
<td align="right">0.3702323</td>
<td align="right">0.3730968</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="right">0.1886936</td>
<td align="right">0.1643634</td>
<td align="right">0.1610919</td>
<td align="right">0.2887750</td>
<td align="right">0.3290075</td>
<td align="right">0.3313922</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="right">0.1844417</td>
<td align="right">0.1617823</td>
<td align="right">0.1507598</td>
<td align="right">0.3029541</td>
<td align="right">0.3475032</td>
<td align="right">0.3375451</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="right">0.1919745</td>
<td align="right">0.1652189</td>
<td align="right">0.1611023</td>
<td align="right">0.3088848</td>
<td align="right">0.3302515</td>
<td align="right">0.3364526</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="enrichment" class="section level4">
<h4>Enrichment</h4>
<p>We compute the ratio of the proportion of errors at CpGs to the
proportion of heterozygous locations that are at CpGs.</p>
<pre class="r"><code>df_vcftools %&gt;%
  summarize(mean_switch_eagle = mean((n_other_cpg_eagle / n_other_eagle) / ( n_het_cpg / n_hets)),
            mean_switch_beagle = mean((n_other_cpg_beagle / n_other_beagle) / ( n_het_cpg / n_hets)),
            mean_switch_shapeit = mean((n_other_cpg_shapeit / n_other_shapeit) / ( n_het_cpg / n_hets) ),
            mean_flip_eagle = mean((n_flip_cpg_eagle / n_flip_eagle) / ( n_het_cpg / n_hets)),
            mean_flip_beagle = mean((n_flip_cpg_beagle / n_flip_beagle) / ( n_het_cpg / n_hets)),
            mean_flip_shapeit = mean((n_flip_cpg_shapeit / n_flip_shapeit) / ( n_het_cpg / n_hets))) %&gt;%
  knitr::kable()</code></pre>
<table style="width:100%;">
<colgroup>
<col width="16%" />
<col width="17%" />
<col width="18%" />
<col width="14%" />
<col width="15%" />
<col width="16%" />
</colgroup>
<thead>
<tr class="header">
<th align="right">mean_switch_eagle</th>
<th align="right">mean_switch_beagle</th>
<th align="right">mean_switch_shapeit</th>
<th align="right">mean_flip_eagle</th>
<th align="right">mean_flip_beagle</th>
<th align="right">mean_flip_shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">1.355444</td>
<td align="right">1.127716</td>
<td align="right">1.088649</td>
<td align="right">2.121156</td>
<td align="right">2.389601</td>
<td align="right">2.381002</td>
</tr>
</tbody>
</table>
<p>Alternative view: odds ratios</p>
<pre class="r"><code>df_vcftools %&gt;%
  mutate( prop_cpg_beagle =  n_other_cpg_beagle / n_other_beagle,
             prop_cpg_eagle =  n_other_cpg_eagle / n_other_eagle,
             prop_cpg_shapeit =  n_other_cpg_shapeit / n_other_shapeit) %&gt;%
  mutate(ods_beagle = prop_cpg_beagle / (1 - prop_cpg_beagle),
         ods_eagle = prop_cpg_eagle / (1 - prop_cpg_eagle),
         ods_shapeit = prop_cpg_shapeit / (1 - prop_cpg_shapeit),
         ods_het = ( n_het_cpg / n_hets) / (1 - ( n_het_cpg / n_hets))) %&gt;%
  mutate(Beagle = ods_beagle / ods_het,
         Eagle = ods_eagle / ods_het,
         SHAPEIT = ods_shapeit / ods_het) %&gt;%
  select(pop, Beagle, Eagle, SHAPEIT) %&gt;%
  pivot_longer(-pop, names_to = &quot;Method&quot;, values_to = &quot;OR&quot;) %&gt;%
  ggplot(aes(x = Method, y = OR, colour = pop)) +
  geom_boxplot() +
  ylab(&quot;Odds CpG Switch / Odds CpG Het&quot;) +
  ggtitle(&quot;Odds Ratio of CpG Status for Switches&quot;) +
  theme_classic() +
  labs(colour = &quot;Population&quot;) +
  scale_color_manual(values = cbPalette) +
  geom_hline(yintercept = 1, linetype=2)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/7f7e2da6f25f457b0fe9eb8b6b6b22b03f3be64e/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-16-1.png" target="_blank">7f7e2da</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/f08aad384c913bead6cd6e38f457e223f33ad7ec/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-16-1.png" target="_blank">f08aad3</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/0e0d17276b9bf7d37f7876f92a57bafb98d9ab15/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-16-1.png" target="_blank">0e0d172</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_vcftools %&gt;%
  mutate( prop_cpg_beagle =  n_other_cpg_beagle / n_other_beagle,
             prop_cpg_eagle =  n_other_cpg_eagle / n_other_eagle,
             prop_cpg_shapeit =  n_other_cpg_shapeit / n_other_shapeit) %&gt;%
  mutate(ods_beagle = prop_cpg_beagle / (1 - prop_cpg_beagle),
         ods_eagle = prop_cpg_eagle / (1 - prop_cpg_eagle),
         ods_shapeit = prop_cpg_shapeit / (1 - prop_cpg_shapeit),
         ods_het = ( n_het_cpg / n_hets) / (1 - ( n_het_cpg / n_hets))) %&gt;%
  mutate(Beagle = ods_beagle / ods_het,
         Eagle = ods_eagle / ods_het,
         SHAPEIT = ods_shapeit / ods_het) %&gt;%
  select(pop, Beagle, Eagle, SHAPEIT) %&gt;%
  pivot_longer(-pop, names_to = &quot;Method&quot;, values_to = &quot;OR&quot;) %&gt;%
  ggplot(aes(x = Method, y = OR)) +
  geom_boxplot() +
  ylab(&quot;Odds CpG Switch / Odds CpG Het&quot;) +
  ggtitle(&quot;Odds Ratio of CpG Status for Switches&quot;) +
  theme_classic() +
  scale_color_manual(values = cbPalette) +
  geom_hline(yintercept = 1, linetype=2)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-16-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>And flips:</p>
<pre class="r"><code>df_vcftools %&gt;%
  mutate( prop_cpg_beagle =  n_flip_cpg_beagle / n_flip_beagle,
             prop_cpg_eagle =  n_flip_cpg_eagle / n_flip_eagle,
             prop_cpg_shapeit =  n_flip_cpg_shapeit / n_flip_shapeit) %&gt;%
  mutate(ods_beagle = prop_cpg_beagle / (1 - prop_cpg_beagle),
         ods_eagle = prop_cpg_eagle / (1 - prop_cpg_eagle),
         ods_shapeit = prop_cpg_shapeit / (1 - prop_cpg_shapeit),
         ods_het = ( n_het_cpg / n_hets) / (1 - ( n_het_cpg / n_hets))) %&gt;%
  mutate(Beagle = ods_beagle / ods_het,
         Eagle = ods_eagle / ods_het,
         SHAPEIT = ods_shapeit / ods_het) %&gt;%
  select(pop, Beagle, Eagle, SHAPEIT) %&gt;%
  pivot_longer(-pop, names_to = &quot;Method&quot;, values_to = &quot;OR&quot;) %&gt;%
  ggplot(aes(x = Method, y = OR, colour = pop)) +
  geom_boxplot() +
  ylab(&quot;Odds CpG Switch / Odds CpG Het&quot;) +
  ggtitle(&quot;Odds Ratio of CpG Status for Switches&quot;) +
  theme_classic() +
  labs(colour = &quot;Population&quot;) +
  scale_color_manual(values = cbPalette) +
  geom_hline(yintercept = 1, linetype=2)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/7f7e2da6f25f457b0fe9eb8b6b6b22b03f3be64e/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-17-1.png" target="_blank">7f7e2da</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/f08aad384c913bead6cd6e38f457e223f33ad7ec/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-17-1.png" target="_blank">f08aad3</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/0e0d17276b9bf7d37f7876f92a57bafb98d9ab15/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-17-1.png" target="_blank">0e0d172</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_vcftools %&gt;%
  mutate( prop_cpg_beagle =  n_other_cpg_beagle / n_other_beagle,
             prop_cpg_eagle =  n_other_cpg_eagle / n_other_eagle,
             prop_cpg_shapeit =  n_other_cpg_shapeit / n_other_shapeit) %&gt;%
  mutate(ods_beagle = prop_cpg_beagle / (1 - prop_cpg_beagle),
         ods_eagle = prop_cpg_eagle / (1 - prop_cpg_eagle),
         ods_shapeit = prop_cpg_shapeit / (1 - prop_cpg_shapeit),
         ods_het = ( n_het_cpg / n_hets) / (1 - ( n_het_cpg / n_hets))) %&gt;%
  mutate(Beagle = ods_beagle / ods_het,
         Eagle = ods_eagle / ods_het,
         SHAPEIT = ods_shapeit / ods_het) %&gt;%
  select(pop, Beagle, Eagle, SHAPEIT) %&gt;%
  pivot_longer(-pop, names_to = &quot;Method&quot;, values_to = &quot;OR&quot;) %&gt;%
  ggplot(aes(x = Method, y = OR)) +
  geom_boxplot() +
  ylab(&quot;Odds CpG Switch / Odds CpG Het&quot;) +
  ggtitle(&quot;Odds Ratio of CpG Status for Switches&quot;) +
  theme_classic() +
  scale_color_manual(values = cbPalette) +
  geom_hline(yintercept = 1, linetype=2)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-17-2.png" width="672" style="display: block; margin: auto;" /></p>
<div id="by-population-2" class="section level5">
<h5>By Population</h5>
<pre class="r"><code>df_vcftools %&gt;%
  group_by(pop) %&gt;%
  summarize(mean_switch_eagle = mean((n_other_cpg_eagle / n_other_eagle) / ( n_het_cpg / n_hets)),
            mean_switch_beagle = mean((n_other_cpg_beagle / n_other_beagle) / ( n_het_cpg / n_hets)),
            mean_switch_shapeit = mean((n_other_cpg_shapeit / n_other_shapeit) / ( n_het_cpg / n_hets) ),
            mean_flip_eagle = mean((n_flip_cpg_eagle / n_flip_eagle) / ( n_het_cpg / n_hets)),
            mean_flip_beagle = mean((n_flip_cpg_beagle / n_flip_beagle) / ( n_het_cpg / n_hets)),
            mean_flip_shapeit = mean((n_flip_cpg_shapeit / n_flip_shapeit) / ( n_het_cpg / n_hets))) %&gt;%
  knitr::kable()</code></pre>
<table>
<colgroup>
<col width="3%" />
<col width="16%" />
<col width="16%" />
<col width="17%" />
<col width="14%" />
<col width="15%" />
<col width="16%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">pop</th>
<th align="right">mean_switch_eagle</th>
<th align="right">mean_switch_beagle</th>
<th align="right">mean_switch_shapeit</th>
<th align="right">mean_flip_eagle</th>
<th align="right">mean_flip_beagle</th>
<th align="right">mean_flip_shapeit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AFR</td>
<td align="right">1.352531</td>
<td align="right">1.030998</td>
<td align="right">0.9928695</td>
<td align="right">1.952522</td>
<td align="right">2.257366</td>
<td align="right">2.203883</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="right">1.452104</td>
<td align="right">1.153275</td>
<td align="right">1.1254495</td>
<td align="right">2.321056</td>
<td align="right">2.612145</td>
<td align="right">2.632413</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="right">1.320305</td>
<td align="right">1.150108</td>
<td align="right">1.1272621</td>
<td align="right">2.020877</td>
<td align="right">2.302090</td>
<td align="right">2.319491</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="right">1.303263</td>
<td align="right">1.142911</td>
<td align="right">1.0656169</td>
<td align="right">2.140890</td>
<td align="right">2.455553</td>
<td align="right">2.385270</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="right">1.349017</td>
<td align="right">1.161286</td>
<td align="right">1.1320457</td>
<td align="right">2.170436</td>
<td align="right">2.320850</td>
<td align="right">2.363954</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="figures" class="section level2">
<h2>Figures</h2>
<div id="total-errors" class="section level3">
<h3>Total Errors</h3>
<pre class="r"><code>df_vcftools %&gt;%
  mutate(Beagle = n_other_beagle + n_flip_beagle,
         Eagle = n_other_eagle + n_flip_eagle, 
         SHAPEIT = n_other_shapeit + n_flip_shapeit) %&gt;%
  ggplot(aes(x = Beagle, y = Eagle, color = pop)) +
  geom_point() +
  ggtitle(&quot;Total Errors&quot;) +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette) +
  labs(color = &quot;Population&quot;)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-1">
Past versions of unnamed-chunk-19-1.png
</button>
</p>
<div id="fig-unnamed-chunk-19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/7f7e2da6f25f457b0fe9eb8b6b6b22b03f3be64e/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-19-1.png" target="_blank">7f7e2da</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/f08aad384c913bead6cd6e38f457e223f33ad7ec/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-19-1.png" target="_blank">f08aad3</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/0e0d17276b9bf7d37f7876f92a57bafb98d9ab15/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-19-1.png" target="_blank">0e0d172</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_vcftools %&gt;%
  mutate(Beagle = n_other_beagle + n_flip_beagle,
         Eagle = n_other_eagle + n_flip_eagle, 
         SHAPEIT = n_other_shapeit + n_flip_shapeit) %&gt;%
  ggplot(aes(x = Beagle, y = SHAPEIT, color = pop)) +
  geom_point() +
  ggtitle(&quot;Total Errors&quot;) +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/7f7e2da6f25f457b0fe9eb8b6b6b22b03f3be64e/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-20-1.png" target="_blank">7f7e2da</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/f08aad384c913bead6cd6e38f457e223f33ad7ec/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-20-1.png" target="_blank">f08aad3</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/0e0d17276b9bf7d37f7876f92a57bafb98d9ab15/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-20-1.png" target="_blank">0e0d172</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_vcftools %&gt;%
  mutate(Beagle = n_other_beagle + n_flip_beagle,
         Eagle = n_other_eagle + n_flip_eagle, 
         SHAPEIT = n_other_shapeit + n_flip_shapeit) %&gt;%
  ggplot(aes(x = Eagle, y = SHAPEIT, color = pop)) +
  geom_point() +
  ggtitle(&quot;Total Errors&quot;) +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/7f7e2da6f25f457b0fe9eb8b6b6b22b03f3be64e/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-21-1.png" target="_blank">7f7e2da</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/f08aad384c913bead6cd6e38f457e223f33ad7ec/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-21-1.png" target="_blank">f08aad3</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/0e0d17276b9bf7d37f7876f92a57bafb98d9ab15/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-21-1.png" target="_blank">0e0d172</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="error-rates-per-heterozygous-site" class="section level4">
<h4>Error rates (per heterozygous site)</h4>
<pre class="r"><code>df_vcftools %&gt;%
  mutate(Beagle = (n_other_beagle + n_flip_beagle) / n_hets,
         Eagle = (n_other_eagle + n_flip_eagle) / n_hets, 
         SHAPEIT = (n_other_shapeit + n_flip_shapeit) / n_hets) %&gt;%
  ggplot(aes(x = Beagle, y = Eagle, color = pop)) +
  geom_point() +
  ggtitle(&quot;Total Errors per Heterozygous Site&quot;) +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-22-1">
Past versions of unnamed-chunk-22-1.png
</button>
</p>
<div id="fig-unnamed-chunk-22-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/7f7e2da6f25f457b0fe9eb8b6b6b22b03f3be64e/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-22-1.png" target="_blank">7f7e2da</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/f08aad384c913bead6cd6e38f457e223f33ad7ec/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-22-1.png" target="_blank">f08aad3</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/0e0d17276b9bf7d37f7876f92a57bafb98d9ab15/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-22-1.png" target="_blank">0e0d172</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_vcftools %&gt;%
  mutate(Beagle = (n_other_beagle + n_flip_beagle) / n_hets,
         Eagle = (n_other_eagle + n_flip_eagle) / n_hets, 
         SHAPEIT = (n_other_shapeit + n_flip_shapeit) / n_hets) %&gt;%
  ggplot(aes(x = Beagle, y = SHAPEIT, color = pop)) +
  geom_point() +
  ggtitle(&quot;Total Errors per Heterozygous Site&quot;) +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-23-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-1">
Past versions of unnamed-chunk-23-1.png
</button>
</p>
<div id="fig-unnamed-chunk-23-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/7f7e2da6f25f457b0fe9eb8b6b6b22b03f3be64e/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-23-1.png" target="_blank">7f7e2da</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/f08aad384c913bead6cd6e38f457e223f33ad7ec/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-23-1.png" target="_blank">f08aad3</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/0e0d17276b9bf7d37f7876f92a57bafb98d9ab15/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-23-1.png" target="_blank">0e0d172</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_vcftools %&gt;%
  mutate(Beagle = (n_other_beagle + n_flip_beagle) / n_hets,
         Eagle = (n_other_eagle + n_flip_eagle) / n_hets, 
         SHAPEIT = (n_other_shapeit + n_flip_shapeit) / n_hets) %&gt;%
  ggplot(aes(x = Eagle, y = SHAPEIT, color = pop)) +
  geom_point() +
  ggtitle(&quot;Total Errors per Heterozygous Site&quot;) +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-1">
Past versions of unnamed-chunk-24-1.png
</button>
</p>
<div id="fig-unnamed-chunk-24-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/7f7e2da6f25f457b0fe9eb8b6b6b22b03f3be64e/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-24-1.png" target="_blank">7f7e2da</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/f08aad384c913bead6cd6e38f457e223f33ad7ec/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-24-1.png" target="_blank">f08aad3</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/0e0d17276b9bf7d37f7876f92a57bafb98d9ab15/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-24-1.png" target="_blank">0e0d172</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="non-flip-switches" class="section level3">
<h3>Non-flip switches</h3>
<pre class="r"><code>df_vcftools %&gt;%
  mutate(Beagle = (n_other_beagle),
         Eagle = (n_other_eagle ), 
         SHAPEIT = (n_other_shapeit)) %&gt;%
  ggplot(aes(x = Beagle, y = Eagle, color = pop)) +
  geom_point() +
  ggtitle(&quot;Switches&quot;) +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-25-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-1">
Past versions of unnamed-chunk-25-1.png
</button>
</p>
<div id="fig-unnamed-chunk-25-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/7f7e2da6f25f457b0fe9eb8b6b6b22b03f3be64e/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-25-1.png" target="_blank">7f7e2da</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/f08aad384c913bead6cd6e38f457e223f33ad7ec/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-25-1.png" target="_blank">f08aad3</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/0e0d17276b9bf7d37f7876f92a57bafb98d9ab15/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-25-1.png" target="_blank">0e0d172</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_vcftools %&gt;%
   mutate(Beagle = (n_other_beagle),
         Eagle = (n_other_eagle ), 
         SHAPEIT = (n_other_shapeit)) %&gt;%
  ggplot(aes(x = Beagle, y = SHAPEIT, color = pop)) +
  geom_point() +
  ggtitle(&quot;Switches&quot;) +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-26-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-26-1">
Past versions of unnamed-chunk-26-1.png
</button>
</p>
<div id="fig-unnamed-chunk-26-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/7f7e2da6f25f457b0fe9eb8b6b6b22b03f3be64e/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-26-1.png" target="_blank">7f7e2da</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/f08aad384c913bead6cd6e38f457e223f33ad7ec/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-26-1.png" target="_blank">f08aad3</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/0e0d17276b9bf7d37f7876f92a57bafb98d9ab15/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-26-1.png" target="_blank">0e0d172</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_vcftools %&gt;%
   mutate(Beagle = (n_other_beagle),
         Eagle = (n_other_eagle ), 
         SHAPEIT = (n_other_shapeit)) %&gt;%
  ggplot(aes(x = Eagle, y = SHAPEIT, color = pop)) +
  geom_point() +
  ggtitle(&quot;Switches&quot;) +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-27-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-27-1">
Past versions of unnamed-chunk-27-1.png
</button>
</p>
<div id="fig-unnamed-chunk-27-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/7f7e2da6f25f457b0fe9eb8b6b6b22b03f3be64e/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-27-1.png" target="_blank">7f7e2da</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/f08aad384c913bead6cd6e38f457e223f33ad7ec/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-27-1.png" target="_blank">f08aad3</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/0e0d17276b9bf7d37f7876f92a57bafb98d9ab15/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-27-1.png" target="_blank">0e0d172</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="per-het.-site" class="section level4">
<h4>Per het. site</h4>
<pre class="r"><code>df_vcftools %&gt;%
  mutate(Beagle = (n_other_beagle) / n_hets,
         Eagle = (n_other_eagle ) / n_hets, 
         SHAPEIT = (n_other_shapeit) / n_hets) %&gt;%
  ggplot(aes(x = Beagle, y = Eagle, color = pop)) +
  geom_point() +
  ggtitle(&quot;Switches per Heterozygous Site&quot;) +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-28-1">
Past versions of unnamed-chunk-28-1.png
</button>
</p>
<div id="fig-unnamed-chunk-28-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/7f7e2da6f25f457b0fe9eb8b6b6b22b03f3be64e/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-28-1.png" target="_blank">7f7e2da</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/f08aad384c913bead6cd6e38f457e223f33ad7ec/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-28-1.png" target="_blank">f08aad3</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_vcftools %&gt;%
  mutate(Beagle = (n_other_beagle) / n_hets,
         Eagle = (n_other_eagle ) / n_hets, 
         SHAPEIT = (n_other_shapeit) / n_hets) %&gt;%
  ggplot(aes(x = Beagle, y = SHAPEIT, color = pop)) +
  geom_point() +
  ggtitle(&quot;Switches per Heterozygous Site&quot;) +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-29-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-29-1">
Past versions of unnamed-chunk-29-1.png
</button>
</p>
<div id="fig-unnamed-chunk-29-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/7f7e2da6f25f457b0fe9eb8b6b6b22b03f3be64e/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-29-1.png" target="_blank">7f7e2da</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_vcftools %&gt;%
  mutate(Beagle = (n_other_beagle) / n_hets,
         Eagle = (n_other_eagle ) / n_hets, 
         SHAPEIT = (n_other_shapeit) / n_hets) %&gt;%
  ggplot(aes(x = Eagle, y = SHAPEIT, color = pop)) +
  geom_point() +
  ggtitle(&quot;Switches per Heterozygous Site&quot;) +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-30-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-30-1">
Past versions of unnamed-chunk-30-1.png
</button>
</p>
<div id="fig-unnamed-chunk-30-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/7f7e2da6f25f457b0fe9eb8b6b6b22b03f3be64e/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-30-1.png" target="_blank">7f7e2da</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="flips" class="section level3">
<h3>Flips</h3>
<pre class="r"><code>df_vcftools %&gt;%
  mutate(Beagle = (n_flip_beagle),
         Eagle = (n_flip_eagle ), 
         SHAPEIT = (n_flip_shapeit)) %&gt;%
  ggplot(aes(x = Beagle, y = Eagle, color = pop)) +
  geom_point() +
  ggtitle(&quot;Flips&quot;) +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-31-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-31-1">
Past versions of unnamed-chunk-31-1.png
</button>
</p>
<div id="fig-unnamed-chunk-31-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/7f7e2da6f25f457b0fe9eb8b6b6b22b03f3be64e/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-31-1.png" target="_blank">7f7e2da</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_vcftools %&gt;%
   mutate(Beagle = (n_flip_beagle),
         Eagle = (n_flip_eagle ), 
         SHAPEIT = (n_flip_shapeit)) %&gt;%
  ggplot(aes(x = Beagle, y = SHAPEIT, color = pop)) +
  geom_point() +
  ggtitle(&quot;Flips&quot;) +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-32-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-32-1">
Past versions of unnamed-chunk-32-1.png
</button>
</p>
<div id="fig-unnamed-chunk-32-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/af46d78ebdfe3df4a908c936e9fd9cb3c6a24173/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-32-1.png" target="_blank">af46d78</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_vcftools %&gt;%
   mutate(Beagle = (n_flip_beagle),
         Eagle = (n_flip_eagle ), 
         SHAPEIT = (n_flip_shapeit)) %&gt;%
  ggplot(aes(x = Eagle, y = SHAPEIT, color = pop)) +
  geom_point() +
  ggtitle(&quot;Flips&quot;) +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-33-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-33-1">
Past versions of unnamed-chunk-33-1.png
</button>
</p>
<div id="fig-unnamed-chunk-33-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/8041a90f4735210b3ffd2e9a1231fd4b9735a442/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-33-1.png" target="_blank">8041a90</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/aa8704bbf7c47c6b3b4c265552120bc446fa974a/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-33-1.png" target="_blank">aa8704b</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="per-het.-site-1" class="section level4">
<h4>Per het. site</h4>
<pre class="r"><code>df_vcftools %&gt;%
  mutate(Beagle = (n_flip_beagle) / n_hets,
         Eagle = (n_flip_eagle ) / n_hets, 
         SHAPEIT = (n_flip_shapeit) / n_hets) %&gt;%
  ggplot(aes(x = Beagle, y = Eagle, color = pop)) +
  geom_point() +
  ggtitle(&quot;Flips per Heterozygous Site&quot;) +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-34-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-34-1">
Past versions of unnamed-chunk-34-1.png
</button>
</p>
<div id="fig-unnamed-chunk-34-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/6172512f57cc03f295aefc7978f0e9f6439169c9/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-34-1.png" target="_blank">6172512</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/8041a90f4735210b3ffd2e9a1231fd4b9735a442/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-34-1.png" target="_blank">8041a90</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/aa8704bbf7c47c6b3b4c265552120bc446fa974a/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-34-1.png" target="_blank">aa8704b</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_vcftools %&gt;%
  mutate(Beagle = (n_flip_beagle) / n_hets,
         Eagle = (n_flip_eagle ) / n_hets, 
         SHAPEIT = (n_flip_shapeit) / n_hets) %&gt;%
  ggplot(aes(x = Beagle, y = SHAPEIT, color = pop)) +
  geom_point() +
  ggtitle(&quot;Flips per Heterozygous Site&quot;) +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-35-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-35-1">
Past versions of unnamed-chunk-35-1.png
</button>
</p>
<div id="fig-unnamed-chunk-35-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/6172512f57cc03f295aefc7978f0e9f6439169c9/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-35-1.png" target="_blank">6172512</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/8041a90f4735210b3ffd2e9a1231fd4b9735a442/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-35-1.png" target="_blank">8041a90</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/aa8704bbf7c47c6b3b4c265552120bc446fa974a/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-35-1.png" target="_blank">aa8704b</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_vcftools %&gt;%
  mutate(Beagle = (n_flip_beagle) / n_hets,
         Eagle = (n_flip_eagle ) / n_hets, 
         SHAPEIT = (n_flip_shapeit) / n_hets) %&gt;%
  ggplot(aes(x = Eagle, y = SHAPEIT, color = pop)) +
  geom_point() +
  ggtitle(&quot;Flips per Heterozygous Site&quot;) +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-36-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-36-1">
Past versions of unnamed-chunk-36-1.png
</button>
</p>
<div id="fig-unnamed-chunk-36-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/6172512f57cc03f295aefc7978f0e9f6439169c9/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-36-1.png" target="_blank">6172512</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/8041a90f4735210b3ffd2e9a1231fd4b9735a442/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-36-1.png" target="_blank">8041a90</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/aa8704bbf7c47c6b3b4c265552120bc446fa974a/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-36-1.png" target="_blank">aa8704b</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="proportion-of-errors-at-cpg" class="section level3">
<h3>Proportion of errors at CpG</h3>
<pre class="r"><code>df_vcftools %&gt;%
  mutate(Beagle = (n_other_cpg_beagle / n_other_beagle) / (n_het_cpg / n_hets),
         Eagle = (n_other_cpg_eagle / n_other_eagle) / (n_het_cpg / n_hets),
         SHAPEIT = (n_other_cpg_shapeit / n_other_shapeit) / (n_het_cpg / n_hets)) %&gt;%
  select(Beagle, Eagle, SHAPEIT) %&gt;%
  pivot_longer(everything(), names_to = &quot;method&quot;, values_to = &quot;enrich&quot;) %&gt;%
  ggplot(aes(x = method, y = enrich)) +
  geom_boxplot() +
  ggtitle(&quot;Enrichment of Switches at CpG Sites&quot;) +
  xlab(&quot;Method&quot;) +
  ylab(&quot;Enrichment&quot;) +
  theme_classic() +
  geom_hline(yintercept = 1)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-37-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-37-1">
Past versions of unnamed-chunk-37-1.png
</button>
</p>
<div id="fig-unnamed-chunk-37-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/6172512f57cc03f295aefc7978f0e9f6439169c9/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-37-1.png" target="_blank">6172512</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/8041a90f4735210b3ffd2e9a1231fd4b9735a442/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-37-1.png" target="_blank">8041a90</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/aa8704bbf7c47c6b3b4c265552120bc446fa974a/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-37-1.png" target="_blank">aa8704b</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_vcftools %&gt;%
  mutate(Beagle = (n_flip_cpg_beagle / n_flip_beagle) / (n_het_cpg / n_hets),
         Eagle = (n_flip_cpg_eagle / n_flip_eagle) / (n_het_cpg / n_hets),
         SHAPEIT = (n_flip_cpg_shapeit / n_flip_shapeit) / (n_het_cpg / n_hets)) %&gt;%
  select(Beagle, Eagle, SHAPEIT) %&gt;%
  pivot_longer(everything(), names_to = &quot;method&quot;, values_to = &quot;enrich&quot;) %&gt;%
  ggplot(aes(x = method, y = enrich)) +
  geom_boxplot() +
  ggtitle(&quot;Enrichment of Flips at CpG Sites&quot;) +
  xlab(&quot;Method&quot;) +
  ylab(&quot;Enrichment&quot;) +
  theme_classic() +
  geom_hline(yintercept = 1)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-37-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_vcftools %&gt;%
  mutate(Beagle = (n_other_cpg_beagle / n_other_beagle) / (n_het_cpg / n_hets),
         Eagle = (n_other_cpg_eagle / n_other_eagle) / (n_het_cpg / n_hets),
         SHAPEIT = (n_other_cpg_shapeit / n_other_shapeit) / (n_het_cpg / n_hets)) %&gt;%
  select(pop, Beagle, Eagle, SHAPEIT) %&gt;%
  pivot_longer(-pop, names_to = &quot;method&quot;, values_to = &quot;enrich&quot;) %&gt;%
  ggplot(aes(x = method, y = enrich, colour = pop)) +
  geom_boxplot() +
  ggtitle(&quot;Enrichment of Switches at CpG Sites&quot;) +
  xlab(&quot;Method&quot;) +
  ylab(&quot;Enrichment&quot;) +
  theme_classic() +
  geom_hline(yintercept = 1) +
  scale_color_manual(values=cbPalette) +
  labs(color = &quot;Population&quot;)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-37-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_vcftools %&gt;%
  mutate(Beagle = (n_flip_cpg_beagle / n_flip_beagle) / (n_het_cpg / n_hets),
         Eagle = (n_flip_cpg_eagle / n_flip_eagle) / (n_het_cpg / n_hets),
         SHAPEIT = (n_flip_cpg_shapeit / n_flip_shapeit) / (n_het_cpg / n_hets)) %&gt;%
  select(pop, Beagle, Eagle, SHAPEIT) %&gt;%
  pivot_longer(-pop, names_to = &quot;method&quot;, values_to = &quot;enrich&quot;) %&gt;%
  ggplot(aes(x = method, y = enrich, color = pop)) +
  geom_boxplot() +
  ggtitle(&quot;Enrichment of Flips at CpG Sites&quot;) +
  xlab(&quot;Method&quot;) +
  ylab(&quot;Enrichment&quot;) +
  theme_classic() +
  geom_hline(yintercept = 1) +
  scale_color_manual(values=cbPalette) +
  labs(color = &quot;Population&quot;)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-37-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="maf-bins" class="section level2">
<h2>MAF Bins</h2>
<p>In the <code>maf_prop_df</code> data frame, we have for each
synthetic diploid the proportion of heterozygous sites that are: rare
variants (maf &lt; 0.01); uncommon variants (0.01 &lt; maf &lt; 0.05);
and common variants (maf &gt; 0.05). We also have these proportions for
each type of error for each method (switch; flip start; flip end;
correct).</p>
<p>Let’s first look at the distributions of these proportions for all
heterozygous positions within each individual:</p>
<pre class="r"><code>prop_maf_fig &lt;- function(cat_type, title_text){
  df &lt;- maf_prop_df %&gt;%
    filter(str_detect(category, paste0(&quot;^&quot;,cat_type, &quot;_[rcu]&quot;))) %&gt;%
    mutate(category = factor(category)) %&gt;%
    mutate(category = fct_recode(category, &quot;Rare&quot; = paste0(cat_type, &quot;_rare&quot;),
                               &quot;Uncommon&quot; = paste0(cat_type, &quot;_uncommon&quot;),
                               &quot;Common&quot; = paste0(cat_type, &quot;_common&quot;))) %&gt;%
    mutate(category = fct_relevel(category, c(&quot;Rare&quot;, &quot;Uncommon&quot;, &quot;Common&quot;)))
  
  p &lt;- df %&gt;%
    ggplot(aes(x = category, y = proportion, color = SP)) +
    geom_boxplot() +
    ggtitle(paste0(title_text, &quot; in MAF Bins&quot;)) +
    xlab(&quot;&quot;) +
    ylab(&quot;Proportion of Sites&quot;) +
    theme_classic() +
    labs(color = &quot;Population&quot;) +
    scale_color_manual(values = cbPalette)
  
  tab_prop &lt;- df %&gt;%
    group_by(SP, category) %&gt;%
    summarize(mean_prop = mean(proportion))
  
  return(list(p = p, tab_prop = tab_prop))
}

res_obj &lt;- prop_maf_fig(&quot;het&quot;, &quot;Heterozygous Sites&quot;)</code></pre>
<pre><code>`summarise()` has grouped output by &#39;SP&#39;. You can override using the `.groups`
argument.</code></pre>
<pre class="r"><code>print(res_obj$p)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-38-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-38-1">
Past versions of unnamed-chunk-38-1.png
</button>
</p>
<div id="fig-unnamed-chunk-38-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/6172512f57cc03f295aefc7978f0e9f6439169c9/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-38-1.png" target="_blank">6172512</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/8041a90f4735210b3ffd2e9a1231fd4b9735a442/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-38-1.png" target="_blank">8041a90</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/aa8704bbf7c47c6b3b4c265552120bc446fa974a/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-38-1.png" target="_blank">aa8704b</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res_obj$tab_prop %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">SP</th>
<th align="left">category</th>
<th align="right">mean_prop</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AFR</td>
<td align="left">Rare</td>
<td align="right">0.0965162</td>
</tr>
<tr class="even">
<td align="left">AFR</td>
<td align="left">Uncommon</td>
<td align="right">0.2067589</td>
</tr>
<tr class="odd">
<td align="left">AFR</td>
<td align="left">Common</td>
<td align="right">0.6967250</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="left">Rare</td>
<td align="right">0.0423491</td>
</tr>
<tr class="odd">
<td align="left">AMR</td>
<td align="left">Uncommon</td>
<td align="right">0.0781212</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="left">Common</td>
<td align="right">0.8795297</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="left">Rare</td>
<td align="right">0.0558559</td>
</tr>
<tr class="even">
<td align="left">EAS</td>
<td align="left">Uncommon</td>
<td align="right">0.0414441</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="left">Common</td>
<td align="right">0.9027000</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="left">Rare</td>
<td align="right">0.0469553</td>
</tr>
<tr class="odd">
<td align="left">EUR</td>
<td align="left">Uncommon</td>
<td align="right">0.0574450</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="left">Common</td>
<td align="right">0.8955997</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="left">Rare</td>
<td align="right">0.0591361</td>
</tr>
<tr class="even">
<td align="left">SAS</td>
<td align="left">Uncommon</td>
<td align="right">0.0428479</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="left">Common</td>
<td align="right">0.8980160</td>
</tr>
</tbody>
</table>
<div id="beagle-errors" class="section level3">
<h3>Beagle Errors</h3>
<div id="switches" class="section level4">
<h4>Switches</h4>
<pre class="r"><code>cat_type &lt;- &quot;beagle_switch&quot;
title_text &lt;- &quot;Beagle Switches&quot;

res_obj &lt;- prop_maf_fig(cat_type, title_text)</code></pre>
<pre><code>`summarise()` has grouped output by &#39;SP&#39;. You can override using the `.groups`
argument.</code></pre>
<pre class="r"><code>print(res_obj$p)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-39-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-39-1">
Past versions of unnamed-chunk-39-1.png
</button>
</p>
<div id="fig-unnamed-chunk-39-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/6172512f57cc03f295aefc7978f0e9f6439169c9/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-39-1.png" target="_blank">6172512</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/8041a90f4735210b3ffd2e9a1231fd4b9735a442/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-39-1.png" target="_blank">8041a90</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/aa8704bbf7c47c6b3b4c265552120bc446fa974a/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-39-1.png" target="_blank">aa8704b</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res_obj$tab_prop %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">SP</th>
<th align="left">category</th>
<th align="right">mean_prop</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AFR</td>
<td align="left">Rare</td>
<td align="right">0.1992096</td>
</tr>
<tr class="even">
<td align="left">AFR</td>
<td align="left">Uncommon</td>
<td align="right">0.2291918</td>
</tr>
<tr class="odd">
<td align="left">AFR</td>
<td align="left">Common</td>
<td align="right">0.5715986</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="left">Rare</td>
<td align="right">0.1945078</td>
</tr>
<tr class="odd">
<td align="left">AMR</td>
<td align="left">Uncommon</td>
<td align="right">0.1714602</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="left">Common</td>
<td align="right">0.6340321</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="left">Rare</td>
<td align="right">0.2380430</td>
</tr>
<tr class="even">
<td align="left">EAS</td>
<td align="left">Uncommon</td>
<td align="right">0.1072100</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="left">Common</td>
<td align="right">0.6547471</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="left">Rare</td>
<td align="right">0.2028975</td>
</tr>
<tr class="odd">
<td align="left">EUR</td>
<td align="left">Uncommon</td>
<td align="right">0.1426834</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="left">Common</td>
<td align="right">0.6544191</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="left">Rare</td>
<td align="right">0.2431572</td>
</tr>
<tr class="even">
<td align="left">SAS</td>
<td align="left">Uncommon</td>
<td align="right">0.0870414</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="left">Common</td>
<td align="right">0.6698014</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="flips-1" class="section level3">
<h3>Flips</h3>
<div id="start" class="section level4">
<h4>Start</h4>
<pre class="r"><code>cat_type &lt;- &quot;beagle_flip&quot;
title_text &lt;- &quot;Beagle Flip Starts&quot;

res_obj &lt;- prop_maf_fig(cat_type, title_text)</code></pre>
<pre><code>`summarise()` has grouped output by &#39;SP&#39;. You can override using the `.groups`
argument.</code></pre>
<pre class="r"><code>print(res_obj$p)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-40-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-40-1">
Past versions of unnamed-chunk-40-1.png
</button>
</p>
<div id="fig-unnamed-chunk-40-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/6172512f57cc03f295aefc7978f0e9f6439169c9/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-40-1.png" target="_blank">6172512</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/8041a90f4735210b3ffd2e9a1231fd4b9735a442/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-40-1.png" target="_blank">8041a90</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/aa8704bbf7c47c6b3b4c265552120bc446fa974a/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-40-1.png" target="_blank">aa8704b</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res_obj$tab_prop %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">SP</th>
<th align="left">category</th>
<th align="right">mean_prop</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AFR</td>
<td align="left">Rare</td>
<td align="right">0.5380834</td>
</tr>
<tr class="even">
<td align="left">AFR</td>
<td align="left">Uncommon</td>
<td align="right">0.1343066</td>
</tr>
<tr class="odd">
<td align="left">AFR</td>
<td align="left">Common</td>
<td align="right">0.3276100</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="left">Rare</td>
<td align="right">0.5878307</td>
</tr>
<tr class="odd">
<td align="left">AMR</td>
<td align="left">Uncommon</td>
<td align="right">0.1107531</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="left">Common</td>
<td align="right">0.3014162</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="left">Rare</td>
<td align="right">0.5724462</td>
</tr>
<tr class="even">
<td align="left">EAS</td>
<td align="left">Uncommon</td>
<td align="right">0.0986035</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="left">Common</td>
<td align="right">0.3289503</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="left">Rare</td>
<td align="right">0.5621629</td>
</tr>
<tr class="odd">
<td align="left">EUR</td>
<td align="left">Uncommon</td>
<td align="right">0.1089259</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="left">Common</td>
<td align="right">0.3289112</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="left">Rare</td>
<td align="right">0.5865144</td>
</tr>
<tr class="even">
<td align="left">SAS</td>
<td align="left">Uncommon</td>
<td align="right">0.0877529</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="left">Common</td>
<td align="right">0.3257327</td>
</tr>
</tbody>
</table>
</div>
<div id="end" class="section level4">
<h4>End</h4>
<pre class="r"><code>cat_type &lt;- &quot;beagle_flip end&quot;
title_text &lt;- &quot;Beagle Flip Ends&quot;

res_obj &lt;- prop_maf_fig(cat_type, title_text)</code></pre>
<pre><code>`summarise()` has grouped output by &#39;SP&#39;. You can override using the `.groups`
argument.</code></pre>
<pre class="r"><code>print(res_obj$p)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-41-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-41-1">
Past versions of unnamed-chunk-41-1.png
</button>
</p>
<div id="fig-unnamed-chunk-41-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/6172512f57cc03f295aefc7978f0e9f6439169c9/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-41-1.png" target="_blank">6172512</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/8041a90f4735210b3ffd2e9a1231fd4b9735a442/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-41-1.png" target="_blank">8041a90</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/aa8704bbf7c47c6b3b4c265552120bc446fa974a/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-41-1.png" target="_blank">aa8704b</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res_obj$tab_prop %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">SP</th>
<th align="left">category</th>
<th align="right">mean_prop</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AFR</td>
<td align="left">Rare</td>
<td align="right">0.1963912</td>
</tr>
<tr class="even">
<td align="left">AFR</td>
<td align="left">Uncommon</td>
<td align="right">0.2455003</td>
</tr>
<tr class="odd">
<td align="left">AFR</td>
<td align="left">Common</td>
<td align="right">0.5581084</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="left">Rare</td>
<td align="right">0.1713416</td>
</tr>
<tr class="odd">
<td align="left">AMR</td>
<td align="left">Uncommon</td>
<td align="right">0.2070938</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="left">Common</td>
<td align="right">0.6215647</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="left">Rare</td>
<td align="right">0.2756562</td>
</tr>
<tr class="even">
<td align="left">EAS</td>
<td align="left">Uncommon</td>
<td align="right">0.1296203</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="left">Common</td>
<td align="right">0.5947235</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="left">Rare</td>
<td align="right">0.2297552</td>
</tr>
<tr class="odd">
<td align="left">EUR</td>
<td align="left">Uncommon</td>
<td align="right">0.1748198</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="left">Common</td>
<td align="right">0.5954250</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="left">Rare</td>
<td align="right">0.2749819</td>
</tr>
<tr class="even">
<td align="left">SAS</td>
<td align="left">Uncommon</td>
<td align="right">0.1033867</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="left">Common</td>
<td align="right">0.6216314</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="eagle-errors" class="section level3">
<h3>Eagle Errors</h3>
<div id="switches-1" class="section level4">
<h4>Switches</h4>
<pre class="r"><code>cat_type &lt;- &quot;eagle_switch&quot;
title_text &lt;- &quot;Eagle Switches&quot;

res_obj &lt;- prop_maf_fig(cat_type, title_text)</code></pre>
<pre><code>`summarise()` has grouped output by &#39;SP&#39;. You can override using the `.groups`
argument.</code></pre>
<pre class="r"><code>print(res_obj$p)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-42-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-42-1">
Past versions of unnamed-chunk-42-1.png
</button>
</p>
<div id="fig-unnamed-chunk-42-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/6172512f57cc03f295aefc7978f0e9f6439169c9/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-42-1.png" target="_blank">6172512</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/8041a90f4735210b3ffd2e9a1231fd4b9735a442/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-42-1.png" target="_blank">8041a90</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/aa8704bbf7c47c6b3b4c265552120bc446fa974a/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-42-1.png" target="_blank">aa8704b</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res_obj$tab_prop %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">SP</th>
<th align="left">category</th>
<th align="right">mean_prop</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AFR</td>
<td align="left">Rare</td>
<td align="right">0.3136962</td>
</tr>
<tr class="even">
<td align="left">AFR</td>
<td align="left">Uncommon</td>
<td align="right">0.2118531</td>
</tr>
<tr class="odd">
<td align="left">AFR</td>
<td align="left">Common</td>
<td align="right">0.4744507</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="left">Rare</td>
<td align="right">0.3039367</td>
</tr>
<tr class="odd">
<td align="left">AMR</td>
<td align="left">Uncommon</td>
<td align="right">0.1677582</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="left">Common</td>
<td align="right">0.5283050</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="left">Rare</td>
<td align="right">0.3275337</td>
</tr>
<tr class="even">
<td align="left">EAS</td>
<td align="left">Uncommon</td>
<td align="right">0.1047851</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="left">Common</td>
<td align="right">0.5676812</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="left">Rare</td>
<td align="right">0.3019794</td>
</tr>
<tr class="odd">
<td align="left">EUR</td>
<td align="left">Uncommon</td>
<td align="right">0.1409590</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="left">Common</td>
<td align="right">0.5570616</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="left">Rare</td>
<td align="right">0.3389934</td>
</tr>
<tr class="even">
<td align="left">SAS</td>
<td align="left">Uncommon</td>
<td align="right">0.0851283</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="left">Common</td>
<td align="right">0.5758783</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="flips-2" class="section level3">
<h3>Flips</h3>
<div id="start-1" class="section level4">
<h4>Start</h4>
<pre class="r"><code>cat_type &lt;- &quot;eagle_flip&quot;
title_text &lt;- &quot;Eagle Flips&quot;

res_obj &lt;- prop_maf_fig(cat_type, title_text)</code></pre>
<pre><code>`summarise()` has grouped output by &#39;SP&#39;. You can override using the `.groups`
argument.</code></pre>
<pre class="r"><code>print(res_obj$p)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-43-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-43-1">
Past versions of unnamed-chunk-43-1.png
</button>
</p>
<div id="fig-unnamed-chunk-43-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/101156547ea5cf5ab7d26907d94ac857ff87c51a/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-43-1.png" target="_blank">1011565</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res_obj$tab_prop %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">SP</th>
<th align="left">category</th>
<th align="right">mean_prop</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AFR</td>
<td align="left">Rare</td>
<td align="right">0.4982103</td>
</tr>
<tr class="even">
<td align="left">AFR</td>
<td align="left">Uncommon</td>
<td align="right">0.1529594</td>
</tr>
<tr class="odd">
<td align="left">AFR</td>
<td align="left">Common</td>
<td align="right">0.3488304</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="left">Rare</td>
<td align="right">0.5662804</td>
</tr>
<tr class="odd">
<td align="left">AMR</td>
<td align="left">Uncommon</td>
<td align="right">0.1175883</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="left">Common</td>
<td align="right">0.3161313</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="left">Rare</td>
<td align="right">0.5769485</td>
</tr>
<tr class="even">
<td align="left">EAS</td>
<td align="left">Uncommon</td>
<td align="right">0.0999720</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="left">Common</td>
<td align="right">0.3230796</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="left">Rare</td>
<td align="right">0.5552438</td>
</tr>
<tr class="odd">
<td align="left">EUR</td>
<td align="left">Uncommon</td>
<td align="right">0.1036375</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="left">Common</td>
<td align="right">0.3411187</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="left">Rare</td>
<td align="right">0.6033717</td>
</tr>
<tr class="even">
<td align="left">SAS</td>
<td align="left">Uncommon</td>
<td align="right">0.0866393</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="left">Common</td>
<td align="right">0.3099890</td>
</tr>
</tbody>
</table>
</div>
<div id="end-1" class="section level4">
<h4>End</h4>
<pre class="r"><code>cat_type &lt;- &quot;eagle_flip end&quot;
title_text &lt;- &quot;Eagle Flip Ends&quot;

res_obj &lt;- prop_maf_fig(cat_type, title_text)</code></pre>
<pre><code>`summarise()` has grouped output by &#39;SP&#39;. You can override using the `.groups`
argument.</code></pre>
<pre class="r"><code>print(res_obj$p)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-44-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-44-1">
Past versions of unnamed-chunk-44-1.png
</button>
</p>
<div id="fig-unnamed-chunk-44-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/phasing_clean/blob/101156547ea5cf5ab7d26907d94ac857ff87c51a/docs/figure/synthetic_diploid_results.Rmd/unnamed-chunk-44-1.png" target="_blank">1011565</a>
</td>
<td>
Andy Beck
</td>
<td>
2024-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res_obj$tab_prop %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">SP</th>
<th align="left">category</th>
<th align="right">mean_prop</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AFR</td>
<td align="left">Rare</td>
<td align="right">0.1687960</td>
</tr>
<tr class="even">
<td align="left">AFR</td>
<td align="left">Uncommon</td>
<td align="right">0.2274438</td>
</tr>
<tr class="odd">
<td align="left">AFR</td>
<td align="left">Common</td>
<td align="right">0.6037601</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="left">Rare</td>
<td align="right">0.1465582</td>
</tr>
<tr class="odd">
<td align="left">AMR</td>
<td align="left">Uncommon</td>
<td align="right">0.1772875</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="left">Common</td>
<td align="right">0.6761544</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="left">Rare</td>
<td align="right">0.2369988</td>
</tr>
<tr class="even">
<td align="left">EAS</td>
<td align="left">Uncommon</td>
<td align="right">0.1086514</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="left">Common</td>
<td align="right">0.6543498</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="left">Rare</td>
<td align="right">0.1862986</td>
</tr>
<tr class="odd">
<td align="left">EUR</td>
<td align="left">Uncommon</td>
<td align="right">0.1470046</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="left">Common</td>
<td align="right">0.6666968</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="left">Rare</td>
<td align="right">0.2322660</td>
</tr>
<tr class="even">
<td align="left">SAS</td>
<td align="left">Uncommon</td>
<td align="right">0.0897263</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="left">Common</td>
<td align="right">0.6780077</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="shapeit-errors" class="section level3">
<h3>SHAPEIT Errors</h3>
<div id="switches-2" class="section level4">
<h4>Switches</h4>
<pre class="r"><code>cat_type &lt;- &quot;shapeit_switch&quot;
title_text &lt;- &quot;SHAPEIT Switches&quot;

res_obj &lt;- prop_maf_fig(cat_type, title_text)</code></pre>
<pre><code>`summarise()` has grouped output by &#39;SP&#39;. You can override using the `.groups`
argument.</code></pre>
<pre class="r"><code>print(res_obj$p)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-45-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>res_obj$tab_prop %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">SP</th>
<th align="left">category</th>
<th align="right">mean_prop</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AFR</td>
<td align="left">Rare</td>
<td align="right">0.1912941</td>
</tr>
<tr class="even">
<td align="left">AFR</td>
<td align="left">Uncommon</td>
<td align="right">0.2316338</td>
</tr>
<tr class="odd">
<td align="left">AFR</td>
<td align="left">Common</td>
<td align="right">0.5770722</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="left">Rare</td>
<td align="right">0.1757003</td>
</tr>
<tr class="odd">
<td align="left">AMR</td>
<td align="left">Uncommon</td>
<td align="right">0.1686226</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="left">Common</td>
<td align="right">0.6556771</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="left">Rare</td>
<td align="right">0.2234550</td>
</tr>
<tr class="even">
<td align="left">EAS</td>
<td align="left">Uncommon</td>
<td align="right">0.1026105</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="left">Common</td>
<td align="right">0.6739345</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="left">Rare</td>
<td align="right">0.1916693</td>
</tr>
<tr class="odd">
<td align="left">EUR</td>
<td align="left">Uncommon</td>
<td align="right">0.1342722</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="left">Common</td>
<td align="right">0.6740585</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="left">Rare</td>
<td align="right">0.2283214</td>
</tr>
<tr class="even">
<td align="left">SAS</td>
<td align="left">Uncommon</td>
<td align="right">0.0840477</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="left">Common</td>
<td align="right">0.6876309</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="flips-3" class="section level3">
<h3>Flips</h3>
<div id="start-2" class="section level4">
<h4>Start</h4>
<pre class="r"><code>cat_type &lt;- &quot;shapeit_flip&quot;
title_text &lt;- &quot;SHAPEIT Flips&quot;

res_obj &lt;- prop_maf_fig(cat_type, title_text)</code></pre>
<pre><code>`summarise()` has grouped output by &#39;SP&#39;. You can override using the `.groups`
argument.</code></pre>
<pre class="r"><code>print(res_obj$p)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-46-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>res_obj$tab_prop %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">SP</th>
<th align="left">category</th>
<th align="right">mean_prop</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AFR</td>
<td align="left">Rare</td>
<td align="right">0.5215304</td>
</tr>
<tr class="even">
<td align="left">AFR</td>
<td align="left">Uncommon</td>
<td align="right">0.1478400</td>
</tr>
<tr class="odd">
<td align="left">AFR</td>
<td align="left">Common</td>
<td align="right">0.3306296</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="left">Rare</td>
<td align="right">0.5917000</td>
</tr>
<tr class="odd">
<td align="left">AMR</td>
<td align="left">Uncommon</td>
<td align="right">0.1141980</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="left">Common</td>
<td align="right">0.2941019</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="left">Rare</td>
<td align="right">0.5545337</td>
</tr>
<tr class="even">
<td align="left">EAS</td>
<td align="left">Uncommon</td>
<td align="right">0.1022251</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="left">Common</td>
<td align="right">0.3432412</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="left">Rare</td>
<td align="right">0.5599597</td>
</tr>
<tr class="odd">
<td align="left">EUR</td>
<td align="left">Uncommon</td>
<td align="right">0.1053868</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="left">Common</td>
<td align="right">0.3346536</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="left">Rare</td>
<td align="right">0.5830829</td>
</tr>
<tr class="even">
<td align="left">SAS</td>
<td align="left">Uncommon</td>
<td align="right">0.0900896</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="left">Common</td>
<td align="right">0.3268275</td>
</tr>
</tbody>
</table>
</div>
<div id="end-2" class="section level4">
<h4>End</h4>
<pre class="r"><code>cat_type &lt;- &quot;shapeit_flip end&quot;
title_text &lt;- &quot;SHAPEIT Flip Ends&quot;

res_obj &lt;- prop_maf_fig(cat_type, title_text)</code></pre>
<pre><code>`summarise()` has grouped output by &#39;SP&#39;. You can override using the `.groups`
argument.</code></pre>
<pre class="r"><code>print(res_obj$p)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-47-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>res_obj$tab_prop %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">SP</th>
<th align="left">category</th>
<th align="right">mean_prop</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AFR</td>
<td align="left">Rare</td>
<td align="right">0.1917881</td>
</tr>
<tr class="even">
<td align="left">AFR</td>
<td align="left">Uncommon</td>
<td align="right">0.2464622</td>
</tr>
<tr class="odd">
<td align="left">AFR</td>
<td align="left">Common</td>
<td align="right">0.5617497</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="left">Rare</td>
<td align="right">0.1685153</td>
</tr>
<tr class="odd">
<td align="left">AMR</td>
<td align="left">Uncommon</td>
<td align="right">0.2094346</td>
</tr>
<tr class="even">
<td align="left">AMR</td>
<td align="left">Common</td>
<td align="right">0.6220501</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="left">Rare</td>
<td align="right">0.2745675</td>
</tr>
<tr class="even">
<td align="left">EAS</td>
<td align="left">Uncommon</td>
<td align="right">0.1317415</td>
</tr>
<tr class="odd">
<td align="left">EAS</td>
<td align="left">Common</td>
<td align="right">0.5936910</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="left">Rare</td>
<td align="right">0.2245520</td>
</tr>
<tr class="odd">
<td align="left">EUR</td>
<td align="left">Uncommon</td>
<td align="right">0.1731890</td>
</tr>
<tr class="even">
<td align="left">EUR</td>
<td align="left">Common</td>
<td align="right">0.6022591</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="left">Rare</td>
<td align="right">0.2704505</td>
</tr>
<tr class="even">
<td align="left">SAS</td>
<td align="left">Uncommon</td>
<td align="right">0.1057007</td>
</tr>
<tr class="odd">
<td align="left">SAS</td>
<td align="left">Common</td>
<td align="right">0.6238488</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="odds-ratios" class="section level3">
<h3>Odds Ratios</h3>
<p>Here we’ll compoute the ratio of odds of being in the rare variant
category for errors and all heterozygous positions within each
individual.</p>
<pre class="r"><code>maf_prop_df %&gt;%
  select(category, proportion, id, SP) %&gt;%
  pivot_wider(names_from = category, values_from = proportion) %&gt;%
  mutate(or_het = het_rare / (1 - het_rare),
         or_b_s = beagle_switch_rare / (1 - beagle_switch_rare),
         or_e_s = eagle_switch_rare / (1 - eagle_switch_rare),
         or_s_s = shapeit_switch_rare / (1 - shapeit_switch_rare)) %&gt;%
  mutate(Beagle = or_b_s / or_het,
         Eagle = or_e_s / or_het,
         SHAPEIT = or_s_s / or_het) %&gt;%
  select(id, SP, Beagle, Eagle, SHAPEIT) %&gt;%
  pivot_longer(!id &amp; !SP, names_to = &quot;Method&quot;, values_to = &quot;OR&quot;) %&gt;%
  ggplot(aes(x = Method, y = OR, colour = SP)) +
  geom_boxplot() +
  ggtitle(&quot;Odds Ratio of Rare Variant&quot;) +
  ylab(&quot;Odds Rare Switch / Odds Rare Heterozygous&quot;) +
  theme_classic() +
  labs(colour = &quot;Population&quot;) +
  geom_hline(yintercept = 1, linetype = 2)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-48-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>maf_prop_df %&gt;%
  select(category, proportion, id, SP) %&gt;%
  pivot_wider(names_from = category, values_from = proportion) %&gt;%
  mutate(or_het = het_rare / (1 - het_rare),
         or_b_s = beagle_flip_rare / (1 - beagle_flip_rare),
         or_e_s = eagle_flip_rare / (1 - beagle_flip_rare),
         or_s_s = shapeit_flip_rare / (1 - beagle_flip_rare)) %&gt;%
  mutate(Beagle = or_b_s / or_het,
         Eagle = or_e_s / or_het,
         SHAPEIT = or_s_s / or_het) %&gt;%
  select(id, SP, Beagle, Eagle, SHAPEIT) %&gt;%
  pivot_longer(!id &amp; !SP, names_to = &quot;Method&quot;, values_to = &quot;OR&quot;) %&gt;%
  ggplot(aes(x = Method, y = OR, colour = SP)) +
  geom_boxplot() +
  ggtitle(&quot;Odds Ratio of Rare Variant&quot;) +
  ylab(&quot;Odds Rare Flip / Odds Rare Heterozygous&quot;) +
  theme_classic() +
  labs(colour = &quot;Population&quot;) +
  geom_hline(yintercept = 1, linetype = 2)</code></pre>
<p><img src="figure/synthetic_diploid_results.Rmd/unnamed-chunk-49-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 20.04.6 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3;  LAPACK version 3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: America/New_York
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] yaml_2.3.8      reactable_0.4.4 janitor_2.2.0   lubridate_1.9.3
 [5] forcats_1.0.0   stringr_1.5.1   dplyr_1.1.4     purrr_1.0.2    
 [9] readr_2.1.5     tidyr_1.3.1     tibble_3.2.1    ggplot2_3.5.0  
[13] tidyverse_2.0.0 workflowr_1.7.1

loaded via a namespace (and not attached):
 [1] gtable_0.3.4      xfun_0.42         bslib_0.7.0       htmlwidgets_1.6.4
 [5] processx_3.8.3    callr_3.7.5       tzdb_0.4.0        vctrs_0.6.5      
 [9] tools_4.4.0       ps_1.7.6          generics_0.1.3    parallel_4.4.0   
[13] fansi_1.0.6       highr_0.10        pkgconfig_2.0.3   lifecycle_1.0.4  
[17] farver_2.1.1      compiler_4.4.0    git2r_0.33.0      munsell_0.5.1    
[21] getPass_0.2-4     snakecase_0.11.1  httpuv_1.6.14     htmltools_0.5.8  
[25] sass_0.4.9        later_1.3.2       pillar_1.9.0      crayon_1.5.2     
[29] jquerylib_0.1.4   whisker_0.4.1     cachem_1.0.8      tidyselect_1.2.0 
[33] digest_0.6.34     stringi_1.8.3     labeling_0.4.3    rprojroot_2.0.4  
[37] fastmap_1.1.1     grid_4.4.0        archive_1.1.7     colorspace_2.1-0 
[41] cli_3.6.2         magrittr_2.0.3    utf8_1.2.4        withr_3.0.0      
[45] scales_1.3.0      promises_1.2.1    bit64_4.0.5       timechange_0.3.0 
[49] rmarkdown_2.26    httr_1.4.7        bit_4.0.5         hms_1.1.3        
[53] evaluate_0.23     knitr_1.45        rlang_1.1.3       Rcpp_1.0.12      
[57] glue_1.7.0        rstudioapi_0.15.0 vroom_1.6.5       jsonlite_1.8.8   
[61] R6_2.5.1          fs_1.6.3         </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
