---
title: "Synthetic Diploid Results"
author: "Andy Beck"
date: "2024-05-15"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Libraries and Directories

```{r}
library(tidyverse)
library(janitor)
library(reactable)
source("code/common_functions.R")
library(yaml)
library(gtsummary)
library(infer)

library(grid)
library(gridExtra) 

config_obj <- yaml::read_yaml("_config.yaml")

eagle_switch_dir <-   paste0(config_obj$base_dir,"/output/switch_errors/switch_errors/eagle/annotated/")
shapeit_switch_dir <- paste0(config_obj$base_dir,"/output/switch_errors/switch_errors/shapeit/annotated/")
beagle_switch_dir <-  paste0(config_obj$base_dir,"/output/switch_errors/switch_errors/beagle/annotated/")
num_sites_dir <-      paste0(config_obj$base_dir,"/output/switch_errors/vcf_n_sites/")
het_loc_dir <-        paste0(config_obj$base_dir,"/output/switch_errors/het_loc/")

sample_info_df <- read_csv("data/1kgp/subject_info.csv") %>%
  select(SAMPLE_NAME, POPULATION, SUPER)

pair_info_df <- read_delim("data/sample_pairs.csv", col_names = c("POP", "ID1", "ID2"))
pair_info_df <- left_join(pair_info_df, sample_info_df, by = c("ID1"="SAMPLE_NAME")) %>%
  rename(SP = SUPER) %>%
  select(-POPULATION)

# data/chrX_gc1kb_pilot.bed
gc_content_1kb <- read_tsv("data/chrX_gc1kb_pilot.bed")
colnames(gc_content_1kb) <- c("CHR", "START", "END", "AT", "GC", "A", "C", "G", "T", "TOTAL", "OTHER", "LENGTH")
gc_content_1kb  <- gc_content_1kb %>%
  mutate(bin_id = (START / 1000) + 1)

cbPalette <- c("#999999", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7", "#000")

maf_prop_df <- read_csv(paste0(config_obj$base_dir,"/output/switch_errors/maf_props.csv"))
pair_info_df$id <- 1:1000
maf_prop_df <- left_join(maf_prop_df, {pair_info_df %>% select(id, SP, POP)})
```


## Introduction

In this document, we present summaries and figures for the distributions of errors in phasing of our synthetic X chromosome diploids.

## Load Switch and Flips Errors

```{r}
df_vcftools <- read_csv(paste0(config_obj$base_dir,"/output/switch_errors/switch_errors/summary.csv"))

df_vcftools$n_total_beagle <- (df_vcftools$n_flip_beagle + df_vcftools$n_other_beagle)
df_vcftools$n_total_eagle <- (df_vcftools$n_flip_eagle + df_vcftools$n_other_eagle)
df_vcftools$n_total_shapeit <- (df_vcftools$n_flip_shapeit + df_vcftools$n_other_shapeit)

sum(df_vcftools$n_total_shapeit < df_vcftools$n_total_eagle)
sum(df_vcftools$n_total_beagle < df_vcftools$n_total_eagle)
```

## Nov 2025: Remove rare variant analysis

```{r}
df_switch_noRare <- read_tsv("output/X_noRare/whatshap/X_noRare_errors_summary.tsv")
df_switch_noRare <- df_switch_noRare %>% 
  mutate(switch_rate = switches / n_het,
         flip_rate = flips / n_het,
         total_rate = (switches + flips) / n_het)

df_switch_noRare$pop <- rep(c("EUR", "AFR", "AMR", "EAS", "SAS"), each = 600)
```

```{r}
df_vcftools_sub <- df_vcftools %>%
  select(pair_id, n_hets, n_other_beagle, n_other_eagle, n_other_shapeit, n_flip_beagle, n_flip_eagle, n_flip_shapeit) %>%
  rename(sample_id = pair_id, n_het = n_hets,
    switches_beagle = n_other_beagle, switches_eagle = n_other_eagle, switches_shapeit = n_other_shapeit,
    flips_beagle = n_flip_beagle, flips_eagle = n_flip_eagle, flips_shapeit = n_flip_shapeit) %>%
  pivot_longer(cols = -c(sample_id, n_het),
               names_to = c("error_type", "method"),
               names_sep = "_",
               values_to = "count") %>%
  pivot_wider(names_from = error_type, values_from = count) %>%
  mutate(total_errors = switches + flips) %>%
  select(sample_id, method, switches, flips, total_errors, n_het) %>%
  mutate(switch_rate = switches / n_het,
         flip_rate = flips / n_het)

df_vcftools_sub$rare <- "with_rare"
df_switch_noRare$rare <- "no_rare"
combined_df <- bind_rows(df_vcftools_sub, df_switch_noRare)

# pivot wider so each sample_id, method pair has a single row, with columns for switches, flips, total_errors, n_het, switch_rate, flip_rate, total_rate for both rare and no_rare
combined_df_wide <- combined_df %>%
  pivot_longer(cols = c(switches, flips, total_errors, n_het, switch_rate, flip_rate, total_rate),
               names_to = "metric",
               values_to = "value") %>%
  unite(temp, metric, rare) %>%
  pivot_wider(names_from = temp, values_from = value) 
```

Let's look at how switch rates and flip rates compare within methods.

```{r}
combined_df_wide %>%
  ggplot(aes(x = switch_rate_with_rare, y = switch_rate_no_rare, color = method)) +
  geom_point() +
  geom_abline(slope = 1, intercept = 0, linetype = "dashed") +
  theme_classic() +
  labs(x = "Switch Rate (With Rare Variants)",
       y = "Switch Rate (No Rare Variants)",
       color = "Method") +
  scale_color_manual(values = ggpubfigs::friendly_pal("vibrant_seven")) +
  theme(legend.position = "top") + # larger axis text
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18))
```

And for flip rates:

```{r}
combined_df_wide %>%
  ggplot(aes(x = flip_rate_with_rare, y = flip_rate_no_rare, color = method)) +
  geom_point() +
  geom_abline(slope = 1, intercept = 0, linetype = "dashed") +
  theme_classic() +
  labs(x = "Flip Rate (With Rare Variants)",
       y = "Flip Rate (No Rare Variants)",
       color = "Method") +
  scale_color_manual(values = ggpubfigs::friendly_pal("vibrant_seven")) +
  theme(legend.position = "top") + # larger axis text
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18))
```

For just the results without rare variants, I now want to do pairwise comparisons of methods for switch rates and flip rates.

First, scatter plots for switch rates:

```{r}
df_switch_noRare %>%
  select(sample_id, pop, method, switch_rate) %>%
  pivot_wider(names_from = method, values_from = switch_rate) %>%
  ggplot(aes(x = beagle, y = eagle, colour = pop)) +
  geom_point() +
  geom_abline(slope = 1, intercept = 0, linetype = "dashed") +
  theme_classic() +
  labs(x = "Beagle Switch Rate (No Rare Variants)",
       y = "Eagle Switch Rate (No Rare Variants)") +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18)) +
  scale_color_manual(values = ggpubfigs::friendly_pal("vibrant_seven"))

df_switch_noRare %>%
  select(sample_id, pop, method, switch_rate) %>%
  pivot_wider(names_from = method, values_from = switch_rate) %>%
  ggplot(aes(x = beagle, y = shapeit, colour = pop)) +
  geom_point() +
  geom_abline(slope = 1, intercept = 0, linetype = "dashed") +
  theme_classic() +
  labs(x = "Beagle Switch Rate (No Rare Variants)",
       y = "SHAPEIT Switch Rate (No Rare Variants)") +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18)) +
  scale_color_manual(values = ggpubfigs::friendly_pal("vibrant_seven"))

df_switch_noRare %>%
  select(sample_id, pop, method, switch_rate) %>%
  pivot_wider(names_from = method, values_from = switch_rate) %>%
  ggplot(aes(x = eagle, y = shapeit, colour = pop)) +
  geom_point() +
  geom_abline(slope = 1, intercept = 0, linetype = "dashed") +
  theme_classic() +
  labs(x = "Eagle Switch Rate (No Rare Variants)",
       y = "SHAPEIT Switch Rate (No Rare Variants)") +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18)) +
  scale_color_manual(values = ggpubfigs::friendly_pal("vibrant_seven"))
```

And now flips:

```{r}
df_switch_noRare %>%
  select(sample_id, pop, method, flip_rate) %>%
  pivot_wider(names_from = method, values_from = flip_rate) %>%
  ggplot(aes(x = beagle, y = eagle, colour = pop)) +
  geom_point() +
  geom_abline(slope = 1, intercept = 0, linetype = "dashed") +
  theme_classic() +
  labs(x = "Beagle Flip Rate (No Rare Variants)",
       y = "Eagle Flip Rate (No Rare Variants)") +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18)) +
  scale_color_manual(values = ggpubfigs::friendly_pal("vibrant_seven"))

df_switch_noRare %>%
  select(sample_id, pop, method, flip_rate) %>%
  pivot_wider(names_from = method, values_from = flip_rate) %>%
  ggplot(aes(x = beagle, y = shapeit, colour = pop)) +
  geom_point() +
  geom_abline(slope = 1, intercept = 0, linetype = "dashed") +
  theme_classic() +
  labs(x = "Beagle Flip Rate (No Rare Variants)",
       y = "SHAPEIT Flip Rate (No Rare Variants)") +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18)) +
  scale_color_manual(values = ggpubfigs::friendly_pal("vibrant_seven"))

df_switch_noRare %>%
  select(sample_id, pop, method, flip_rate) %>%
  pivot_wider(names_from = method, values_from = flip_rate) %>%
  ggplot(aes(x = eagle, y = shapeit, colour = pop)) +
  geom_point() +
  geom_abline(slope = 1, intercept = 0, linetype = "dashed") +
  theme_classic() +
  labs(x = "Eagle Flip Rate (No Rare Variants)",
       y = "SHAPEIT Flip Rate (No Rare Variants)") +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18)) +
  scale_color_manual(values = ggpubfigs::friendly_pal("vibrant_seven"))
```

## Oct 2025: Text Revisions

1. What is the distribution of heterozygous sites per sample?

```{r}
df_vcftools %>% 
  summarize(mean_n_hets = mean(n_hets),
            sd_n_hets = sd(n_hets),
            min_n_hets = min(n_hets),
            max_n_hets = max(n_hets),
            median_n_hets = median(n_hets),
            n = n())

mean(df_vcftools$n_hets)
# stratify summary by pop
df_vcftools %>%
  group_by(pop) %>%
  summarize(mean_n_hets = mean(n_hets),
            sd_n_hets = sd(n_hets),
            min_n_hets = min(n_hets),
            max_n_hets = max(n_hets),
            median_n_hets = median(n_hets),
            n = n())
```

2. What are the average error rates per sample?

```{r}
options(pillar.sigfig = 7)
df_vcftools %>%
  mutate(Beagle_rate = (n_other_beagle + n_flip_beagle) / n_hets,
         Beagle_count = (n_other_beagle + n_flip_beagle), 
         Eagle_rate = (n_other_eagle + n_flip_eagle) / n_hets,
         Eagle_count = (n_other_eagle + n_flip_eagle),
         SHAPEIT_rate = (n_other_shapeit + n_flip_shapeit) / n_hets,
         SHAPEIT_count = (n_other_shapeit + n_flip_shapeit)) %>%
  summarize(mean_Beagle_rate = mean(Beagle_rate),
            mean_Beagle_count = mean(Beagle_count),
            mean_Eagle_rate = mean(Eagle_rate),
            mean_Eagle_count = mean(Eagle_count),
            mean_SHAPEIT_rate = mean(SHAPEIT_rate),
            mean_SHAPEIT_count = mean(SHAPEIT_count)) %>%
   print(n = Inf, width = Inf)
```

3. Is there a significant difference in error rates between methods? Let's do a paired t-test between each pair of methods.

```{r}
df_vcftools %>%
  mutate(Beagle = (n_other_beagle + n_flip_beagle) / n_hets ,
         Eagle = (n_other_eagle + n_flip_eagle) / n_hets ,
         SHAPEIT = (n_other_shapeit + n_flip_shapeit) / n_hets ) %>%
  select(Beagle, Eagle, SHAPEIT) %>%
  pivot_longer(cols = everything(), names_to = "method", values_to = "error_rate") %>%
  group_by(method) %>%
  summarize(mean_error_rate = mean(error_rate),
            sd_error_rate = sd(error_rate),
            n = n()) %>%
  ungroup()

fit <- df_vcftools %>%
  mutate(Beagle = (n_other_beagle + n_flip_beagle) / n_hets ,
         Eagle = (n_other_eagle + n_flip_eagle) / n_hets ,
         SHAPEIT = (n_other_shapeit + n_flip_shapeit) / n_hets ) %>%
  select(pair_id, Beagle, Eagle, SHAPEIT) %>%
  pivot_longer(cols = -pair_id, names_to = "method", values_to = "error_rate") %>%
  {lmerTest::lmer(error_rate ~ method + (1|pair_id), data = .)}
anova(fit)
emmeans::emmeans(fit, ~ method) %>% pairs()
```

4. Do switch error rates differ across methods?

```{r}
fit <- df_vcftools %>%
  mutate(Beagle = n_other_beagle / n_hets ,
         Eagle = n_other_eagle / n_hets ,
         SHAPEIT = n_other_shapeit / n_hets ) %>%
  select(pair_id, Beagle, Eagle, SHAPEIT) %>%
  pivot_longer(cols = -pair_id, names_to = "method", values_to = "error_rate") %>%
  {lmerTest::lmer(error_rate ~ method + (1|pair_id), data = .)}
anova(fit)
emmeans::emmeans(fit, ~ method) %>% pairs()

# Get mean switch rates for each method
df_vcftools %>%
  mutate(Beagle = n_other_beagle / n_hets ,
         Eagle = n_other_eagle / n_hets ,
         SHAPEIT = n_other_shapeit / n_hets ) %>%
  select(Beagle, Eagle, SHAPEIT) %>%
  summarize(mean_Beagle = mean(Beagle),
            mean_Eagle = mean(Eagle),
            mean_SHAPEIT = mean(SHAPEIT))
```

5. Do flip error rates differ across methods?

```{r}
fit <- df_vcftools %>%
  mutate(Beagle = n_flip_beagle / n_hets ,
         Eagle = n_flip_eagle / n_hets ,
         SHAPEIT = n_flip_shapeit / n_hets ) %>%
  select(pair_id, Beagle, Eagle, SHAPEIT) %>%
  pivot_longer(cols = -pair_id, names_to = "method", values_to = "error_rate") %>%
  {lmerTest::lmer(error_rate ~ method + (1|pair_id), data = .)}
anova(fit)
emmeans::emmeans(fit, ~ method) %>% pairs()

# Get average flip rates for each method
df_vcftools %>%
  mutate(Beagle = n_flip_beagle / n_hets ,
         Eagle = n_flip_eagle / n_hets ,
         SHAPEIT = n_flip_shapeit / n_hets ) %>%
  select(Beagle, Eagle, SHAPEIT) %>%
  summarize(mean_Beagle = mean(Beagle),
            mean_Eagle = mean(Eagle),
            mean_SHAPEIT = mean(SHAPEIT))
```

6. Do switch error rates differ by population?

```{r}
mean_df <- df_vcftools %>%
  group_by(pop) %>%
  summarize(mean_switch_beagle = mean(n_other_beagle / n_hets),
            sd_switch_beagle = sd(n_other_beagle / n_hets),
            mean_switch_eagle = mean(n_other_eagle / n_hets),
            sd_switch_eagle = sd(n_other_eagle / n_hets),
            mean_switch_shapeit = mean(n_other_shapeit / n_hets),
            sd_switch_shapeit = sd(n_other_shapeit / n_hets))
mean_df

fit <- df_vcftools %>%
  mutate(Beagle = n_other_beagle / n_hets ,
         Eagle = n_other_eagle / n_hets ,
         SHAPEIT = n_other_shapeit / n_hets ) %>%
  select(pair_id, pop, Beagle, Eagle, SHAPEIT) %>%
  pivot_longer(cols = -c(pair_id, pop), names_to = "method", values_to = "error_rate") %>%
  {lmerTest::lmer(error_rate ~ method * pop + (1|pair_id), data = .)}
anova(fit)
```

## Tables

### Mean error rates

Total errors:

```{r}
df_vcftools %>%
  mutate(Beagle = (n_other_beagle + n_flip_beagle) ,
         Eagle = (n_other_eagle + n_flip_eagle) ,
         SHAPEIT = (n_other_shapeit + n_flip_shapeit) ) %>%
  tbl_summary(include = c(Beagle, Eagle, SHAPEIT),
              statistic = list(
                all_continuous() ~ "{mean} ({sd})",
                all_categorical() ~ "{n} / {N} ({p}%)")
              )
```

By population:

```{r}
df_vcftools %>%
  mutate(Beagle = (n_other_beagle + n_flip_beagle) ,
         Eagle = (n_other_eagle + n_flip_eagle) ,
         SHAPEIT = (n_other_shapeit + n_flip_shapeit) ) %>%
  tbl_summary(by = pop,
              include = c(Beagle, Eagle, SHAPEIT),
              statistic = list(
                all_continuous() ~ "{mean} ({sd})",
                all_categorical() ~ "{n} / {N} ({p}%)")
              )
```

Errors per het:

```{r}
df_vcftools %>%
  mutate(Beagle = (n_other_beagle + n_flip_beagle) / n_hets ,
         Eagle = (n_other_eagle + n_flip_eagle) / n_hets ,
         SHAPEIT = (n_other_shapeit + n_flip_shapeit) / n_hets ) %>%
  tbl_summary(include = c(Beagle, Eagle, SHAPEIT),
              statistic = list(
                all_continuous() ~ "{mean} ({sd})",
                all_categorical() ~ "{n} / {N} ({p}%)")
              )

df_vcftools %>%
  mutate(Beagle = (n_other_beagle + n_flip_beagle) / n_hets ,
         Eagle = (n_other_eagle + n_flip_eagle) / n_hets ,
         SHAPEIT = (n_other_shapeit + n_flip_shapeit) / n_hets ) %>%
  tbl_summary(by = pop,
              include = c(Beagle, Eagle, SHAPEIT),
              statistic = list(
                all_continuous() ~ "{mean} ({sd})",
                all_categorical() ~ "{n} / {N} ({p}%)")
              )
```

Errors per MB:

```{r}
df_vcftools %>%
  mutate(Beagle = (n_other_beagle + n_flip_beagle) / size_mb ,
         Eagle = (n_other_eagle + n_flip_eagle) / size_mb ,
         SHAPEIT = (n_other_shapeit + n_flip_shapeit) / size_mb ) %>%
  tbl_summary(include = c(Beagle, Eagle, SHAPEIT),
              statistic = list(
                all_continuous() ~ "{mean} ({sd})",
                all_categorical() ~ "{n} / {N} ({p}%)")
              )

df_vcftools %>%
  mutate(Beagle = (n_other_beagle + n_flip_beagle) / size_mb ,
         Eagle = (n_other_eagle + n_flip_eagle) / size_mb ,
         SHAPEIT = (n_other_shapeit + n_flip_shapeit) / size_mb ) %>%
  tbl_summary(by = pop,
              include = c(Beagle, Eagle, SHAPEIT),
              statistic = list(
                all_continuous() ~ "{mean} ({sd})",
                all_categorical() ~ "{n} / {N} ({p}%)")
              )
```

#### F/T tests for differences

Errors per het

```{r}
fit <- df_vcftools %>%
  mutate(Beagle = (n_other_beagle + n_flip_beagle) / n_hets ,
         Eagle = (n_other_eagle + n_flip_eagle) / n_hets ,
         SHAPEIT = (n_other_shapeit + n_flip_shapeit) / n_hets ) %>%
  select(Beagle, Eagle, SHAPEIT) %>%
  mutate(unit_id = 1:1000) %>%
  pivot_longer(Beagle:SHAPEIT, names_to = "method", values_to = "error_rate") %>%
  {lmerTest::lmer(error_rate ~ method + (1|unit_id), data = .)}

anova(fit)
report::report(fit)
emmeans::emmeans(fit, ~ method) %>% pairs()


int_fit <- df_vcftools %>%
  mutate(Beagle = (n_other_beagle + n_flip_beagle) / n_hets ,
         Eagle = (n_other_eagle + n_flip_eagle) / n_hets ,
         SHAPEIT = (n_other_shapeit + n_flip_shapeit) / n_hets ) %>%
  select(Beagle, Eagle, SHAPEIT, pop) %>%
  mutate(unit_id = 1:1000) %>%
  pivot_longer(Beagle:SHAPEIT, names_to = "method", values_to = "error_rate") %>%
  {lmerTest::lmer(error_rate ~ method * pop + (1|unit_id), data = .)}

anova(int_fit) %>% broom::tidy()
emmeans::emmeans(int_fit, ~ method | pop) %>% pairs()


df_vcftools %>%
  mutate(Beagle = (n_other_beagle + n_flip_beagle) / n_hets ,
         Eagle = (n_other_eagle + n_flip_eagle) / n_hets ,
         SHAPEIT = (n_other_shapeit + n_flip_shapeit) / n_hets ) %>%
  mutate(diff = Eagle - Beagle) %>%
  t_test(response = diff, mu = 0)

df_vcftools %>%
  mutate(Beagle = (n_other_beagle + n_flip_beagle) / n_hets ,
         Eagle = (n_other_eagle + n_flip_eagle) / n_hets ,
         SHAPEIT = (n_other_shapeit + n_flip_shapeit) / n_hets ) %>%
  mutate(diff = SHAPEIT - Eagle) %>%
  t_test(response = diff, mu = 0)

df_vcftools %>%
  mutate(Beagle = (n_other_beagle + n_flip_beagle) / n_hets ,
         Eagle = (n_other_eagle + n_flip_eagle) / n_hets ,
         SHAPEIT = (n_other_shapeit + n_flip_shapeit) / n_hets ) %>%
  mutate(diff = SHAPEIT - Beagle) %>%
  t_test(response = diff, mu = 0)

mean_diff <- df_vcftools %>%
  mutate(Beagle = (n_other_beagle + n_flip_beagle) / n_hets ,
         Eagle = (n_other_eagle + n_flip_eagle) / n_hets ,
         SHAPEIT = (n_other_shapeit + n_flip_shapeit) / n_hets ) %>%
  mutate(diff = SHAPEIT - Eagle) %>%
  pull(diff) 

mean(mean_diff) / (sd(mean_diff)/sqrt(1000))
BSDA::z.test(mean_diff, sigma.x = sd(mean_diff))

mean_diff <- df_vcftools %>%
  mutate(Beagle = (n_other_beagle + n_flip_beagle) / size_mb ,
         Eagle = (n_other_eagle + n_flip_eagle) / size_mb ,
         SHAPEIT = (n_other_shapeit + n_flip_shapeit) / size_mb ) %>%
  mutate(diff = Beagle - Eagle) %>%
  pull(diff) 

mean(mean_diff) / (sd(mean_diff)/sqrt(1000))
BSDA::z.test(mean_diff, sigma.x = sd(mean_diff))
```

Difference Table:

```{r}
df_vcftools %>%
  mutate(Beagle = (n_other_beagle + n_flip_beagle) / n_hets ,
         Eagle = (n_other_eagle + n_flip_eagle) / n_hets ,
         SHAPEIT = (n_other_shapeit + n_flip_shapeit) / n_hets ) %>%
  mutate(B_E = Beagle - Eagle,
         B_S = Beagle - SHAPEIT,
         E_S = Eagle - SHAPEIT) %>%
  select(pop, B_E, B_S, E_S) %>%
  tbl_summary(by = pop,
              include = c(B_E, B_S, E_S),
              statistic = list(
                all_continuous() ~ "{mean} ({sd})",
                all_categorical() ~ "{n} / {N} ({p}%)")
              )
```

Population-specific T Tests

```{r}
df_vcftools %>%
  mutate(Beagle = (n_other_beagle + n_flip_beagle) / n_hets ,
         Eagle = (n_other_eagle + n_flip_eagle) / n_hets ,
         SHAPEIT = (n_other_shapeit + n_flip_shapeit) / n_hets ) %>%
  mutate(B_E = Beagle - Eagle,
         B_S = Beagle - SHAPEIT,
         E_S = Eagle - SHAPEIT) %>%
  select(pop, B_E, B_S, E_S) %>%
  group_by(pop) %>%
  summarize(m_B_E = mean(B_E), 
            m_B_S = mean(B_S), 
            m_E_S = mean(E_S),
            s_B_E = sd(B_E),
            s_B_S = sd(B_S),
            s_E_S = sd(E_S)) %>%
  rowwise() %>%
  mutate(z_B_E = m_B_E / (s_B_E / sqrt(200)),
         z_B_S = m_B_S / (s_B_S / sqrt(200)),
         z_E_S = m_E_S / (s_E_S / sqrt(200))) %>%
  mutate(p_B_E = 2 * pnorm(-abs(z_B_E)),
         p_B_S = 2 * pnorm(-abs(z_B_S)),
         p_E_S = 2 * pnorm(-abs(z_E_S))) %>%
  select(pop,
         m_B_E, z_B_E, p_B_E, 
         m_B_S, z_B_S, p_B_S,
         m_E_S, z_E_S, p_E_S) %>%
  knitr::kable()
```



### Accuracy

```{r}
df_vcftools %>%
  mutate(Beagle = (n_hets - n_other_beagle - ( 2 * n_flip_beagle) ) / n_hets ,
         Eagle = (n_hets - n_other_eagle - ( 2* n_flip_eagle)) / n_hets ,
         SHAPEIT = (n_hets - n_other_shapeit - (2* n_flip_shapeit)) / n_hets ) %>%
  tbl_summary(include = c(Beagle, Eagle, SHAPEIT),
              statistic = list(
                all_continuous() ~ "{mean} ({sd})",
                all_categorical() ~ "{n} / {N} ({p}%)")
              )

df_vcftools %>%
  mutate(Beagle = (n_hets - n_other_beagle - (2 * n_flip_beagle)) / n_hets ,
         Eagle = (n_hets - n_other_eagle - (2 * n_flip_eagle)) / n_hets ,
         SHAPEIT = (n_hets - n_other_shapeit - (2 * n_flip_shapeit)) / n_hets ) %>%
  tbl_summary(by = pop,
              include = c(Beagle, Eagle, SHAPEIT),
              statistic = list(
                all_continuous() ~ "{mean} ({sd})",
                all_categorical() ~ "{n} / {N} ({p}%)")
              )

fit <- df_vcftools %>%
  mutate(Beagle = (n_hets - n_other_beagle - (2 * n_flip_beagle)) / n_hets ,
         Eagle = (n_hets - n_other_eagle - (2 * n_flip_eagle)) / n_hets ,
         SHAPEIT = (n_hets - n_other_shapeit - (2 * n_flip_shapeit)) / n_hets ) %>%
  select(Beagle, Eagle, SHAPEIT, pop) %>%
  mutate(unit_id = 1:1000) %>%
  pivot_longer(Beagle:SHAPEIT, names_to = "method", values_to = "accuracy") %>%
  {lmerTest::lmer(accuracy ~ method + (1|unit_id), data = .)}
summary(fit)
anova(fit)
report::report(fit)
emmeans::emmeans(fit, ~ method) %>% pairs()

fit <- df_vcftools %>%
  mutate(Beagle = (n_hets - n_other_beagle - (2 * n_flip_beagle)) / n_hets ,
         Eagle = (n_hets - n_other_eagle - (2 * n_flip_eagle)) / n_hets ,
         SHAPEIT = (n_hets - n_other_shapeit - (2 * n_flip_shapeit)) / n_hets ) %>%
  select(Beagle, Eagle, SHAPEIT, pop) %>%
  mutate(unit_id = 1:1000) %>%
  pivot_longer(Beagle:SHAPEIT, names_to = "method", values_to = "accuracy") %>%
  {lmerTest::lmer(accuracy ~ method*pop + (1|unit_id), data = .)}
summary(fit)
anova(fit) %>% broom::tidy()
report::report(fit)
emmeans::emmeans(fit, ~ method | pop) %>% pairs()
```


### Mean number of switches and flips

```{r}
df_vcftools %>%
  summarize(mean_switch_eagle = mean(n_other_eagle),
            mean_switch_beagle = mean(n_other_beagle),
            mean_switch_shapeit = mean(n_other_shapeit),
            mean_flip_eagle = mean(n_flip_eagle),
            mean_flip_beagle = mean(n_flip_beagle),
            mean_flip_shapeit = mean(n_flip_shapeit)) %>%
  knitr::kable()

df_vcftools %>%
  mutate(B_E = n_other_beagle - n_other_eagle,
         B_S = n_other_beagle - n_other_shapeit,
         E_S = n_other_eagle - n_other_shapeit) %>%
  select(B_E, B_S, E_S) %>%
  as.matrix() %>%
  apply(MARGIN = 2, FUN = t.test)

df_vcftools %>%
  select(n_other_beagle, n_other_eagle) %>%
  {t.test(.$n_other_beagle, .$n_other_eagle, paired = T)}
```

#### By population

```{r}
df_vcftools %>%
  group_by(pop) %>%
  summarize(mean_switch_eagle = mean(n_other_eagle),
            mean_switch_beagle = mean(n_other_beagle),
            mean_switch_shapeit = mean(n_other_shapeit),
            mean_flip_eagle = mean(n_flip_eagle),
            mean_flip_beagle = mean(n_flip_beagle),
            mean_flip_shapeit = mean(n_flip_shapeit)) %>%
  knitr::kable()
```

### Mean number of switches and flips per MB

```{r}
df_vcftools %>%
  mutate(Beagle_switch = n_other_beagle / size_mb,
         Eagle_switch = n_other_eagle / size_mb,
         SHAPEIT_switch = n_other_shapeit / size_mb,
         Beagle_flip = n_flip_beagle / size_mb,
         Eagle_flip = n_flip_eagle / size_mb,
         SHAPEIT_flip = n_flip_shapeit / size_mb) %>%
  tbl_summary(include = c(Beagle_switch, Eagle_switch, SHAPEIT_switch,
                          Beagle_flip, Eagle_flip, SHAPEIT_flip),
              statistic = list(
                all_continuous() ~ "{mean} ({sd})",
                all_categorical() ~ "{n} / {N} ({p}%)")
              )
```

#### By population

```{r}
df_vcftools %>%
  mutate(Beagle_switch = n_other_beagle / size_mb,
         Eagle_switch = n_other_eagle / size_mb,
         SHAPEIT_switch = n_other_shapeit / size_mb,
         Beagle_flip = n_flip_beagle / size_mb,
         Eagle_flip = n_flip_eagle / size_mb,
         SHAPEIT_flip = n_flip_shapeit / size_mb) %>%
  tbl_summary(by = pop,
    include = c(Beagle_switch, Eagle_switch, SHAPEIT_switch,
                          Beagle_flip, Eagle_flip, SHAPEIT_flip),
              statistic = list(
                all_continuous() ~ "{mean} ({sd})",
                all_categorical() ~ "{n} / {N} ({p}%)")
              )
```


### Mean number of switches and flips per heterozygous site

```{r}
df_vcftools %>%
  mutate(Beagle_switch = n_other_beagle / n_hets,
         Eagle_switch = n_other_eagle / n_hets,
         SHAPEIT_switch = n_other_shapeit / n_hets,
         Beagle_flip = n_flip_beagle / n_hets,
         Eagle_flip = n_flip_eagle / n_hets,
         SHAPEIT_flip = n_flip_shapeit / n_hets) %>%
  tbl_summary(include = c(Beagle_switch, Eagle_switch, SHAPEIT_switch,
                          Beagle_flip, Eagle_flip, SHAPEIT_flip),
              statistic = list(
                all_continuous() ~ "{mean} ({sd})",
                all_categorical() ~ "{n} / {N} ({p}%)")
              )

fit <- df_vcftools %>%
  mutate(Beagle = n_other_beagle / n_hets,
         Eagle = n_other_eagle / n_hets,
         SHAPEIT = n_other_shapeit / n_hets) %>%
  select(Beagle, Eagle, SHAPEIT, pop) %>%
  mutate(unit_id = 1:1000) %>%
  pivot_longer(Beagle:SHAPEIT, names_to = "method", values_to = "switch_rate") %>%
  {lmerTest::lmer(switch_rate ~ method + (1|unit_id), data = .)}

anova(fit) %>% broom::tidy()
emmeans::emmeans(fit, ~ method) %>% pairs()

fit <- df_vcftools %>%
  mutate(Beagle = n_other_beagle / n_hets,
         Eagle = n_other_eagle / n_hets,
         SHAPEIT = n_other_shapeit / n_hets) %>%
  select(Beagle, Eagle, SHAPEIT, pop) %>%
  mutate(unit_id = 1:1000) %>%
  pivot_longer(Beagle:SHAPEIT, names_to = "method", values_to = "switch_rate") %>%
  {lmerTest::lmer(switch_rate ~ method * pop + (1|unit_id), data = .)}
anova(fit) %>% broom::tidy()

emmeans::emmeans(fit, ~ method | pop) %>% pairs()

fit <- df_vcftools %>%
  mutate(Beagle = n_flip_beagle / n_hets,
         Eagle = n_flip_eagle / n_hets,
         SHAPEIT = n_flip_shapeit / n_hets) %>%
  select(Beagle, Eagle, SHAPEIT, pop) %>%
  mutate(unit_id = 1:1000) %>%
  pivot_longer(Beagle:SHAPEIT, names_to = "method", values_to = "flip_rate") %>%
  {lmerTest::lmer(flip_rate ~ method + (1|unit_id), data = .)}

anova(fit) %>% broom::tidy()
emmeans::emmeans(fit, ~ method) %>% pairs()

fit <- df_vcftools %>%
  mutate(Beagle = n_flip_beagle / n_hets,
         Eagle = n_flip_eagle / n_hets,
         SHAPEIT = n_flip_shapeit / n_hets) %>%
  select(Beagle, Eagle, SHAPEIT, pop) %>%
  mutate(unit_id = 1:1000) %>%
  pivot_longer(Beagle:SHAPEIT, names_to = "method", values_to = "flip_rate") %>%
  {lmerTest::lmer(flip_rate ~ method * pop + (1|unit_id), data = .)}

anova(fit) %>% broom::tidy() 

emmeans::emmeans(fit, ~ method | pop) %>% pairs()
```

T-tests:

```{r}
df_vcftools %>%
  mutate(Beagle_switch = n_other_beagle / n_hets,
         Eagle_switch = n_other_eagle / n_hets,
         SHAPEIT_switch = n_other_shapeit / n_hets) %>%
  {t.test(.$Beagle_switch, .$Eagle_switch, paired = T)}

df_vcftools %>%
  mutate(Beagle_switch = n_other_beagle / n_hets,
         Eagle_switch = n_other_eagle / n_hets,
         SHAPEIT_switch = n_other_shapeit / n_hets) %>%
  {t.test(.$Beagle_switch, .$SHAPEIT_switch, paired = T)}

df_vcftools %>%
  mutate(Beagle_switch = n_other_beagle / n_hets,
         Eagle_switch = n_other_eagle / n_hets,
         SHAPEIT_switch = n_other_shapeit / n_hets) %>%
  {t.test(.$Eagle_switch, .$SHAPEIT_switch, paired = T)}

df_vcftools %>%
  mutate(Beagle_flip = n_flip_beagle / n_hets,
         Eagle_flip = n_flip_eagle / n_hets,
         SHAPEIT_flip = n_flip_shapeit / n_hets) %>%
  {t.test(.$Beagle_flip, .$Eagle_flip, paired = T)}

df_vcftools %>%
  mutate(Beagle_flip = n_flip_beagle / n_hets,
         Eagle_flip = n_flip_eagle / n_hets,
         SHAPEIT_flip = n_flip_shapeit / n_hets) %>%
  {t.test(.$Beagle_flip, .$SHAPEIT_flip, paired = T)}

df_vcftools %>%
  mutate(Beagle_flip = n_flip_beagle / n_hets,
         Eagle_flip = n_flip_eagle / n_hets,
         SHAPEIT_flip = n_flip_shapeit / n_hets) %>%
  {t.test(.$Eagle_flip, .$SHAPEIT_flip, paired = T)}
```


#### By population

```{r}
df_vcftools %>%
  mutate(Beagle_switch = n_other_beagle / n_hets,
         Eagle_switch = n_other_eagle / n_hets,
         SHAPEIT_switch = n_other_shapeit / n_hets,
         Beagle_flip = n_flip_beagle / n_hets,
         Eagle_flip = n_flip_eagle / n_hets,
         SHAPEIT_flip = n_flip_shapeit / n_hets) %>%
  tbl_summary(by = pop,
    include = c(Beagle_switch, Eagle_switch, SHAPEIT_switch,
                          Beagle_flip, Eagle_flip, SHAPEIT_flip),
              statistic = list(
                all_continuous() ~ "{mean} ({sd})",
                all_categorical() ~ "{n} / {N} ({p}%)")
              )
```


### Mean median number of heterozygous sites between errors

```{r}
df_vcftools %>%
  group_by(pop) %>%
  summarize(mean_het_switch_beagle = mean(med_hets_switch_beagle),
            mean_het_switch_eagle = mean(med_hets_switch_eagle),
            mean_het_switch_shapeit = mean(med_hets_switch_shapeit)) %>%
  knitr::kable()
```

```{r}
df_vcftools %>%
  group_by(pop) %>%
  summarize(mean_het_flip_beagle = mean(med_hets_flip_beagle),
            mean_het_flip_eagle = mean(med_hets_flip_eagle),
            mean_het_flip_shapeit = mean(med_hets_flip_shapeit)) %>%
  knitr::kable()
```

Note here that the populations will also differ in regards to the mean number of heterozygous sites in each pseudo-diploid:

```{r}
df_vcftools %>%
  group_by(pop) %>%
  summarize(mean_n_het = mean(n_hets),
            sd_n_het = sd(n_hets)) %>%
  knitr::kable()
```

### Mean mean number of heterozygous sites between errors

```{r}
df_vcftools %>%
  group_by(pop) %>%
  summarize(mean_het_switch_beagle = mean(mean_hets_switch_beagle),
            mean_het_switch_eagle = mean(mean_hets_switch_eagle),
            mean_het_switch_shapeit = mean(mean_hets_switch_shapeit)) %>%
  knitr::kable()
```

```{r}
df_vcftools %>%
  group_by(pop) %>%
  summarize(mean_het_flip_beagle = mean(mean_hets_flip_beagle),
            mean_het_flip_eagle = mean(mean_hets_flip_eagle),
            mean_het_flip_shapeit = mean(mean_hets_flip_shapeit)) %>%
  knitr::kable()
```

### Errors at CpGs

#### Total

```{r}
df_vcftools %>%
  summarize(mean_switch_eagle = mean(n_other_cpg_eagle),
            mean_switch_beagle = mean(n_other_cpg_beagle),
            mean_switch_shapeit = mean(n_other_cpg_shapeit),
            mean_flip_eagle = mean(n_flip_cpg_eagle),
            mean_flip_beagle = mean(n_flip_cpg_beagle),
            mean_flip_shapeit = mean(n_flip_cpg_shapeit)) %>%
  knitr::kable()
```

##### By Population

```{r}
df_vcftools %>%
  group_by(pop) %>%
  summarize(mean_switch_eagle = mean(n_other_cpg_eagle),
            mean_switch_beagle = mean(n_other_cpg_beagle),
            mean_switch_shapeit = mean(n_other_cpg_shapeit),
            mean_flip_eagle = mean(n_flip_cpg_eagle),
            mean_flip_beagle = mean(n_flip_cpg_beagle),
            mean_flip_shapeit = mean(n_flip_cpg_shapeit)) %>%
  knitr::kable()
```

#### Proportions

```{r}
df_vcftools %>%
  summarize(mean_switch_eagle = mean(n_other_cpg_eagle / n_other_eagle),
            mean_switch_beagle = mean(n_other_cpg_beagle / n_other_beagle),
            mean_switch_shapeit = mean(n_other_cpg_shapeit / n_other_shapeit),
            mean_flip_eagle = mean(n_flip_cpg_eagle / n_flip_eagle),
            mean_flip_beagle = mean(n_flip_cpg_beagle / n_flip_beagle),
            mean_flip_shapeit = mean(n_flip_cpg_shapeit / n_flip_shapeit)) %>%
  knitr::kable()
```

##### By Population

```{r}
df_vcftools %>%
  group_by(pop) %>%
  summarize(mean_switch_eagle = mean(n_other_cpg_eagle / n_other_eagle),
            mean_switch_beagle = mean(n_other_cpg_beagle / n_other_beagle),
            mean_switch_shapeit = mean(n_other_cpg_shapeit / n_other_shapeit),
            mean_flip_eagle = mean(n_flip_cpg_eagle / n_flip_eagle),
            mean_flip_beagle = mean(n_flip_cpg_beagle / n_flip_beagle),
            mean_flip_shapeit = mean(n_flip_cpg_shapeit / n_flip_shapeit)) %>% 
  knitr::kable()
```

#### Enrichment

We compute the ratio of the proportion of errors at CpGs to the proportion of heterozygous locations that are at CpGs.

```{r}
df_vcftools %>%
  summarize(mean_switch_eagle = mean((n_other_cpg_eagle / n_other_eagle) / ( n_het_cpg / n_hets)),
            mean_switch_beagle = mean((n_other_cpg_beagle / n_other_beagle) / ( n_het_cpg / n_hets)),
            mean_switch_shapeit = mean((n_other_cpg_shapeit / n_other_shapeit) / ( n_het_cpg / n_hets) ),
            mean_flip_eagle = mean((n_flip_cpg_eagle / n_flip_eagle) / ( n_het_cpg / n_hets)),
            mean_flip_beagle = mean((n_flip_cpg_beagle / n_flip_beagle) / ( n_het_cpg / n_hets)),
            mean_flip_shapeit = mean((n_flip_cpg_shapeit / n_flip_shapeit) / ( n_het_cpg / n_hets))) %>%
  knitr::kable()
```

Alternative view: odds ratios

```{r}
df_vcftools %>%
  mutate( prop_cpg_beagle =  n_other_cpg_beagle / n_other_beagle,
             prop_cpg_eagle =  n_other_cpg_eagle / n_other_eagle,
             prop_cpg_shapeit =  n_other_cpg_shapeit / n_other_shapeit) %>%
  mutate(ods_beagle = prop_cpg_beagle / (1 - prop_cpg_beagle),
         ods_eagle = prop_cpg_eagle / (1 - prop_cpg_eagle),
         ods_shapeit = prop_cpg_shapeit / (1 - prop_cpg_shapeit),
         ods_het = ( n_het_cpg / n_hets) / (1 - ( n_het_cpg / n_hets))) %>%
  mutate(Beagle = ods_beagle / ods_het,
         Eagle = ods_eagle / ods_het,
         SHAPEIT = ods_shapeit / ods_het) %>%
  select(pop, Beagle, Eagle, SHAPEIT) %>%
  pivot_longer(-pop, names_to = "Method", values_to = "OR") %>%
  ggplot(aes(x = Method, y = OR, colour = pop)) +
  geom_boxplot() +
  ylab("Odds CpG Switch / Odds CpG Het") +
  ggtitle("Odds Ratio of CpG Status for Switches") +
  theme_classic() +
  labs(colour = "Population") +
  scale_color_manual(values = cbPalette) +
  geom_hline(yintercept = 1, linetype=2)

df_vcftools %>%
  mutate( prop_cpg_beagle =  n_other_cpg_beagle / n_other_beagle,
             prop_cpg_eagle =  n_other_cpg_eagle / n_other_eagle,
             prop_cpg_shapeit =  n_other_cpg_shapeit / n_other_shapeit) %>%
  mutate(ods_beagle = prop_cpg_beagle / (1 - prop_cpg_beagle),
         ods_eagle = prop_cpg_eagle / (1 - prop_cpg_eagle),
         ods_shapeit = prop_cpg_shapeit / (1 - prop_cpg_shapeit),
         ods_het = ( n_het_cpg / n_hets) / (1 - ( n_het_cpg / n_hets))) %>%
  mutate(Beagle = ods_beagle / ods_het,
         Eagle = ods_eagle / ods_het,
         SHAPEIT = ods_shapeit / ods_het) %>%
  select(pop, Beagle, Eagle, SHAPEIT) %>%
  pivot_longer(-pop, names_to = "Method", values_to = "OR") %>%
  ggplot(aes(x = Method, y = OR)) +
  geom_boxplot() +
  ylab("Odds CpG Switch / Odds CpG Het") +
  ggtitle("Odds Ratio of CpG Status for Switches") +
  theme_classic() +
  scale_color_manual(values = cbPalette) +
  geom_hline(yintercept = 1, linetype=2)
```

And flips:

```{r}
df_vcftools %>%
  mutate( prop_cpg_beagle =  n_flip_cpg_beagle / n_flip_beagle,
             prop_cpg_eagle =  n_flip_cpg_eagle / n_flip_eagle,
             prop_cpg_shapeit =  n_flip_cpg_shapeit / n_flip_shapeit) %>%
  mutate(ods_beagle = prop_cpg_beagle / (1 - prop_cpg_beagle),
         ods_eagle = prop_cpg_eagle / (1 - prop_cpg_eagle),
         ods_shapeit = prop_cpg_shapeit / (1 - prop_cpg_shapeit),
         ods_het = ( n_het_cpg / n_hets) / (1 - ( n_het_cpg / n_hets))) %>%
  mutate(Beagle = ods_beagle / ods_het,
         Eagle = ods_eagle / ods_het,
         SHAPEIT = ods_shapeit / ods_het) %>%
  select(pop, Beagle, Eagle, SHAPEIT) %>%
  pivot_longer(-pop, names_to = "Method", values_to = "OR") %>%
  ggplot(aes(x = Method, y = OR, colour = pop)) +
  geom_boxplot() +
  ylab("Odds CpG Flip / Odds CpG Het") +
  ggtitle("Odds Ratio of CpG Status for Flips") +
  theme_classic() +
  labs(colour = "Population") +
  scale_color_manual(values = cbPalette) +
  geom_hline(yintercept = 1, linetype=2)

df_vcftools %>%
  mutate( prop_cpg_beagle =  n_flip_cpg_beagle / n_flip_beagle,
             prop_cpg_eagle =  n_flip_cpg_eagle / n_flip_eagle,
             prop_cpg_shapeit =  n_flip_cpg_shapeit / n_flip_shapeit) %>%
  mutate(ods_beagle = prop_cpg_beagle / (1 - prop_cpg_beagle),
         ods_eagle = prop_cpg_eagle / (1 - prop_cpg_eagle),
         ods_shapeit = prop_cpg_shapeit / (1 - prop_cpg_shapeit),
         ods_het = ( n_het_cpg / n_hets) / (1 - ( n_het_cpg / n_hets))) %>%
  mutate(Beagle = ods_beagle / ods_het,
         Eagle = ods_eagle / ods_het,
         SHAPEIT = ods_shapeit / ods_het) %>%
  select(pop, Beagle, Eagle, SHAPEIT) %>%
  pivot_longer(-pop, names_to = "Method", values_to = "OR") %>%
  ggplot(aes(x = Method, y = OR)) +
  geom_boxplot() +
  ylab("Odds CpG Flip / Odds CpG Het") +
  ggtitle("Odds Ratio of CpG Status for Flips") +
  theme_classic() +
  scale_color_manual(values = cbPalette) +
  geom_hline(yintercept = 1, linetype=2)

```

```{r}
df_vcftools %>%
  summarize(mean_switch_eagle = mean((n_other_cpg_eagle / n_other_eagle) / ( n_het_cpg / n_hets)),
            mean_switch_beagle = mean((n_other_cpg_beagle / n_other_beagle) / ( n_het_cpg / n_hets)),
            mean_switch_shapeit = mean((n_other_cpg_shapeit / n_other_shapeit) / ( n_het_cpg / n_hets) ),
            mean_flip_eagle = mean((n_flip_cpg_eagle / n_flip_eagle) / ( n_het_cpg / n_hets)),
            mean_flip_beagle = mean((n_flip_cpg_beagle / n_flip_beagle) / ( n_het_cpg / n_hets)),
            mean_flip_shapeit = mean((n_flip_cpg_shapeit / n_flip_shapeit) / ( n_het_cpg / n_hets))) %>%
  knitr::kable()

perform_one_sided_ttest_purrr <- function(column_data) {
  if (!is.numeric(column_data) || length(na.omit(column_data)) < 2) {
    return(tibble(t.statistic = NA, p.value = NA, mean = NA, conf.low = NA, conf.high = NA))
  }
  
  test_result <- t.test(column_data, mu = 1, alternative = "greater")
  
  # Return results as a tibble (dataframe row) for easier combining
  tibble(
    t.statistic = test_result$statistic,
    p.value = test_result$p.value,
    mean = test_result$estimate,
    conf.low = test_result$conf.int[1],
    conf.high = test_result$conf.int[2]
  )
}
  
# t tests
df_vcftools %>%
  mutate(or_switch_eagle = ((n_other_cpg_eagle / n_other_eagle) / ( n_het_cpg / n_hets)),
            or_switch_beagle = ((n_other_cpg_beagle / n_other_beagle) / ( n_het_cpg / n_hets)),
            or_switch_shapeit = ((n_other_cpg_shapeit / n_other_shapeit) / ( n_het_cpg / n_hets) ),
            or_flip_eagle = ((n_flip_cpg_eagle / n_flip_eagle) / ( n_het_cpg / n_hets)),
            or_flip_beagle = ((n_flip_cpg_beagle / n_flip_beagle) / ( n_het_cpg / n_hets)),
            or_flip_shapeit = ((n_flip_cpg_shapeit / n_flip_shapeit) / ( n_het_cpg / n_hets))) %>%
  select(starts_with("or_")) %>%
  purrr::map_df(perform_one_sided_ttest_purrr, .id = "ColumnName") %>%
  knitr::kable()
```

##### By Population

```{r}
df_vcftools %>%
  group_by(pop) %>%
  summarize(mean_switch_eagle = mean((n_other_cpg_eagle / n_other_eagle) / ( n_het_cpg / n_hets)),
            mean_switch_beagle = mean((n_other_cpg_beagle / n_other_beagle) / ( n_het_cpg / n_hets)),
            mean_switch_shapeit = mean((n_other_cpg_shapeit / n_other_shapeit) / ( n_het_cpg / n_hets) ),
            mean_flip_eagle = mean((n_flip_cpg_eagle / n_flip_eagle) / ( n_het_cpg / n_hets)),
            mean_flip_beagle = mean((n_flip_cpg_beagle / n_flip_beagle) / ( n_het_cpg / n_hets)),
            mean_flip_shapeit = mean((n_flip_cpg_shapeit / n_flip_shapeit) / ( n_het_cpg / n_hets))) %>%
  knitr::kable()
```

## Figures

### Total Errors

```{r include=FALSE}
b_l_e <- df_vcftools %>%
  mutate(Beagle = n_other_beagle + n_flip_beagle,
         Eagle = n_other_eagle + n_flip_eagle, 
         SHAPEIT = n_other_shapeit + n_flip_shapeit) %>%
  mutate(b_l_e = Beagle < Eagle) %>%
  pull(b_l_e) %>%
  sum()

s_l_e <- df_vcftools %>%
  mutate(Beagle = n_other_beagle + n_flip_beagle,
         Eagle = n_other_eagle + n_flip_eagle, 
         SHAPEIT = n_other_shapeit + n_flip_shapeit) %>%
  mutate(b_l_e = SHAPEIT < Eagle) %>%
  pull(b_l_e) %>%
  sum()

s_l_b <- df_vcftools %>%
  mutate(Beagle = n_other_beagle + n_flip_beagle,
         Eagle = n_other_eagle + n_flip_eagle, 
         SHAPEIT = n_other_shapeit + n_flip_shapeit) %>%
  mutate(b_l_e = SHAPEIT < Beagle) %>%
  pull(b_l_e) %>%
  sum()

b_l_s <- df_vcftools %>%
  mutate(Beagle = n_other_beagle + n_flip_beagle,
         Eagle = n_other_eagle + n_flip_eagle, 
         SHAPEIT = n_other_shapeit + n_flip_shapeit) %>%
  mutate(b_l_s = SHAPEIT > Beagle) %>%
  pull(b_l_s) %>%
  sum()
```


Comparing Eagle to Beagle, we observed a higher number of errors in Eagle in `r b_l_e` synthetic diploids, while for SHAPEIT and Eagle we see more errors in Eagle than in SHAPEIT in `r s_l_e` synthetic diploids.

#### Boxplots

```{r}
df_vcftools %>%
  mutate(Beagle = n_flip_beagle + n_other_beagle,
         Eagle = n_other_eagle + n_flip_eagle,
         SHAPEIT = n_other_shapeit + n_flip_shapeit) %>%
  select(n_hets, Beagle, Eagle, SHAPEIT) %>%
  colMeans()
```


Across a mean number of 68929.5 heterozygous sites, we observed an mean number of 197.7 errors for Beagle, 236.5 for Eagle and 201.2 for SHAPEIT. Thus, the three algorithms reconstructed X haplotypes to a high degree of accuracy, with error rates of 0.0031, 0.0037, and 0.0031 errors per heterozygous position for Beagle, Eagle, and SHAPEIT, respectively. While similar, these error rates were found to be significantly different ($F(2, 1998) = 2542$, $p < 0.001$), with Eagle having a higher error rate than both Beagle (mean difference $5.87\times 10^{-4}$, $p < 0.001$) and SHAPEIT (mean difference $5.4 \times 10^{-4}$, $p < 0.001$), and SHAPEIT having a higher error rate than Beagle (mean difference $4.6 \times 10^{-5}$, $p < 0.001$).

```{r}
df_vcftools %>%
  mutate(Beagle = n_other_beagle + n_flip_beagle,
         Eagle = n_other_eagle + n_flip_eagle, 
         SHAPEIT = n_other_shapeit + n_flip_shapeit) %>%
  select(pair_id, Beagle, Eagle, SHAPEIT) %>%
  pivot_longer(-pair_id, names_to = "method", values_to = "errors") %>%
  ggplot(aes(x = method, y = errors)) + 
  geom_boxplot() +
  xlab("Method") +
  ylab("Errors") +
  theme_classic()
```

```{r}
df_vcftools %>%
  mutate(Beagle = (n_other_beagle + n_flip_beagle) / n_hets,
         Eagle = (n_other_eagle + n_flip_eagle) / n_hets, 
         SHAPEIT =( n_other_shapeit + n_flip_shapeit) / n_hets) %>%
  select(pair_id, Beagle, Eagle, SHAPEIT) %>%
  pivot_longer(-pair_id, names_to = "method", values_to = "errors") %>%
  ggplot(aes(x = method, y = errors)) + 
  geom_boxplot() +
  xlab("Method") +
  ylab("Error Rate per Heterozygous Position") +
  theme_classic()

df_vcftools %>%
  mutate(Beagle = (n_other_beagle + n_flip_beagle) / n_hets,
         Eagle = (n_other_eagle + n_flip_eagle) / n_hets, 
         SHAPEIT =( n_other_shapeit + n_flip_shapeit) / n_hets) %>%
  select(pair_id, Beagle, Eagle, SHAPEIT) %>%
  pivot_longer(-pair_id, names_to = "method", values_to = "errors") %>%
  ggplot(aes(x = method, y = errors)) + 
  geom_violin(draw_quantiles = c(0.5)) +
  xlab("Method") +
  ylab("Error Rate per Heterozygous Position") +
  theme_classic()
```

```{r}
df_vcftools %>%
  mutate(Beagle = (n_other_beagle + n_flip_beagle) / n_hets,
         Eagle = (n_other_eagle + n_flip_eagle) / n_hets, 
         SHAPEIT =( n_other_shapeit + n_flip_shapeit) / n_hets) %>%
  select(pair_id, pop, Beagle, Eagle, SHAPEIT) %>%
  pivot_longer(Beagle:SHAPEIT, names_to = "method", values_to = "errors") %>%
  ggplot(aes(x = method, y = errors, colour = pop)) + 
  geom_boxplot() +
  xlab("Method") +
  ylab("Error Rate per Heterozygous Position") +
  scale_color_manual(values = cbPalette) +
  theme_classic()
```

```{r}
df_vcftools %>%
  mutate(Beagle = n_other_beagle / n_hets,
         Eagle = n_other_eagle / n_hets,
         SHAPEIT = n_other_shapeit / n_hets,
         type = "Switches") %>%
  select(pair_id, pop, type, Beagle, Eagle, SHAPEIT) %>%
  bind_rows({df_vcftools %>%
      mutate(Beagle = n_flip_beagle / n_hets,
             Eagle = n_flip_eagle / n_hets,
             SHAPEIT = n_flip_shapeit / n_hets, 
             type = "Flips") %>%
      select(pair_id, pop, type, Beagle, Eagle, SHAPEIT)}) %>%
  pivot_longer(Beagle:SHAPEIT, names_to = "method", values_to = "errors") %>%
  ggplot(aes(x = method, y = errors)) +
  geom_boxplot() +
  xlab("") +
  ylab("Error Rate") +
  theme_bw() +
  facet_grid(cols = vars(type)) +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank())
```

```{r}
df_vcftools %>%
  mutate(Beagle = n_other_beagle / n_hets,
         Eagle = n_other_eagle / n_hets,
         SHAPEIT = n_other_shapeit / n_hets,
         type = "Switches") %>%
  select(pair_id, pop, type, Beagle, Eagle, SHAPEIT) %>%
  bind_rows({df_vcftools %>%
      mutate(Beagle = n_flip_beagle / n_hets,
             Eagle = n_flip_eagle / n_hets,
             SHAPEIT = n_flip_shapeit / n_hets, 
             type = "Flips") %>%
      select(pair_id, pop, type, Beagle, Eagle, SHAPEIT)}) %>%
  pivot_longer(Beagle:SHAPEIT, names_to = "method", values_to = "errors") %>%
  ggplot(aes(x = method, y = errors, colour = pop)) +
  geom_boxplot() +
  xlab("") +
  ylab("Error Rate") +
  theme_bw() +
  scale_color_manual(values = cbPalette) +
  labs(colour = "Population") +
  facet_grid(cols = vars(type)) +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank())
```

```{r}
df_vcftools %>%
  mutate(Beagle = n_other_beagle / n_hets,
         Eagle = n_other_eagle / n_hets,
         SHAPEIT = n_other_shapeit / n_hets,
         type = "Switches") %>%
  select(pair_id, pop, type, Beagle, Eagle, SHAPEIT) %>%
  bind_rows({df_vcftools %>%
      mutate(Beagle = n_flip_beagle / n_hets,
             Eagle = n_flip_eagle / n_hets,
             SHAPEIT = n_flip_shapeit / n_hets, 
             type = "Flips") %>%
      select(pair_id, pop, type, Beagle, Eagle, SHAPEIT)}) %>%
  pivot_longer(Beagle:SHAPEIT, names_to = "method", values_to = "errors") %>%
  ggplot(aes(x = method, y = errors, colour = type)) +
  geom_boxplot() +
  xlab("") +
  ylab("Error Rate") +
  theme_bw() +
  scale_color_manual(values = cbPalette) +
  labs(colour = "Error Type") +
  facet_grid(cols = vars(pop)) +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank()) + # rotate xaxis text labels
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```

```{r}
df_vcftools %>%
  mutate(Beagle = n_flip_beagle + n_other_beagle,
         Eagle = n_other_eagle + n_flip_eagle,
         SHAPEIT = n_other_shapeit + n_flip_shapeit) %>%
  select(pop, subpop, n_hets, Beagle, Eagle, SHAPEIT) %>%
  group_by(pop, subpop) %>%
  summarize(n_het = mean(n_hets),
            Beagle = mean(Beagle),
            Eagle = mean(Eagle),
            SHAPEIT = mean(SHAPEIT)) %>%
  arrange(desc(Beagle)) %>%
  knitr::kable()

df_vcftools %>%
  mutate(Beagle = n_flip_beagle + n_other_beagle,
         Eagle = n_other_eagle + n_flip_eagle,
         SHAPEIT = n_other_shapeit + n_flip_shapeit) %>%
  select(pop, subpop, n_hets, Beagle, Eagle, SHAPEIT) %>%
  group_by(pop, subpop) %>%
  summarize(n_het = mean(n_hets),
            Beagle = mean(Beagle),
            Eagle = mean(Eagle),
            SHAPEIT = mean(SHAPEIT)) %>%
  pivot_longer(Beagle:SHAPEIT, names_to = "method", values_to = "errors") %>%
  ggplot(aes(x = n_het, y = errors, colour = method)) +
  geom_point() +
  xlab("Mean Heterozygous Positions") +
  ylab("Mean Number Errors") +
  labs(colour = "Method") +
  facet_wrap(vars(pop))
```


Comparing across populations, both the number of errors and the number of heterozygous positions differed greatly. While e.g. YRA samples had the largest number of errors, they also had the largest number of heterozygous sites, resulting in a low overall error rate.  Overall, EAS and SAS had the two highest error rates across all three methods (Table ~\ref{tab:mean_err_pop}). The differences between methods were also found to be significantly different between populations ($F(8, 1990) = 71.08$, $p < 0.001$). Comparing methods within populations we observed that in all populations Beagle had the lowest error rate, followed by SHAPEIT while Eagle had the largest error rate. and while many pairwise comparisons of methods remain significant within each population with no change in ranking between methods, the difference between Beagle and SHAPEIT is not significant in the EAS ($t(1990)=1.42$, $p = 0.331$) and SAS ($t(1990) = 1.51$, $p = 0.286$) populations. Within each method, error rates were found to be significantly different between all pairs of populations.


```{r}
df_vcftools %>%
  mutate(Beagle = n_other_beagle / (n_other_beagle + n_flip_beagle),
         Eagle = n_other_eagle / (n_other_eagle + n_flip_eagle),
         SHAPEIT = n_other_shapeit / (n_other_shapeit + n_flip_shapeit),
         Total = (n_other_beagle + n_other_eagle + n_other_shapeit) / (n_other_beagle + n_flip_beagle + n_other_eagle + n_flip_eagle + n_other_shapeit + n_flip_shapeit)) %>%
  select(Beagle:Total) %>%
  colMeans()
```


We next separated overall errors into switch errors and flip errors. Overall, switch errors constituted 54.3$\%$ of the overall errors. Switch rates differed significantly between the three methods ($F(2, 1998) = 4651.30$, $p < 0.001$), with SHAPEIT ($1.57 \times 10^{-3}$ switches/heterozygous position) having a lower average switch rate than both Beagle ($1.70 \times 10^{-3}$, $t(1998) = 16.84$, $p < 0.01$) and Eagle ($2.29 \times 10^{-3}$, $t(1998) = 90.67$, $p < 0.001$).  Flip rates also varried significantly between the three methods ($F(2, 1998) = 638.99$, $p < 0.001$), but unlike switches, SHAPEIT ($1.56 \times 10^{-3}$ flips/het) was found to have a significantly higher flip rate than Beagle ($1.38 \times 10^{-3}$ flip/het; $t(1998) = 30.57$, $p < 0.001$) and Eagle ($1.37 \times 10^{-3}$ flips/het; $t(1998) = 31.34$, $p < 0.001$), whereas the difference between Beagle and Eagles flip rates was not significant ($t(1998) = 0.77$, $p = 0.72$). Both switch rates ($F(4, 995) = 382.04$, $p < 0.001$) and flip rates ($F(4, 995) = 256.57$, $p < 0.001$) were found to have significant differences between populations, but only switch rates were found to have a significant difference in method contrasts between populations ($F(8, 1990) = 89.15$, $p < 0.001$). The ordering of the methods by switch rates and flip rates was found to be consistent across the five populations. For both switch and flip errors, the population with the highest average rate is EAS, while the population with the lowest average rate is AFR.

```{r}
df_vcftools %>%
  mutate(Beagle = n_other_beagle,
         Eagle = n_other_eagle ,
         SHAPEIT = n_other_shapeit ) %>%
  select(pop, subpop, n_hets, Beagle, Eagle, SHAPEIT) %>%
  group_by(pop, subpop) %>%
  summarize(n_het = mean(n_hets),
            Beagle = mean(Beagle),
            Eagle = mean(Eagle),
            SHAPEIT = mean(SHAPEIT)) %>%
  arrange(desc(Eagle)) %>%
  knitr::kable()

df_vcftools %>%
  mutate(Beagle = n_other_beagle,
         Eagle = n_other_eagle ,
         SHAPEIT = n_other_shapeit ) %>%
  select(pop, subpop, n_hets, Beagle, Eagle, SHAPEIT) %>%
  group_by(pop, subpop) %>%
  summarize(n_het = mean(n_hets),
            Beagle = mean(Beagle),
            Eagle = mean(Eagle),
            SHAPEIT = mean(SHAPEIT)) %>%
  pivot_longer(Beagle:SHAPEIT, names_to = "method", values_to = "switch") %>%
  ggplot(aes(x = n_het, y = switch, colour = pop)) +
  geom_point() +
  facet_wrap(vars(method)) +
  theme_bw() + # remove grid
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank())

df_vcftools %>%
  mutate(Beagle = n_flip_beagle,
         Eagle = n_flip_eagle ,
         SHAPEIT = n_flip_shapeit ) %>%
  select(pop, subpop, n_hets, Beagle, Eagle, SHAPEIT) %>%
  group_by(pop, subpop) %>%
  summarize(n_het = mean(n_hets),
            Beagle = mean(Beagle),
            Eagle = mean(Eagle),
            SHAPEIT = mean(SHAPEIT)) %>%
  pivot_longer(Beagle:SHAPEIT, names_to = "method", values_to = "flip") %>%
  ggplot(aes(x = n_het, y = flip, colour = pop)) +
  geom_point() +
  facet_wrap(vars(method)) + 
  theme_bw() + 
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank())
```


#### Scatterplots

```{r}
df_vcftools %>%
  mutate(Beagle = n_other_beagle + n_flip_beagle,
         Eagle = n_other_eagle + n_flip_eagle, 
         SHAPEIT = n_other_shapeit + n_flip_shapeit) %>%
  ggplot(aes(x = Beagle, y = Eagle, color = pop)) +
  geom_point() +
  ggtitle("Total Errors") +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette) +
  labs(color = "Population")
```


```{r}
df_vcftools %>%
  mutate(Beagle = n_other_beagle + n_flip_beagle,
         Eagle = n_other_eagle + n_flip_eagle, 
         SHAPEIT = n_other_shapeit + n_flip_shapeit) %>%
  ggplot(aes(x = Beagle, y = SHAPEIT, color = pop)) +
  geom_point() +
  ggtitle("Total Errors") +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette)
```

```{r}
df_vcftools %>%
  mutate(Beagle = n_other_beagle + n_flip_beagle,
         Eagle = n_other_eagle + n_flip_eagle, 
         SHAPEIT = n_other_shapeit + n_flip_shapeit) %>%
  ggplot(aes(x = Eagle, y = SHAPEIT, color = pop)) +
  geom_point() +
  ggtitle("Total Errors") +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette)
```

All in one figure:

```{r}
cross_meth_plot <- function(df, p_title) {
  p1 <- df %>%
    ggplot(aes(x = Beagle, y = Eagle, color = pop)) +
    geom_point() +
    theme_classic() +
    geom_abline(slope = 1, intercept = 0) +
    scale_color_manual(values = cbPalette) +
    labs(color = "Population")
  
  p_legend <- cowplot::get_legend(p1)
  
  p1 <- p1 + guides(colour="none")
  
  p2 <- df %>%
    ggplot(aes(x = Beagle, y = SHAPEIT, color = pop)) +
    geom_point() +
    ggtitle(p_title) +
    theme_classic() +
    geom_abline(slope = 1, intercept = 0) +
    scale_color_manual(values = cbPalette) +
    guides(colour="none")
  
  p3 <- df %>%
    ggplot(aes(x = SHAPEIT, y = Eagle, color = pop)) +
    geom_point() +
    theme_classic() +
    geom_abline(slope = 1, intercept = 0) +
    scale_color_manual(values = cbPalette) +
    guides(colour="none")
  
  return(grid.arrange(p2, p_legend, p1, p3, ncol = 2))
  }
df_vcftools %>%
  mutate(Beagle = n_other_beagle + n_flip_beagle,
         Eagle = n_other_eagle + n_flip_eagle, 
         SHAPEIT = n_other_shapeit + n_flip_shapeit) %>%
  cross_meth_plot(p_title = "Total Errors")
```

#### Error rates (per heterozygous site)

```{r}
df_vcftools %>%
  mutate(Beagle = (n_other_beagle + n_flip_beagle) / n_hets,
         Eagle = (n_other_eagle + n_flip_eagle) / n_hets, 
         SHAPEIT = (n_other_shapeit + n_flip_shapeit) / n_hets) %>%
  ggplot(aes(x = Beagle, y = Eagle, color = pop)) +
  geom_point() +
  ggtitle("Total Errors per Heterozygous Site") +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette)
```

```{r}
df_vcftools %>%
  mutate(Beagle = (n_other_beagle + n_flip_beagle) / n_hets,
         Eagle = (n_other_eagle + n_flip_eagle) / n_hets, 
         SHAPEIT = (n_other_shapeit + n_flip_shapeit) / n_hets) %>%
  ggplot(aes(x = Beagle, y = SHAPEIT, color = pop)) +
  geom_point() +
  ggtitle("Total Errors per Heterozygous Site") +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette)
```

```{r}
df_vcftools %>%
  mutate(Beagle = (n_other_beagle + n_flip_beagle) / n_hets,
         Eagle = (n_other_eagle + n_flip_eagle) / n_hets, 
         SHAPEIT = (n_other_shapeit + n_flip_shapeit) / n_hets) %>%
  ggplot(aes(x = Eagle, y = SHAPEIT, color = pop)) +
  geom_point() +
  ggtitle("Total Errors per Heterozygous Site") +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette)
```

All in one figure:

```{r}
df_vcftools %>%
  mutate(Beagle = (n_other_beagle + n_flip_beagle) / n_hets,
         Eagle = (n_other_eagle + n_flip_eagle) / n_hets, 
         SHAPEIT = (n_other_shapeit + n_flip_shapeit) / n_hets) %>%
  cross_meth_plot(p_title = "Total Error Rates")
```

```{r}
df_vcftools %>%
  mutate(Beagle = (n_other_beagle + n_flip_beagle) / n_hets,
         Eagle = (n_other_eagle + n_flip_eagle) / n_hets, 
         SHAPEIT = (n_other_shapeit + n_flip_shapeit) / n_hets) %>%
  select(pair_id, pop, Beagle, Eagle, SHAPEIT) %>%
  pivot_longer(Beagle:SHAPEIT, names_to = "method", values_to = "error_rate") %>%
  ggplot(aes(x = method, y = error_rate)) +
  geom_boxplot() +
  xlab("Method") +
  ylab("Error Rate")

df_vcftools %>%
  mutate(Beagle = (n_other_beagle + n_flip_beagle) / n_hets,
         Eagle = (n_other_eagle + n_flip_eagle) / n_hets, 
         SHAPEIT = (n_other_shapeit + n_flip_shapeit) / n_hets) %>%
  select(pair_id, pop, Beagle, Eagle, SHAPEIT) %>%
  pivot_longer(Beagle:SHAPEIT, names_to = "method", values_to = "error_rate") %>%
  ggplot(aes(x = pop, y = error_rate, colour = method)) +
  geom_boxplot() +
  xlab("Population") +
  ylab("Error Rate") +
  scale_color_manual(values = cbPalette) +
  labs(color = "Method") +
  theme_classic()
```

```{r}
df_vcftools %>%
  mutate(Beagle = (n_other_beagle + n_flip_beagle) / n_hets,
         Eagle = (n_other_eagle + n_flip_eagle) / n_hets, 
         SHAPEIT = (n_other_shapeit + n_flip_shapeit) / n_hets) %>%
  select(Beagle, Eagle, SHAPEIT) %>%
  corrr::correlate() %>%
  corrr::shave()
```



### Non-flip switches

```{r include=FALSE}
s_l_b <- df_vcftools %>%
  mutate(Beagle = (n_other_beagle),
         Eagle = (n_other_eagle ), 
         SHAPEIT = (n_other_shapeit)) %>%
  mutate(s_l_b = SHAPEIT < Beagle) %>%
  pull(s_l_b) %>%
  sum()

s_l_e<- df_vcftools %>%
  mutate(Beagle = (n_other_beagle),
         Eagle = (n_other_eagle ), 
         SHAPEIT = (n_other_shapeit)) %>%
  mutate(s_l_e = SHAPEIT < Eagle) %>%
  pull(s_l_e) %>%
  sum()

b_l_e<- df_vcftools %>%
  mutate(Beagle = (n_other_beagle),
         Eagle = (n_other_eagle ), 
         SHAPEIT = (n_other_shapeit)) %>%
  mutate(b_l_e = Beagle < Eagle) %>%
  pull(b_l_e) %>%
  sum()
```

For switches, we observe fewer errors in SHAPEIT than in Beagle in `r s_l_b` synthetic diploids, and in `r s_l_e` synthetic diploids we observe fewer errors in SHAPEIT than in Eagle. Comparing Beagle to Eagle, we observe fewer errors in Beagle than in Eagle in `r b_l_e` synthetic diploids.

```{r}
df_vcftools %>%
  mutate(Beagle = (n_other_beagle),
         Eagle = (n_other_eagle ), 
         SHAPEIT = (n_other_shapeit)) %>%
  ggplot(aes(x = Beagle, y = Eagle, color = pop)) +
  geom_point() +
  ggtitle("Switches") +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette)
```

```{r}
df_vcftools %>%
   mutate(Beagle = (n_other_beagle),
         Eagle = (n_other_eagle ), 
         SHAPEIT = (n_other_shapeit)) %>%
  ggplot(aes(x = Beagle, y = SHAPEIT, color = pop)) +
  geom_point() +
  ggtitle("Switches") +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette)
```

```{r}
df_vcftools %>%
   mutate(Beagle = (n_other_beagle),
         Eagle = (n_other_eagle ), 
         SHAPEIT = (n_other_shapeit)) %>%
  ggplot(aes(x = Eagle, y = SHAPEIT, color = pop)) +
  geom_point() +
  ggtitle("Switches") +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette)
```

```{r}
df_vcftools %>%
   mutate(Beagle = (n_other_beagle),
         Eagle = (n_other_eagle ), 
         SHAPEIT = (n_other_shapeit)) %>%
  cross_meth_plot(p_title = "Switches")
```


#### Per het. site

```{r}
df_vcftools %>%
  mutate(Beagle = (n_other_beagle) / n_hets,
         Eagle = (n_other_eagle ) / n_hets, 
         SHAPEIT = (n_other_shapeit) / n_hets) %>%
  ggplot(aes(x = Beagle, y = Eagle, color = pop)) +
  geom_point() +
  ggtitle("Switches per Heterozygous Site") +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette)
```

```{r}
df_vcftools %>%
  mutate(Beagle = (n_other_beagle) / n_hets,
         Eagle = (n_other_eagle ) / n_hets, 
         SHAPEIT = (n_other_shapeit) / n_hets) %>%
  ggplot(aes(x = Beagle, y = SHAPEIT, color = pop)) +
  geom_point() +
  ggtitle("Switches per Heterozygous Site") +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette)
```

```{r}
df_vcftools %>%
  mutate(Beagle = (n_other_beagle) / n_hets,
         Eagle = (n_other_eagle ) / n_hets, 
         SHAPEIT = (n_other_shapeit) / n_hets) %>%
  ggplot(aes(x = Eagle, y = SHAPEIT, color = pop)) +
  geom_point() +
  ggtitle("Switches per Heterozygous Site") +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette)
```

```{r}
df_vcftools %>%
  mutate(Beagle = (n_other_beagle) / n_hets,
         Eagle = (n_other_eagle ) / n_hets, 
         SHAPEIT = (n_other_shapeit) / n_hets) %>%
  cross_meth_plot(p_title = "Switch Rate")
```

```{r}
df_vcftools %>%
  mutate(Beagle = (n_other_beagle) / n_hets,
         Eagle = (n_other_eagle ) / n_hets, 
         SHAPEIT = (n_other_shapeit) / n_hets) %>%
  select(pair_id, pop, Beagle, Eagle, SHAPEIT) %>%
  pivot_longer(Beagle:SHAPEIT, names_to = "method", values_to = "error_rate") %>%
  ggplot(aes(x = method, y = error_rate)) +
  geom_boxplot() +
  xlab("Method") +
  ylab("Switch Rate") +
  theme_classic()
```

```{r}
df_vcftools %>%
  mutate(Beagle = (n_other_beagle) / n_hets,
         Eagle = (n_other_eagle ) / n_hets, 
         SHAPEIT = (n_other_shapeit) / n_hets) %>%
  select(pair_id, pop, Beagle, Eagle, SHAPEIT) %>%
  pivot_longer(Beagle:SHAPEIT, names_to = "method", values_to = "error_rate") %>%
  ggplot(aes(x = pop, y = error_rate, colour = method)) +
  geom_boxplot() +
  xlab("Population") +
  ylab("Switch Rate") +
  scale_color_manual(values = cbPalette) +
  labs(color = "Method") +
  theme_classic()
```

```{r}
df_vcftools %>%
  mutate(Beagle = (n_other_beagle) / n_hets,
         Eagle = (n_other_eagle ) / n_hets, 
         SHAPEIT = (n_other_shapeit) / n_hets) %>%
  select(Beagle, Eagle, SHAPEIT) %>%
  corrr::correlate() %>%
  corrr::shave()
```


### Flips

```{r include=FALSE}
b_l_s <- df_vcftools %>%
  mutate(Beagle = (n_flip_beagle),
         Eagle = (n_flip_eagle ), 
         SHAPEIT = (n_flip_shapeit)) %>%
  mutate(b_l_s = SHAPEIT > Beagle) %>%
  pull(b_l_s) %>%
  sum()

e_l_s<- df_vcftools %>%
  mutate(Beagle = (n_flip_beagle),
         Eagle = (n_flip_eagle ), 
         SHAPEIT = (n_flip_shapeit)) %>%
  mutate(e_l_s = SHAPEIT > Eagle) %>%
  pull(e_l_s) %>%
  sum()

e_l_b<- df_vcftools %>%
  mutate(Beagle = (n_flip_beagle),
         Eagle = (n_flip_eagle ), 
         SHAPEIT = (n_flip_shapeit)) %>%
  mutate(e_l_b = Beagle > Eagle) %>%
  pull(e_l_b) %>%
  sum()
```

For switches, we observe fewer errors in Beagle than in SHAPEIT in `r b_l_s` synthetic diploids, and in `r e_l_s` synthetic diploids we observe fewer errors in Eagle than in SHAPEIT Comparing Beagle to Eagle, we observe fewer errors in Eagle than in Beagle in `r e_l_b` synthetic diploids.

```{r}
df_vcftools %>%
  mutate(Beagle = (n_flip_beagle),
         Eagle = (n_flip_eagle ), 
         SHAPEIT = (n_flip_shapeit)) %>%
  ggplot(aes(x = Beagle, y = Eagle, color = pop)) +
  geom_point() +
  ggtitle("Flips") +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette)
```

```{r}
df_vcftools %>%
   mutate(Beagle = (n_flip_beagle),
         Eagle = (n_flip_eagle ), 
         SHAPEIT = (n_flip_shapeit)) %>%
  ggplot(aes(x = Beagle, y = SHAPEIT, color = pop)) +
  geom_point() +
  ggtitle("Flips") +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette)
```

```{r}
df_vcftools %>%
   mutate(Beagle = (n_flip_beagle),
         Eagle = (n_flip_eagle ), 
         SHAPEIT = (n_flip_shapeit)) %>%
  ggplot(aes(x = Eagle, y = SHAPEIT, color = pop)) +
  geom_point() +
  ggtitle("Flips") +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette)
```

```{r}
df_vcftools %>%
  mutate(Beagle = (n_flip_beagle),
         Eagle = (n_flip_eagle ), 
         SHAPEIT = (n_flip_shapeit))  %>%
  cross_meth_plot(p_title = "Flip Errors")
```

#### Per het. site

```{r}
df_vcftools %>%
  mutate(Beagle = (n_flip_beagle) / n_hets,
         Eagle = (n_flip_eagle ) / n_hets, 
         SHAPEIT = (n_flip_shapeit) / n_hets) %>%
  ggplot(aes(x = Beagle, y = Eagle, color = pop)) +
  geom_point() +
  ggtitle("Flips per Heterozygous Site") +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette)
```

```{r}
df_vcftools %>%
  mutate(Beagle = (n_flip_beagle) / n_hets,
         Eagle = (n_flip_eagle ) / n_hets, 
         SHAPEIT = (n_flip_shapeit) / n_hets) %>%
  ggplot(aes(x = Beagle, y = SHAPEIT, color = pop)) +
  geom_point() +
  ggtitle("Flips per Heterozygous Site") +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette)
```

```{r}
df_vcftools %>%
  mutate(Beagle = (n_flip_beagle) / n_hets,
         Eagle = (n_flip_eagle ) / n_hets, 
         SHAPEIT = (n_flip_shapeit) / n_hets) %>%
  ggplot(aes(x = Eagle, y = SHAPEIT, color = pop)) +
  geom_point() +
  ggtitle("Flips per Heterozygous Site") +
  theme_classic() +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_manual(values = cbPalette)
```

```{r}
df_vcftools %>%
  mutate(Beagle = (n_flip_beagle) / n_hets,
         Eagle = (n_flip_eagle ) / n_hets, 
         SHAPEIT = (n_flip_shapeit) / n_hets) %>%
  cross_meth_plot(p_title = "Flip Error Rates")
```

```{r}
df_vcftools %>%
  mutate(Beagle = (n_flip_beagle) / n_hets,
         Eagle = (n_flip_eagle ) / n_hets, 
         SHAPEIT = (n_flip_shapeit) / n_hets) %>%
  select(pair_id, pop, Beagle, Eagle, SHAPEIT) %>%
  pivot_longer(Beagle:SHAPEIT, names_to = "method", values_to = "error_rate") %>%
  ggplot(aes(x = method, y = error_rate)) +
  geom_boxplot() +
  xlab("Method") +
  ylab("Flip Rate") +
  theme_classic()
```

```{r}
df_vcftools %>%
  mutate(Beagle = (n_flip_beagle) / n_hets,
         Eagle = (n_flip_eagle ) / n_hets, 
         SHAPEIT = (n_flip_shapeit) / n_hets ) %>%
  select(pair_id, pop, Beagle, Eagle, SHAPEIT) %>%
  pivot_longer(Beagle:SHAPEIT, names_to = "method", values_to = "error_rate") %>%
  ggplot(aes(x = pop, y = error_rate, colour = method)) +
  geom_boxplot() +
  xlab("Population") +
  ylab("Flip Rate") +
  scale_color_manual(values = cbPalette) +
  labs(color = "Method") +
  theme_classic()
```

```{r}
df_vcftools %>%
  mutate(Beagle = (n_flip_beagle) / n_hets,
         Eagle = (n_flip_eagle ) / n_hets, 
         SHAPEIT = (n_flip_shapeit) / n_hets) %>%
  select(Beagle, Eagle, SHAPEIT) %>%
  corrr::correlate() %>%
  corrr::shave()
```


### Proportion of errors at CpG

#### T Tests

Null: enrichment = 1

```{r}
df_vcftools %>%
  mutate(Beagle = (n_switch_cpg_beagle / n_switch_beagle) / (n_het_cpg / n_hets),
         Eagle = (n_switch_cpg_eagle / n_switch_eagle) / (n_het_cpg / n_hets),
         SHAPEIT = (n_switch_cpg_shapeit / n_switch_shapeit) / (n_het_cpg / n_hets)) %>%
  select(Beagle, Eagle, SHAPEIT) %>%
  purrr::map_df(perform_one_sided_ttest_purrr, .id = "ColumnName") %>%
  knitr::kable()
```


#### Figures

```{r}
df_vcftools %>%
  mutate(Beagle = (n_switch_cpg_beagle / n_switch_beagle),
         Eagle = (n_switch_cpg_eagle / n_switch_eagle),
         SHAPEIT = (n_switch_cpg_shapeit / n_switch_shapeit)) %>%
  select(Beagle, Eagle, SHAPEIT) %>%
  pivot_longer(everything(), names_to = "method", values_to = "enrich") %>%
  ggplot(aes(x = method, y = enrich)) +
  geom_boxplot() +
  ggtitle("Errors at CpG Sites") +
  xlab("Method") +
  ylab("Proportion") +
  theme_classic()

df_vcftools %>%
  mutate(Beagle = (n_other_cpg_beagle / n_other_beagle),
         Eagle = (n_other_cpg_eagle / n_other_eagle),
         SHAPEIT = (n_other_cpg_shapeit / n_other_shapeit)) %>%
  select(Beagle, Eagle, SHAPEIT) %>%
  pivot_longer(everything(), names_to = "method", values_to = "enrich") %>%
  ggplot(aes(x = method, y = enrich)) +
  geom_boxplot() +
  ggtitle("Switches at CpG Sites") +
  xlab("Method") +
  ylab("Proportion") +
  theme_classic()

df_vcftools %>%
  mutate(Beagle = (n_flip_cpg_beagle / n_flip_beagle),
         Eagle = (n_flip_cpg_eagle / n_flip_eagle),
         SHAPEIT = (n_flip_cpg_shapeit / n_flip_shapeit)) %>%
  select(Beagle, Eagle, SHAPEIT) %>%
  pivot_longer(everything(), names_to = "method", values_to = "enrich") %>%
  ggplot(aes(x = method, y = enrich)) +
  geom_boxplot() +
  ggtitle("Flips at CpG Sites") +
  xlab("Method") +
  ylab("Proportion") +
  theme_classic()

df_vcftools %>%
  mutate(Beagle = (n_switch_cpg_beagle / n_switch_beagle) / (n_het_cpg / n_hets),
         Eagle = (n_switch_cpg_eagle / n_switch_eagle) / (n_het_cpg / n_hets),
         SHAPEIT = (n_switch_cpg_shapeit / n_switch_shapeit) / (n_het_cpg / n_hets)) %>%
  select(Beagle, Eagle, SHAPEIT) %>%
  pivot_longer(everything(), names_to = "method", values_to = "enrich") %>%
  ggplot(aes(x = method, y = enrich)) +
  geom_boxplot() +
  ggtitle("Enrichment of Errors at CpG Sites") +
  xlab("Method") +
  ylab("Enrichment") +
  theme_classic() +
  geom_hline(yintercept = 1)

df_vcftools %>%
  mutate(Beagle = (n_other_cpg_beagle / n_other_beagle) / (n_het_cpg / n_hets),
         Eagle = (n_other_cpg_eagle / n_other_eagle) / (n_het_cpg / n_hets),
         SHAPEIT = (n_other_cpg_shapeit / n_other_shapeit) / (n_het_cpg / n_hets)) %>%
  select(Beagle, Eagle, SHAPEIT) %>%
  pivot_longer(everything(), names_to = "method", values_to = "enrich") %>%
  ggplot(aes(x = method, y = enrich)) +
  geom_boxplot() +
  ggtitle("Enrichment of Switches at CpG Sites") +
  xlab("Method") +
  ylab("Enrichment") +
  theme_classic() +
  geom_hline(yintercept = 1)

df_vcftools %>%
  mutate(Beagle = (n_flip_cpg_beagle / n_flip_beagle) / (n_het_cpg / n_hets),
         Eagle = (n_flip_cpg_eagle / n_flip_eagle) / (n_het_cpg / n_hets),
         SHAPEIT = (n_flip_cpg_shapeit / n_flip_shapeit) / (n_het_cpg / n_hets)) %>%
  select(Beagle, Eagle, SHAPEIT) %>%
  pivot_longer(everything(), names_to = "method", values_to = "enrich") %>%
  ggplot(aes(x = method, y = enrich)) +
  geom_boxplot() +
  ggtitle("Enrichment of Flips at CpG Sites") +
  xlab("Method") +
  ylab("Enrichment") +
  theme_classic() +
  geom_hline(yintercept = 1)

df_vcftools %>%
  mutate(Beagle = (n_switch_cpg_beagle / n_switch_beagle) / (n_het_cpg / n_hets),
         Eagle = (n_switch_cpg_eagle / n_switch_eagle) / (n_het_cpg / n_hets),
         SHAPEIT = (n_switch_cpg_shapeit / n_switch_shapeit) / (n_het_cpg / n_hets)) %>%
  select(pop, Beagle, Eagle, SHAPEIT) %>%
  pivot_longer(-pop, names_to = "method", values_to = "enrich") %>%
  ggplot(aes(x = method, y = enrich, colour = pop)) +
  geom_boxplot() +
  ggtitle("Enrichment of Errors at CpG Sites") +
  xlab("Method") +
  ylab("Enrichment") +
  theme_classic() +
  geom_hline(yintercept = 1) +
  scale_color_manual(values=cbPalette) +
  labs(color = "Population")

df_vcftools %>%
  mutate(Beagle = (n_other_cpg_beagle / n_other_beagle) / (n_het_cpg / n_hets),
         Eagle = (n_other_cpg_eagle / n_other_eagle) / (n_het_cpg / n_hets),
         SHAPEIT = (n_other_cpg_shapeit / n_other_shapeit) / (n_het_cpg / n_hets)) %>%
  select(pop, Beagle, Eagle, SHAPEIT) %>%
  pivot_longer(-pop, names_to = "method", values_to = "enrich") %>%
  ggplot(aes(x = method, y = enrich, colour = pop)) +
  geom_boxplot() +
  ggtitle("Enrichment of Switches at CpG Sites") +
  xlab("Method") +
  ylab("Enrichment") +
  theme_classic() +
  geom_hline(yintercept = 1) +
  scale_color_manual(values=cbPalette) +
  labs(color = "Population")

df_vcftools %>%
  mutate(Beagle = (n_flip_cpg_beagle / n_flip_beagle) / (n_het_cpg / n_hets),
         Eagle = (n_flip_cpg_eagle / n_flip_eagle) / (n_het_cpg / n_hets),
         SHAPEIT = (n_flip_cpg_shapeit / n_flip_shapeit) / (n_het_cpg / n_hets)) %>%
  select(pop, Beagle, Eagle, SHAPEIT) %>%
  pivot_longer(-pop, names_to = "method", values_to = "enrich") %>%
  ggplot(aes(x = method, y = enrich, color = pop)) +
  geom_boxplot() +
  ggtitle("Enrichment of Flips at CpG Sites") +
  xlab("Method") +
  ylab("Enrichment") +
  theme_classic() +
  geom_hline(yintercept = 1) +
  scale_color_manual(values=cbPalette) +
  labs(color = "Population")
```

We observe a mild enrichment of switches and a more profound enrichment of flips at CpGs across all methods. We also observe some variation across populations within each method, for example: AFR has almost no enrichment for switches in both Beagle and SHAPEIT.

```{r}
df_vcftools %>%
  mutate(Beagle = (n_other_cpg_beagle / n_other_beagle) / (n_het_cpg / n_hets),
         Eagle = (n_other_cpg_eagle / n_other_eagle) / (n_het_cpg / n_hets),
         SHAPEIT = (n_other_cpg_shapeit / n_other_shapeit) / (n_het_cpg / n_hets)) %>%
  cross_meth_plot(p_title = "CpG Enrichment: Switches")

df_vcftools %>%
  mutate(Beagle = (n_flip_cpg_beagle / n_flip_beagle) / (n_het_cpg / n_hets),
         Eagle = (n_flip_cpg_eagle / n_flip_eagle) / (n_het_cpg / n_hets),
         SHAPEIT = (n_flip_cpg_shapeit / n_flip_shapeit) / (n_het_cpg / n_hets)) %>%
  cross_meth_plot(p_title = "CpG Enrichment: Flips")
```

#### Tables

##### Total errors

```{r}
df_vcftools %>%
  mutate(Beagle = (n_switch_cpg_beagle / n_switch_beagle),
         Eagle = (n_switch_cpg_eagle / n_switch_eagle),
         SHAPEIT = (n_switch_cpg_shapeit / n_switch_shapeit)) %>%
  tbl_summary(include = c(Beagle, Eagle, SHAPEIT),
              statistic = list(
                all_continuous() ~ "{mean} ({sd})",
                all_categorical() ~ "{n} / {N} ({p}%)")
              )

df_vcftools %>%
  mutate(Beagle = (n_switch_cpg_beagle / n_switch_beagle),
         Eagle = (n_switch_cpg_eagle / n_switch_eagle),
         SHAPEIT = (n_switch_cpg_shapeit / n_switch_shapeit)) %>%
  tbl_summary(by = pop,
              include = c(Beagle, Eagle, SHAPEIT),
              statistic = list(
                all_continuous() ~ "{mean} ({sd})",
                all_categorical() ~ "{n} / {N} ({p}%)")
              )

df_vcftools %>%
  mutate(Beagle = (n_switch_cpg_beagle / n_switch_beagle) / (n_het_cpg / n_hets),
         Eagle = (n_switch_cpg_eagle / n_switch_eagle) / (n_het_cpg / n_hets),
         SHAPEIT = (n_switch_cpg_shapeit / n_switch_shapeit) / (n_het_cpg / n_hets)) %>%
  tbl_summary(include = c(Beagle, Eagle, SHAPEIT),
              statistic = list(
                all_continuous() ~ "{mean} ({sd})",
                all_categorical() ~ "{n} / {N} ({p}%)")
              )

df_vcftools %>%
  mutate(Beagle = (n_switch_cpg_beagle / n_switch_beagle) / (n_het_cpg / n_hets),
         Eagle = (n_switch_cpg_eagle / n_switch_eagle) / (n_het_cpg / n_hets),
         SHAPEIT = (n_switch_cpg_shapeit / n_switch_shapeit) / (n_het_cpg / n_hets)) %>%
  tbl_summary(by = pop,
              include = c(Beagle, Eagle, SHAPEIT),
              statistic = list(
                all_continuous() ~ "{mean} ({sd})",
                all_categorical() ~ "{n} / {N} ({p}%)")
              )
```

##### Switches

```{r}
df_vcftools %>%
  mutate(Beagle = (n_other_cpg_beagle / n_other_beagle),
         Eagle = (n_other_cpg_eagle / n_other_eagle),
         SHAPEIT = (n_other_cpg_shapeit / n_other_shapeit)) %>%
  tbl_summary(include = c(Beagle, Eagle, SHAPEIT),
              statistic = list(
                all_continuous() ~ "{mean} ({sd})",
                all_categorical() ~ "{n} / {N} ({p}%)")
              )

df_vcftools %>%
  mutate(Beagle = (n_other_cpg_beagle / n_other_beagle),
         Eagle = (n_other_cpg_eagle / n_other_eagle),
         SHAPEIT = (n_other_cpg_shapeit / n_other_shapeit)) %>%
  tbl_summary(by = pop,
              include = c(Beagle, Eagle, SHAPEIT),
              statistic = list(
                all_continuous() ~ "{mean} ({sd})",
                all_categorical() ~ "{n} / {N} ({p}%)")
              )

df_vcftools %>%
  mutate(Beagle = (n_other_cpg_beagle / n_other_beagle) / (n_het_cpg / n_hets),
         Eagle = (n_other_cpg_eagle / n_other_eagle) / (n_het_cpg / n_hets),
         SHAPEIT = (n_other_cpg_shapeit / n_other_shapeit) / (n_het_cpg / n_hets)) %>%
  tbl_summary(include = c(Beagle, Eagle, SHAPEIT),
              statistic = list(
                all_continuous() ~ "{mean} ({sd})",
                all_categorical() ~ "{n} / {N} ({p}%)")
              )

df_vcftools %>%
  mutate(Beagle = (n_other_cpg_beagle / n_other_beagle) / (n_het_cpg / n_hets),
         Eagle = (n_other_cpg_eagle / n_other_eagle) / (n_het_cpg / n_hets),
         SHAPEIT = (n_other_cpg_shapeit / n_other_shapeit) / (n_het_cpg / n_hets)) %>%
  tbl_summary(by = pop,
              include = c(Beagle, Eagle, SHAPEIT),
              statistic = list(
                all_continuous() ~ "{mean} ({sd})",
                all_categorical() ~ "{n} / {N} ({p}%)")
              )
```

##### Flips

```{r}
df_vcftools %>%
  mutate(Beagle = (n_flip_cpg_beagle / n_flip_beagle),
         Eagle = (n_flip_cpg_eagle / n_flip_eagle),
         SHAPEIT = (n_flip_cpg_shapeit / n_flip_shapeit)) %>%
  tbl_summary(include = c(Beagle, Eagle, SHAPEIT),
              statistic = list(
                all_continuous() ~ "{mean} ({sd})",
                all_categorical() ~ "{n} / {N} ({p}%)")
              )

df_vcftools %>%
  mutate(Beagle = (n_flip_cpg_beagle / n_flip_beagle),
         Eagle = (n_flip_cpg_eagle / n_flip_eagle),
         SHAPEIT = (n_flip_cpg_shapeit / n_flip_shapeit)) %>%
  tbl_summary(by = pop,
              include = c(Beagle, Eagle, SHAPEIT),
              statistic = list(
                all_continuous() ~ "{mean} ({sd})",
                all_categorical() ~ "{n} / {N} ({p}%)")
              )

df_vcftools %>%
  mutate(Beagle = (n_flip_cpg_beagle / n_flip_beagle) / (n_het_cpg / n_hets),
         Eagle = (n_flip_cpg_eagle / n_flip_eagle) / (n_het_cpg / n_hets),
         SHAPEIT = (n_flip_cpg_shapeit / n_flip_shapeit) / (n_het_cpg / n_hets)) %>%
  tbl_summary(include = c(Beagle, Eagle, SHAPEIT),
              statistic = list(
                all_continuous() ~ "{mean} ({sd})",
                all_categorical() ~ "{n} / {N} ({p}%)")
              )

df_vcftools %>%
  mutate(Beagle = (n_flip_cpg_beagle / n_flip_beagle) / (n_het_cpg / n_hets),
         Eagle = (n_flip_cpg_eagle / n_flip_eagle) / (n_het_cpg / n_hets),
         SHAPEIT = (n_flip_cpg_shapeit / n_flip_shapeit) / (n_het_cpg / n_hets)) %>%
  tbl_summary(by = pop,
              include = c(Beagle, Eagle, SHAPEIT),
              statistic = list(
                all_continuous() ~ "{mean} ({sd})",
                all_categorical() ~ "{n} / {N} ({p}%)")
              )
```

## MAF Bins

### Draft Revision

1. What is the average enrichment of variants at rare heterozygous positions?

```{r}
maf_prop_df %>%
  pivot_wider(names_from = category, values_from = proportion) %>%
  mutate(Beagle = beagle_switch_rare / het_rare,
         Eagle = eagle_switch_rare / het_rare,
         SHAPEIT = shapeit_switch_rare / het_rare) %>%
  select(id, SP, Beagle:SHAPEIT) %>%
  summarize(Beagle = mean(Beagle),
            Eagle = mean(Eagle),
            SHAPEIT = mean(SHAPEIT)) %>%
  knitr::kable()
```


```{r}
maf_prop_df %>%
  pivot_wider(names_from = category, values_from = proportion) %>%
  mutate(Beagle = beagle_switch_rare / het_rare,
         Eagle = eagle_switch_rare / het_rare,
         SHAPEIT = shapeit_switch_rare / het_rare) %>%
  select(id, SP, Beagle:SHAPEIT) %>%
  group_by(SP) %>%
  summarize(Beagle = mean(Beagle),
            Eagle = mean(Eagle),
            SHAPEIT = mean(SHAPEIT)) %>%
  knitr::kable()
```

```{r}
maf_prop_df %>%
  pivot_wider(names_from = category, values_from = proportion) %>%
  mutate(Beagle = beagle_flip_rare / het_rare,
         Eagle = eagle_flip_rare / het_rare,
         SHAPEIT = shapeit_flip_rare / het_rare) %>%
  select(id, SP, Beagle:SHAPEIT) %>%
  summarize(Beagle = mean(Beagle),
            Eagle = mean(Eagle),
            SHAPEIT = mean(SHAPEIT)) %>%
  knitr::kable()
```


```{r}
maf_prop_df %>%
  pivot_wider(names_from = category, values_from = proportion) %>%
  mutate(Beagle = beagle_switch_rare / het_rare,
         Eagle = eagle_switch_rare / het_rare,
         SHAPEIT = shapeit_switch_rare / het_rare) %>%
  select(id, SP, Beagle:SHAPEIT) %>%
  group_by(SP) %>%
  summarize(Beagle = mean(Beagle),
            Eagle = mean(Eagle),
            SHAPEIT = mean(SHAPEIT)) %>%
  knitr::kable()
```

### Old Work

In the `maf_prop_df` data frame, we have for each synthetic diploid the proportion of heterozygous sites that are: rare variants (maf < 0.01); uncommon variants (0.01 < maf < 0.05); and common variants (maf > 0.05). We also have these proportions for each type of error for each method (switch; flip start; flip end; correct).

Let's first look at the distributions of these proportions for all heterozygous positions within each individual:

```{r}
prop_maf_fig <- function(cat_type, title_text){
  df <- maf_prop_df %>%
    filter(str_detect(category, paste0("^",cat_type, "_[rcu]"))) %>%
    mutate(category = factor(category)) %>%
    mutate(category = fct_recode(category, "Rare" = paste0(cat_type, "_rare"),
                               "Uncommon" = paste0(cat_type, "_uncommon"),
                               "Common" = paste0(cat_type, "_common"))) %>%
    mutate(category = fct_relevel(category, c("Rare", "Uncommon", "Common")))
  
  p <- df %>%
    ggplot(aes(x = category, y = proportion, color = SP)) +
    geom_boxplot() +
    ggtitle(paste0(title_text, " in MAF Bins")) +
    xlab("") +
    ylab("Proportion of Sites") +
    theme_classic() +
    labs(color = "Population") +
    scale_color_manual(values = cbPalette)
  
  tab_prop <- df %>%
    group_by(SP, category) %>%
    summarize(mean_prop = mean(proportion)) %>%
    pivot_wider(id_cols = SP, names_from = category, values_from = mean_prop)
  
  return(list(p = p, tab_prop = tab_prop))
}

res_obj <- prop_maf_fig("het", "Heterozygous Sites")
print(res_obj$p)

res_obj$tab_prop %>%
  knitr::kable()
```

### Beagle Errors

#### Switches

```{r}
cat_type <- "beagle_switch"
title_text <- "Beagle Switches"

res_obj <- prop_maf_fig(cat_type, title_text)
print(res_obj$p)

res_obj$tab_prop %>%
  knitr::kable()
```

### Flips

#### Start

```{r}
cat_type <- "beagle_flip"
title_text <- "Beagle Flip Starts"

res_obj <- prop_maf_fig(cat_type, title_text)
print(res_obj$p)

res_obj$tab_prop %>%
  knitr::kable()
```

#### End

```{r}
cat_type <- "beagle_flip end"
title_text <- "Beagle Flip Ends"

res_obj <- prop_maf_fig(cat_type, title_text)
print(res_obj$p)

res_obj$tab_prop %>%
  knitr::kable()
```

### Eagle Errors

#### Switches

```{r}
cat_type <- "eagle_switch"
title_text <- "Eagle Switches"

res_obj <- prop_maf_fig(cat_type, title_text)
print(res_obj$p)

res_obj$tab_prop %>%
  knitr::kable()
```

### Flips

#### Start

```{r}
cat_type <- "eagle_flip"
title_text <- "Eagle Flips"

res_obj <- prop_maf_fig(cat_type, title_text)
print(res_obj$p)

res_obj$tab_prop %>%
  knitr::kable()
```

#### End

```{r}
cat_type <- "eagle_flip end"
title_text <- "Eagle Flip Ends"

res_obj <- prop_maf_fig(cat_type, title_text)
print(res_obj$p)

res_obj$tab_prop %>%
  knitr::kable()
```

### SHAPEIT Errors

#### Switches

```{r}
cat_type <- "shapeit_switch"
title_text <- "SHAPEIT Switches"

res_obj <- prop_maf_fig(cat_type, title_text)
print(res_obj$p)

res_obj$tab_prop %>%
  knitr::kable()
```

### Flips

#### Start

```{r}
cat_type <- "shapeit_flip"
title_text <- "SHAPEIT Flips"

res_obj <- prop_maf_fig(cat_type, title_text)
print(res_obj$p)

res_obj$tab_prop %>%
  knitr::kable()
```

#### End

```{r}
cat_type <- "shapeit_flip end"
title_text <- "SHAPEIT Flip Ends"

res_obj <- prop_maf_fig(cat_type, title_text)
print(res_obj$p)

res_obj$tab_prop %>%
  knitr::kable()
```

### Plot all

```{r}
df_switch_maf <- maf_prop_df %>%
  filter(str_detect(category, "switch")) %>%
  separate(category, into=c("method", "type", "maf_c"), sep="_") %>%
  select(-type) %>%
  bind_rows({maf_prop_df %>%
      filter(str_starts(category, "het")) %>%
      separate(category, into=c("method", "maf_c"), sep = "_") }) %>%
  mutate(method = factor(method),
         maf_c = factor(maf_c)) %>%
  mutate(method = fct_recode(method, "Beagle" = "beagle",
                             "Eagle" = "eagle",
                             "SHAPEIT" = "shapeit",
                             "Heterozygous" = "het"),
         maf_c = fct_recode(maf_c, "Rare" = "rare",
                            "Uncommon" = "uncommon",
                            "Common" = "common")) %>%
  mutate(method = fct_relevel(method, "Heterozygous", "Beagle", "Eagle", "SHAPEIT"),
         maf_c = fct_relevel(maf_c, "Rare", "Uncommon", "Common"))

df_switch_maf %>%
  ggplot(aes(x = maf_c, y = proportion, colour = method)) +
  geom_boxplot() +
  xlab("") +
  ylab("Proportion") +
  scale_color_manual(values = cbPalette) +
  labs(color = "Method") +
  ggtitle("Switch Errors") +
  theme_classic()

df_switch_maf %>%
  group_by(method, maf_c) %>%
  summarize(proportion = mean(proportion))

df_switch_maf %>%
  filter(maf_c == "Rare") %>%
  group_by(id, method) %>%
  summarize(odds_r = proportion / (1 - proportion)) %>%
  pivot_wider(id_cols = id, names_from = "method", values_from = "odds_r") %>%
  mutate(Beagle = Beagle / Heterozygous,
         Eagle = Eagle / Heterozygous,
         SHAPEIT = SHAPEIT / Heterozygous) %>%
  ungroup() %>%
  select(Beagle, Eagle, SHAPEIT) %>%
  purrr::map_df(perform_one_sided_ttest_purrr, .id = "ColumnName") %>%
  knitr::kable()
```

```{r}
df_flip_maf <- maf_prop_df %>%
  filter(str_detect(category, "flip_")) %>%
  separate(category, into=c("method", "type", "maf_c"), sep="_") %>%
  select(-type) %>%
  bind_rows({maf_prop_df %>%
      filter(str_starts(category, "het")) %>%
      separate(category, into=c("method", "maf_c"), sep = "_") }) %>%
  mutate(method = factor(method),
         maf_c = factor(maf_c)) %>%
  mutate(method = fct_recode(method, "Beagle" = "beagle",
                             "Eagle" = "eagle",
                             "SHAPEIT" = "shapeit",
                             "Heterozygous" = "het"),
         maf_c = fct_recode(maf_c, "Rare" = "rare",
                            "Uncommon" = "uncommon",
                            "Common" = "common")) %>%
  mutate(method = fct_relevel(method, "Heterozygous", "Beagle", "Eagle", "SHAPEIT"),
         maf_c = fct_relevel(maf_c, "Rare", "Uncommon", "Common")) 

df_flip_maf %>%
  ggplot(aes(x = maf_c, y = proportion, colour = method)) +
  geom_boxplot() +
  xlab("") +
  ylab("Proportion") +
  scale_color_manual(values = cbPalette) +
  labs(color = "Method") +
  ggtitle("Flip Errors") +
  theme_classic()

df_flip_maf %>%
  group_by(method, maf_c) %>%
  summarize(proportion = mean(proportion)) 

df_flip_maf %>%
  filter(maf_c == "Rare") %>%
  group_by(id, method) %>%
  summarize(odds_r = proportion / (1 - proportion)) %>%
  pivot_wider(id_cols = id, names_from = "method", values_from = "odds_r") %>%
  mutate(Beagle = Beagle / Heterozygous,
         Eagle = Eagle / Heterozygous,
         SHAPEIT = SHAPEIT / Heterozygous) %>%
  ungroup() %>%
  select(Beagle, Eagle, SHAPEIT) %>%
  purrr::map_df(perform_one_sided_ttest_purrr, .id = "ColumnName") %>%
  knitr::kable()
```


### Enrichment of Rare

#### Enrichment of Switches at Rare

```{r}
maf_prop_df %>%
  filter(category == "het_rare" | str_detect(category, "switch_rare")) %>%
  select(id, category, proportion,SP) %>%
  pivot_wider(id_cols = c("id", "SP"), names_from = category, values_from = proportion) %>%
  mutate(Beagle = beagle_switch_rare / het_rare,
         Eagle = eagle_switch_rare / het_rare,
         SHAPEIT = shapeit_switch_rare / het_rare) %>%
  select(id, SP, Eagle, Beagle, SHAPEIT) %>%
  pivot_longer(Eagle:SHAPEIT, names_to = "method", values_to = "enrich") %>%
  ggplot(aes(x = method, y = enrich, color = SP)) +
  geom_boxplot() +
  theme_classic() +
  ggtitle("Enrichment of Switches at Rare Variants") + 
  xlab("Method") +
  ylab("Prop. rare switches / Prop. rare heterozygous sites")
```

Same plot as above, but don't separate by population:

```{r}
maf_prop_df %>%
  filter(category == "het_rare" | str_detect(category, "switch_rare")) %>%
  select(id, category, proportion) %>%
  pivot_wider(id_cols = c("id"), names_from = category, values_from = proportion) %>%
  mutate(Beagle = beagle_switch_rare / het_rare,
         Eagle = eagle_switch_rare / het_rare,
         SHAPEIT = shapeit_switch_rare / het_rare) %>%
  select(id, Eagle, Beagle, SHAPEIT) %>%
  pivot_longer(Eagle:SHAPEIT, names_to = "method", values_to = "enrich") %>%
  ggplot(aes(x = method, y = enrich)) +
  geom_boxplot() +
  theme_classic() +
  ggtitle("Enrichment of Switches at Rare Variants") + 
  xlab("Method") +
  ylab("Prop. rare switches / Prop. rare heterozygous sites") + # make axis labels bigger
  theme(axis.text = element_text(size = 14)) +
  theme(axis.title = element_text(size = 16)) +
  theme(plot.title = element_text(size = 18, hjust = 0.5))
  
```


#### Enrichment of Flips at Rare

```{r}
maf_prop_df %>%
  filter(category == "het_rare" | str_detect(category, "flip_rare")) %>%
  select(id, category, proportion,SP) %>%
  pivot_wider(id_cols = c("id", "SP"), names_from = category, values_from = proportion) %>%
  mutate(Beagle = beagle_flip_rare / het_rare,
         Eagle = eagle_flip_rare / het_rare,
         SHAPEIT = shapeit_flip_rare / het_rare) %>%
  select(id, SP, Eagle, Beagle, SHAPEIT) %>%
  pivot_longer(Eagle:SHAPEIT, names_to = "method", values_to = "enrich") %>%
  ggplot(aes(x = method, y = enrich, color = SP)) +
  geom_boxplot() +
  theme_classic() +
  ggtitle("Enrichment of Flips at Rare Variants") + 
  xlab("Method") +
  ylab("Prop. rare flips / Prop. rare heterozygous sites")
```

Same plot as above, but don't separate by population:

```{r}
maf_prop_df %>%
  filter(category == "het_rare" | str_detect(category, "flip_rare")) %>%
  select(id, category, proportion) %>%
  pivot_wider(id_cols = c("id"), names_from = category, values_from = proportion) %>%
  mutate(Beagle = beagle_flip_rare / het_rare,
         Eagle = eagle_flip_rare / het_rare,
         SHAPEIT = shapeit_flip_rare / het_rare) %>%
  select(id, Eagle, Beagle, SHAPEIT) %>%
  pivot_longer(Eagle:SHAPEIT, names_to = "method", values_to = "enrich") %>%
  ggplot(aes(x = method, y = enrich)) +
  geom_boxplot() +
  theme_classic() +
  ggtitle("Enrichment of Flips at Rare Variants") + 
  xlab("Method") +
  ylab("Prop. rare flips / Prop. rare heterozygous sites") + # make axis labels bigger
  theme(axis.text = element_text(size = 14)) +
  theme(axis.title = element_text(size = 16)) +
  theme(plot.title = element_text(size = 18, hjust = 0.5))
```


### Odds Ratios

Here we'll compute the ratio of odds of being in the rare variant category for errors and all heterozygous positions within each individual.

```{r}
maf_prop_df %>%
  select(category, proportion, id, SP) %>%
  pivot_wider(names_from = category, values_from = proportion) %>%
  mutate(or_het = het_rare / (1 - het_rare),
         or_b_s = beagle_switch_rare / (1 - beagle_switch_rare),
         or_e_s = eagle_switch_rare / (1 - eagle_switch_rare),
         or_s_s = shapeit_switch_rare / (1 - shapeit_switch_rare)) %>%
  mutate(Beagle = or_b_s / or_het,
         Eagle = or_e_s / or_het,
         SHAPEIT = or_s_s / or_het) %>%
  select(id, SP, Beagle, Eagle, SHAPEIT) %>%
  pivot_longer(!id & !SP, names_to = "Method", values_to = "OR") %>%
  ggplot(aes(x = Method, y = OR, colour = SP)) +
  geom_boxplot() +
  ggtitle("Odds Ratio of Rare Variant") +
  ylab("Odds Rare Switch / Odds Rare Heterozygous") +
  theme_classic() +
  labs(colour = "Population") +
  geom_hline(yintercept = 1, linetype = 2)

maf_prop_df %>%
  select(category, proportion, id, SP) %>%
  pivot_wider(names_from = category, values_from = proportion) %>%
  mutate(or_het = het_rare / (1 - het_rare),
         or_b_s = beagle_switch_rare / (1 - beagle_switch_rare),
         or_e_s = eagle_switch_rare / (1 - eagle_switch_rare),
         or_s_s = shapeit_switch_rare / (1 - shapeit_switch_rare)) %>%
  mutate(Beagle = or_b_s / or_het,
         Eagle = or_e_s / or_het,
         SHAPEIT = or_s_s / or_het) %>%
  tbl_summary(include = c(Beagle, Eagle, SHAPEIT),
              statistic = list(
                all_continuous() ~ "{mean} ({sd})",
                all_categorical() ~ "{n} / {N} ({p}%)")
              )

maf_prop_df %>%
  select(category, proportion, id, SP) %>%
  pivot_wider(names_from = category, values_from = proportion) %>%
  mutate(or_het = het_rare / (1 - het_rare),
         or_b_s = beagle_switch_rare / (1 - beagle_switch_rare),
         or_e_s = eagle_switch_rare / (1 - eagle_switch_rare),
         or_s_s = shapeit_switch_rare / (1 - shapeit_switch_rare)) %>%
  mutate(Beagle = or_b_s / or_het,
         Eagle = or_e_s / or_het,
         SHAPEIT = or_s_s / or_het) %>%
  tbl_summary(by = SP,
              include = c(Beagle, Eagle, SHAPEIT),
              statistic = list(
                all_continuous() ~ "{mean} ({sd})",
                all_categorical() ~ "{n} / {N} ({p}%)")
              )
```

Let's use a T test to compare the observed odds ratios to 1:

```{r}
maf_prop_df %>%
  select(category, proportion, id, SP) %>%
  pivot_wider(names_from = category, values_from = proportion) %>%
  mutate(or_het = het_rare / (1 - het_rare),
         or_b_s = beagle_switch_rare / (1 - beagle_switch_rare),
         or_e_s = eagle_switch_rare / (1 - eagle_switch_rare),
         or_s_s = shapeit_switch_rare / (1 - shapeit_switch_rare)) %>%
  mutate(Beagle = or_b_s / or_het,
         Eagle = or_e_s / or_het,
         SHAPEIT = or_s_s / or_het) %>%
  select(Beagle, Eagle, SHAPEIT) %>%
  purrr::map_df(perform_one_sided_ttest_purrr, .id = "ColumnName") %>%
  knitr::kable()


maf_prop_df %>%
  select(category, proportion, id, SP) %>%
  pivot_wider(names_from = category, values_from = proportion) %>%
  mutate(or_het = het_rare / (1 - het_rare),
         or_b_s = beagle_flip_rare / (1 - beagle_flip_rare),
         or_e_s = eagle_flip_rare / (1 - beagle_flip_rare),
         or_s_s = shapeit_flip_rare / (1 - beagle_flip_rare)) %>%
  mutate(Beagle = or_b_s / or_het,
         Eagle = or_e_s / or_het,
         SHAPEIT = or_s_s / or_het) %>%
  select(Beagle, Eagle, SHAPEIT) %>%
  purrr::map_df(perform_one_sided_ttest_purrr, .id = "ColumnName") %>%
  knitr::kable()
```


```{r}
maf_prop_df %>%
  select(category, proportion, id, SP) %>%
  pivot_wider(names_from = category, values_from = proportion) %>%
  mutate(or_het = het_rare / (1 - het_rare),
         or_b_s = beagle_flip_rare / (1 - beagle_flip_rare),
         or_e_s = eagle_flip_rare / (1 - beagle_flip_rare),
         or_s_s = shapeit_flip_rare / (1 - beagle_flip_rare)) %>%
  mutate(Beagle = or_b_s / or_het,
         Eagle = or_e_s / or_het,
         SHAPEIT = or_s_s / or_het) %>%
  select(id, SP, Beagle, Eagle, SHAPEIT) %>%
  pivot_longer(!id & !SP, names_to = "Method", values_to = "OR") %>%
  ggplot(aes(x = Method, y = OR, colour = SP)) +
  geom_boxplot() +
  ggtitle("Odds Ratio of Rare Variant") +
  ylab("Odds Rare Flip / Odds Rare Heterozygous") +
  theme_classic() +
  labs(colour = "Population") +
  geom_hline(yintercept = 1, linetype = 2)

maf_prop_df %>%
  select(category, proportion, id, SP) %>%
  pivot_wider(names_from = category, values_from = proportion) %>%
  mutate(or_het = het_rare / (1 - het_rare),
         or_b_s = beagle_flip_rare / (1 - beagle_flip_rare),
         or_e_s = eagle_flip_rare / (1 - beagle_flip_rare),
         or_s_s = shapeit_flip_rare / (1 - beagle_flip_rare)) %>%
  mutate(Beagle = or_b_s / or_het,
         Eagle = or_e_s / or_het,
         SHAPEIT = or_s_s / or_het) %>%
  tbl_summary(include = c(Beagle, Eagle, SHAPEIT),
              statistic = list(
                all_continuous() ~ "{mean} ({sd})",
                all_categorical() ~ "{n} / {N} ({p}%)")
              )

maf_prop_df %>%
  select(category, proportion, id, SP) %>%
  pivot_wider(names_from = category, values_from = proportion) %>%
  mutate(or_het = het_rare / (1 - het_rare),
         or_b_s = beagle_flip_rare / (1 - beagle_flip_rare),
         or_e_s = eagle_flip_rare / (1 - beagle_flip_rare),
         or_s_s = shapeit_flip_rare / (1 - beagle_flip_rare)) %>%
  mutate(Beagle = or_b_s / or_het,
         Eagle = or_e_s / or_het,
         SHAPEIT = or_s_s / or_het) %>%
  tbl_summary(by = SP,
              include = c(Beagle, Eagle, SHAPEIT),
              statistic = list(
                all_continuous() ~ "{mean} ({sd})",
                all_categorical() ~ "{n} / {N} ({p}%)")
              )
```

