---
title: "trio_phasing_results"
author: "Andy Beck"
date: "2024-05-20"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction

Here in this document we present figures, tables, and statistics for the phasing evaluation on chromosome 15. This sample consists of 602 individuals phased using a reference panel of unrelated individuals from the 1000 genomes project. The phasing results are compared to a pedigree-adjusted phased genome for each sample.

## Libraries and Data

```{r}
library(tidyverse)
library(pander)
library(reactable)
library(yaml)

config_obj <- yaml::read_yaml("_config.yaml")

cbPalette <- c("#999999", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")

df_subj <- read_csv("data/1kgp/subject_info.csv")
ped_df <- read_table("data/1kgp/1kGP.3202_samples.pedigree_info.txt")
child_ids <- ped_df %>% filter(fatherID != "0" & motherID != "0") %>% pull(sampleID)
df_subj_rel <- df_subj %>%
  filter(SAMPLE_NAME %in% child_ids)
df_subj_rel$id2 <- 1:602

df_switch_trio <- read_csv(paste0(config_obj$base_dir,"/",
                                  config_obj$trio_result_dir, "/switch_errors/summary.csv"))

df_switch_trio_22 <- read_csv(paste0(config_obj$base_dir,
                                     "/output/trio_phase_22/switch_errors/summary.csv"))

df_het <- read_tsv(paste0(config_obj$base_dir, "/", config_obj$trio_result_dir, "/het_pos_count.tsv"),
                   col_names = c("id", "n_het"))
df_het_22 <- read_tsv(paste0(config_obj$base_dir, "/output/trio_phase_22/het_pos_count.tsv"),
                   col_names = c("id", "n_het22"))

n_var_site <- 1627002
n_var_site_22 <- 767007

df_het$frac_het <- df_het$n_het / n_var_site
df_het_22$frac_het22 <- df_het_22$n_het22 / n_var_site_22

df_subj_rel <- left_join(df_subj_rel, df_het, by = c("id2" = "id")) %>%
  left_join(df_het_22, by = c("id2" = "id"))

```

### Chr X Results

We'll also pull in the results on chromosome X for the synthetic diploids to allow for comparisons of rates between 15 and X.

```{r}
sd_pair_info_df <- read_delim("data/sample_pairs.csv", col_names = c("POP", "ID1", "ID2"))
sd_pair_info_df <- left_join(sd_pair_info_df, df_subj, by = c("ID1"="SAMPLE_NAME")) %>%
  rename(SP = SUPER) %>%
  select(-POPULATION)

df_switch_x <- read_csv(paste0(config_obj$base_dir,"/output/switch_errors/switch_errors/summary.csv"))
df_het_x <- read_tsv(paste0(config_obj$base_dir, "/output/switch_errors/het_pos_count.tsv"), col_names = c("id", "n_het"))
n_var_site_x <- 1722954
df_het_x$frac_het <- df_het_x$n_het / n_var_site_x

sd_pair_info_df$n_het <-df_het_x$n_het
sd_pair_info_df$frac_het <-df_het_x$frac_het
```


## Subject Information

The 1000 Genomes collection of 3,202 genomes is sampled from 26 different populations, each belonging to 1 of 5 distinct super-populations (AFR, AMR, EAS, EUR, SAS). All 26 populations are represented in the 2,504 1kGP phase 3 subjects, but not all are present in the 602 trios sequenced by the New York Genome Center:

```{r}
df_subj %>%
  group_by(SUPER, POPULATION) %>%
  summarize(N = n()) %>%
  reactable( 
            groupBy = c("SUPER"),
            columns = list(
              N = colDef(aggregate = "sum")),
    bordered = TRUE,
    defaultExpanded = TRUE)

df_subj_rel %>%
  group_by(SUPER, POPULATION) %>%
  summarize(N = n()) %>%
  reactable( 
            groupBy = c("SUPER"),
            columns = list(
              N = colDef(aggregate = "sum")),
    bordered = TRUE,
    defaultExpanded = TRUE)
```

For completeness, we also tabulate other summary information about the 602 trios (which is just sex; 1 - male, 2 - female):

```{r}
df_subj_rel %>%
  group_by(sex) %>%
  summarize(N = n()) %>%
  reactable( 
            groupBy = c("sex"),
            columns = list(
              N = colDef(aggregate = "sum")),
    bordered = TRUE,
    defaultExpanded = TRUE)
```

## Chromosome 15 Information

Things I'd like:

* Density of heterozygous SNPs per individual
* GC content of heterozygous positions

First let's look at the density of heterozygous positions in each sample, i.e. what fraction of the sites are heterozygous.

```{r}
df_subj_rel %>%
  ggplot() +
  stat_density(aes(x = frac_het, colour = SUPER),
                  geom="line",position="identity") +
  theme_classic() +
  ggtitle("Fraction of Sites Heterozygous", "Chromosome 15") +
  labs(colour="Population") +
  xlab("Fraction of Sites Heterozygous") +
  scale_colour_manual(values=cbPalette)

df_subj_rel %>%
  ggplot() +
  geom_boxplot(aes(x = frac_het, y = SUPER , fill = SUPER), show.legend = FALSE) +
  theme_classic() +
  ggtitle("Fraction of Sites Heterozygous", "Chromosome 15") +
  labs(fill="Population") +
  xlab("Fraction of Sites Heterozygous") +
  ylab("Population") +
  scale_fill_manual(values=cbPalette)

df_subj_rel %>%
  ggplot() +
  geom_boxplot(aes(x = n_het, y = SUPER , fill = SUPER), show.legend = FALSE) +
  theme_classic() +
  ggtitle("Number of Heterozygous Sites", "Chromosome 15") +
  labs(fill="Population") +
  xlab("Number of Heterozygous Sites") +
  ylab("Population") +
  scale_fill_manual(values=cbPalette)
```

How does this compare to what we observe in the X chromosome?

```{r}
sd_pair_info_df %>%
  ggplot() +
  stat_density(aes(x = frac_het, colour = SP),
                  geom="line",position="identity") +
  theme_classic() +
  ggtitle("Fraction of Sites Heterozygous", "Chromosome X") +
  labs(colour="Population") +
  xlab("Fraction of Sites Heterozygous") +
  scale_colour_manual(values=cbPalette)

sd_pair_info_df %>%
  ggplot() +
  geom_boxplot(aes(x = frac_het, y = SP , fill = SP), show.legend = FALSE) +
  theme_classic() +
  ggtitle("Fraction of Sites Heterozygous", "Chromosome X") +
  labs(fill="Population") +
  xlab("Fraction of Sites Heterozygous") +
  ylab("Population") +
  scale_fill_manual(values=cbPalette)

sd_pair_info_df %>%
  ggplot() +
  geom_boxplot(aes(x = n_het, y = SP , fill = SP), show.legend = FALSE) +
  theme_classic() +
  ggtitle("Number of Heterozygous Sites", "Chromosome X") +
  labs(fill="Population") +
  xlab("Number of Heterozygous Sites") +
  ylab("Population") +
  scale_fill_manual(values=cbPalette)
```

And for another comparison, we'll summarize heterozygosity by super population:

```{r}
df_subj_rel %>%
  group_by(SUPER) %>%
  summarize(mean_het_15 = mean(frac_het)) %>%
  left_join({ sd_pair_info_df %>%
      group_by(SP) %>%
      summarize(mean_het_x = mean(frac_het))
  }, by = c("SUPER" = "SP")) %>%
  mutate(x_15_ratio = mean_het_x/mean_het_15) %>%
  knitr::kable()
```

And finally, since samples are shared between 15 and 22, let's look at how fraction of heterozygous sites looks like across chromosomes:

```{r}
df_subj_rel %>%
  ggplot(aes(frac_het, frac_het22, color = SUPER)) +
  geom_point() +
  geom_abline(slope = 1, intercept = 0) +
  theme_classic() +
  xlab("Chromosome 15 % Heterozygous") +
  ylab("Chromosome 22 % Heterozygous") +
  labs(color = "Population") +
  scale_color_manual(values = cbPalette)
  
```

## Switch Errors

### Total Errors

```{r}
plot_total_errors <- function(df, m1, m2, pop){
  p <- df %>%
    mutate(Beagle = n_switch_beagle,
         Eagle = n_switch_eagle,
         SHAPEIT = n_switch_shapeit) %>%
    ggplot(aes(x = !! sym(m1), y = !! sym(m2), colour = !! sym(pop))) +
    geom_point() +
    geom_abline(slope = 1, intercept = 0) +
    ggtitle("Total Errors", "Chromosome 15") +
    theme_classic() +
    xlab(m1) +
    ylab(m2) +
    labs(color="Population") +
    scale_color_manual(values = cbPalette)
  return(p)
}

plot_total_errors(df_switch_trio, "Beagle", "Eagle", "SUPER")
plot_total_errors(df_switch_trio, "Beagle", "SHAPEIT", "SUPER")
plot_total_errors(df_switch_trio, "SHAPEIT", "Eagle", "SUPER")
```

And now chromosome 22:

```{r}
plot_total_errors(df_switch_trio_22, "Beagle", "Eagle", "SUPER")
plot_total_errors(df_switch_trio_22, "Beagle", "SHAPEIT", "SUPER")
plot_total_errors(df_switch_trio_22, "SHAPEIT", "Eagle", "SUPER")
```

And finally, chromosome X:

```{r}
plot_total_errors(df_switch_x, "Beagle", "Eagle", "pop")
plot_total_errors(df_switch_x, "Beagle", "SHAPEIT", "pop")
plot_total_errors(df_switch_x, "SHAPEIT", "Eagle", "pop")
```


#### Error Rate

```{r}
plot_total_rate <- function(df, m1, m2, pop){
  p <- df %>%
    mutate(Beagle = n_switch_beagle / n_hets,
         Eagle = n_switch_eagle / n_hets,
         SHAPEIT = n_switch_shapeit / n_hets) %>%
    ggplot(aes(x = !! sym(m1), y = !! sym(m2), colour = !! sym(pop))) +
    geom_point() +
    geom_abline(slope = 1, intercept = 0) +
    ggtitle("Total Errors", "Chromosome 15") +
    theme_classic() +
    xlab(m1) +
    ylab(m2) +
    labs(color="Population") +
    scale_color_manual(values = cbPalette)
  return(p)
}

plot_total_rate(df_switch_trio, "Beagle", "Eagle", "SUPER")
plot_total_rate(df_switch_trio, "Beagle", "SHAPEIT", "SUPER")
plot_total_rate(df_switch_trio, "SHAPEIT", "Eagle", "SUPER")
```

And now chromosome 22:

```{r}
plot_total_rate(df_switch_trio_22, "Beagle", "Eagle", "SUPER")
plot_total_rate(df_switch_trio_22, "Beagle", "SHAPEIT", "SUPER")
plot_total_rate(df_switch_trio_22, "SHAPEIT", "Eagle", "SUPER")
```

And finally, chromosome X:

```{r}
plot_total_rate(df_switch_x, "Beagle", "Eagle", "pop")
plot_total_rate(df_switch_x, "Beagle", "SHAPEIT", "pop")
plot_total_rate(df_switch_x, "SHAPEIT", "Eagle", "pop")
```

### Non-flip Switches

```{r}
plot_switches <- function(df, m1, m2, pop){
  p <- df %>%
    mutate(Beagle = n_other_beagle,
         Eagle = n_other_eagle,
         SHAPEIT = n_other_shapeit) %>%
    ggplot(aes(x = !! sym(m1), y = !! sym(m2), colour = !! sym(pop))) +
    geom_point() +
    geom_abline(slope = 1, intercept = 0) +
    ggtitle("Total Errors", "Chromosome 15") +
    theme_classic() +
    xlab(m1) +
    ylab(m2) +
    labs(color="Population") +
    scale_color_manual(values = cbPalette)
  return(p)
}

plot_switches(df_switch_trio, "Beagle", "Eagle", "SUPER")
plot_switches(df_switch_trio, "Beagle", "SHAPEIT", "SUPER")
plot_switches(df_switch_trio, "SHAPEIT", "Eagle", "SUPER")
```

And now chromosome 22:

```{r}
plot_switches(df_switch_trio_22, "Beagle", "Eagle", "SUPER")
plot_switches(df_switch_trio_22, "Beagle", "SHAPEIT", "SUPER")
plot_switches(df_switch_trio_22, "SHAPEIT", "Eagle", "SUPER")
```

And finally, chromosome X:

```{r}
plot_switches(df_switch_x, "Beagle", "Eagle", "pop")
plot_switches(df_switch_x, "Beagle", "SHAPEIT", "pop")
plot_switches(df_switch_x, "SHAPEIT", "Eagle", "pop")
```

#### Error Rate

```{r}
plot_switch_rate <- function(df, m1, m2, pop){
  p <- df %>%
    mutate(Beagle = n_other_beagle / n_hets,
         Eagle = n_other_eagle / n_hets,
         SHAPEIT = n_other_shapeit / n_hets) %>%
    ggplot(aes(x = !! sym(m1), y = !! sym(m2), colour = !! sym(pop))) +
    geom_point() +
    geom_abline(slope = 1, intercept = 0) +
    ggtitle("Total Errors", "Chromosome 15") +
    theme_classic() +
    xlab(m1) +
    ylab(m2) +
    labs(color="Population") +
    scale_color_manual(values = cbPalette)
  return(p)
}

plot_switch_rate(df_switch_trio, "Beagle", "Eagle", "SUPER")
plot_switch_rate(df_switch_trio, "Beagle", "SHAPEIT", "SUPER")
plot_switch_rate(df_switch_trio, "SHAPEIT", "Eagle", "SUPER")
```

And now chromosome 22:

```{r}
plot_switch_rate(df_switch_trio_22, "Beagle", "Eagle", "SUPER")
plot_switch_rate(df_switch_trio_22, "Beagle", "SHAPEIT", "SUPER")
plot_switch_rate(df_switch_trio_22, "SHAPEIT", "Eagle", "SUPER")
```

And finally, chromosome X:

```{r}
plot_switch_rate(df_switch_x, "Beagle", "Eagle", "pop")
plot_switch_rate(df_switch_x, "Beagle", "SHAPEIT", "pop")
plot_switch_rate(df_switch_x, "SHAPEIT", "Eagle", "pop")
```

### Flips

```{r}
plot_flips <- function(df, m1, m2, pop){
  p <- df %>%
    mutate(Beagle = n_flip_beagle,
         Eagle = n_flip_eagle,
         SHAPEIT = n_flip_shapeit) %>%
    ggplot(aes(x = !! sym(m1), y = !! sym(m2), colour = !! sym(pop))) +
    geom_point() +
    geom_abline(slope = 1, intercept = 0) +
    ggtitle("Total Errors", "Chromosome 15") +
    theme_classic() +
    xlab(m1) +
    ylab(m2) +
    labs(color="Population") +
    scale_color_manual(values = cbPalette)
  return(p)
}

plot_flips(df_switch_trio, "Beagle", "Eagle", "SUPER")
plot_flips(df_switch_trio, "Beagle", "SHAPEIT", "SUPER")
plot_flips(df_switch_trio, "SHAPEIT", "Eagle", "SUPER")
```

And now chromosome 22:

```{r}
plot_flips(df_switch_trio_22, "Beagle", "Eagle", "SUPER")
plot_flips(df_switch_trio_22, "Beagle", "SHAPEIT", "SUPER")
plot_flips(df_switch_trio_22, "SHAPEIT", "Eagle", "SUPER")
```

And finally, chromosome X:

```{r}
plot_flips(df_switch_x, "Beagle", "Eagle", "pop")
plot_flips(df_switch_x, "Beagle", "SHAPEIT", "pop")
plot_flips(df_switch_x, "SHAPEIT", "Eagle", "pop")
```

#### Error Rate

```{r}
plot_flip_rate <- function(df, m1, m2, pop){
  p <- df %>%
    mutate(Beagle = n_flip_beagle / n_hets,
         Eagle = n_flip_eagle / n_hets,
         SHAPEIT = n_flip_shapeit / n_hets) %>%
    ggplot(aes(x = !! sym(m1), y = !! sym(m2), colour = !! sym(pop))) +
    geom_point() +
    geom_abline(slope = 1, intercept = 0) +
    ggtitle("Total Errors", "Chromosome 15") +
    theme_classic() +
    xlab(m1) +
    ylab(m2) +
    labs(color="Population") +
    scale_color_manual(values = cbPalette)
  return(p)
}

plot_flip_rate(df_switch_trio, "Beagle", "Eagle", "SUPER")
plot_flip_rate(df_switch_trio, "Beagle", "SHAPEIT", "SUPER")
plot_flip_rate(df_switch_trio, "SHAPEIT", "Eagle", "SUPER")
```

And now chromosome 22:

```{r}
plot_flip_rate(df_switch_trio_22, "Beagle", "Eagle", "SUPER")
plot_flip_rate(df_switch_trio_22, "Beagle", "SHAPEIT", "SUPER")
plot_flip_rate(df_switch_trio_22, "SHAPEIT", "Eagle", "SUPER")
```

And finally, chromosome X:

```{r}
plot_flip_rate(df_switch_x, "Beagle", "Eagle", "pop")
plot_flip_rate(df_switch_x, "Beagle", "SHAPEIT", "pop")
plot_flip_rate(df_switch_x, "SHAPEIT", "Eagle", "pop")
```
